

<!DOCTYPE html>
<html lang="zh-CN" data-default-color-scheme=&#34;auto&#34;>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/favicon.png">
  <link rel="icon" href="/img/favicon.png">
  <meta name="viewport"
        content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  
    <meta name="description" content="">
  
  <meta name="author" content="John Doe">
  <meta name="keywords" content="">
  
  <title>20210728 R practice - Hexo</title>

  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" />


  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/github-markdown-css@4.0.0/github-markdown.min.css" />
  <link  rel="stylesheet" href="/lib/hint/hint.min.css" />

  
    
    
      
      <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@10.4.0/styles/github-gist.min.css" />
    
  

  
    <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.css" />
  



<!-- 主题依赖的图标库，不要自行修改 -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_ba1fz6golrf.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_kmeydafke9r.css">


<link  rel="stylesheet" href="/css/main.css" />

<!-- 自定义样式保持在最底部 -->


  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    var CONFIG = {"hostname":"example.com","root":"/","version":"1.8.8","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"right","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"copy_btn":true,"image_zoom":{"enable":true},"toc":{"enable":true,"headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"onlypost":false},"web_analytics":{"enable":false,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":null,"app_key":null,"server_url":null}}};
  </script>
  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
<meta name="generator" content="Hexo 5.4.0"></head>


<body>
  <header style="height: 70vh;">
    <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand"
       href="/">&nbsp;<strong>Liu Keqin</strong>&nbsp;</a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/">
                <i class="iconfont icon-home-fill"></i>
                首页
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/">
                <i class="iconfont icon-archive-fill"></i>
                归档
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/">
                <i class="iconfont icon-category-fill"></i>
                分类
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/">
                <i class="iconfont icon-tags-fill"></i>
                标签
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/">
                <i class="iconfont icon-user-fill"></i>
                关于
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" data-toggle="modal" data-target="#modalSearch">&nbsp;<i
                class="iconfont icon-search"></i>&nbsp;</a>
          </li>
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" href="javascript:">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

    <div class="banner" id="banner" parallax=true
         style="background: url('/img/default.png') no-repeat center center;
           background-size: cover;">
      <div class="full-bg-img">
        <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
          <div class="page-header text-center fade-in-up">
            <span class="h2" id="subtitle" title="20210728 R practice">
              
            </span>

            
              <div class="mt-3">
  
  
    <span class="post-meta">
      <i class="iconfont icon-date-fill" aria-hidden="true"></i>
      <time datetime="2021-07-28 17:18" pubdate>
        2021年7月28日 下午
      </time>
    </span>
  
</div>

<div class="mt-1">
  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-chart"></i>
      7.8k 字
    </span>
  

  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-clock-fill"></i>
      
      
      153
       分钟
    </span>
  

  
  
</div>

            
          </div>

          
        </div>
      </div>
    </div>
  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="d-none d-lg-block col-lg-2"></div>
    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div class="py-5" id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">20210728 R practice</h1>
            
            <div class="markdown-body">
              <h1 id="R-practice"><a href="#R-practice" class="headerlink" title="R practice"></a>R practice</h1><p>R version 4.0.2 (2020-06-22)</p>
<h2 id="tibble"><a href="#tibble" class="headerlink" title="tibble"></a>tibble</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br></pre></td><td class="code"><pre><code class="hljs R">library(tidyverse)<br>as_tibble(iris)<br><br>tibble(<br>  x = <span class="hljs-number">1</span>:<span class="hljs-number">5</span>,<br>  y = <span class="hljs-number">1</span>,<br>  z = x^<span class="hljs-number">2</span><br>)<br><br>tibble(<br>  col_1 = <span class="hljs-string">&#x27;feature1&#x27;</span>,<br>  col_2 = <span class="hljs-string">&#x27;feature2&#x27;</span>,<br>  col_3 = <span class="hljs-string">&#x27;feature3&#x27;</span><br>)<br><br>tribble(<br>  ~feature_1, ~feature_2, ~feature_3,<br>  <span class="hljs-comment">#---------|-----------|-----------|</span><br>  <span class="hljs-string">&#x27;a&#x27;</span>,<span class="hljs-string">&#x27;b&#x27;</span>,<span class="hljs-string">&#x27;c&#x27;</span>,<br>  <span class="hljs-string">&#x27;d&#x27;</span>,<span class="hljs-string">&#x27;e&#x27;</span>,<span class="hljs-string">&#x27;f&#x27;</span><br>)<br><br>tibble(<br>  a = lubridate::now() + runif(<span class="hljs-number">1e3</span>) * <span class="hljs-number">86400</span>,<br>  b = lubridate::today() + runif(<span class="hljs-number">1e3</span>) * <span class="hljs-number">30</span>,<br>  <span class="hljs-built_in">c</span> = <span class="hljs-number">1</span>:<span class="hljs-number">1e3</span>,<br>  d = runif(<span class="hljs-number">1e3</span>),<br>  e = sample(<span class="hljs-built_in">letters</span>, <span class="hljs-number">1e3</span>, replace = <span class="hljs-literal">TRUE</span>)<br>)%&gt;%<br>print(n = <span class="hljs-number">20</span>, width = <span class="hljs-literal">Inf</span>)<br><br>tibble(<br>  a = lubridate::now() + runif(<span class="hljs-number">1e3</span>) * <span class="hljs-number">86400</span>,<br>  b = lubridate::today() + runif(<span class="hljs-number">1e3</span>) * <span class="hljs-number">30</span>,<br>  <span class="hljs-built_in">c</span> = <span class="hljs-number">1</span>:<span class="hljs-number">1e3</span>,<br>  d = runif(<span class="hljs-number">1e3</span>),<br>  e = sample(<span class="hljs-built_in">letters</span>, <span class="hljs-number">1e3</span>, replace = <span class="hljs-literal">TRUE</span>)<br>)%&gt;%<br>  View()<br><br><br>options(tibble.print_max = <span class="hljs-number">20</span>, tibble.print_min = <span class="hljs-number">8</span>)<br>tibble(<br>  a = lubridate::now() + runif(<span class="hljs-number">1e3</span>) * <span class="hljs-number">86400</span>,<br>  b = lubridate::today() + runif(<span class="hljs-number">1e3</span>) * <span class="hljs-number">30</span>,<br>  <span class="hljs-built_in">c</span> = <span class="hljs-number">1</span>:<span class="hljs-number">1e3</span>,<br>  d = runif(<span class="hljs-number">1e3</span>),<br>  e = sample(<span class="hljs-built_in">letters</span>, <span class="hljs-number">1e3</span>, replace = <span class="hljs-literal">TRUE</span>)<br>)<br><br>df = tibble(<br>  x = runif(<span class="hljs-number">5</span>),<br>  y = rnorm(<span class="hljs-number">5</span>)<br>)<br>df$x  <br>df$y<br>df[[<span class="hljs-number">1</span>]]<br>df %&gt;% .$x<br>df %&gt;% .[[<span class="hljs-number">1</span>]]<br><br><span class="hljs-built_in">class</span>(as.data.frame(df))<br><br>as_tibble(mtcars)%&gt;%<br>  View()<br>mtcars$cyl<br>mtcars$wt<br>mtcars[[<span class="hljs-number">1</span>]]<br><br>test = tibble(mtcars)<br>cars = tibble(mpg = mtcars$mpg, qsec = mtcars$qsec) <br>ggplot(cars, aes(x = mpg, y = qsec))+<br>  geom_point()<br><br>test1 = tibble(mtcars)<br>test2 = tibble(col1 = test1[[<span class="hljs-number">1</span>]], col2 = test1[[<span class="hljs-number">2</span>]], col3 = test1[[<span class="hljs-number">1</span>]]/test[[<span class="hljs-number">2</span>]]) %&gt;%<br>  View()<br></code></pre></td></tr></table></figure>
<h2 id="string"><a href="#string" class="headerlink" title="string"></a>string</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><code class="hljs R">library(tidyverse)<br>library(stringr)<br><br>x &lt;- <span class="hljs-string">&quot;this is a sentence&quot;</span> <br>str_view_all(x, boundary(<span class="hljs-string">&quot;word&quot;</span>))<br><br>str_split(x, <span class="hljs-string">&quot; &quot;</span>, simplify = <span class="hljs-literal">TRUE</span>)<br><br>str_view_all(<span class="hljs-string">&quot;ababab&quot;</span>, <span class="hljs-string">&quot;ab&quot;</span>)<br><br>str_locate(<span class="hljs-string">&quot;abababa&quot;</span>, <span class="hljs-string">&quot;ab&quot;</span>)<br><br>month_levels &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;Jan&quot;</span>, <span class="hljs-string">&quot;Feb&quot;</span>, <span class="hljs-string">&quot;Mar&quot;</span>, <span class="hljs-string">&quot;Apr&quot;</span>, <span class="hljs-string">&quot;May&quot;</span>, <span class="hljs-string">&quot;Jun&quot;</span>, <br>                  <span class="hljs-string">&quot;Jul&quot;</span>, <span class="hljs-string">&quot;Aug&quot;</span>, <span class="hljs-string">&quot;Sep&quot;</span>, <span class="hljs-string">&quot;Oct&quot;</span>, <span class="hljs-string">&quot;Nov&quot;</span>, <span class="hljs-string">&quot;Dec&quot;</span>)<br>x1 &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;Dec&quot;</span>, <span class="hljs-string">&quot;Feb&quot;</span>, <span class="hljs-string">&quot;Jan&quot;</span>, <span class="hljs-string">&quot;Oct&quot;</span>)<br>y1 &lt;- factor(x1, levels = month_levels)<br>y1<br>sort(y1)<br><br>View(flights)<br>library(ggplot2)<br>flights1 &lt;- flights<br>  ggplot(flights1)+<br>  geom_point(mapping = aes(air_time,distance))<br></code></pre></td></tr></table></figure>
<h2 id="read-r"><a href="#read-r" class="headerlink" title="read r"></a>read r</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br></pre></td><td class="code"><pre><code class="hljs R">library(tidyverse)<br>diamonds_tibble &lt;- read_csv(<span class="hljs-string">&quot;diamonds.csv&quot;</span>)<br>View(diamonds_tibble)<br><br>read_csv(<br>  <span class="hljs-string">&quot;a,b,c</span><br><span class="hljs-string">  1,2,3</span><br><span class="hljs-string">  4,5,6&quot;</span><br>)<br><br>read_csv(<br>  <span class="hljs-string">&quot;the first line of metadata</span><br><span class="hljs-string">  the second line of metadata</span><br><span class="hljs-string">  x,y,z</span><br><span class="hljs-string">  1,2,3&quot;</span>,<br>  skip=<span class="hljs-number">2</span><br>)<br><br>read_csv(<br>  <span class="hljs-string">&quot;#a comment i want to skip</span><br><span class="hljs-string">  x,y,z</span><br><span class="hljs-string">  1,2,3&quot;</span>,<br>  comment=<span class="hljs-string">&#x27;#&#x27;</span><br>)<br><br>read_csv(<br>  <span class="hljs-string">&quot;1,2,3</span><br><span class="hljs-string">  4,5,6&quot;</span>,<br>  col_names = <span class="hljs-literal">FALSE</span><br>)<br><br>read_csv(<br>  <span class="hljs-string">&quot;1,2,3</span><br><span class="hljs-string">  4,5,6&quot;</span>,<br>  col_names = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;x&quot;</span>,<span class="hljs-string">&quot;y&quot;</span>,<span class="hljs-string">&quot;z&quot;</span>)<br>)<br><br>read_csv(<br>  <span class="hljs-string">&quot;1,2,3\n4,5,6\n7,8,9&quot;</span>,<br>  col_names = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;test1&quot;</span>,<span class="hljs-string">&quot;test2&quot;</span>,<span class="hljs-string">&quot;test3&quot;</span>)<br>)<br><br>read_csv(<br>  <span class="hljs-string">&quot;1,2,3\n4,5,.&quot;</span>,<br>  na = <span class="hljs-string">&quot;.&quot;</span><br>)<br><br>read_csv(<br>  <span class="hljs-string">&quot;a,b\n1,2,3&quot;</span><br>)<br><br>read_csv(<br>  <span class="hljs-string">&quot;a,b,c\n1,2&quot;</span><br>)<br><br>read_csv(<br>  <span class="hljs-string">&quot;a,b\n\1,2,3&quot;</span><br>)<br></code></pre></td></tr></table></figure>
<h2 id="parse"><a href="#parse" class="headerlink" title="parse"></a>parse</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br></pre></td><td class="code"><pre><code class="hljs R">library(tidyverse)<br>str(parse_logical(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;TRUE&quot;</span>,<span class="hljs-string">&quot;FALSE&quot;</span>,<span class="hljs-string">&quot;NA&quot;</span>)))<br><br>str(parse_integer(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;1&quot;</span>,<span class="hljs-string">&quot;2&quot;</span>,<span class="hljs-string">&quot;3&quot;</span>)))<br><br>str(parse_date(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;2020-01-01&quot;</span>,<span class="hljs-string">&quot;1999-02-23&quot;</span>)))<br><br>parse_integer(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;1&quot;</span>,<span class="hljs-string">&quot;2&quot;</span>,<span class="hljs-string">&quot;.&quot;</span>,<span class="hljs-string">&quot;1&quot;</span>), na=<span class="hljs-string">&quot;.&quot;</span>)<br><br>x &lt;- parse_integer(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;1&quot;</span>,<span class="hljs-string">&quot;2&quot;</span>,<span class="hljs-string">&quot;3&quot;</span>,<span class="hljs-string">&quot;q&quot;</span>))<br>problems(x)<br><br>parse_double(<span class="hljs-string">&quot;1.23&quot;</span>)<br>parse_double(<span class="hljs-string">&quot;1,23&quot;</span>, locale = locale(decimal_mark = <span class="hljs-string">&quot;,&quot;</span>))<br><br>parse_number(<span class="hljs-string">&quot;$20&quot;</span>)<br>parse_number(<span class="hljs-string">&quot;56%&quot;</span>)<br>parse_number(<span class="hljs-string">&quot;it costs $618347&quot;</span>)<br>parse_number(<span class="hljs-string">&quot;111.234.566&quot;</span>, locale = locale(grouping_mark = <span class="hljs-string">&quot;.&quot;</span>))<br>parse_number(<span class="hljs-string">&quot;111&#x27;678&#x27;980&quot;</span>, locale = locale(grouping_mark = <span class="hljs-string">&quot;&#x27;&quot;</span>))<br><br>charToRaw(<span class="hljs-string">&quot;Hadley&quot;</span>)<br><br>x1 &lt;- <span class="hljs-string">&quot;EL Ni\xf1o was particularly bad this year&quot;</span><br>parse_character(x1, locale = locale(encoding = <span class="hljs-string">&quot;Latin1&quot;</span>))<br><br>guess_encoding(charToRaw(x1))<br><br>fruit &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;apple&quot;</span>, <span class="hljs-string">&quot;banana&quot;</span>)<br>parse_factor(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;apple&quot;</span>,<span class="hljs-string">&quot;banana&quot;</span>), levels = fruit)<br><br>fruit &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;apple&quot;</span>, <span class="hljs-string">&quot;banana&quot;</span>)<br>parse_factor(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;apple&quot;</span>,<span class="hljs-string">&quot;banana&quot;</span>,<span class="hljs-string">&quot;lemon&quot;</span>), levels = fruit)<br><br>parse_datetime(<span class="hljs-string">&quot;2010-10-02T2010&quot;</span>)<br><br>parse_datetime(<span class="hljs-string">&quot;20201010&quot;</span>)<br><br>parse_date(<span class="hljs-string">&quot;2020-10-20&quot;</span>)<br><br>library(hms)<br>parse_time(<span class="hljs-string">&quot;01:10:00 am&quot;</span>)<br>parse_time(<span class="hljs-string">&quot;10:20:01&quot;</span>)<br><br>parse_date(<span class="hljs-string">&quot;19/09/10&quot;</span>,<span class="hljs-string">&quot;%y/%m/%d&quot;</span>)<br>parse_date(<span class="hljs-string">&quot;99/08/03&quot;</span>,<span class="hljs-string">&quot;%y/%d/%m&quot;</span>)<br></code></pre></td></tr></table></figure>
<h2 id="guess-parse"><a href="#guess-parse" class="headerlink" title="guess parse"></a>guess parse</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br></pre></td><td class="code"><pre><code class="hljs R">library(readr)<br>guess_parser(<span class="hljs-built_in">c</span>(<span class="hljs-number">2020</span>-<span class="hljs-number">12</span>-<span class="hljs-number">23</span>))<br>guess_parser(<span class="hljs-string">&quot;2020-12-23&quot;</span>)<br>str(parse_guess(<span class="hljs-string">&quot;2020-12-23&quot;</span>))<br>guess_parser(<span class="hljs-string">&quot;15:02&quot;</span>)<br>guess_parser(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;1&quot;</span>,<span class="hljs-string">&quot;2&quot;</span>,<span class="hljs-string">&quot;3&quot;</span>))<br>guess_parser(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;1214,214,1234&quot;</span>))<br><br><br>challenge &lt;- read_csv(readr_example(<span class="hljs-string">&quot;challenge.csv&quot;</span>),<br>col_types = cols(x = col_date(),y = col_logical()))<br>challenge &lt;- read_csv(readr_example(<span class="hljs-string">&quot;challenge.csv&quot;</span>),<br>col_types = cols(x = col_double(),y = col_date()))<br>tail(challenge)<br><br><br>challenge1 &lt;- read_csv(readr_example(<span class="hljs-string">&quot;challenge.csv&quot;</span>),<br>                       guess_max = <span class="hljs-number">1001</span>)<br><br>library(tidyverse)<br>df = tribble(<br>  ~x, ~y,<br>  <span class="hljs-string">&quot;1&quot;</span>,<span class="hljs-string">&quot;1.234&quot;</span>,<br>  <span class="hljs-string">&quot;2&quot;</span>,<span class="hljs-string">&quot;2.344&quot;</span>,<br>  <span class="hljs-string">&quot;3&quot;</span>,<span class="hljs-string">&quot;3.455&quot;</span><br>)<br>type_convert(df)<br><br>write_csv(challenge, <span class="hljs-string">&quot;challenge.csv&quot;</span>) <span class="hljs-comment">#unreliable</span><br>write_excel_csv(challenge, <span class="hljs-string">&quot;challlenge.xlsx&quot;</span>) <span class="hljs-comment">#can not open</span><br>write_excel_csv(challenge, <span class="hljs-string">&quot;challenge.xls&quot;</span>) <span class="hljs-comment">#can be open but damaged</span><br><br>read_csv(<span class="hljs-string">&quot;challenge.csv&quot;</span>)<br><br>write_rds(challenge, <span class="hljs-string">&quot;challenge.rds&quot;</span>)<br>read_rds(<span class="hljs-string">&quot;challenge.rds&quot;</span>)<br><br>library(feather)<br>write_feather(challenge, <span class="hljs-string">&quot;challenge.feather&quot;</span>)<br>read_feather(<span class="hljs-string">&quot;challenge.feather&quot;</span>)<br></code></pre></td></tr></table></figure>
<h2 id="gene-expression-analysis-example"><a href="#gene-expression-analysis-example" class="headerlink" title="gene expression analysis example"></a>gene expression analysis example</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br></pre></td><td class="code"><pre><code class="hljs R">gene_data = read.csv(<span class="hljs-string">&quot;gene_exp_exampple/gene_expression_data.csv&quot;</span>) <span class="hljs-comment">#get gene data from csv file</span><br>View(gene_data) <span class="hljs-comment"># see the data in tibble frame</span><br><span class="hljs-built_in">dim</span>(gene_data) <span class="hljs-comment"># see the dimensions of the table</span><br>head(gene_data) <span class="hljs-comment"># show the first 6 rows of the table </span><br>summary(gene_data) <span class="hljs-comment"># show the information about the different columns of the table</span><br><br>count_cell_types = table(gene_data[,<span class="hljs-string">&quot;cell_type&quot;</span>])<br><span class="hljs-comment"># We can use the command “table” on the column “cell_type” of the data we loaded </span><br><span class="hljs-comment"># to obtain a count of each of the different cell types present in this data in an easy to read form.</span><br><span class="hljs-comment"># in this case we want to abtain a count of cell_types</span><br>count_cell_types<br><br><span class="hljs-comment"># what do we know about the data?</span><br><span class="hljs-comment"># this table list 305 samples of 3 type of cells</span><br><span class="hljs-comment"># and 2 kinds of genes(A and B)</span><br><span class="hljs-comment"># and there are 184 blood.cells, 108 fibroblasts and 13 IPSCs</span><br><br><span class="hljs-comment"># draw the pie plot to show the ratio of each type of cells</span><br>pie(count_cell_types)<br><span class="hljs-comment"># draw the bar plot to show the counts of each type of cells</span><br>barplot(count_cell_types)<br><br><span class="hljs-comment"># how to change the defaut colors of plots?</span><br>cell_colors = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;gray&quot;</span>, <span class="hljs-string">&quot;purple&quot;</span>, <span class="hljs-string">&quot;orange&quot;</span>)<br>pie(count_cell_types, col=cell_colors)<br>barplot(count_cell_types, col=cell_colors)<br><br><span class="hljs-comment"># draw a point plot of there types of cells&#x27; different expression</span><br>plot(<br>  x=gene_data[,<span class="hljs-string">&quot;geneA&quot;</span>], <span class="hljs-comment"># data in the x axis</span><br>  y=gene_data[,<span class="hljs-string">&quot;geneB&quot;</span>]  <span class="hljs-comment"># data in the y axis</span><br>)<br><br><span class="hljs-comment"># change the default colors of pointplot (1)</span><br>cell_type_data = as.factor(gene_data[,<span class="hljs-string">&quot;cell_type&quot;</span>]) <br><span class="hljs-comment"># save this column to a variable and subsequently transform it to a factor.</span><br>plot(<br>  x=gene_data[,<span class="hljs-string">&quot;geneA&quot;</span>], <span class="hljs-comment"># data in the x axis</span><br>  y=gene_data[,<span class="hljs-string">&quot;geneB&quot;</span>], <span class="hljs-comment"># data in the y axis</span><br>  col=cell_type_data<br>)<br><br><span class="hljs-comment"># change the default colors of pointplot (2)</span><br>plot(<br>  x=gene_data[,<span class="hljs-string">&quot;geneA&quot;</span>], <span class="hljs-comment"># data in the x axis</span><br>  y=gene_data[,<span class="hljs-string">&quot;geneB&quot;</span>], <span class="hljs-comment"># data in the y axis</span><br>  col=cell_colors[cell_type_data] <span class="hljs-comment"># Because of how factors work </span><br>  <span class="hljs-comment"># we can use them to subset the colour vector we created (cell_colours) </span><br>  <span class="hljs-comment"># with the categories of the factor itself (the different cell types). </span><br>  <span class="hljs-comment"># This makes it so that each of the cell types has the colour we want for it.</span><br>)<br><br><span class="hljs-comment"># change the default shape of point</span><br>plot(<br>  x=gene_data[,<span class="hljs-string">&quot;geneA&quot;</span>],<br>  y=gene_data[,<span class="hljs-string">&quot;geneB&quot;</span>],<br>  col=cell_colors[cell_type_data],<br>  pch=<span class="hljs-number">19</span><br>)<br><br><span class="hljs-comment"># add a title for the pointplot</span><br>plot(<br>  x=gene_data[,<span class="hljs-string">&quot;geneA&quot;</span>],<br>  y=gene_data[,<span class="hljs-string">&quot;geneB&quot;</span>],<br>  col=cell_colors[cell_type_data],<br>  pch=<span class="hljs-number">19</span>,<br>  main=<span class="hljs-string">&quot;Gene A vs Gene B sactterplot&quot;</span>,<br>  ylab=<span class="hljs-string">&quot;Gene B expression&quot;</span>,<br>  xlab=<span class="hljs-string">&quot;Gene A expression&quot;</span><br>)<br><br><span class="hljs-comment"># add a legend at the topright for the pointplot</span><br>plot(<br>  x=gene_data[,<span class="hljs-string">&quot;geneA&quot;</span>],<br>  y=gene_data[,<span class="hljs-string">&quot;geneB&quot;</span>],<br>  col=cell_colors[cell_type_data],<br>  pch=<span class="hljs-number">19</span>,<br>  main=<span class="hljs-string">&quot;Gene A vs Gene B sactterplot&quot;</span>,<br>  ylab=<span class="hljs-string">&quot;Gene B expression&quot;</span>,<br>  xlab=<span class="hljs-string">&quot;Gene A expression&quot;</span><br>)<br>legend(<br>  x=<span class="hljs-string">&quot;topright&quot;</span>,<br>  legend=levels(cell_type_data),<br>  col=cell_colors,<br>  pch=<span class="hljs-number">19</span><br>)<br><br><span class="hljs-comment"># draw a boxplot to describe geneA and geneB expresison across all the cells</span><br>boxplot(<br>  x=<span class="hljs-built_in">list</span>(<span class="hljs-string">&quot;geneA&quot;</span>=gene_data[,<span class="hljs-string">&quot;geneA&quot;</span>], <span class="hljs-string">&quot;geneB&quot;</span>=gene_data[,<span class="hljs-string">&quot;geneB&quot;</span>]),<br>  col=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;purple&quot;</span>,<span class="hljs-string">&quot;orange&quot;</span>),<br>  main=<span class="hljs-string">&quot;GeneA and GeneB expression across all samples&quot;</span>,<br>  ylab=<span class="hljs-string">&quot;Gene Expression&quot;</span>,<br>  xlab=<span class="hljs-string">&quot;Genes&quot;</span>,<br>  pch=<span class="hljs-number">19</span><br>)<br><br><span class="hljs-comment"># draw a boxplot to describe gene B expression of 3 types cell </span><br>boxplot(<br>  formula=gene_data[,<span class="hljs-string">&quot;geneB&quot;</span>]~gene_data[,<span class="hljs-string">&quot;cell_type&quot;</span>],<br>  col=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;cyan&quot;</span>,<span class="hljs-string">&quot;purple&quot;</span>,<span class="hljs-string">&quot;orange&quot;</span>),<br>  main=<span class="hljs-string">&quot;Gene B expression by cell type&quot;</span>,<br>  ylab=<span class="hljs-string">&quot;Gene B Expression&quot;</span>,<br>  xlab=<span class="hljs-string">&quot;cell type&quot;</span>,<br>  pch=<span class="hljs-number">20</span><br>)<br><br><span class="hljs-comment"># draw a boxplot to describe gene A expression of 3 types cell </span><br>boxplot(<br>  formula=gene_data[,<span class="hljs-string">&quot;geneB&quot;</span>]~gene_data[,<span class="hljs-string">&quot;cell_type&quot;</span>],<br>  col=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;cyan&quot;</span>,<span class="hljs-string">&quot;purple&quot;</span>,<span class="hljs-string">&quot;orange&quot;</span>),<br>  main=<span class="hljs-string">&quot;Gene A expression by cell type&quot;</span>,<br>  ylab=<span class="hljs-string">&quot;Gene A Expression&quot;</span>,<br>  xlab=<span class="hljs-string">&quot;cell type&quot;</span>,<br>  pch=<span class="hljs-number">20</span><br>)<br></code></pre></td></tr></table></figure>
<h2 id="ggplot-save"><a href="#ggplot-save" class="headerlink" title="ggplot save"></a>ggplot save</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br></pre></td><td class="code"><pre><code class="hljs R">library(nycflights13)<br>library(tidyverse)<br>flights<br>filter(flights, month==<span class="hljs-number">1</span>, day==<span class="hljs-number">1</span>)<br><br>(dec25 &lt;- filter(flights, month==<span class="hljs-number">12</span>, day==<span class="hljs-number">25</span>))<br><br>nycflights &lt;- flights %&gt;%<br>  group_by(dest) %&gt;%<br>  summarise(count=n(),<br>            dist = mean(flights,na.rm=<span class="hljs-literal">TRUE</span>),<br>            delay = mean(arr_delay, na.rm=<span class="hljs-literal">TRUE</span>)<br>            ) %&gt;%<br>  filter(count&gt;<span class="hljs-number">20</span>, dest!=<span class="hljs-string">&#x27;HNL&#x27;</span>)not_cancelled &lt;- flights %&gt;%<br>  filter(!<span class="hljs-built_in">is.na</span>(dep_delay), !<span class="hljs-built_in">is.na</span>(arr_delay))<br><br>not_cancelled %&gt;%<br>  group_by(year, month, day) %&gt;%<br>  summarize(mean = mean(dep_delay))<br><br>ggplot(data = diamonds)+<br>  geom_histogram(mapping = aes(x = carat),binwidth = <span class="hljs-number">0.5</span>)<br>ggplot(data = diamonds)+<br>  geom_boxplot(mapping = aes(x = cut, y = price))<br><br>ggplot(data = mpg)+<br>  geom_boxplot(mapp = aes(x = <span class="hljs-built_in">class</span>, y = hwy))<br><br>ggplot(data = mpg)+<br>  geom_boxplot(mapping = aes(x = reorder(<span class="hljs-built_in">class</span>, hwy, FUN = median), y = hwy))<br><br>ggplot(data = mpg)+<br>  geom_boxplot(mapping = aes(x = reorder(<span class="hljs-built_in">class</span>, hwy, FUN = median), y = hwy))+<br>  coord_flip()<br><br>ggplot(data = diamonds)+<br>  geom_count(mapping = aes(x = cut, y = color))<br><br>diamonds%&gt;%<br>  count(cut, color)%&gt;%<br>  ggplot(mapping = aes(x = color, y = cut))+<br>  geom_tile(mapping = aes(fill = n))<br><br>ggplot(data = diamonds)+<br>  geom_point(mapping = aes(x = carat, y = price))<br><br>diamonds%&gt;%<br>  ggplot(mapping = aes(x = carat, y = price))+<br>  geom_point()<br><br>diamonds%&gt;%<br>  ggplot(mapping = aes(x = carat, y = price))+<br>  geom_point(alpha = <span class="hljs-number">1</span>/<span class="hljs-number">100</span>)<br>  <br>ggplot(data = diamonds)+<br>  geom_bin2d(mapping = aes(x = carat, y = price))<br><br>ggplot(data = diamonds)+<br>  geom_hex(mapping = aes(x = carat, y = price))<br><br>ggplot(data = diamonds, mapping = aes(x = carat, y = price))+<br>  geom_boxplot(mapping = aes(group = cut_width(carat, <span class="hljs-number">0.5</span>)))<br><br>ggplot(data = diamonds,mapping = aes(x = carat, y = price))+<br>  geom_boxplot(mapping = aes(group = cut_number(carat, <span class="hljs-number">20</span>)))<br><br>ggplot(data = faithful)+<br>  geom_point(mapping = aes(x = eruptions, y = waiting))<br></code></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs R">library(tidyverse)<br>ggplot(diamonds, aes(carat, price))+<br>  geom_boxplot(aes(group=cut_number(carat,<span class="hljs-number">20</span>)))<br>ggsave(<span class="hljs-string">&quot;diamonds.pdf&quot;</span>)<br>write.csv(diamonds, <span class="hljs-string">&quot;diamonds.csv&quot;</span>)<br></code></pre></td></tr></table></figure>
<h2 id="dplyr"><a href="#dplyr" class="headerlink" title="dplyr"></a>dplyr</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br></pre></td><td class="code"><pre><code class="hljs R">library(tidyverse)<br>library(nycflights13)<br>View(flights)<br>View(airlines)<br>View(airports)<br>View(planes)<br>View(weather)<br><br>planes %&gt;%<br>  count(tailnum)%&gt;%<br>  filter(n&gt;<span class="hljs-number">1</span>)<br><br>weather%&gt;%<br>  count(year, month, day, hour, origin)%&gt;%<br>  filter(n&gt;<span class="hljs-number">1</span>)<br><br><span class="hljs-comment">#add a agent key to flights</span><br>flights %&gt;%<br>  arrange(year, month, day, sched_dep_time, carrier, flight) %&gt;%<br>  mutate(flight_id = row_number()) %&gt;%<br>  glimpse()<br><br><br><span class="hljs-comment">#drawing amc map</span><br>airports %&gt;%<br>  semi_join(flights, <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;faa&quot;</span> = <span class="hljs-string">&quot;dest&quot;</span>)) %&gt;%<br>  ggplot(aes(lon, lat)) +<br>  borders(<span class="hljs-string">&quot;state&quot;</span>) +<br>  geom_point() +<br>  coord_quickmap()<br><br>avg_dest_delays &lt;-<br>  flights %&gt;%<br>  group_by(dest) %&gt;%<br>  <span class="hljs-comment"># arrival delay NA&#x27;s are cancelled flights</span><br>  summarise(delay = mean(arr_delay, na.rm = <span class="hljs-literal">TRUE</span>)) %&gt;%<br>  inner_join(airports, by = <span class="hljs-built_in">c</span>(dest = <span class="hljs-string">&quot;faa&quot;</span>))<br><br>avg_dest_delays %&gt;%<br>  ggplot(aes(lon, lat, colour = delay)) +<br>  borders(<span class="hljs-string">&quot;state&quot;</span>) +<br>  geom_point() +<br>  coord_quickmap()<br><br>library(tidyverse)<br>flights1 &lt;- flights%&gt;%<br>  select(year:day, hour, origin, dest, tailnum, carrier)<br>View(flights1)<br><br>library(tidyverse)<br>flights2 &lt;- flights%&gt;%<br>  select(-origin, -dest)%&gt;%<br>  left_join(airlines, by=<span class="hljs-string">&quot;carrier&quot;</span>)<br>View(flights2)<br><br><span class="hljs-comment">#inner_join</span><br>x &lt;- tribble(<br>  ~key, ~val_x,<br>  <span class="hljs-number">1</span>, <span class="hljs-string">&quot;x1&quot;</span>,<br>  <span class="hljs-number">2</span>, <span class="hljs-string">&quot;x2&quot;</span>,<br>  <span class="hljs-number">3</span>, <span class="hljs-string">&quot;x3&quot;</span><br>)<br>y &lt;- tribble(<br>  ~key, ~val_y,<br>  <span class="hljs-number">1</span>, <span class="hljs-string">&quot;y1&quot;</span>,<br>  <span class="hljs-number">2</span>, <span class="hljs-string">&quot;y2&quot;</span>,<br>  <span class="hljs-number">3</span>, <span class="hljs-string">&quot;y3&quot;</span><br>)<br>x%&gt;%<br>  inner_join(y, by = <span class="hljs-string">&quot;key&quot;</span>)<br><br><span class="hljs-comment">#full_join</span><br>x &lt;- tribble(<br>  ~key, ~val_x,<br>  <span class="hljs-number">1</span>, <span class="hljs-string">&quot;x1&quot;</span>,<br>  <span class="hljs-number">2</span>, <span class="hljs-string">&quot;x2&quot;</span>,<br>  <span class="hljs-number">3</span>, <span class="hljs-string">&quot;x3&quot;</span><br>)<br>y &lt;- tribble(<br>  ~key, ~val_y,<br>  <span class="hljs-number">1</span>, <span class="hljs-string">&quot;y1&quot;</span>,<br>  <span class="hljs-number">2</span>, <span class="hljs-string">&quot;y2&quot;</span>,<br>  <span class="hljs-number">4</span>, <span class="hljs-string">&quot;y3&quot;</span><br>)<br>x%&gt;%<br>  full_join(y, by = <span class="hljs-string">&quot;key&quot;</span>)<br><br><span class="hljs-comment">#left_join</span><br>x &lt;- tribble(<br>  ~key, ~val_x,<br>  <span class="hljs-number">1</span>, <span class="hljs-string">&quot;x1&quot;</span>,<br>  <span class="hljs-number">2</span>, <span class="hljs-string">&quot;x2&quot;</span>,<br>  <span class="hljs-number">3</span>, <span class="hljs-string">&quot;x3&quot;</span><br>)<br>y &lt;- tribble(<br>  ~key, ~val_y,<br>  <span class="hljs-number">1</span>, <span class="hljs-string">&quot;y1&quot;</span>,<br>  <span class="hljs-number">2</span>, <span class="hljs-string">&quot;y2&quot;</span>,<br>  <span class="hljs-number">4</span>, <span class="hljs-string">&quot;y3&quot;</span><br>)<br>x%&gt;%<br>  left_join(y, by = <span class="hljs-string">&quot;key&quot;</span>)<br><br><span class="hljs-comment">#right_join</span><br>x &lt;- tribble(<br>  ~key, ~val_x,<br>  <span class="hljs-number">1</span>, <span class="hljs-string">&quot;x1&quot;</span>,<br>  <span class="hljs-number">2</span>, <span class="hljs-string">&quot;x2&quot;</span>,<br>  <span class="hljs-number">3</span>, <span class="hljs-string">&quot;x3&quot;</span><br>)<br>y &lt;- tribble(<br>  ~key, ~val_y,<br>  <span class="hljs-number">1</span>, <span class="hljs-string">&quot;y1&quot;</span>,<br>  <span class="hljs-number">2</span>, <span class="hljs-string">&quot;y2&quot;</span>,<br>  <span class="hljs-number">4</span>, <span class="hljs-string">&quot;y3&quot;</span><br>)<br>x%&gt;%<br>  right_join(y, by = <span class="hljs-string">&quot;key&quot;</span>)<br><br>flights2%&gt;%<br>  left_join(weather)<br><br>flights2%&gt;%<br>  left_join(planes, by = <span class="hljs-string">&quot;tailnum&quot;</span>)<br><br>library(tidyverse)<br>library(nycflights13)<br>flights2 &lt;- flights<br>flights2 %&gt;%<br>  left_join(airports, <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;dest&quot;</span> = <span class="hljs-string">&quot;faa&quot;</span>))<br><br>flights2%&gt;%<br>  left_join(airports,<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;origin&quot;</span> = <span class="hljs-string">&quot;faa&quot;</span>))<br><br>top_dest &lt;- flights2%&gt;%<br>  count(dest, sort = <span class="hljs-literal">TRUE</span>)%&gt;%<br>  head(<span class="hljs-number">10</span>)<br>top_dest<br><br>flights%&gt;%<br>  semi_join(top_dest)<br><br>flights%&gt;%<br>  anti_join(planes, by = <span class="hljs-string">&quot;tailnum&quot;</span>)%&gt;%<br>  count(tailnum, sort = <span class="hljs-literal">TRUE</span>)<br><br>top_flight &lt;- subset(flights2, flight&gt;=<span class="hljs-number">100</span>)<br>top_flight<br><br><span class="hljs-comment">#</span><br>str(planes)<br>plane_ages &lt;-<br>  planes %&gt;%<br>  mutate(age = <span class="hljs-number">2018</span> - year) %&gt;%<br>  select(tailnum, age)<br><br>flights %&gt;%<br>  inner_join(plane_ages, by = <span class="hljs-string">&quot;tailnum&quot;</span>) %&gt;%<br>  group_by(age) %&gt;%<br>  filter(!<span class="hljs-built_in">is.na</span>(dep_delay)) %&gt;%<br>  summarise(delay = mean(dep_delay)) %&gt;%<br>  ggplot(aes(x = age, y = delay)) +<br>  geom_point() +<br>  geom_line()<br><br><span class="hljs-comment">#</span><br>str(weather)<br>flight_weather &lt;-<br>  flights %&gt;%<br>  inner_join(weather, by = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;origin&quot;</span> = <span class="hljs-string">&quot;origin&quot;</span>,<br>                             <span class="hljs-string">&quot;year&quot;</span> = <span class="hljs-string">&quot;year&quot;</span>,<br>                             <span class="hljs-string">&quot;month&quot;</span> = <span class="hljs-string">&quot;month&quot;</span>,<br>                             <span class="hljs-string">&quot;day&quot;</span> = <span class="hljs-string">&quot;day&quot;</span>,<br>                             <span class="hljs-string">&quot;hour&quot;</span> = <span class="hljs-string">&quot;hour&quot;</span>))<br><br>flight_weather %&gt;%<br>  group_by(precip) %&gt;%<br>  summarise(delay = mean(dep_delay, na.rm = <span class="hljs-literal">TRUE</span>)) %&gt;%<br>  ggplot(aes(x = precip, y = delay)) +<br>  geom_line() + geom_point()<br><br><span class="hljs-comment">#</span><br>flights %&gt;%<br>  group_by(year, month, day) %&gt;%<br>  summarise(total_24 = <span class="hljs-built_in">sum</span>(dep_delay, na.rm = <span class="hljs-literal">TRUE</span>) + <span class="hljs-built_in">sum</span>(arr_delay, na.rm = <span class="hljs-literal">TRUE</span>)) %&gt;%<br>  mutate(total_48 = total_24 + lag(total_24)) %&gt;%<br>  arrange(desc(total_48))<br></code></pre></td></tr></table></figure>
<h2 id="Bio-Computational-course"><a href="#Bio-Computational-course" class="headerlink" title="Bio-Computational course"></a>Bio-Computational course</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># boxplot </span><br>boxplot(count ~ spray, data = InsectSprays, col = <span class="hljs-string">&quot;bisque&quot;</span>)<br><br><span class="hljs-comment"># 扰动图</span><br>library(ggplot2)<br>ggplot(data = ToothGrowth,aes(x = dose, y = len, fill = supp)) +<br>  geom_boxplot() +<br>  geom_point(position = position_jitterdodge())<br><br><span class="hljs-comment"># 无重复样本：MAplot</span><br><span class="hljs-comment"># define ma_plot function</span><br>ma_plot &lt;- <span class="hljs-keyword">function</span>(dat)&#123;<br>  df &lt;- data.frame(a=dat[,<span class="hljs-number">1</span>],b=dat[,<span class="hljs-number">1</span>],<span class="hljs-built_in">c</span>=dat[,<span class="hljs-number">1</span>])<br>  <span class="hljs-comment"># transform data and add the pvalue</span><br>  log_condition1 &lt;- <span class="hljs-built_in">log</span>(dat[,<span class="hljs-number">1</span>]);<br>  log_condition2 &lt;- <span class="hljs-built_in">log</span>(dat[,<span class="hljs-number">2</span>])<br>  df[,<span class="hljs-number">1</span>] &lt;- (log_condition1 + log_condition2)/<span class="hljs-number">2</span><br>  df[,<span class="hljs-number">2</span>] &lt;- (log_condition1 - log_condition2)<br>  df[,<span class="hljs-number">3</span>] &lt;- (log_condition1 - log_condition2)<br>  <br>  colnames(df) &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;AveExpr&quot;</span>,<span class="hljs-string">&quot;logFC&quot;</span>,<span class="hljs-string">&quot;pvalue&quot;</span>)<br>  <span class="hljs-comment">#p &lt;- ggplot(data = df,aes(x=AveExpr, y=logFC))</span><br>  p &lt;- ggplot(data = df,aes(x=AveExpr, y=logFC, colour=pvalue))<br>  p &lt;- p + geom_point(alpha = <span class="hljs-number">0.4</span>, size = <span class="hljs-number">1</span>) + theme_classic() + theme(aspect.ratio = <span class="hljs-number">1</span>) +<br>    geom_hline(aes(yintercept = <span class="hljs-number">0</span>),color = <span class="hljs-string">&quot;black&quot;</span>,size = <span class="hljs-number">0.3</span> ,alpha = <span class="hljs-number">0.6</span>) +<br>    geom_hline(aes(yintercept = -<span class="hljs-number">1</span>),color = <span class="hljs-string">&quot;greenyellow&quot;</span>,size = <span class="hljs-number">0.5</span> ,alpha = <span class="hljs-number">0.8</span>) +<br>    geom_hline(aes(yintercept = <span class="hljs-number">1</span>),color = <span class="hljs-string">&quot;greenyellow&quot;</span>,size = <span class="hljs-number">0.5</span> ,alpha = <span class="hljs-number">0.8</span>) +<br>    labs(x = <span class="hljs-string">&quot;Mean expression&quot;</span>, y = <span class="hljs-string">&quot;Log2 Fold Change&quot;</span>) +<br>    scale_colour_gradientn(colors = rev(<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;darkblue&quot;</span>,<span class="hljs-string">&quot;dodgerblue&quot;</span>,<span class="hljs-string">&quot;white&quot;</span>,<span class="hljs-string">&quot;orange&quot;</span>,<span class="hljs-string">&quot;red&quot;</span>)))<br>  print(p)<br>  <span class="hljs-built_in">return</span>(df)<br>&#125;<br><span class="hljs-comment"># 读入数据</span><br>ori_dat &lt;- read.delim(<span class="hljs-string">&quot;/Users/keqinliu/MAP_testing.tsv&quot;</span>); <span class="hljs-comment">#read.delim()</span><br><span class="hljs-comment"># R最基本的数据结构是data.frame</span><br>dat &lt;- data.frame(condition1 = ori_dat[,<span class="hljs-number">2</span>],condition2 = ori_dat[,<span class="hljs-number">3</span>])<br>rownames(dat) &lt;- ori_dat[,<span class="hljs-number">1</span>]<br>res &lt;- ma_plot(dat)<br><span class="hljs-comment"># finding the most signaficant genes</span><br>rownames(res) &lt;- rownames(dat)<br>head( res[order(<span class="hljs-built_in">abs</span>( res$logFC) ,decreasing = <span class="hljs-built_in">T</span>),] )<br><br><span class="hljs-comment"># 1st kind: fold-change </span><br>e &lt;- <span class="hljs-built_in">log</span>(dat[,<span class="hljs-number">1</span>]);b &lt;- <span class="hljs-built_in">log</span>(dat[,<span class="hljs-number">2</span>]);<br>dat[,<span class="hljs-number">3</span>] &lt;- (e - b) / <span class="hljs-built_in">min</span>(e,b)<br>colnames(dat)[<span class="hljs-number">3</span>] &lt;- <span class="hljs-string">&quot;FC&quot;</span><br>head( dat[order(<span class="hljs-built_in">abs</span>( dat$FC) ,decreasing = <span class="hljs-built_in">T</span>),] )<br><span class="hljs-comment"># calculate p-value and draw hist plot</span><br>value &lt;- dat$FC<br>summary(value) <span class="hljs-comment">#基于当前假设，得到比当前统计量更极端的值的概率</span><br>hist(value[<span class="hljs-number">1</span>:<span class="hljs-number">100</span>])<br><br><span class="hljs-comment"># </span><br>ori_dat &lt;- read.delim(<span class="hljs-string">&quot;/Users/keqinliu/GDS2676.rnaExp&quot;</span>);<br>sampleInfor &lt;- read.delim(<span class="hljs-string">&quot;/Users/keqinliu/sampleInfor.txt&quot;</span>,header = <span class="hljs-built_in">F</span>);<br>condition1 &lt;- sampleInfor[,<span class="hljs-number">1</span>]<br>condition2 &lt;- setdiff(colnames(ori_dat),condition1)<br>condition2 &lt;- setdiff(condition2,<span class="hljs-string">&quot;GENE&quot;</span>)<br>ori_dat$logFC &lt;- 1<br>ori_dat$Pvalue &lt;- 1<br><span class="hljs-comment">#这里用for循环，没有用apply</span><br><span class="hljs-comment">#增加可读性，实际使用的时候用apply</span><br><span class="hljs-comment">#i &lt;- 1</span><br><span class="hljs-keyword">for</span> (i <span class="hljs-keyword">in</span> <span class="hljs-number">1</span>:nrow(ori_dat)) &#123;<br>  v1 &lt;- <span class="hljs-built_in">as.numeric</span>(ori_dat[i,condition1])<br>  v2 &lt;- <span class="hljs-built_in">as.numeric</span>(ori_dat[i,condition2])<br>  tmp_log1 &lt;- <span class="hljs-built_in">log</span>( mean(v1) );<br>  tmp_log2 &lt;- <span class="hljs-built_in">log</span>( mean(v2) )<br>  ori_dat[i,<span class="hljs-string">&quot;logFC&quot;</span>] &lt;- (tmp_log1 - tmp_log2) / <span class="hljs-built_in">min</span>(tmp_log1,tmp_log2)<br>  ori_dat[i,<span class="hljs-string">&quot;Pvalue&quot;</span>] &lt;- wilcox.test(v1,v2)$p.value;<br>&#125;<br><span class="hljs-comment"># y is -log_pvalue</span><br><span class="hljs-comment"># x is logFC</span><br>plot(ori_dat$logFC,-log10( ori_dat$Pvalue ))<br><span class="hljs-comment">#删除异常值</span><br>toRemove &lt;- which(ori_dat$logFC &lt; -<span class="hljs-number">1000</span>)<br>ori_dat &lt;- ori_dat[-toRemove,]<br>plot(ori_dat$logFC,-log10( ori_dat$Pvalue ))<br></code></pre></td></tr></table></figure>
<h2 id="compare-distrubition-by-t-test"><a href="#compare-distrubition-by-t-test" class="headerlink" title="compare distrubition by t test"></a>compare distrubition by t test</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># use t_test to compare distribution</span><br>gene_data=read.csv(<span class="hljs-string">&quot;gene_exp_exampple/gene_expression_data.csv&quot;</span>)<br><span class="hljs-built_in">dim</span>(gene_data)<br>summary(gene_data)<br>head(gene_data)<br>count_cell_type=table(gene_data[,<span class="hljs-string">&quot;cell_type&quot;</span>])<br>count_cell_type<br><span class="hljs-comment"># check the data</span><br><br>geneB_data &lt;- gene_data[,<span class="hljs-string">&quot;geneB&quot;</span>] <span class="hljs-comment"># substting geneB</span><br>geneA_data &lt;- gene_data[,<span class="hljs-string">&quot;geneA&quot;</span>] <span class="hljs-comment"># substting geneA</span><br>cell_type_data &lt;- gene_data[,<span class="hljs-string">&quot;cell_type&quot;</span>] <span class="hljs-comment"># substting cell_type_data</span><br><br><span class="hljs-comment"># substting geneA with different cells</span><br>bloodcells_geneA_data &lt;- geneA_data[cell_type_data==<span class="hljs-string">&quot;BLOOD.CELLS&quot;</span>]<br>fibroblasts_geneA_data &lt;- geneA_data[cell_type_data==<span class="hljs-string">&quot;FIBROBLASTS&quot;</span>]<br>ipsc_geneA_data &lt;- geneA_data[cell_type_data==<span class="hljs-string">&#x27;IPSC&#x27;</span>]<br><br><span class="hljs-comment"># substting geneB with different cells</span><br>bloodcells_geneB_data &lt;- geneB_data[cell_type_data==<span class="hljs-string">&quot;BLOOD.CELLS&quot;</span>]<br>fibroblasts_geneB_data &lt;- geneB_data[cell_type_data==<span class="hljs-string">&quot;FIBROBLASTS&quot;</span>]<br>ipsc_geneB_data &lt;- geneB_data[cell_type_data==<span class="hljs-string">&#x27;IPSC&#x27;</span>]<br><br><span class="hljs-comment"># run the t-test (geneA)</span><br>res_bloodcells_fibroblasts_geneA &lt;- t.test(bloodcells_geneA_data, fibroblasts_geneA_data)<br>res_bloodcells_ipsc_geneA &lt;- t.test(bloodcells_geneA_data, ipsc_geneA_data)<br>res_fibroblasts_ipsc_geneA &lt;- t.test(fibroblasts_geneB_data, ipsc_geneB_data)<br>res_bloodcells_fibroblasts_geneA<br>res_bloodcells_ipsc_geneA<br>res_bloodcells_fibroblasts_geneA<br><br><span class="hljs-comment"># show the important statistic factors in the t-test(in this case we use res_bloodcells_fibroblasts_geneA as example)</span><br>res_bloodcells_fibroblasts_geneA[<span class="hljs-string">&quot;statistic&quot;</span>] <span class="hljs-comment"># show t value</span><br>res_bloodcells_fibroblasts_geneA[<span class="hljs-string">&quot;p.value&quot;</span>] <span class="hljs-comment"># show p value</span><br><span class="hljs-comment"># how to analyze the result?</span><br><span class="hljs-comment"># the larger the t value is, the more different the 2 groups are</span><br><span class="hljs-comment"># For the p-value, given an interval of confidence of 95% and an alternative hypothesis </span><br><span class="hljs-comment"># that the 2 groups have different mean distributions, </span><br><span class="hljs-comment"># a p-value &lt; 0.05 tells us that this difference did not occur by chance and that, therefore, this difference is significant.</span><br></code></pre></td></tr></table></figure>
<h2 id="RNA-Seq"><a href="#RNA-Seq" class="headerlink" title="RNA-Seq"></a>RNA-Seq</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br><span class="line">232</span><br><span class="line">233</span><br><span class="line">234</span><br><span class="line">235</span><br><span class="line">236</span><br><span class="line">237</span><br><span class="line">238</span><br><span class="line">239</span><br><span class="line">240</span><br><span class="line">241</span><br><span class="line">242</span><br><span class="line">243</span><br><span class="line">244</span><br><span class="line">245</span><br><span class="line">246</span><br><span class="line">247</span><br><span class="line">248</span><br><span class="line">249</span><br><span class="line">250</span><br><span class="line">251</span><br><span class="line">252</span><br><span class="line">253</span><br><span class="line">254</span><br><span class="line">255</span><br><span class="line">256</span><br><span class="line">257</span><br><span class="line">258</span><br><span class="line">259</span><br><span class="line">260</span><br><span class="line">261</span><br><span class="line">262</span><br><span class="line">263</span><br><span class="line">264</span><br><span class="line">265</span><br><span class="line">266</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># Weil Cornell Medical School</span><br><span class="hljs-comment"># RNA-seq Analysis Workshop Course Materials</span><br><span class="hljs-comment"># Normalizing and Transforming read counts </span><br>read.counts &lt;- read.table(<span class="hljs-string">&quot;/Users/keqinliu/studying_document/bioinfo/R_project/RNA_SEQ/featureCounts_result.txt&quot;</span>, header=<span class="hljs-literal">TRUE</span>)<br><span class="hljs-comment"># get the table of read counts by indicating the path to the file</span><br>row.names(read.counts) &lt;- read.counts$Geneid<br><span class="hljs-comment"># correspond to the gene ID and the col.names to the sample names</span><br>read.counts &lt;- read.counts[ , -<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">6</span>)]<br><span class="hljs-comment"># exclude all the columns that do not contain read counts</span><br>orig_names &lt;- <span class="hljs-built_in">names</span>(read.counts)<br><span class="hljs-comment"># assign the sample names</span><br><span class="hljs-built_in">names</span>(read.counts) &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;SNF2_1&quot;</span>, <span class="hljs-string">&quot;SNF2_2&quot;</span>, <span class="hljs-string">&quot;WT_1&quot;</span>, <span class="hljs-string">&quot;WT_2&quot;</span>)<br><span class="hljs-built_in">names</span>(read.counts) &lt;- gsub(<span class="hljs-string">&quot;.*(WT|SNF2)(_[0-2]+).*&quot;</span>,<span class="hljs-string">&quot;\\1\\2&quot;</span>,orig_names)<br><span class="hljs-comment"># check data</span><br>str(read.counts)<br>head(read.counts)<br><br><span class="hljs-comment"># make a data.frame with meta-data where row.names should match the individual sample names</span><br>sample_info &lt;- data.frame(condition = gsub(<span class="hljs-string">&quot;_[0-2]+&quot;</span>, <span class="hljs-string">&quot;&quot;</span>, <span class="hljs-built_in">names</span>(read.counts)), row.names=<span class="hljs-built_in">names</span>(read.counts))<br>sample_info<br><br>library(DESeq2)<br>DESeq.ds &lt;- DESeqDataSetFromMatrix(countData = read.counts, colData = sample_info, design = ~ condition)<br><span class="hljs-comment"># generate the DESeqDataSet</span><br><br><span class="hljs-comment"># check the result by using the accessors described above (these 3 comand will return 3 table result)</span><br>colData(DESeq.ds) %&gt;% head<br>assay(DESeq.ds, <span class="hljs-string">&quot;counts&quot;</span>) %&gt;% head<br>rowData(DESeq.ds) %&gt;% head<br><br>counts(DESeq.ds) %&gt;% str <span class="hljs-comment">#test what counts returns</span><br><br>DESeq.ds &lt;- DESeq.ds[ rowSums(counts(DESeq.ds)) &gt; <span class="hljs-number">0</span>, ] <span class="hljs-comment"># remove genes without any counts</span><br>colSums(counts(DESeq.ds)) <span class="hljs-comment"># investigate different library sizes</span><br><span class="hljs-comment"># should be as same as colSums(read.counts)</span><br>colSums(read.counts)<br><br><span class="hljs-comment"># DESeq2&#x27;s default method can normalize read counts </span><br><span class="hljs-comment"># to account for differences in sequencing depths  is implementeds in estimateSizeFactors() </span><br><span class="hljs-comment"># calculate the size factor and add it to the data set</span><br>DESeq.ds &lt;- estimateSizeFactors(DESeq.ds)<br>sizeFactors(DESeq.ds)<br>colData(DESeq.ds) <span class="hljs-comment"># check colData(DESeq.ds) and find that this now contain the sizeFactors</span><br><br><span class="hljs-comment"># use counts to retrieve the normalized_read_counts</span><br>counts.sf_normalized &lt;- counts(DESeq.ds, normalized = <span class="hljs-literal">TRUE</span>)<br><br>counts(DESeq.ds)[<span class="hljs-number">1</span>,] <span class="hljs-comment"># use counts to check DESeq.ds </span><br><br><span class="hljs-comment"># transformation of sequencing-depth-normalized read counts</span><br><span class="hljs-comment"># if the read counts are transformed to the log scale following normalization</span><br><span class="hljs-comment"># many down-stream analyses(including clustering) will work much better</span><br><span class="hljs-comment"># because the relatively large dynamic range of expression values that RNA-Seq data can cover</span><br><span class="hljs-comment"># compared to log10, log2 are more commonly used</span><br><br><span class="hljs-comment"># log2 transformation of read counts</span><br>log.norm.counts &lt;- log2(counts.sf_normalized + <span class="hljs-number">1</span>)<br>par(mfrow=<span class="hljs-built_in">c</span>(<span class="hljs-number">2</span>,<span class="hljs-number">1</span>), cex.axis=<span class="hljs-number">0.75</span>) <span class="hljs-comment"># draw the following 2 images</span><br>boxplot(counts.sf_normalized, norch=<span class="hljs-literal">TRUE</span>, <br>        main=<span class="hljs-string">&quot;untransformed read counts&quot;</span>,<br>        ylab = <span class="hljs-string">&quot;read counts&quot;</span>)<br>boxplot(log.norm.counts, norch=<span class="hljs-literal">TRUE</span>,<br>        main=<span class="hljs-string">&quot;transformed read counts&quot;</span>,<br>        ylab = <span class="hljs-string">&quot;log2 read counts&quot;</span>)<br><span class="hljs-comment"># if don&#x27;t decrease font size, SNF2_2 won&#x27;t show out</span><br><br><span class="hljs-comment"># Visually exploring normalized read counts</span><br>par(mfrow=<span class="hljs-built_in">c</span>(<span class="hljs-number">2</span>,<span class="hljs-number">1</span>), cex.axis=<span class="hljs-number">0.75</span>)<br>plot(log.norm.counts[,<span class="hljs-number">1</span>:<span class="hljs-number">2</span>], cex=<span class="hljs-number">.1</span>, main=<span class="hljs-string">&quot;Normalized log2(read counts)&quot;</span>)<br>plot(log.norm.counts[,<span class="hljs-number">3</span>:<span class="hljs-number">4</span>], cex=<span class="hljs-number">.1</span>, main=<span class="hljs-string">&quot;Normalized log2(read counts)&quot;</span>)<br><br><span class="hljs-comment"># Visually check for heteroskedasticity: </span><br><span class="hljs-comment"># to plot the mean vs the standard deviation</span><br>library(vsn)<br>library(ggplot2)<br>msd_plot &lt;- meanSdPlot(log.norm.counts, ranks = <span class="hljs-literal">FALSE</span>, plot = <span class="hljs-literal">FALSE</span>)<br>msd_plot$gg +<br>  ggtitle(<span class="hljs-string">&quot;sequencing depth normalized log2(read counts)&quot;</span>) +<br>  ylab(<span class="hljs-string">&quot;standard deviation&quot;</span>)<br><br><span class="hljs-comment"># transformation of read counts including variance shrinkage</span><br><span class="hljs-comment"># obtain regularized log-transformed values</span><br>DESeq.rlog &lt;- rlog(DESeq.ds, blind=<span class="hljs-literal">TRUE</span>)<br>rlog.norm.counts &lt;- assay(DESeq.rlog)<br><br><span class="hljs-comment"># mean-sd plot for rlog-transformed data</span><br>msd_plot &lt;- meanSdPlot(rlog.norm.counts,<br>                       ranks = <span class="hljs-literal">FALSE</span>,<br>                       plot = <span class="hljs-literal">FALSE</span>)<br>msd_plot$gg + <br>  ggtitle(<span class="hljs-string">&quot;rlog-transformed read counts&quot;</span>) +<br>  ylab(<span class="hljs-string">&quot;standard deviation&quot;</span>)<br><br><span class="hljs-comment"># use cor() to calculate the correlation between columns of a matrix</span><br>distance.m_rlog &lt;- as.dist(<span class="hljs-number">1</span> - cor(rlog.norm.counts, method = <span class="hljs-string">&quot;pearson&quot;</span>))<br><br><span class="hljs-comment"># use plot() to interprete the output of the hclust</span><br>par(mfrow=<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">1</span>), cex.axis=<span class="hljs-number">0.75</span>)<br>plot( hclust(distance.m_rlog))<br>labels = colnames(rlog.norm.counts),<br>main = <span class="hljs-string">&quot;rlog transformed read counts/ndistance: Pearson Correlation&quot;</span><br><br><span class="hljs-comment"># PCA by prcomp and plot</span><br>pc &lt;- prcomp(t(rlog.norm.counts))<br>plot(pc$x[,<span class="hljs-number">1</span>], pc$x[,<span class="hljs-number">2</span>],<br>     col = colData(DESeq.ds)[,<span class="hljs-number">1</span>],<br>     main = <span class="hljs-string">&quot;PCA of seq.depth normalized/n and rlog-transformed read counts&quot;</span><br>     )<br><br><span class="hljs-comment"># PCA by DESeq2</span><br><span class="hljs-comment"># use plotPCA</span><br>library(DESeq2)<br>library(ggplot2)<br>p &lt;- plotPCA(DESeq.rlog)<br>P &lt;- P + theme_bw() + ggtitle(<span class="hljs-string">&quot;Rlog transformed counts&quot;</span>)<br>print(p)<br><br><span class="hljs-comment"># DGE</span><br><span class="hljs-comment"># if duplication &lt; 12, we suggest edgeR</span><br><span class="hljs-comment"># if duplication &gt; 12, we suggest DESeq and Limma-voom, they tend to be more conservative than edgeR</span><br><span class="hljs-comment"># in this case we use DESeq2(because I don&#x27;t know how to use edgeR)</span><br>library(DESeq2)<br><span class="hljs-comment"># DESeq2 uses the levels of condition to determine the order of the comparison</span><br>str(colData(DESeq.ds)$condition)<br>colData(DESeq.ds)$condition &lt;- relevel(colData(DESeq.ds)$condition, <span class="hljs-string">&quot;WT&quot;</span>) <span class="hljs-comment"># set WT as the first level condition</span><br>DESeq.ds &lt;- DESeq(DESeq.ds)  <span class="hljs-comment"># run DESeq</span><br><br><span class="hljs-comment"># check DGE result</span><br>DGE_results &lt;- results(DESeq.ds, independentFiltering = <span class="hljs-literal">TRUE</span>, alpha = <span class="hljs-number">0.05</span>)<br>summary(DGE_results)<br>head(DGE_results)<br>table(DGE_results$padj &lt; <span class="hljs-number">0.05</span>)<br>rownames(subset(DGE_results, padj &lt; <span class="hljs-number">0.05</span>))<br><br><span class="hljs-comment"># exploring plots following DGE analysis</span><br><span class="hljs-comment"># Histogram </span><br><span class="hljs-comment"># use hist function</span><br>hist(DGE_results$pvalue,<br>     col = <span class="hljs-string">&quot;purple&quot;</span>, border = <span class="hljs-string">&quot;white&quot;</span>, xlab = <span class="hljs-string">&quot;&quot;</span>, ylab = <span class="hljs-string">&quot;&quot;</span>,<br>     main = <span class="hljs-string">&quot;Frequencies of p-values&quot;</span>)<br><br><span class="hljs-comment"># MA plot</span><br><span class="hljs-comment"># use plotMA function</span><br>plotMA(DGE_results, alpha = <span class="hljs-number">0.05</span>, main = <span class="hljs-string">&quot;WT vs SNF2 mutants&quot;</span>,<br>       ylim = <span class="hljs-built_in">c</span>(-<span class="hljs-number">4</span>,<span class="hljs-number">4</span>))<br>       <br><span class="hljs-comment"># Heatmap</span><br><span class="hljs-comment"># use aheatmap function in NMF package </span><br>library(NMF)<br>DGE.results.sorted &lt;- DGE_results[order(DGE_results$padj), ] <span class="hljs-comment"># sort the results by p-value</span><br>DGEgenes &lt;- rownames(subset(DGE.results.sorted, padj &lt; <span class="hljs-number">0.05</span>)) <span class="hljs-comment"># identify genes with the desired adjusted p-value cut-off</span><br>hm.mat_DGEgenes &lt;- log.norm.counts[DGEgenes, ] <span class="hljs-comment"># extract normalized read counts for DE genes into a matrix</span><br><span class="hljs-comment"># found porblems: if you try to draw these 3 plot together</span><br><span class="hljs-comment"># you will get an error:</span><br><span class="hljs-comment"># Error in gridPLT() : Figure region too small and/or viewport too large</span><br><span class="hljs-comment"># solusion: save each one as pdf or png(pdf will be more clear than png)</span><br>pdf(<span class="hljs-string">&quot;heatmap1.pdf&quot;</span>)<br>aheatmap(hm.mat_DGEgenes, Rowv = <span class="hljs-literal">NA</span>, Colv = <span class="hljs-literal">NA</span>) <span class="hljs-comment"># plot normalized read counts of DE genes sorted by the adjusted p-value</span><br>dev.off()<br><br>png(<span class="hljs-string">&quot;heatmap2.png&quot;</span>)<br>aheatmap(hm.mat_DGEgenes, <br>         Rowv = <span class="hljs-literal">TRUE</span>, Colv = <span class="hljs-literal">TRUE</span>,<br>         distfun = <span class="hljs-string">&quot;euclidean&quot;</span>, hclustfun = <span class="hljs-string">&quot;average&quot;</span>) <span class="hljs-comment"># combine the heatmap with hierarchichal clustering</span><br>dev.off()<br><br>pdf(<span class="hljs-string">&quot;heatmap3.pdf&quot;</span>)<br>aheatmap(hm.mat_DGEgenes,<br>         Rowv = <span class="hljs-literal">TRUE</span>, Colv = <span class="hljs-literal">TRUE</span>,<br>         distfun = <span class="hljs-string">&quot;euclidean&quot;</span>, hclustfun = <span class="hljs-string">&quot;average&quot;</span>, <span class="hljs-comment"># scale the read counts per gene to emphasize the sample-type-specific difference</span><br>         scale = <span class="hljs-string">&quot;row&quot;</span>)<br>dev.off()<br><br><br><span class="hljs-comment"># read counts of single gene</span><br>library(org.Sc.sgd.db)<br>keytypes(org.Sc.sgd.db) <span class="hljs-comment"># list all the types of keywords that are avilable to query the annotation database</span><br>columns(org.Sc.sgd.db) <span class="hljs-comment"># list columns that can be retrieved from the annotation database</span><br>anno &lt;- select(org.Sc.sgd.db, <span class="hljs-comment"># make a batch retriveval for all gene</span><br>               keys = DGEgenes,<br>               keytype = <span class="hljs-string">&quot;ORF&quot;</span>,<br>               columns = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;SGD&quot;</span>, <span class="hljs-string">&quot;GENENAME&quot;</span>, <span class="hljs-string">&quot;CHR&quot;</span>))<br>DGE.results.sorted_logFC &lt;- DGE_results[order(DGE_results$log2FoldChange), ] <span class="hljs-comment"># check </span><br><span class="hljs-comment"># whether SNF2 pops up among the top downregulated genes(in this case SNF2 is the goal gene)</span><br>DGEgenes_logFC &lt;- rownames(subset(DGE.results.sorted_logFC, padj &lt; <span class="hljs-number">0.05</span>))<br>head(anno[match(DGEgenes_logFC, anno$ORF), ])<br><br><span class="hljs-comment"># find the corresponding to SNF2</span><br>subset(anno, GENENAME == <span class="hljs-string">&quot;SNF2&quot;</span>)<br><br><span class="hljs-comment"># DESeq2 offers a wrapper function to plot read counts for single genes</span><br>library(grDevices)<br>plotCounts(dds = DESeq.ds,<br>           gene = <span class="hljs-string">&quot;YOR290C&quot;</span>,<br>           normalized = <span class="hljs-literal">TRUE</span>, transform = <span class="hljs-literal">FALSE</span>,<br>           main = <span class="hljs-built_in">expression</span>( atop(<span class="hljs-string">&quot;Expression of &quot;</span>*italic(<span class="hljs-string">&quot;snf2&quot;</span>), <span class="hljs-string">&quot;(YOR290C)&quot;</span>)))<br>out.df &lt;- merge(as.data.frame(DGE_results), anno,<br>                by.x = <span class="hljs-string">&quot;row.names&quot;</span>, by,y = <span class="hljs-string">&quot;ORF&quot;</span>) <span class="hljs-comment"># get an error:</span><br><span class="hljs-comment"># Error in as.vector(x, mode) : cannot coerce type &#x27;closure&#x27; to vector of type &#x27;any&#x27;</span><br><br><span class="hljs-comment"># use degeR to get DGE results</span><br>library(edgeR)<br>sample_info.edger &lt;- factor(<span class="hljs-built_in">c</span>( <span class="hljs-built_in">rep</span>(<span class="hljs-string">&quot;SNF2&quot;</span>, <span class="hljs-number">2</span>), <span class="hljs-built_in">rep</span>(<span class="hljs-string">&quot;WT&quot;</span>, <span class="hljs-number">2</span>)))<br>sample_info.edger &lt;- relevel(sample_info.edger, ref = <span class="hljs-string">&quot;WT&quot;</span>)<br>edgeR.DGElist &lt;- DGEList(counts = read.counts, group = sample_info.edger)<br>head(edgeR.DGElist$counts) <span class="hljs-comment"># check edgeR.DGElist</span><br>edgeR.DGElist$samples<br><span class="hljs-comment"># keep &lt;- rowSums( cpm(edgeR.DGElist) &gt;= 1) &gt;= 5 </span><br><span class="hljs-comment"># remove genes that do not have one count per million in at least 5 samples </span><br><span class="hljs-comment"># edgeR.DGElist &lt;- edgeR.DGElist[keep, ]</span><br><br><span class="hljs-comment"># recompute library sizes after filtering （error: lib turns to be 0)</span><br><span class="hljs-comment"># edgeR.DGElist$samples$lib.size &lt;- colSums(edgeR.DGElist$counts)</span><br><span class="hljs-comment"># head(edgeR.DGElist$samples)</span><br><br>edgeR.DGElist &lt;- calcNormFactors(edgeR.DGElist, method = <span class="hljs-string">&quot;TMM&quot;</span>) <span class="hljs-comment">#calculate normalizztion factor for the library size</span><br>edgeR.DGElist$samples<br><br>design &lt;- model.matrix(~sample_info.edger)<br>edgeR.DGElist &lt;- estimateDisp(edgeR.DGElist, design) <span class="hljs-comment">#估计所有样本中所有读取计数的离散度</span><br>edger_fit &lt;- glmFit(edgeR.DGElist, design) <span class="hljs-comment">#拟合负二项模型</span><br>edger_lrt &lt;- glmLRT(edger_fit) <span class="hljs-comment">#使用负数对每个基因进行测试，二项式模型</span><br>DGE_results_edgeR &lt;- topTags(edger_lrt, n=<span class="hljs-literal">Inf</span>, <br>                             sort.by = <span class="hljs-string">&quot;PValue&quot;</span>, adjust.method = <span class="hljs-string">&quot;BH&quot;</span>) <span class="hljs-comment">#从edger_lrt表中提取结果和过滤后的p-values</span><br><br><br><span class="hljs-comment"># use Limma to get DGE results</span><br>library(limma)<br>design &lt;- model.matrix(~sample_info.edger) <br>rownames(design) &lt;- colnames(edgeR.DGElist)<br>voomTransformed &lt;- voom(edgeR.DGElist, design, plot = <span class="hljs-literal">FALSE</span>)<br>voomed.fitted &lt;- lmFit(voomTransformed, design = design) <span class="hljs-comment">#为每个基因拟合线性模型</span><br>voomed.fitted &lt;- eBayes(voomed.fitted) <span class="hljs-comment">#计算t和F，</span><br>colnames(design) <span class="hljs-comment">#通过logFC和统计学特征量提取基因列表</span><br>DGE_results_limma &lt;- topTable(voomed.fitted, coef = <span class="hljs-string">&quot;sample_info.edgerSNF2&quot;</span>,<br>                              number = <span class="hljs-literal">Inf</span>, adjust.method = <span class="hljs-string">&quot;BH&quot;</span>,<br>                              sort.by = <span class="hljs-string">&quot;logFC&quot;</span>)<br><br><span class="hljs-comment"># check your DGE result</span><br><span class="hljs-comment"># 3 method: DESeq2, edgeR and limma all need to be compare to each other</span><br>pdf(<span class="hljs-string">&quot;venn.pdf&quot;</span>)<br>DE_list &lt;- <span class="hljs-built_in">list</span>(deseq2 = rownames(subset(DGE_results, padj &lt;= <span class="hljs-number">0.05</span> )),<br>                edger = rownames(subset(DGE_results_edgeR$table, FDR &lt;= <span class="hljs-number">0.05</span>)),<br>                limma = rownames(subset(DGE_results_limma, adj.P.Val &lt;= <span class="hljs-number">0.05</span>)))<br>gplots::venn(DE_list)<br>dev.off()  <span class="hljs-comment">#画出DESeq2，edgeR和limma的韦恩图</span><br><br><span class="hljs-comment"># 当有3个以上的列表时，需要用比韦恩图更复杂的表示方法</span><br>DE_gns &lt;- UpSetR::fromList(DE_list)<br>UpSetR::upset(DE_gns, order.by = <span class="hljs-string">&quot;freq&quot;</span>)<br><br><span class="hljs-comment"># 在3个方法中寻找logFC的相关性</span><br>DE_gns_all &lt;- row.names(DE_gns[rowSums(DE_gns) == <span class="hljs-number">3</span>,])<br>DE_fc &lt;- data.frame(edger = DGE_results_edgeR[DE_gns_all,]$table$logFC,<br>                    limma = DGE_results_limma[DE_gns_all,]$logFC,<br>                    deseq2 = DGE_results[DE_gns_all,]$log2FoldChange,<br>                    row.names = DE_gns_all)<br>pairs(DE_fc)<br>pheatmap::pheatmap( as.matrix(DE_fc) )<br><span class="hljs-comment"># report error: Error in plot.window(...) : &#x27;xlim&#x27;值不能是无限的</span><br><span class="hljs-comment"># reprot error: Error in hclust(d, method = method) : 外接函数调用时不能有NA/NaN/Inf(arg10)</span><br><br><br><br><span class="hljs-comment"># Example downstream analyses</span><br><span class="hljs-comment"># use KEGG or GO</span><br></code></pre></td></tr></table></figure>
<h2 id="basic-analysis-of-gene-expression"><a href="#basic-analysis-of-gene-expression" class="headerlink" title="basic analysis of gene expression"></a>basic analysis of gene expression</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br><span class="line">232</span><br><span class="line">233</span><br><span class="line">234</span><br><span class="line">235</span><br><span class="line">236</span><br><span class="line">237</span><br><span class="line">238</span><br><span class="line">239</span><br><span class="line">240</span><br><span class="line">241</span><br><span class="line">242</span><br><span class="line">243</span><br><span class="line">244</span><br><span class="line">245</span><br><span class="line">246</span><br><span class="line">247</span><br><span class="line">248</span><br><span class="line">249</span><br><span class="line">250</span><br><span class="line">251</span><br><span class="line">252</span><br><span class="line">253</span><br><span class="line">254</span><br><span class="line">255</span><br><span class="line">256</span><br><span class="line">257</span><br><span class="line">258</span><br><span class="line">259</span><br><span class="line">260</span><br><span class="line">261</span><br><span class="line">262</span><br><span class="line">263</span><br><span class="line">264</span><br><span class="line">265</span><br><span class="line">266</span><br><span class="line">267</span><br><span class="line">268</span><br><span class="line">269</span><br><span class="line">270</span><br><span class="line">271</span><br><span class="line">272</span><br><span class="line">273</span><br><span class="line">274</span><br><span class="line">275</span><br><span class="line">276</span><br><span class="line">277</span><br><span class="line">278</span><br><span class="line">279</span><br><span class="line">280</span><br><span class="line">281</span><br><span class="line">282</span><br><span class="line">283</span><br><span class="line">284</span><br><span class="line">285</span><br><span class="line">286</span><br><span class="line">287</span><br><span class="line">288</span><br><span class="line">289</span><br><span class="line">290</span><br><span class="line">291</span><br><span class="line">292</span><br><span class="line">293</span><br><span class="line">294</span><br><span class="line">295</span><br><span class="line">296</span><br><span class="line">297</span><br><span class="line">298</span><br><span class="line">299</span><br><span class="line">300</span><br><span class="line">301</span><br><span class="line">302</span><br><span class="line">303</span><br><span class="line">304</span><br><span class="line">305</span><br><span class="line">306</span><br><span class="line">307</span><br><span class="line">308</span><br><span class="line">309</span><br><span class="line">310</span><br><span class="line">311</span><br><span class="line">312</span><br><span class="line">313</span><br><span class="line">314</span><br><span class="line">315</span><br><span class="line">316</span><br><span class="line">317</span><br><span class="line">318</span><br><span class="line">319</span><br><span class="line">320</span><br><span class="line">321</span><br><span class="line">322</span><br><span class="line">323</span><br><span class="line">324</span><br><span class="line">325</span><br><span class="line">326</span><br><span class="line">327</span><br><span class="line">328</span><br><span class="line">329</span><br><span class="line">330</span><br></pre></td><td class="code"><pre><code class="hljs R">library(Biobase)<br>library(GEOquery)<br>library(limma)<br>library(mogene10sttranscriptcluster.db)<br>library(RColorBrewer)<br>library(gProfileR)<br>library(genefilter)<br>library(openxlsx)<br>library(piano)<br>library(gplots)<br>library(sva)<br>library(msigdbr)<br><br><span class="hljs-comment"># 1 download GEO data</span><br>GEO_List &lt;- getGEO(GEO = <span class="hljs-string">&#x27;GSE76999&#x27;</span>) <span class="hljs-comment">#get GEO data</span><br><span class="hljs-built_in">class</span>(GEO_List) <span class="hljs-comment">#get GEO_List type</span><br>View(GEO_List) <span class="hljs-comment">#see GEO_List</span><br><span class="hljs-built_in">length</span>(GEO_List) <span class="hljs-comment">#see GEO_List length</span><br><span class="hljs-built_in">names</span>(GEO_List)<br><br><span class="hljs-comment"># 1.1 see the data kind and labels</span><br>eset &lt;- GEO_List[[<span class="hljs-number">1</span>]] <span class="hljs-comment">#assign GEO_List into another variable</span><br>eset<br>varLabels(eset) <span class="hljs-comment">#to see labels kind of GEO_List</span><br>eset$title <span class="hljs-comment">#to see the names of experiment</span><br>eset$organism_ch1 <span class="hljs-comment">#to see the organism of experiment</span><br><br><span class="hljs-comment">#2 process the expression table for analysis</span><br>sel &lt;- 1:<span class="hljs-number">12</span> <span class="hljs-comment">#select the samples we need(the first 12)</span><br>eset &lt;- eset[ , sel]  <br>exprs(eset) &lt;- exprs(eset)[ ,sel] <span class="hljs-comment">#These generic functions access the expression and error measurements </span><br><span class="hljs-comment">#of assay data stored in an object derived from the eset class</span><br>labels &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;BM_MOs&quot;</span>, <span class="hljs-string">&quot;FL_MOs&quot;</span>, <span class="hljs-string">&quot;YS_Macs&quot;</span>) <span class="hljs-comment">#set labels</span><br>sel_groups &lt;- factor(<span class="hljs-built_in">rep</span>(labels, each=<span class="hljs-number">4</span>)) <span class="hljs-comment">#factor function uses for stroing class</span><br>palette(rainbow(<span class="hljs-number">12</span>)) <span class="hljs-comment">#palette default has 8 colors &quot;black&quot;   &quot;red&quot;     &quot;green3&quot;  &quot;blue&quot;    &quot;cyan&quot;    &quot;magenta&quot; &quot;yellow&quot; </span><br><span class="hljs-comment"># and  &quot;gray&quot; we can use number to get access to these colors. For example: palette(rainbow(12)) or palette(gray(c(0.1, 0.2, 0.3)))</span><br>boxplot(exprs(eset), col = sel_groups)<br>boxplot(exprs(eset), col = sel_groups, las=<span class="hljs-number">2</span>) <span class="hljs-comment"># las=2:rotate the labels of the boxplot</span><br>boxplot(exprs(eset), col = sel_groups, las=<span class="hljs-number">2</span>, cex.axis=<span class="hljs-number">0.7</span>) <span class="hljs-comment"># ces.axis:make labels smaller</span><br>boxplot(exprs(eset), col = sel_groups, las=<span class="hljs-number">2</span>, cex.axis=<span class="hljs-number">0.7</span>, boxwex=<span class="hljs-number">0.6</span>, outline=<span class="hljs-literal">FALSE</span>) <span class="hljs-comment"># boxwex=0.6, outline=FALSE:make bar thiner</span><br><span class="hljs-comment"># The boxplot indicates all samples have the similar distribution and are normalized.</span><br><br><span class="hljs-comment"># 2.1 get barplot more orderly(to make sure the data get normalized)</span><br><span class="hljs-comment"># normalize the data (if you need) </span><br>exprs_matrix.norm &lt;- normalizeBetweenArrays(exprs(eset)) <span class="hljs-comment"># normalizeBetweenArrays is a function in Limma</span><br>boxplot(exprs_matrix.norm, col=sel_groups, las=<span class="hljs-number">2</span>, cex.axis=<span class="hljs-number">0.7</span>, boxwex=<span class="hljs-number">0.6</span>, outline=<span class="hljs-literal">FALSE</span>)<br><span class="hljs-comment"># now you can get the barplot more orderly</span><br><br><span class="hljs-comment"># 3 differential expression analysis by Limma </span><br>ex &lt;- exprs(eset)<br>exprs(eset) &lt;- log2(ex) <span class="hljs-comment">#transform into log2 scale</span><br>eset$description &lt;- sel_groups <span class="hljs-comment"># replace the long description by short labels</span><br>design &lt;- model.matrix(~ description + <span class="hljs-number">0</span>, eset) <span class="hljs-comment"># model.matrix uses to create a matrix， </span><br><span class="hljs-comment"># Construct design matrices by model.matrix</span><br><span class="hljs-comment"># 0 defines the way we compare the samples</span><br>colnames(design) &lt;- levels(sel_groups)<br>eset$description <span class="hljs-comment"># to see the short new description of eset</span><br>design <span class="hljs-comment"># to see the design list</span><br><br><span class="hljs-comment"># 3.1 fit a liner model, perform differential analysis and estimate p-values</span><br>fit &lt;- lmFit(eset, design) <span class="hljs-comment"># fit a linear model for each gene given a series of arrays</span><br><span class="hljs-comment"># lmFit is the basic computing engines called by lm used to fit linear models</span><br>cont.matrix &lt;- makeContrasts(BM_MOs-FL_MOs, BM_MOs-YS_Macs, FL_MOs-YS_Macs, levels=design)<br><span class="hljs-comment"># build comparision and compute the statistics</span><br><span class="hljs-comment"># makeContrasts：Construct the contrast matrix corresponding to specified contrasts of a set of parameters.</span><br>fit2 &lt;- contrasts.fit(fit, cont.matrix)<br><span class="hljs-comment"># contrast.fit ：Compute Contrasts From Linear Model Fit</span><br>fit2 &lt;- eBayes(fit2, <span class="hljs-number">0.05</span>)<br><span class="hljs-comment"># eBayes ：Empirical Bayes Statistics For Differential Expression， </span><br><span class="hljs-comment"># Given a microarray linear model fit, compute moderated t-statistics, moderated F-statistic, and log-odds </span><br><span class="hljs-comment"># of differential expression by empirical Bayes moderation of the standard errors towards a common value.</span><br><br><span class="hljs-comment"># 3.2 Extract the sig. DE genes, coef number represents the order of column in cont.matrix:</span><br>de_genes_BM_MOs_FL_MOs &lt;- topTable(fit2, coef=<span class="hljs-number">1</span>, adjust=<span class="hljs-string">&quot;fdr&quot;</span>, p.value=<span class="hljs-number">0.05</span>, number=<span class="hljs-literal">Inf</span>)<br><span class="hljs-comment"># coef=1  BM_MOs-FL_MOs</span><br>de_genes_BM_MOs_YS_Macs &lt;- topTable(fit2, coef=<span class="hljs-number">2</span>, adjust=<span class="hljs-string">&quot;fdr&quot;</span>, p.value=<span class="hljs-number">0.05</span>, number=<span class="hljs-literal">Inf</span>)<br><span class="hljs-comment"># coef=2  BM_MOs-YS_Macs</span><br>de_genes_FL_MOs_YS_Macs &lt;- topTable(fit2, coef=<span class="hljs-number">3</span>, adjust=<span class="hljs-string">&quot;fdr&quot;</span>, p.value=<span class="hljs-number">0.05</span>, number=<span class="hljs-literal">Inf</span>)<br><span class="hljs-comment"># coef=3  FL_MOs-YS_Macs</span><br><span class="hljs-comment"># topTable: is the Table Of Top Genes From Linear Model Fit, </span><br><span class="hljs-comment"># it can Extract a table of the top-ranked genes from a linear model fit.</span><br>head(de_genes_BM_MOs_FL_MOs) <span class="hljs-comment"># show top de genes in BM_MOs vs FL_MOs comparirson</span><br>all_de_genes &lt;- topTable(fit2, adjust=<span class="hljs-string">&quot;fdr&quot;</span>, p.value=<span class="hljs-number">0.05</span>, number=<span class="hljs-literal">Inf</span>)<br>all_genes_BM_FOs_FL_MOs &lt;- topTable(fit2, coef=<span class="hljs-number">1</span>, adjust=<span class="hljs-string">&quot;fdr&quot;</span>, p.value=<span class="hljs-number">1</span>, number=<span class="hljs-literal">Inf</span>)<br><br><span class="hljs-comment"># 3.3 output de_genes list to excel list (and csv file)</span><br>wb &lt;- createWorkbook()<br><span class="hljs-comment"># create each sheet</span><br>addWorksheet(wb, <span class="hljs-string">&quot;de_genes_BM_MOs_FL_MOs&quot;</span>)<br>addWorksheet(wb, <span class="hljs-string">&quot;de_genes_BM_MOs_YS_Macs&quot;</span>)<br>addWorksheet(wb, <span class="hljs-string">&quot;de_genes_FL_MOs_YS_Macs&quot;</span>)<br><span class="hljs-comment"># write the data into each sheet</span><br>writeData(wb, <span class="hljs-string">&quot;de_genes_BM_MOs_FL_MOs&quot;</span>, de_genes_BM_MOs_FL_MOs, rowNames = <span class="hljs-literal">FALSE</span>)<br>writeData(wb, <span class="hljs-string">&quot;de_genes_BM_MOs_YS_Macs&quot;</span>, de_genes_BM_MOs_YS_Macs, rowNames = <span class="hljs-literal">FALSE</span>)<br>writeData(wb, <span class="hljs-string">&quot;de_genes_FL_MOs_YS_Macs&quot;</span>, de_genes_FL_MOs_YS_Macs, rowNames = <span class="hljs-literal">FALSE</span>)<br>saveWorkbook(wb, <span class="hljs-string">&quot;DE_genes_GSE76999.xlsx&quot;</span>, overwrite = <span class="hljs-literal">TRUE</span>)<br><span class="hljs-comment"># to save csv file</span><br>write.table(de_genes_BM_MOs_FL_MOs, file=<span class="hljs-string">&quot;de_genes_BM_MOs_FL_MOs.csv&quot;</span>, row.names=<span class="hljs-built_in">F</span>, sep=<span class="hljs-string">&quot;\t&quot;</span>)<br>write.table(de_genes_BM_MOs_YS_Macs, file=<span class="hljs-string">&quot;de_genes_BM_MOs_YS_Macs.csv&quot;</span>, row.names=<span class="hljs-built_in">F</span>, sep=<span class="hljs-string">&quot;\t&quot;</span>)<br>write.table(de_genes_FL_MOs_YS_Macs, file=<span class="hljs-string">&quot;de_genes_FL_MOs_YS_Macs.csv&quot;</span>, row.names=<span class="hljs-built_in">F</span>, sep=<span class="hljs-string">&quot;\t&quot;</span>)<br><br><span class="hljs-comment"># 3.4 get gene annotation(turn gene ID into gene name)</span><br><span class="hljs-comment"># you can get a particular R package to for this mapping task </span><br><span class="hljs-comment"># here we use pakcage mogene10sttranscriptcluster.db which corresponds the Affymetrix mouse genes 10 described in GSE76999</span><br>columns(mogene10sttranscriptcluster.db)<br><span class="hljs-comment"># mogene10sttranscriptcluster.db package contains the annotaion of Affymetrix mouse genes 10</span><br>ID &lt;- featureNames(eset) <span class="hljs-comment"># get all the probe ID in our ExpressionSet</span><br>tmp &lt;- select(mogene10sttranscriptcluster.db, ID, <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;SYMBOL&quot;</span>, <span class="hljs-string">&quot;ENTREZID&quot;</span>))<br><span class="hljs-comment"># insert a column for symbols which match the ID for each row</span><br>all_genes_BM_MOs_FL_MOs$ENTREZID &lt;- tmp$ENTREZID[match(all_genes_BM_MOs_FL_MOs$ID, tmp$PROBEID)]<br>all_de_genes$symbol &lt;- tmp$SYMBOL[match(all_de_genes$ID, tmp$PROBEID)]<br>de_genes_BM_MOs_FL_MOs$ENTREZID &lt;- tmp$ENTREZID[match(de_genes_BM_MOs_FL_MOs$ID, tmp$PROBEID)]<br>de_genes_BM_MOs_YS_Macs$ENTREZID &lt;- tmp$ENTREZID[match(de_genes_BM_MOs_YS_Macs$ID, tmp$PROBEID)]<br>de_genes_FL_MOs_YS_Macs$ENTREZID &lt;- tmp$ENTREZID[match(de_genes_FL_MOs_YS_Macs$ID, tmp$PROBEID)]<br><br><span class="hljs-comment"># 3.5 drawing MA plot and volcano plot</span><br><span class="hljs-comment"># drawing MA plot(BM_MOs - FL_MOs) By Limma build-in function</span><br>ma &lt;- fit2[,<span class="hljs-string">&quot;BM_MOs - FL_MOs&quot;</span>]<br>plotMA(ma)<br><span class="hljs-comment"># drawing volcano plot(BM_MOs-FL_MOs)</span><br>volcanoplot(fit2, coeff=<span class="hljs-number">1</span>, xlim=<span class="hljs-built_in">c</span>(-<span class="hljs-number">2</span>,<span class="hljs-number">2</span>), col=<span class="hljs-string">&quot;gray&quot;</span>)<br>title(<span class="hljs-string">&quot;BM_MOs - FL_MOs&quot;</span>)<br><br><span class="hljs-comment"># drawing MA plot(BM_MOs-YS_Macs)</span><br>ma &lt;- fit2[,<span class="hljs-string">&quot;BM_MOs-YS_Macs&quot;</span>]<br>plotMA(ma)<br><span class="hljs-comment"># drawing volcano plot(BM_MOs-FL_MOs)</span><br>volcanoplot(fit2, coeff=<span class="hljs-number">2</span>, xlim=<span class="hljs-built_in">c</span>(-<span class="hljs-number">2</span>,<span class="hljs-number">2</span>), col=<span class="hljs-string">&quot;gray&quot;</span>)<br>title(<span class="hljs-string">&quot;BM_MOs - FL_MOs&quot;</span>)<br><br><span class="hljs-comment"># drawing MA plot(FL_MOs-YS_Macs)</span><br>ma &lt;- fit2[,<span class="hljs-string">&quot;FL_MOs-YS_Macs&quot;</span>]<br>plotMA(ma)<br><span class="hljs-comment"># drawing volcano plot(FL_MOs-YS_Macs)</span><br>volcanoplot(fit2, coeff=<span class="hljs-number">3</span>, xlim=<span class="hljs-built_in">c</span>(-<span class="hljs-number">2</span>,<span class="hljs-number">2</span>), col=<span class="hljs-string">&quot;gray&quot;</span>)<br>title(<span class="hljs-string">&quot;FL_MOs-YS_Macs&quot;</span>)<br><br><span class="hljs-comment"># how to read a MA plot?</span><br><span class="hljs-comment"># log-fold-change can be realized as expression difference. </span><br><span class="hljs-comment"># For example, BM_MOs - FL_MOs means (BM_MOs expression)/(FL_MOs expression), the ratio=0 means expression is equal,</span><br><span class="hljs-comment"># the ratio&gt;0 means (BM_MOs expression) is higher than (FL_MOs expression)</span><br><span class="hljs-comment"># the ratio&lt;0 meand (BM_MOs expression) is lower than (FL_MOs expression)</span><br><span class="hljs-comment"># average expression is the average expression of 2 cell types</span><br><span class="hljs-comment"># how to read a volcanoplot?</span><br><span class="hljs-comment"># -log10(P-value) can be realized as significance, the higher the significance is, the higher the -log10(P-value) is</span><br><span class="hljs-comment"># the log2-fold-change is as same as above </span><br><br><span class="hljs-comment"># 4 PCA Principle component analysis</span><br><span class="hljs-comment"># what is PCA</span><br><span class="hljs-comment"># Principal component analysis (PCA) is a mathematical algorithm </span><br><span class="hljs-comment"># that reduces the dimensionality of the data while retaining most of the variation in the data set1. </span><br><span class="hljs-comment"># It accomplishes this reduction by identifying directions, called principal components, </span><br><span class="hljs-comment"># along which the variation in the data is maximal. By using a few components, </span><br><span class="hljs-comment"># each sample can be represented by relatively few numbers instead of by values for thousands of variables. </span><br><span class="hljs-comment"># Samples can then be plotted, making it possible to visually assess similarities and differences </span><br><span class="hljs-comment"># between samples and determine whether samples can be grouped </span><br><span class="hljs-comment"># (see Nature Biotechnology 26, 303 - 304 (2008) “What is principal component analysis?”).</span><br>pca &lt;- prcomp(t(ex))<br><span class="hljs-comment"># Performs a principalcomponents analysis on the givendata matrix and returns the results as anobject of class prcomp.</span><br>plot(pca$x[, <span class="hljs-number">1</span>:<span class="hljs-number">2</span>], col=sel_groups, pch=<span class="hljs-number">19</span>)<br><span class="hljs-comment"># take pc1 and pc2 for the plot</span><br>legend(<span class="hljs-string">&#x27;topright&#x27;</span>, inset=<span class="hljs-number">.05</span>, labels, pch=<span class="hljs-number">19</span>, col=<span class="hljs-number">1</span>:<span class="hljs-number">3</span>, horiz=<span class="hljs-literal">TRUE</span>)<br><span class="hljs-comment"># include a legend for points</span><br><br><span class="hljs-comment"># 5 clustering by heatmap</span><br><span class="hljs-comment"># 5.1 filter ex by the list in all_de_genes</span><br>sel &lt;- match(all_de_genes$ID, rownames(ex))<br><span class="hljs-comment"># ex is the matrix for all the expression values</span><br><span class="hljs-comment"># all_de_genes is all the DE genes</span><br><span class="hljs-comment"># Get the index of DE genes in ex</span><br>sel_ex &lt;- ex[sel,]<br><span class="hljs-comment"># Select those rows only</span><br><br><span class="hljs-comment"># 5.2 create a few functions defining the use of Pearson correlation as a distance matrix and complete linkage for cluster</span><br>clust &lt;- <span class="hljs-keyword">function</span>(x) hclust(x, method=<span class="hljs-string">&quot;complete&quot;</span>)<br><span class="hljs-comment"># perfirm complete linkage clustering</span><br>dist &lt;- <span class="hljs-keyword">function</span>(x) as.dist((<span class="hljs-number">1</span>-cor(t(x)))/<span class="hljs-number">2</span>)<br><span class="hljs-comment"># use the inverse of correlation as distance</span><br>hmcol &lt;- rev(brewer.pal(<span class="hljs-number">11</span>, <span class="hljs-string">&quot;RdBu&quot;</span>))<br><br><span class="hljs-comment"># 5.3 drawing heatmap</span><br>View(sel_ex) <span class="hljs-comment">#find out that sel_ex has NA gene ID row title and 0 gene expression row </span><br>sel_ex_filter &lt;- sel_ex <span class="hljs-comment"># assign sel_ex to sel_ex_filter</span><br>sel_ex_no0 &lt;- sel_ex_filter[which(rowSums(sel_ex_filter==<span class="hljs-number">0</span>)==<span class="hljs-number">0</span>),] <br><span class="hljs-comment"># delete NA gene ID row title and 0 gene expression row from sel_ex_filter and assign to sel_ex_no0</span><br>sel_ex_no0 <span class="hljs-comment"># check sel_ex_no0</span><br>type(sel_ex_no0) <span class="hljs-comment"># check sel_ex_no0 type</span><br>hp &lt;- heatmap.2(sel_ex_no0, <span class="hljs-comment"># input matrix(sel_ex_no0), heatmap.2 only accept numeric matrix without NA/NaN/Inf(arg10)</span><br>                col=hmcol, <span class="hljs-comment"># define the customized colormap</span><br>                scale=<span class="hljs-string">&quot;row&quot;</span>, <span class="hljs-comment"># scale the values for each gene (row)</span><br>                labRow=<span class="hljs-built_in">F</span>, <span class="hljs-comment"># Not showing row labels</span><br>                density.info=<span class="hljs-string">&quot;none&quot;</span>, <span class="hljs-comment"># Not showing density info</span><br>                trace=<span class="hljs-string">&quot;none&quot;</span>, <span class="hljs-comment"># Not showing trace line</span><br>                margins = <span class="hljs-built_in">c</span>(<span class="hljs-number">8</span>,<span class="hljs-number">3</span>), <span class="hljs-comment"># modify the margin to fit the labels</span><br>                labCol = <span class="hljs-built_in">rep</span>(labels,each=<span class="hljs-number">4</span>), <span class="hljs-comment"># add the column labels</span><br>                <span class="hljs-comment"># define hierarchical clustering method</span><br>                hclust = clust,<br>                <span class="hljs-comment"># Using correlation coefficient for distance function</span><br>                distfun = dist<br>)<br><span class="hljs-comment"># what does z-score mean?</span><br><span class="hljs-comment"># z-score = 0 :equal to average expression</span><br><span class="hljs-comment"># z-score &gt; 0 :higher than average expression</span><br><span class="hljs-comment"># z-score &lt; 0: lower than average expression</span><br><br><br><span class="hljs-comment"># 5.4 get genes from clustering</span><br><span class="hljs-comment"># get the clusters from the dendrogram</span><br>hc &lt;- as.hclust(hp$rowDendrogram)<br><span class="hljs-comment"># Extract the hierarchical cluster from heatmap to class &quot;hclust&quot;</span><br>clusters &lt;- cutree(hc, k=<span class="hljs-number">5</span>) <span class="hljs-comment"># k is the argument used to cut a dendrogram tree into several groups</span><br><span class="hljs-comment"># in this case, we use k=5 because we want 5 groups</span><br><span class="hljs-comment"># Cut the tree by the dedired number of groups</span><br>clustered_genes &lt;- <span class="hljs-built_in">list</span>(all_de_genes$symbol[match(<span class="hljs-built_in">names</span>(clusters)[clusters==<span class="hljs-number">1</span>],all_de_genes$ID)],<br>                        all_de_genes$symbol[match(<span class="hljs-built_in">names</span>(clusters)[clusters==<span class="hljs-number">2</span>],all_de_genes$ID)],<br>                        all_de_genes$symbol[match(<span class="hljs-built_in">names</span>(clusters)[clusters==<span class="hljs-number">3</span>],all_de_genes$ID)],<br>                        all_de_genes$symbol[match(<span class="hljs-built_in">names</span>(clusters)[clusters==<span class="hljs-number">4</span>],all_de_genes$ID)],<br>                        all_de_genes$symbol[match(<span class="hljs-built_in">names</span>(clusters)[clusters==<span class="hljs-number">5</span>],all_de_genes$ID)])<br><span class="hljs-comment"># converting the ID into gene symbol into a list.</span><br>clustered_genes &lt;- <span class="hljs-built_in">list</span>(all_de_genes[<span class="hljs-built_in">names</span>(clusters[clusters==<span class="hljs-number">1</span>]),]$symbol,<br>                        all_de_genes[<span class="hljs-built_in">names</span>(clusters[clusters==<span class="hljs-number">2</span>]),]$symbol,<br>                        all_de_genes[<span class="hljs-built_in">names</span>(clusters[clusters==<span class="hljs-number">3</span>]),]$symbol,<br>                        all_de_genes[<span class="hljs-built_in">names</span>(clusters[clusters==<span class="hljs-number">4</span>]),]$symbol,<br>                        all_de_genes[<span class="hljs-built_in">names</span>(clusters[clusters==<span class="hljs-number">5</span>]),]$symbol)<br>wb &lt;- createWorkbook()<br>addWorksheet(wb, <span class="hljs-string">&quot;cluster 1&quot;</span>)<br>addWorksheet(wb, <span class="hljs-string">&quot;cluster 2&quot;</span>)<br>addWorksheet(wb, <span class="hljs-string">&quot;cluster 3&quot;</span>)<br>addWorksheet(wb, <span class="hljs-string">&quot;cluster 4&quot;</span>)<br>addWorksheet(wb, <span class="hljs-string">&quot;cluster 5&quot;</span>)<br>writeData(wb, <span class="hljs-string">&quot;cluster 1&quot;</span>, clustered_genes[[<span class="hljs-number">1</span>]], rowNames = <span class="hljs-literal">FALSE</span>)<br>writeData(wb, <span class="hljs-string">&quot;cluster 2&quot;</span>, clustered_genes[[<span class="hljs-number">2</span>]], rowNames = <span class="hljs-literal">FALSE</span>)<br>writeData(wb, <span class="hljs-string">&quot;cluster 3&quot;</span>, clustered_genes[[<span class="hljs-number">3</span>]], rowNames = <span class="hljs-literal">FALSE</span>)<br>writeData(wb, <span class="hljs-string">&quot;cluster 4&quot;</span>, clustered_genes[[<span class="hljs-number">4</span>]], rowNames = <span class="hljs-literal">FALSE</span>)<br>writeData(wb, <span class="hljs-string">&quot;cluster 5&quot;</span>, clustered_genes[[<span class="hljs-number">5</span>]], rowNames = <span class="hljs-literal">FALSE</span>)<br>saveWorkbook(wb, <span class="hljs-string">&quot;gene_clusters.xlsx&quot;</span>, overwrite = <span class="hljs-literal">TRUE</span>)<br><br><span class="hljs-comment"># 6 GO (Gene Ontology) Enrichment Analysis</span><br><span class="hljs-comment"># 6.1 we will look at genes up-regulated or down-regulated in the comparirsosn BM-MOs (G0) v.s. YS-Macs (G2).</span><br>go_enrich_up &lt;- gprofiler(as.vector(all_de_genes$symbol[all_de_genes$BM_MOs...YS_Macs &gt; <span class="hljs-number">0</span>]),<br>                          organism=<span class="hljs-string">&quot;mmusculus&quot;</span>)  <span class="hljs-comment">#up_regulated</span><br>go_enrich_down &lt;- gprofiler(as.vector(all_de_genes$symbol[all_de_genes$BM_MOs...YS_Macs &lt; <span class="hljs-number">0</span>]), <br>                          organism=<span class="hljs-string">&quot;mmusculus&quot;</span>)  <span class="hljs-comment">#down_regulated</span><br>View(go_enrich_up)<br>View(go_enrich_down)<br><span class="hljs-comment"># save result to excel file</span><br>wb &lt;- createWorkbook()<br>addWorksheet(wb, <span class="hljs-string">&quot;GO_BM_MOs_YS_Macs_UP&quot;</span>)<br>addWorksheet(wb, <span class="hljs-string">&quot;GO_BM_MOs_YS_Macs_DOWN&quot;</span>)<br>writeData(wb, <span class="hljs-string">&quot;GO_BM_MOs_YS_Macs_UP&quot;</span>, go_enrich_up, rowNames = <span class="hljs-literal">FALSE</span>)<br>writeData(wb, <span class="hljs-string">&quot;GO_BM_MOs_YS_Macs_DOWN&quot;</span>, go_enrich_down, rowNames = <span class="hljs-literal">FALSE</span>)<br>saveWorkbook(wb, <span class="hljs-string">&quot;GO_analysis.xlsx&quot;</span>, overwrite = <span class="hljs-literal">TRUE</span>)<br><br><span class="hljs-comment"># 7 GSEA</span><br><span class="hljs-comment"># test a specific (maybe self-defined) gene set to see </span><br><span class="hljs-comment"># whether they are up-regulated or down-regulated compared to other genes in terms of differential expression</span><br><span class="hljs-comment"># assume the set of first 50 genes is the interested gene set</span><br>interested_gene_set &lt;- rownames(exprs(eset))[<span class="hljs-number">1</span>:<span class="hljs-number">50</span>]<br>res.gsea &lt;- camera(exprs(eset), <span class="hljs-built_in">list</span>(set1 = interested_gene_set), design)<br>res.gsea <span class="hljs-comment">#print res.gsea</span><br><br><span class="hljs-comment"># 8 combing other dataset for PCA and clustering  (but there is no cluster point in the polt)</span><br><span class="hljs-comment"># 8.1 combine Imun_gen.ex.Rdump and ex</span><br>ex_imun_gen &lt;- dget(<span class="hljs-string">&quot;material/Imun_gen.ex.Rdump&quot;</span>) <span class="hljs-comment"># get Imun_gen.ex.Rdump from material </span><br>phenoData_table &lt;- read.table(file=<span class="hljs-string">&quot;material/pheno.table.csv&quot;</span>, header=<span class="hljs-built_in">T</span>, sep=<span class="hljs-string">&#x27;,&#x27;</span>, <span class="hljs-built_in">quote</span>=<span class="hljs-string">&#x27;&#x27;</span>) <span class="hljs-comment"># get pheno.table.csv from material</span><br>ex_imun_gen <span class="hljs-comment"># check content of the ex_imun_gen</span><br>phenoData_table <span class="hljs-comment"># check content of the phenoData_table</span><br>ex_combine.raw &lt;- merge(ex_imun_gen, ex, by=<span class="hljs-string">&#x27;row.names&#x27;</span>) <span class="hljs-comment">#combine ex_imun_gen and ex</span><br>rownames(ex_combine.raw) &lt;- ex_combine.raw$<span class="hljs-string">&quot;Row.names&quot;</span><br>ex_combine.raw$<span class="hljs-string">&quot;Row.names&quot;</span>&lt;- <span class="hljs-literal">NULL</span> <span class="hljs-comment"># now ex_combine.raw is data.frame</span><br>ex_combine.raw &lt;- as.matrix(ex_combine.raw) <span class="hljs-comment"># turn ex_combine.raw from data.frame into matrix</span><br><br><span class="hljs-comment"># 8.2 Do PCA on merged data and draw merged PCA plot</span><br>pca_im &lt;- prcomp(t(ex_combine.raw))  <span class="hljs-comment"># perform PCA</span><br>exp_names &lt;- paste0(phenoData_table$cell, <span class="hljs-string">&quot;_&quot;</span>, phenoData_table$Tissue) <span class="hljs-comment"># combine cell and tissue data. </span><br><span class="hljs-comment"># Take PC1 and PC2 for the plot</span><br><span class="hljs-comment">#par(mai = c(1,  1, .25, 2)) </span><br>plot(pca_im$x[,<span class="hljs-number">1</span>:<span class="hljs-number">2</span>], <br>     pch = <span class="hljs-built_in">c</span>(<span class="hljs-number">16</span>, <span class="hljs-number">17</span>)[<span class="hljs-built_in">as.numeric</span>(phenoData_table$cell)],  <span class="hljs-comment"># cell type with distinct symbols</span><br>     col=brewer.pal(<span class="hljs-number">10</span>,<span class="hljs-string">&quot;Set1&quot;</span>)[<span class="hljs-built_in">as.numeric</span>(phenoData_table$Tissue)])  <span class="hljs-comment"># cell tissue with distinct colors</span><br><span class="hljs-comment"># Legend outside the plot on right</span><br>legend(<span class="hljs-string">&quot;top&quot;</span>,             <span class="hljs-comment"># Location of legend </span><br>       xpd = <span class="hljs-literal">TRUE</span>,                       <span class="hljs-comment"># Allow drawing outside plot area</span><br>       xjust = <span class="hljs-number">0</span>,                        <span class="hljs-comment"># Left justify legend box on x</span><br>       yjust = <span class="hljs-number">1</span>,                        <span class="hljs-comment"># Center legend box on y</span><br>       legend = unique(phenoData_table$Tissue),<br>       col = brewer.pal(<span class="hljs-number">10</span>,<span class="hljs-string">&quot;Set1&quot;</span>),      <span class="hljs-comment"># Legend Element Colors</span><br>       pch = <span class="hljs-built_in">c</span>(<span class="hljs-number">16</span>),                      <span class="hljs-comment"># Legend Element Styles</span><br>       title = <span class="hljs-string">&quot;Tissue&quot;</span>)                 <span class="hljs-comment"># Legend Title</span><br><br><span class="hljs-comment"># Legend outside the plot on bottom</span><br>legend(<span class="hljs-string">&quot;bottom&quot;</span>,            <span class="hljs-comment"># Location of legend</span><br>       xpd = <span class="hljs-literal">TRUE</span>,                       <span class="hljs-comment"># Allow drawing outside plot area</span><br>       xjust = <span class="hljs-number">0</span>,                        <span class="hljs-comment"># Left justify legend box on x</span><br>       yjust = <span class="hljs-number">0</span>,                        <span class="hljs-comment"># Center legend box on y</span><br>       legend = unique(phenoData_table$cell),<br>       col = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;black&quot;</span>, gray(<span class="hljs-number">.2</span>)),       <span class="hljs-comment"># Legend Element Colors</span><br>       pch = <span class="hljs-built_in">c</span>(<span class="hljs-number">16</span>, <span class="hljs-number">17</span>),                  <span class="hljs-comment"># Symbol styles for legend elements</span><br>       title = <span class="hljs-string">&quot;Cell&quot;</span>)                   <span class="hljs-comment"># Legend Title</span><br><br>phenoData_table$Tissue<br>phenoData_table$cell<br><br>modcombat = model.matrix(~<span class="hljs-number">1</span>, data=phenoData_table)<br>ex_combined.batch_removed = ComBat(dat=ex_combine.raw, <br>                                   batch=phenoData_table$batch, mod=modcombat, par.prior=<span class="hljs-literal">TRUE</span>)<br><br><br><span class="hljs-comment"># Select the de genes from the expression matrix after batch effect removal</span><br>sel &lt;- match(all_de_genes$ID, rownames(ex_combined.batch_removed))<br>sel &lt;- sel[!<span class="hljs-built_in">is.na</span>(sel)] <span class="hljs-comment"># remove the NAs</span><br><br>clust &lt;- <span class="hljs-keyword">function</span>(x) hclust(x, method=<span class="hljs-string">&quot;complete&quot;</span>) <span class="hljs-comment"># perfrom complete linkage clustering</span><br>dist &lt;- <span class="hljs-keyword">function</span>(x) as.dist((<span class="hljs-number">1</span>-cor(t(x)))/<span class="hljs-number">2</span>)  <span class="hljs-comment"># use the inverse of correlation as distance. </span><br>hmcol &lt;- rev(brewer.pal(<span class="hljs-number">11</span>, <span class="hljs-string">&quot;RdBu&quot;</span>))<br><br><span class="hljs-comment"># heatmap.2 is a powerful and complicated function, please read its help message</span><br>hp &lt;- heatmap.2(ex_combined.batch_removed[sel,], <span class="hljs-comment"># input matrix </span><br>                col=hmcol, <span class="hljs-comment"># define the customized colormap</span><br>                scale=<span class="hljs-string">&quot;row&quot;</span>, <span class="hljs-comment"># scale the values for each gene (row)</span><br>                labRow=<span class="hljs-built_in">F</span>, <span class="hljs-comment"># Not showing row labels</span><br>                density.info=<span class="hljs-string">&quot;none&quot;</span>, <span class="hljs-comment"># Not showing density info</span><br>                trace=<span class="hljs-string">&quot;none&quot;</span>, <span class="hljs-comment"># Not showing trace line</span><br>                margins = <span class="hljs-built_in">c</span>(<span class="hljs-number">11</span>,<span class="hljs-number">3</span>), <span class="hljs-comment"># modify the margin to fit the labels</span><br>                labCol=paste0(phenoData_table$cell, <span class="hljs-string">&quot; / &quot;</span>, phenoData_table$Tissue), <span class="hljs-comment"># add the column labels</span><br>                <span class="hljs-comment"># define hierarchical clustering method</span><br>                hclust = clust,<br>                <span class="hljs-comment"># Using correlation coefficient for distance function</span><br>                distfun = dist<br>)<br></code></pre></td></tr></table></figure>
<h2 id="Analysis-of-cancer-genome-atlas-in-r"><a href="#Analysis-of-cancer-genome-atlas-in-r" class="headerlink" title="Analysis of cancer genome atlas in r"></a>Analysis of cancer genome atlas in r</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br></pre></td><td class="code"><pre><code class="hljs r"><span class="hljs-comment">#Analysis of cancer genome atlas in r</span><br><br><span class="hljs-comment"># 1. loading libraries</span><br>library(TCGAbiolinks)<br>library(limma)<br>library(edgeR)<br>library(glmnet)<br>library(FactoMineR)<br>library(caret)<br>library(SummarizedExperiment)<br>library(gplots)<br>library(survival)<br>library(survminer)<br>library(RColorBrewer)<br>library(gProfileR)<br>library(genefilter)<br><br><span class="hljs-comment"># get TCGA project</span><br>GDCprojects =  getGDCprojects()<br><span class="hljs-comment"># use head check data</span><br>head(GDCprojects[<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;project_id&quot;</span>, <span class="hljs-string">&quot;name&quot;</span>)])<br><span class="hljs-comment"># use TCGAbiolinks:::getProjectSummary to get information from id:TCGA-LIHC</span><br>TCGAbiolinks:::getProjectSummary(<span class="hljs-string">&quot;TCGA-LIHC&quot;</span>)<br><span class="hljs-comment"># query:TCGA-LIHC</span><br>query_TCGA = GDCquery(<br>  project = <span class="hljs-string">&quot;TCGA-LIHC&quot;</span>,<br>  data.category = <span class="hljs-string">&quot;Transcriptome Profiling&quot;</span>, <span class="hljs-comment"># parameter enforced by GDCquery</span><br>  experimental.strategy = <span class="hljs-string">&quot;RNA-Seq&quot;</span>,<br>  workflow.type = <span class="hljs-string">&quot;HTSeq - Counts&quot;</span>)<br><span class="hljs-comment"># use getResult to visualize the query results in a more readable way</span><br>lihc_result &lt;- getResults(query_TCGA)<br>colnames(lihc_result) <span class="hljs-comment"># try sample_type</span><br>head(lihc_result$sample_type) <span class="hljs-comment"># find &quot;Primary Tumor&quot;</span><br><span class="hljs-comment"># download this sample type data</span><br>query_TCGA = GDCquery(<br>  project = <span class="hljs-string">&quot;TCGA-LIHC&quot;</span>,<br>  data.category = <span class="hljs-string">&quot;Transcriptome Profiling&quot;</span>, <span class="hljs-comment"># parameter enforced by GDCquery</span><br>  experimental.strategy = <span class="hljs-string">&quot;RNA-Seq&quot;</span>,<br>  workflow.type = <span class="hljs-string">&quot;HTSeq - Counts&quot;</span>,<br>  sample.type = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;Primary Tumor&quot;</span>, <span class="hljs-string">&quot;Solid Tissue Normal&quot;</span>))<br>new_lihc_result &lt;- getResults(query_TCGA)<br>head(new_lihc_result)<br>GDCdownload(query_TCGA) <span class="hljs-comment"># download TCGA data</span><br><br>tcga_data = GDCprepare(query_TCGA) <span class="hljs-comment"># load rna-seq into r</span><br><span class="hljs-built_in">dim</span>(tcga_data) <span class="hljs-comment"># check the size of data</span><br><br><span class="hljs-comment"># There are 3 functions that allow us to access to most important data present in this object, these are: </span><br><span class="hljs-comment"># colData () ，rowData () ，assays ()</span><br>colnames(colData(tcga_data))<br>View(tcga_data)<br><span class="hljs-comment"># The table() function (in this context) produces a small summary with the sum of each of the factors present in a given column.</span><br>table(tcga_data@colData$vital_status) <span class="hljs-comment"># the sum umber of alive and dead patient </span><br>table(tcga_data@colData$tumor_stage) <span class="hljs-comment"># the stage of tumor</span><br>table(tcga_data@colData$definition) <span class="hljs-comment"># the sample number of tumor and normal tissue</span><br>table(tcga_data@colData$tissue_or_organ_of_origin) <span class="hljs-comment"># check the number of tissue and organ</span><br>table(tcga_data@colData$race) <span class="hljs-comment"># race report</span><br><br><span class="hljs-comment"># Is there a particular column (feature) that allows you to distinguish tumor tissue from normal tissue?</span><br><span class="hljs-built_in">dim</span>(assay(tcga_data)) <span class="hljs-comment"># get gene expression matrix</span><br>head(assay(tcga_data)[<span class="hljs-number">1</span>:<span class="hljs-number">5</span>,<span class="hljs-number">1</span>:<span class="hljs-number">10</span>]) <span class="hljs-comment"># 查看前10个样本的前5个基因</span><br>head(rowData(tcga_data)) <span class="hljs-comment"># check 前6个基因的ensembl id and gene id </span><br><br><span class="hljs-comment"># save tcga data as rds</span><br>saveRDS(object = tcga_data,<br>        file = <span class="hljs-string">&quot;tcga_data.RDS&quot;</span>,<br>        compress = <span class="hljs-literal">FALSE</span>)<br>        <br><br><span class="hljs-comment"># 2. Normalize RNA data</span><br><span class="hljs-comment"># uncompleted</span><br></code></pre></td></tr></table></figure>

            </div>
            <hr>
            <div>
              <div class="post-metas mb-3">
                
                  <div class="post-meta mr-3">
                    <i class="iconfont icon-category"></i>
                    
                      <a class="hover-with-bg" href="/categories/LEARNING-RECORDS/">LEARNING RECORDS</a>
                    
                  </div>
                
                
                  <div class="post-meta">
                    <i class="iconfont icon-tags"></i>
                    
                      <a class="hover-with-bg" href="/tags/R/">R</a>
                    
                  </div>
                
              </div>
              
                <p class="note note-warning">
                  
                    本博客所有文章除特别声明外，均采用 <a target="_blank" href="https://creativecommons.org/licenses/by-sa/4.0/deed.zh" rel="nofollow noopener noopener">CC BY-SA 4.0 协议</a> ，转载请注明出处！
                  
                </p>
              
              
                <div class="post-prevnext">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/2021/08/09/20210809%E5%AE%9E%E9%AA%8C%E8%AE%B0%E5%BD%95%E4%BA%BA%E6%BA%90%E5%8C%96%E5%B0%8F%E9%BC%A0%E5%85%A8%E8%A1%80DNA%E6%8A%BD%E6%8F%90/">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">20210809实验记录人源化小鼠全血DNA抽提</span>
                        <span class="visible-mobile">上一篇</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2021/07/19/20210719%E5%AE%9E%E9%AA%8C%E8%AE%B0%E5%BD%95B16%E5%A4%9A%E8%82%BD%E6%BA%B6%E6%B6%B2%E9%85%8D%E7%BD%AE%E6%96%B9%E6%B3%95/">
                        <span class="hidden-mobile">20210719实验记录B16多肽溶液配置方法</span>
                        <span class="visible-mobile">下一篇</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
              <!-- Comments -->
              <article class="comments" id="comments">
                
                  
                
                
  <div class="disqus" style="width:100%">
    <div id="disqus_thread"></div>
    <script>
      Fluid.utils.waitElementVisible('disqus_thread', function() {
        Fluid.utils.createCssLink('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqusjs.css');
        Fluid.utils.createScript('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqus.js', function () {
          new DisqusJS({
            shortname: 'fluid',
            apikey: ''
          });
        });
      });
    </script>
    <noscript>Please enable JavaScript to view the
      <a target="_blank" href="https://disqus.com/?ref_noscript" rel="nofollow noopener noopener">comments powered by Disqus.</a>
    </noscript>
  </div>


              </article>
            
          </article>
        </div>
      </div>
    </div>
    
      <div class="d-none d-lg-block col-lg-2 toc-container" id="toc-ctn">
        <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;目录</p>
  <div class="toc-body" id="toc-body"></div>
</div>

      </div>
    
  </div>
</div>

<!-- Custom -->


    

    
      <a id="scroll-top-button" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">搜索</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v"
                 for="local-search-input">关键词</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>
    

    
  </main>

  <footer class="text-center mt-5 py-3">
  <div class="footer-content">
     <a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a> <i class="iconfont icon-love"></i> <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener"><span>Fluid</span></a> 
  </div>
  

  

  
</footer>

<!-- SCRIPTS -->

  <script  src="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" ></script>
<script  src="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/js/bootstrap.min.js" ></script>
<script  src="/js/debouncer.js" ></script>
<script  src="/js/events.js" ></script>
<script  src="/js/plugins.js" ></script>

<!-- Plugins -->


  
    <script  src="/js/lazyload.js" ></script>
  



  



  <script  src="https://cdn.jsdelivr.net/npm/tocbot@4.12.0/dist/tocbot.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/anchor-js@4.3.0/anchor.min.js" ></script>



  <script defer src="https://cdn.jsdelivr.net/npm/clipboard@2.0.6/dist/clipboard.min.js" ></script>






  <script  src="https://cdn.jsdelivr.net/npm/typed.js@2.0.11/lib/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var title = document.getElementById('subtitle').title;
      
      typing(title)
      
    })(window, document);
  </script>



  <script  src="/js/local-search.js" ></script>
  <script>
    (function () {
      var path = "/local-search.xml";
      $('#local-search-input').on('click', function() {
        searchFunc(path, 'local-search-input', 'local-search-result');
      });
      $('#modalSearch').on('shown.bs.modal', function() {
        $('#local-search-input').focus();
      });
    })()
  </script>





  

  
    <!-- MathJax -->
    <script>
      MathJax = {
        tex: {
          inlineMath: [['$', '$'], ['\\(', '\\)']]
        },
        options: {
          renderActions: {
            findScript: [10, doc => {
              document.querySelectorAll('script[type^="math/tex"]').forEach(node => {
                const display = !!node.type.match(/; *mode=display/);
                const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display);
                const text = document.createTextNode('');
                node.parentNode.replaceChild(text, node);
                math.start = { node: text, delim: '', n: 0 };
                math.end = { node: text, delim: '', n: 0 };
                doc.math.push(math);
              });
            }, '', false],
            insertedScript: [200, () => {
              document.querySelectorAll('mjx-container').forEach(node => {
                let target = node.parentNode;
                if (target.nodeName.toLowerCase() === 'li') {
                  target.parentNode.classList.add('has-jax');
                }
              });
            }, '', false]
          }
        }
      };
    </script>

    <script async src="https://cdn.jsdelivr.net/npm/mathjax@3.1.2/es5/tex-svg.js" ></script>

  











<!-- 主题的启动项 保持在最底部 -->
<script  src="/js/boot.js" ></script>



</body>
</html>
