

<!DOCTYPE html>
<html lang="zh-CN" data-default-color-scheme=&#34;auto&#34;>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/favicon.png">
  <link rel="icon" href="/img/favicon.png">
  <meta name="viewport"
        content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  
    <meta name="description" content="">
  
  <meta name="author" content="John Doe">
  <meta name="keywords" content="">
  
  <title>2021.12.06-12.10 文献阅读：自闭症光谱常见遗传风险变异的识别 - Hexo</title>

  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" />


  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/github-markdown-css@4.0.0/github-markdown.min.css" />
  <link  rel="stylesheet" href="/lib/hint/hint.min.css" />

  
    
    
      
      <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@10.4.0/styles/github-gist.min.css" />
    
  

  
    <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.css" />
  



<!-- 主题依赖的图标库，不要自行修改 -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_ba1fz6golrf.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_kmeydafke9r.css">


<link  rel="stylesheet" href="/css/main.css" />

<!-- 自定义样式保持在最底部 -->


  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    var CONFIG = {"hostname":"example.com","root":"/","version":"1.8.8","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"right","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"copy_btn":true,"image_zoom":{"enable":true},"toc":{"enable":true,"headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"onlypost":false},"web_analytics":{"enable":false,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":null,"app_key":null,"server_url":null}}};
  </script>
  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
<meta name="generator" content="Hexo 5.4.0"></head>


<body>
  <header style="height: 70vh;">
    <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand"
       href="/">&nbsp;<strong>Liu Keqin</strong>&nbsp;</a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/">
                <i class="iconfont icon-home-fill"></i>
                首页
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/">
                <i class="iconfont icon-archive-fill"></i>
                归档
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/">
                <i class="iconfont icon-category-fill"></i>
                分类
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/">
                <i class="iconfont icon-tags-fill"></i>
                标签
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/">
                <i class="iconfont icon-user-fill"></i>
                关于
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" data-toggle="modal" data-target="#modalSearch">&nbsp;<i
                class="iconfont icon-search"></i>&nbsp;</a>
          </li>
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" href="javascript:">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

    <div class="banner" id="banner" parallax=true
         style="background: url('/img/default.png') no-repeat center center;
           background-size: cover;">
      <div class="full-bg-img">
        <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
          <div class="page-header text-center fade-in-up">
            <span class="h2" id="subtitle" title="2021.12.06-12.10 文献阅读：自闭症光谱常见遗传风险变异的识别">
              
            </span>

            
              <div class="mt-3">
  
  
    <span class="post-meta">
      <i class="iconfont icon-date-fill" aria-hidden="true"></i>
      <time datetime="2021-12-10 12:18" pubdate>
        2021年12月10日 中午
      </time>
    </span>
  
</div>

<div class="mt-1">
  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-chart"></i>
      18.7k 字
    </span>
  

  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-clock-fill"></i>
      
      
      246
       分钟
    </span>
  

  
  
</div>

            
          </div>

          
        </div>
      </div>
    </div>
  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="d-none d-lg-block col-lg-2"></div>
    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div class="py-5" id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">2021.12.06-12.10 文献阅读：自闭症光谱常见遗传风险变异的识别</h1>
            
            <div class="markdown-body">
              <h1><span id="2021-12-06-12-10-wen-xian-yue-du-zi-bi-zheng-guang-pu-chang-jian-yi-chuan-feng-xian-bian-yi-de-shi-bie">2021.12.06-12.10 文献阅读：自闭症光谱常见遗传风险变异的识别</span><a href="#2021-12-06-12-10-wen-xian-yue-du-zi-bi-zheng-guang-pu-chang-jian-yi-chuan-feng-xian-bian-yi-de-shi-bie" class="header-anchor">#</a></h1><p><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/">Identification of common genetic risk variants for autism spectrum disorder</a></p>
<div class="toc">

<!-- toc -->
<ul>
<li><a href="#zhai-yao">摘要</a></li>
<li><a href="#yao-dian">要点</a></li>
<li><a href="#jian-yao-jie-shao">简要介绍</a></li>
<li><a href="#jie-guo">结果</a><ul>
<li><a href="#1-gwas-quan-ji-yin-zu-xian-zhu-wei-dian-snp-hou-xuan-ji-yin-zuo">1.GWAS-全基因组显著位点+SNP+候选基因座</a></li>
<li><a href="#2-duo-xing-zhuang-gwas-yu-qi-ta-xing-zhuang-yi-yu-zheng-chong-du-yi-yu-zheng-jing-shen-fen-lie-zheng-zhu-yi-li-que-xian-duo-dong-zhang-ai-de-xiang-guan-fen-xi-fang-fa-ld-hub-yu-ld-score-regression">2.多性状 GWAS 与其他性状【抑郁症、重度抑郁症、精神分裂症、注意力缺陷/多动障碍】的相关分析（方法：LD hub与LD score regression）</a><ul>
<li><a href="#matg-fang-fa-gu-ji-asd-yu-qi-ta-biao-xing-guan-lian-xing-generalizes-the-standard-inverse-variance-weighted-meta-analysis-for-multiple-phenotypes-gai-gua-liao-biao-zhun-de-fan-fang-chai-jia-quan-yuan-fen-xi-gua-yong-yu-duo-chong-biao-xing">MATG方法估计ASD与其他表性关联性：generalizes the standard inverse-variance weighted meta-analysis for multiple phenotypes（概括了标准的反方差加权元分析，适用于多种表型）</a></li>
</ul>
</li>
<li><a href="#3-ji-yin-yu-ji-yin-ji-fen-xi">3.基因与基因集分析</a></li>
<li><a href="#4-duo-ji-yin-gou-zhu-de-pou-xi">4.多基因构筑的剖析</a><ul>
<li><a href="#4-1-zai-8-ge-gu-du-zheng-biao-xing-zhong-zhi-you-ren-zhi-biao-xing-shou-jiao-yu-cheng-du-yu-iq-chu-xian-liao-qiang-lie-de-yi-zhi-xing-you-zhi-li-zhang-ai-de-dui-zhao-zu-de-fen-zui-di-qi-ci-shi-you-zhi-li-zhang-ai-de-asd-bing-li-wu-zhi-li-zhang-ai-de-asd-bing-li-de-fen-you-ming-xian-gao-yu-qi-ta-zu">4.1 在8个孤独症表型中，只有认知表型（受教育程度与IQ）出现了强烈的异质性-有智力障碍的对照组得分最低，其次是有智力障碍的ASD病例，无智力障碍的ASD病例得分又明显高于其他组</a></li>
<li><a href="#4-2-shou-jiao-yu-cheng-du-yu-a-si-bo-ge-he-er-tong-gu-du-zheng-gong-xiang-xiang-si-de-yi-chuan-jie-gou-er-qie-zai-asd-shou-shi-zhe-zhong-guan-cha-dao-de-yu-jiao-yu-cheng-du-zheng-xiang-guan-de-deng-wei-ji-yin-guo-duo-jin-xian-yu-a-si-bo-ge-zheng-he-er-tong-zi-bi-zheng">4.2 受教育程度与阿斯伯格和儿童孤独症共享相似的遗传结构，而且在ASD受试者中观察到的与教育程度正相关的等位基因过多（仅限于阿斯伯格症和儿童自闭症）</a></li>
<li><a href="#4-3-prs-fen-xi-jie-guo-ge-ti-de-asd-feng-xian-yi-ji-liang-yi-lai-de-fang-shi-qu-jue-yu-shu-qian-ge-chang-jian-bian-yi-ti-de-duo-ji-yin-fu-dan-shui-ping-zhe-ke-yi-tong-guo-zeng-jia-asd-xiang-guan-xing-zhuang-de-snp-quan-chong-er-de-dao-jia-qiang">4.3 PRS分析结果-个体的ASD风险以剂量依赖的方式取决于数千个常见变异体的多基因负担水平，这可以通过增加ASD相关性状的SNP权重而得到加强</a></li>
</ul>
</li>
<li><a href="#5-gong-neng-zhu-shi">5.功能注释</a><ul>
<li><a href="#5-1-zai-bao-shou-de-dna-qu-yu-he-h3k4me1-zu-dan-bai-biao-ji-yi-ji-zuo-wei-yi-ge-qun-ti-zai-zhong-shu-shen-jing-xi-tong-cns-xi-bao-lei-xing-zhong-biao-da-de-ji-yin-zhong-yi-chuan-xing-ming-xian-fu-ji-dan-ge-xi-bao-zu-zhi-de-zeng-qiang-zi-xiang-guan-biao-ji-h3k4me1-zai-da-nao-he-shen-jing-yuan-xi-bao-xi-zhong-ming-xian-fu-ji">5.1 在保守的DNA区域和H3K4me1组蛋白标记，以及作为一个群体在中枢神经系统（CNS）细胞类型中表达的基因中，遗传性明显富集+单个细胞/组织的增强子相关标记H3K4me1在大脑和神经元细胞系中明显富集</a></li>
<li><a href="#5-2-hi-c-shu-ju-fen-xi-fa-xian-asd-de-gong-tong-bian-yi-wei-yu-yu-yu-ce-zai-ren-lei-pi-zhi-fa-yu-zhong-huo-yue-de-diao-kong-yuan-jian-gao-du-fu-ji-de-qu-yu-fen-xi-ta-men-de-ping-jun-biao-da-gui-ji-guan-cha-dao-que-ding-de-asd-hou-xuan-ji-yin-zai-tai-er-pi-zhi-fa-yu-guo-cheng-zhong-biao-xian-chu-zui-gao-de-biao-da-liang">5.2 Hi-C数据分析发现ASD的共同变异位于与预测在人类皮质发育中活跃的调控元件高度富集的区域+分析它们的平均表达轨迹，观察到确定的ASD候选基因在胎儿皮质发育过程中表现出最高的表达量</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#tao-lun">讨论</a><ul>
<li><a href="#1-asd-yu-ling-wai-san-lei-biao-xing-ju-you-xiang-dang-cheng-du-de-yi-chuan-xiang-guan-xing-jing-shen-fen-lie-zheng-chong-du-yi-yu-zheng-he-shou-jiao-yu-cheng-du">1.ASD与另外三类表型具有相当程度的遗传相关性：精神分裂症、重度抑郁症和受教育程度</a></li>
<li><a href="#2-dui-asd-yang-ben-hua-fen-wei-bu-tong-zi-lei-qun-guan-cha-dao-yu-qi-ta-xing-zhuang-de-yi-chuan-chong-die-de-qiang-lie-yi-zhi-xing-yu-jiao-gao-zhi-li-he-jiao-yu-cheng-du-xiang-guan-de-deng-wei-ji-yin-de-guo-liang-zhi-zai-jiao-gao-gong-neng-lei-bie-te-bie-shi-a-si-bo-ge-zong-he-zheng-he-mei-you-he-bing-zhi-li-zhang-ai-de-ge-ti-zhong-guan-cha-dao-qi-ta-wei-ming-que-de-pdd-he-fei-dian-xing-zi-bi-zheng-de-shen-jing-zhi-de-prs-ye-ming-xian-gao-yu-er-tong-zi-bi-zheng-he-a-si-bo-ge-zheng">2.对ASD样本划分为不同子类群，观察到与其他形状的遗传重叠的强烈异质性：与较高智力和教育程度相关的等位基因的过量只在较高功能类别（特别是阿斯伯格综合征和没有合并智力障碍的个体）中观察到，其他/未明确的PDD和非典型自闭症的神经质的PRS也明显高于儿童自闭症和阿斯伯格症</a></li>
<li><a href="#3-you-zhi-li-zhang-ai-he-wu-zhi-li-zhang-ai-de-asd-bing-li-zhi-jian-gu-ji-de-snp-yi-chuan-xing-cun-zai-hen-da-chai-yi-er-a-si-bo-ge-zheng-de-yi-chuan-xing-zui-gao-zhe-wei-chang-qi-de-guan-cha-ti-gong-liao-yi-chuan-zheng-ju-pu-tong-bian-yi-ti-zai-gao-gong-neng-asd-bing-li-zhong-de-gong-xian-ke-neng-geng-wei-tu-chu">3.有智力障碍和无智力障碍的ASD病例之间估计的SNP遗传性存在很大差异，而阿斯伯格症的遗传性最高，这为长期的观察提供了遗传证据；普通变异体在高功能ASD病例中的贡献可能更为突出。</a></li>
<li><a href="#4-hi-c-fen-xi-he-da-nao-zhuan-lu-zu-shu-ju-yi-ji-dan-ji-yin-zuo-de-fen-xi-qiang-diao-liao-yu-da-nao-fa-yu-he-shen-jing-yuan-gong-neng-xiang-guan-de-guo-cheng-de-can-yu-tong-shi-zai-dui-xin-de-he-han-jian-bian-yi-ti-de-yan-jiu-zhong-que-ding-liao-yi-xie-ji-yin-yu-asd-feng-xian-you-guan">4.Hi-C分析和大脑转录组数据以及单基因座的分析强调了与大脑发育和神经元功能相关的过程的参与；同时在对新的和罕见变异体的研究中，确定了一些基因与ASD风险有关。</a></li>
</ul>
</li>
<li><a href="#fang-fa">方法</a><ul>
<li><a href="#1-ipsych-yang-ben">1.iPSYCH样本</a></li>
<li><a href="#2-psychiatric-genomic-consortium-pgc-yang-ben">2.Psychiatric Genomic Consortium (PGC)样本</a></li>
<li><a href="#3-tong-ji-fen-xi">3.统计分析</a><ul>
<li><a href="#3-1-gwas-fen-xi">3.1 GWAS 分析</a></li>
<li><a href="#3-2-ji-yu-ji-yin-de-guan-lian-he-ji-yin-ji-fen-xi">3.2 基于基因的关联和基因集分析</a></li>
<li><a href="#3-3-snp-yi-chuan-li-heritability">3.3 SNP遗传力（heritability）</a></li>
<li><a href="#3-3-yi-chuan-xiang-guan-xing-genetic-correlation">3.3 遗传相关性/genetic correlation</a></li>
<li><a href="#3-4-duo-ji-yin-feng-xian-ping-fen-polygenic-risk-score">3.4 多基因风险评分/polygenic risk score</a></li>
<li><a href="#3-5-hi-c-fen-xi">3.5 Hi-C分析</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#zi-yuan">资源</a></li>
<li><a href="#gao-yin-yong-wen-xian">高引用文献</a></li>
</ul>
<!-- tocstop -->
</div>

<h2><span id="zhai-yao">摘要</span><a href="#zhai-yao" class="header-anchor">#</a></h2><p>自闭症谱系障碍（ASD）是一组高度遗传和异质的神经发育表型，被诊断为超过1%的儿童。常见的遗传变异对ASD的易感性有很大的贡献，但到目前为止，还没有个别变异与ASD有密切的联系。随着来自独特的<strong>丹麦</strong>人口资源的样本量明显增加，我们<strong>报告了对18,381个ASD病例和27,969个对照的全基因组关联元分析，确定了五个全基因组的重要位点。利用三个表型（精神分裂症、重度抑郁症和受教育程度）的GWAS结果，以同样严格的意义水平确定了另外七个与其他性状共享的基因座</strong>。<u><strong>剖析多基因结构，我们发现在ASD亚型中既有定量的也有定性的多基因异质性。</strong></u>这些结果突出了生物学上的见解，特别是与神经元功能和皮质生成有关的见解，并确立了在短期内对ASD进行大规模的GWAS研究将更有成效。</p>
<h2><span id="yao-dian">要点</span><a href="#yao-dian" class="header-anchor">#</a></h2><p>一项对18381例奥斯汀谱系障碍（ASD）病例和27969例对照的全基因组关联元分析确定了5个风险位点。作者发现ASD亚型之间存在着定量和定性的多基因异质性。</p>
<h2><span id="jian-yao-jie-shao">简要介绍</span><a href="#jian-yao-jie-shao" class="header-anchor">#</a></h2><p>ASD是一组普遍的神经发育障碍的术语，其特点是社会和交流技能受损，同时有重复和限制性行为。临床表现是非常异质的，包括有严重损伤和智力障碍的人，以及智商高于平均水平、学术和职业功能水平高的人。ASD影响1-1.5%的个体，具有高度的遗传性，常见的和罕见的变异体都对其病因有影响。据估计，<strong>常见的变异体占ASD责任的主要部分</strong>，其他常见的神经精神疾病也是如此。相比之下，<strong>新的突变，主要是拷贝数变异（CNVs）和基因破坏性点突变，具有较大的个体效应，但总体上解释了&lt;5%的整体责任，而且遗传性也小得多</strong>。虽然一些基因通过新发突变的过度统计聚集被令人信服地牵连进来，但迄今为止最大的GWAS（n = 7,387例扫描）——虽然为常见变异的大量贡献提供了令人信服的证据——但并<strong>没有最终确定具有全基因组意义的单一变异</strong>。这<strong>强调了常见变异，就像在其他复杂的疾病如精神分裂症中一样，单个变异的影响不大，需要大量增加样本数量</strong>。</p>
<p>在这里，我们报告了第一个与ASD相关的常见风险变体，与以前的GWAS5-8相比，发现的样本量增加了一倍以上。我们描述了<strong>ASD与其他复杂疾病和性状之间强烈的遗传相关性，证实了共同的病因，并且我们展示了表明ASD各临床亚型的多基因结构差异的结果</strong>。利用这些关系和最近引入的计算技术，我们发现了更多与其他表型共享的新型ASD相关变异。此外，通过与胎儿大脑Hi-C染色质相互作用分析和大脑转录组数据的整合，我们探讨了我们排名靠前的GWAS结果的功能意义。</p>
<blockquote>
<ul>
<li>De Rubeis S et al. Synaptic, transcriptional and chromatin genes disrupted in autism. <em>Nature</em> 515, 209–215 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4402723/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25363760">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Synaptic,+transcriptional+and+chromatin+genes+disrupted+in+autism&amp;author=S+De+Rubeis&amp;volume=515&amp;publication_year=2014&amp;pages=209-215&amp;pmid=25363760&amp;">Google Scholar</a>]</li>
<li>Gaugler T et al. Most genetic risk for autism resides with common variation. <em>Nat Genet</em> 46 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4137411/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25038753">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nat+Genet&amp;title=Most+genetic+risk+for+autism+resides+with+common+variation&amp;author=T+Gaugler&amp;volume=46&amp;publication_year=2014&amp;">Google Scholar</a>]</li>
<li>Gaugler T et al. Most genetic risk for autism resides with common variation. <em>Nat Genet</em> 46 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4137411/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25038753">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nat+Genet&amp;title=Most+genetic+risk+for+autism+resides+with+common+variation&amp;author=T+Gaugler&amp;volume=46&amp;publication_year=2014&amp;">Google Scholar</a>]</li>
<li>Krumm N et al. Excess of rare, inherited truncating mutations in autism. <em>Nat Genet</em> 47 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4449286/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25961944">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nat+Genet&amp;title=Excess+of+rare,+inherited+truncating+mutations+in+autism&amp;author=N+Krumm&amp;volume=47&amp;publication_year=2015&amp;">Google Scholar</a>]</li>
<li>Anney RJL et al. Meta-analysis of gwas of over 16,000 individuals with autism spectrum disorder highlights a novel locus at 10q24.32 and a significant overlap with schizophrenia. <em>Molecular Autism</em> 8, 21 (2017). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5441062/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/28540026">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Molecular+Autism&amp;title=Meta-analysis+of+gwas+of+over+16,000+individuals+with+autism+spectrum+disorder+highlights+a+novel+locus+at+10q24.32+and+a+significant+overlap+with+schizophrenia&amp;author=RJL+Anney&amp;volume=8&amp;publication_year=2017&amp;pages=21&amp;pmid=28540026&amp;">Google Scholar</a>]</li>
<li>Ma D et al. A genome-wide association study of autism reveals a common novel risk locus at 5p14.1. <em>Annals of human genetics</em> 73, 263–273 (2009). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2918410/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/19456320">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Annals+of+human+genetics&amp;title=A+genome-wide+association+study+of+autism+reveals+a+common+novel+risk+locus+at+5p14.1&amp;author=D+Ma&amp;volume=73&amp;publication_year=2009&amp;pages=263-273&amp;pmid=19456320&amp;">Google Scholar</a>]</li>
<li>Devlin B, Melhem N &amp; Roeder K Do common variants play a role in risk for autism? evidence and theoretical musings. <em>Brain research</em> 1380, 78–84 (2011). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3638767/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/21078308">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Brain+research&amp;title=Do+common+variants+play+a+role+in+risk+for+autism?+evidence+and+theoretical+musings&amp;author=B+Devlin&amp;author=N+Melhem&amp;author=K+Roeder&amp;volume=1380&amp;publication_year=2011&amp;pages=78-84&amp;pmid=21078308&amp;">Google Scholar</a>]</li>
<li>Turley P et al. Multi-trait analysis of genome-wide association summary statistics using mtag. <em>Nature Genetics</em> 50, 229–237 (2018). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5805593/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/29292387">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Multi-trait+analysis+of+genome-wide+association+summary+statistics+using+mtag&amp;author=P+Turley&amp;volume=50&amp;publication_year=2018&amp;pages=229-237&amp;pmid=29292387&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<h2><span id="jie-guo">结果</span><a href="#jie-guo" class="header-anchor">#</a></h2><h3><span id="1-gwas-quan-ji-yin-zu-xian-zhu-wei-dian-snp-hou-xuan-ji-yin-zuo">1.GWAS-全基因组显著位点+SNP+候选基因座</span><a href="#1-gwas-quan-ji-yin-zu-xian-zhu-wei-dian-snp-hou-xuan-ji-yin-zuo" class="header-anchor">#</a></h3><p>作为iPSYCH项目的一部分，研究者收集了一个丹麦全国范围内基于人口的病例队列样本，包括几乎所有1981年至2005年期间在丹麦出生并在2014年之前被诊断为ASD（根据ICD-10）的个体，并进行了基因分型。研究者从同一出生队列中随机选择了对照组（补充表1）。研究者以前曾验证过基于登记册的ASD诊断，并证明了从出生后不久收集的血点中提取和扩增的DNA基因分型的准确性。使用Ricopili15对基因型进行处理，对数据进行严格的质量控制，去除相关个体，根据主成分分析排除祖先异常值，并使用1000基因组计划第3阶段参考板进行归因。经过这样的处理，来自iPSYCH样本的13,076个病例和22,664个对照的基因型被纳入分析中。按照现在人类复杂性状基因组学的标准,主要分析是将iPSYCH ASD的结果与来自精神病基因组学联盟（PGC；5,305个病例和5,305个假对照）的五个基于家庭的欧洲血统的三人组样本进行荟萃分析。所有的PGC样本都采用了与这里相同的Ricopili管道进行质量控制、推断和分析。</p>
<p>支持研究设计之间的一致性，iPSYCH基于人群和PGC基于家庭的分析显示了高度的遗传相关性，rG=0.779（SE=0.106；P=1.75×10-13），与其他精神疾病的数据集之间观察到的遗传相关性相似。同样，由多基因风险评分（PRS）评估的多基因性在不同的样本中显示出一致性，支持不同样本和研究设计的效果的同质性（见下面关于PRS对样本的五向分割的结果）。SNP的遗传性（h2G）估计为0.118（SE = 0.010），人口流行率为0.01218。</p>
<p>主要的GWAS荟萃分析共有18381个ASD病例和27969个对照，并应用了反方差加权的固定效应模型。为了确保分析有足够的动力和稳健性，研究者检查了小等位基因频率（MAF）≥0.01，归一化INFO得分≥0.7，并得到有效样本量大于70%的标志物的支持。这个最终的荟萃分析包括了9,112,387个常染色体标记的结果，并在三个独立的基因座中产生了93个全基因组的重要标记（图1；表1a；补充图1-44）。每个基因座都得到了丹麦病例对照和PGC家族数据的有力支持。虽然观察到适度的膨胀（lambda=1.12，lambda1000=1.006），但LD评分回归分析表明，这是由于多基因性（&gt;96%，见方法）而不是混杂引起的。在X染色体上分析的294,911个标记中，最强的信号是P=7.8×10-5。</p>
<p><img src="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/bin/nihms-1015648-f0001.jpg" srcset="/img/loading.gif" alt="An external file that holds a picture, illustration, etc. Object name is nihms-1015648-f0001.jpg"></p>
<blockquote>
<p>图1-曼哈顿图</p>
<p>a：主要的ASD扫描（18,381例和27,969例对照），前景为黄色的后续样本（2,119例和142,379例对照）的综合分析结果。</p>
<p>b-d：对ASD和精神分裂症（34,129例和45,512例对照）、教育程度（N=328,917）和重度抑郁症（111,902例和312,113例对照）的三个MTAG扫描的曼哈顿图。全尺寸图见补充图45-48。在所有面板上，五个分析的综合结果（对每个标记物而言，由五个标记物的最小P值组成）在背景中以灰色显示。</p>
<blockquote>
<ul>
<li>Schizophrenia Working Group of the Psychiatric Genomics Consortium. Biological insights from 108 schizophrenia-associated genetic loci. <em>Nature</em> 511, 421–427 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4112379/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25056061">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Biological+insights+from+108+schizophrenia-associated+genetic+loci&amp;volume=511&amp;publication_year=2014&amp;pages=421-427&amp;pmid=25056061&amp;">Google Scholar</a>]</li>
<li>Okbay A et al. Genome-wide association study identifies 74 loci associated with educational attainment. <em>Nature</em> 533, 539–542 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4883595/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27225129">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Genome-wide+association+study+identifies+74+loci+associated+with+educational+attainment&amp;author=A+Okbay&amp;volume=533&amp;publication_year=2016&amp;pages=539-542&amp;pmid=27225129&amp;">Google Scholar</a>]</li>
<li>Wray NR et al. Genome-wide association analyses identify 44 risk variants and refine the genetic architecture of major depressive disorder. <em>Nature Genetics</em> 50, 668–681 (2018). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5934326/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/29700475">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Genome-wide+association+analyses+identify+44+risk+variants+and+refine+the+genetic+architecture+of+major+depressive+disorder&amp;author=NR+Wray&amp;volume=50&amp;publication_year=2018&amp;pages=668-681&amp;pmid=29700475&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<div class="table-container">
<table>
<thead>
<tr>
<th style="text-align:left"></th>
<th style="text-align:left">Index var</th>
<th style="text-align:center">CHR</th>
<th style="text-align:center">BP</th>
<th style="text-align:center">Analysis</th>
<th style="text-align:center">P</th>
<th style="text-align:center">β</th>
<th style="text-align:center">SE</th>
<th style="text-align:center">A1/A2</th>
<th style="text-align:center">FRQ</th>
<th style="text-align:center">Support from other scans</th>
<th style="text-align:center">Support from other scans</th>
<th style="text-align:center">Support from other scans</th>
<th style="text-align:left">Nearest genes</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left"></td>
<td style="text-align:left"></td>
<td style="text-align:center"></td>
<td style="text-align:center"></td>
<td style="text-align:center"></td>
<td style="text-align:center"></td>
<td style="text-align:center"></td>
<td style="text-align:center"></td>
<td style="text-align:center"></td>
<td style="text-align:center"></td>
<td style="text-align:center"><strong>Scan</strong></td>
<td style="text-align:center"><strong>P</strong></td>
<td style="text-align:center"><strong>β</strong></td>
<td style="text-align:left"></td>
</tr>
<tr>
<td style="text-align:left"><strong>a</strong></td>
<td style="text-align:left">rs910805</td>
<td style="text-align:center">20</td>
<td style="text-align:center">21248116</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">2.04 × 10−9</td>
<td style="text-align:center">−0.096</td>
<td style="text-align:center">0.016</td>
<td style="text-align:center">A/G</td>
<td style="text-align:center">0.76</td>
<td style="text-align:center">ASD-SCZ ASD-Edu*</td>
<td style="text-align:center">1.5 × 10−10 2.0 × 10−8</td>
<td style="text-align:center">−0.069 −0.061</td>
<td style="text-align:left"><em>KIZ, XRN2, NKX2–2, NKX2–4</em></td>
</tr>
<tr>
<td style="text-align:left"><strong>a</strong></td>
<td style="text-align:left">rs10099100</td>
<td style="text-align:center">8</td>
<td style="text-align:center">10576775</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">1.07 × 10−8</td>
<td style="text-align:center">0.084</td>
<td style="text-align:center">0.015</td>
<td style="text-align:center">C/G</td>
<td style="text-align:center">0.331</td>
<td style="text-align:center">Comb ASD ASD-Edu</td>
<td style="text-align:center">9.6 × 10−9 1.6 × 10−8</td>
<td style="text-align:center">0.078 0.056</td>
<td style="text-align:left"><em>C8orf74, SOX7, PINX1</em></td>
</tr>
<tr>
<td style="text-align:left"><strong>a</strong></td>
<td style="text-align:left">rs201910565</td>
<td style="text-align:center">1</td>
<td style="text-align:center">96561801</td>
<td style="text-align:center">Comb ASD</td>
<td style="text-align:center">2.48 × 10−8</td>
<td style="text-align:center">−0.077</td>
<td style="text-align:center">0.014</td>
<td style="text-align:center">A/AT</td>
<td style="text-align:center">0.689</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">3.4 × 10−7</td>
<td style="text-align:center">−0.033</td>
<td style="text-align:left"><em>LOC102723661, PTBP2</em></td>
</tr>
<tr>
<td style="text-align:left"><strong>a</strong></td>
<td style="text-align:left">rs71190156</td>
<td style="text-align:center">20</td>
<td style="text-align:center">14836243</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">2.75 × 10−8</td>
<td style="text-align:center">−0.078</td>
<td style="text-align:center">0.014</td>
<td style="text-align:center">GTTTT TTT/G</td>
<td style="text-align:center">0.481</td>
<td style="text-align:center">Comb ASD ASD-Edu</td>
<td style="text-align:center">3.0 × 10−8 1.2 × 10−8</td>
<td style="text-align:center">−0.072 0.053</td>
<td style="text-align:left"><em>MACROD2</em></td>
</tr>
<tr>
<td style="text-align:left"><strong>a</strong></td>
<td style="text-align:left">rs111931861</td>
<td style="text-align:center">7</td>
<td style="text-align:center">104744219</td>
<td style="text-align:center">Comb ASD</td>
<td style="text-align:center">3.53 × 10−8</td>
<td style="text-align:center">−0.216</td>
<td style="text-align:center">0.039</td>
<td style="text-align:center">A/G</td>
<td style="text-align:center">0.966</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">1.1 × 10−7</td>
<td style="text-align:center">−0.094</td>
<td style="text-align:left"><em>KMT2E, SRPK2</em></td>
</tr>
<tr>
<td style="text-align:left"><strong>b</strong></td>
<td style="text-align:left">rs2388334</td>
<td style="text-align:center">6</td>
<td style="text-align:center">98591622</td>
<td style="text-align:center">ASD-Edu</td>
<td style="text-align:center">3.34 × 10−12</td>
<td style="text-align:center">−0.065</td>
<td style="text-align:center">0.009</td>
<td style="text-align:center">A/G</td>
<td style="text-align:center">0.517</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">1.0 × 10−6</td>
<td style="text-align:center">−0.068</td>
<td style="text-align:left"><em>MMS22L, POU3F2</em></td>
</tr>
<tr>
<td style="text-align:left"><strong>b</strong></td>
<td style="text-align:left">rs325506</td>
<td style="text-align:center">5</td>
<td style="text-align:center">104012303</td>
<td style="text-align:center">ASD-MD</td>
<td style="text-align:center">3.26 × 10−11</td>
<td style="text-align:center">0.057</td>
<td style="text-align:center">0.009</td>
<td style="text-align:center">C/G</td>
<td style="text-align:center">0.423</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">3.5 × 10−7</td>
<td style="text-align:center">0.071</td>
<td style="text-align:left"><em>NUD12</em></td>
</tr>
<tr>
<td style="text-align:left"><strong>b</strong></td>
<td style="text-align:left">rs11787216</td>
<td style="text-align:center">8</td>
<td style="text-align:center">142615222</td>
<td style="text-align:center">ASD-Edu</td>
<td style="text-align:center">1.99 × 10−9</td>
<td style="text-align:center">−0.058</td>
<td style="text-align:center">0.010</td>
<td style="text-align:center">T/C</td>
<td style="text-align:center">0.364</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">2.6 × 10−6</td>
<td style="text-align:center">−0.030</td>
<td style="text-align:left"><em>MROH5</em></td>
</tr>
<tr>
<td style="text-align:left"><strong>b</strong></td>
<td style="text-align:left">rs1452075</td>
<td style="text-align:center">3</td>
<td style="text-align:center">62481063</td>
<td style="text-align:center">ASD-Edu</td>
<td style="text-align:center">3.17 × 10−9</td>
<td style="text-align:center">0.061</td>
<td style="text-align:center">0.010</td>
<td style="text-align:center">T/C</td>
<td style="text-align:center">0.721</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">2.1 × 10−7</td>
<td style="text-align:center">0.035</td>
<td style="text-align:left"><em>CADPS</em></td>
</tr>
<tr>
<td style="text-align:left"><strong>b</strong></td>
<td style="text-align:left">rs1620977</td>
<td style="text-align:center">1</td>
<td style="text-align:center">72729142</td>
<td style="text-align:center">ASD-MD</td>
<td style="text-align:center">6.66 × 10−9</td>
<td style="text-align:center">0.056</td>
<td style="text-align:center">0.010</td>
<td style="text-align:center">A/G</td>
<td style="text-align:center">0.26</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">1.2 × 10−4</td>
<td style="text-align:center">0.062</td>
<td style="text-align:left"><em>NEGR1</em></td>
</tr>
<tr>
<td style="text-align:left"><strong>b</strong></td>
<td style="text-align:left">rs10149470</td>
<td style="text-align:center">14</td>
<td style="text-align:center">104017953</td>
<td style="text-align:center">ASD-MD</td>
<td style="text-align:center">8.52 × 10−9</td>
<td style="text-align:center">−0.049</td>
<td style="text-align:center">0.008</td>
<td style="text-align:center">A/G</td>
<td style="text-align:center">0.487</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">8.5 × 10−5</td>
<td style="text-align:center">−0.056</td>
<td style="text-align:left"><em>MARK3, CKB, TRMT61A, BAG5, APOPT1, KLC1, XRCC3</em></td>
</tr>
<tr>
<td style="text-align:left"><strong>b</strong></td>
<td style="text-align:left">rs16854048</td>
<td style="text-align:center">4</td>
<td style="text-align:center">42123728</td>
<td style="text-align:center">ASD-MD</td>
<td style="text-align:center">1.29 × 10−8</td>
<td style="text-align:center">0.069</td>
<td style="text-align:center">0.012</td>
<td style="text-align:center">A/C</td>
<td style="text-align:center">0.858</td>
<td style="text-align:center">ASD</td>
<td style="text-align:center">5.9 × 10−5</td>
<td style="text-align:center">0.082</td>
<td style="text-align:left"><em>SLC30A9, BEND4, TMEM33, DCAF4L1</em></td>
</tr>
</tbody>
</table>
</div>
<blockquote>
<p>表1-来自ASD扫描和MTAG分析的全基因组显著位点。</p>
<p>面板a显示了单独分析ASD表型时达到全基因组意义的基因座。</p>
<blockquote>
<p>分析 “一栏表示来自原始扫描（ASD）和后续样本（组合ASD）的综合分析的最小P值。</p>
<p>“来自其他扫描的支持 “一栏列出了进一步支持该基因座在全基因组意义上的其他分析（包括MTAG）。对于ASD的扫描结果，它显示了来自其他扫描的该基因座的全基因组显著性结果，而对于Comb ASD，它显示了来自ASD的结果。</p>
</blockquote>
<p>面板b显示了在三个MTAG分析中发现的额外的全基因组显着位点。这三项分析是ASD与精神分裂症（SCZ）、教育程度（Edu）和严重抑郁症（MD）。</p>
<blockquote>
<p>这里的 “分析 “一栏指出了哪个MTAG分析给出的结果（ASD-Edu或ASD-MD），而 “来自其他扫描的支持 “一栏则单独提供了ASD的相应扫描结果。</p>
<p>在这两个板块中，独立的位点被定义为r2<0.1且距离>400kb，索引变体显示在 “Index var “栏中。</0.1且距离></p>
<p>其他栏目有：染色体（CHR）、染色体位置（BP）、等位基因（A1/A2）、A1的等位基因频率（FRQ）、相对于A1的效应估计值（β）、β的标准误差（SE）和索引变体的关联P值（P）。最近的基因 “一栏列出了所有SNP与指数变异体的r2≥0.6的区域50kb内最近的基因。</p>
<blockquote>
<ul>
<li>Schizophrenia Working Group of the Psychiatric Genomics Consortium. Biological insights from 108 schizophrenia-associated genetic loci. <em>Nature</em> 511, 421–427 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4112379/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25056061">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Biological+insights+from+108+schizophrenia-associated+genetic+loci&amp;volume=511&amp;publication_year=2014&amp;pages=421-427&amp;pmid=25056061&amp;">Google Scholar</a>]</li>
<li>Okbay A et al. Genome-wide association study identifies 74 loci associated with educational attainment. <em>Nature</em> 533, 539–542 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4883595/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27225129">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Genome-wide+association+study+identifies+74+loci+associated+with+educational+attainment&amp;author=A+Okbay&amp;volume=533&amp;publication_year=2016&amp;pages=539-542&amp;pmid=27225129&amp;">Google Scholar</a>]</li>
<li>Wray NR et al. Genome-wide association analyses identify 44 risk variants and refine the genetic architecture of major depressive disorder. <em>Nature Genetics</em> 50, 668–681 (2018). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5934326/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/29700475">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Genome-wide+association+analyses+identify+44+risk+variants+and+refine+the+genetic+architecture+of+major+depressive+disorder&amp;author=NR+Wray&amp;volume=50&amp;publication_year=2018&amp;pages=668-681&amp;pmid=29700475&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
</blockquote>
<p>接下来，研究者在五个欧洲血统的队列中获得了P值&lt;1×10-5的前88个基因座的复制数据，共增加了2119个病例和142379个对照（补充表2和3）。观察到效应方向的总体复制（88个中的53个（60%）P＜1×10-5；23个中的16个（70%）P＜1×10-6；符号检验P=0.035和P=0.047，分别），另外两个位点在综合分析中达到全基因组意义（表1a）。关于已确定的基因座的更多细节可在补充表4中找到，选定的候选基因座在表1中描述。</p>
<div class="table-container">
<table>
<thead>
<tr>
<th style="text-align:left">Gene</th>
<th style="text-align:left">Locus<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/table/T2/?report=objectonly#TFN1">*</a> and supporting evidence</th>
<th style="text-align:left">Gene function</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left"><em>NEGR1</em></td>
<td style="text-align:left">Chr1:72,729,142 Shared ASD-MDD locus Locus also significant in depression<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R21">21</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R22">22</a>, educational attainment<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R20">20</a>, intelligence<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R23">23</a>, obesity and BMI<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R24">24</a>–<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R28">28</a> and in an ASD-schizophrenia meta-analysis<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R5">5</a>. <em>NEGR1</em> is the only protein-coding gene in the locus <em>NEGR1</em> is supported by brain Hi-C and eQTL analyses<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R21">21</a></td>
<td style="text-align:left">NEGR1 (neuronal growth regulator 1) is an adhesion molecule modulating synapse formation in hippocampal neurons<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R29">29</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R30">30</a> and neurite outgrowth<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R31">31</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R32">32</a>. It is a member of the IgLON protein family implicated in synaptic plasticity and axon extension<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R33">33</a>–<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R35">35</a>. Predominantly expressed (and developmentally upregulated) in hippocampus and cortex<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R36">36</a> and also hypothalamus<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R37">37</a>.</td>
</tr>
<tr>
<td style="text-align:left"><em>PTBP2</em></td>
<td style="text-align:left">Chr1:96,561,801 ASD locus Locus also significant in BMI<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R24">24</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R25">25</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R27">27</a> weight<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R25">25</a> and educational attainment<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R20">20</a>. In schizophrenia, the locus shows a p-value of 6.5 × 10−6 <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R15">15</a> <em>PTBP2</em> is the nearest protein-coding gene, approx. 625 kb from index SNP. De novo and rare variants in <em>PTBP2</em> have been reported in ASD cases<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R1">1</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R3">3</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R38">38</a>. <em>PTBP2</em> is supported by Hi-C results in this study (<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/figure/F5/">Fig. 5d</a>)</td>
<td style="text-align:left">PTBP2 is also known as nPTB (neuronal PTB) or brPTB (brain PTB) and is a splicing regulator. PTBP1 and its paralog PTBP2 bind to intronic polypyrimidine tracts in pre-mRNAs and target large sets of exons to coordinate alternative splicing programs during development<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R39">39</a>. Several switches in the expression of PTBP1 and PTBP2 regulate alternative splicing during neurogenesis and neuronal differentiation<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R40">40</a>–<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R43">43</a>.</td>
</tr>
<tr>
<td style="text-align:left"><em>CADPS</em></td>
<td style="text-align:left">Chr3:62,481,063 Shared ASD-Educational attainment locus Locus also significant in study of cognitive decline rate<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R44">44</a> <em>CADPS</em> is supported by Hi-C results in this study (<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/figure/F5/">Fig. 5a</a>).</td>
<td style="text-align:left"><em>CADPS</em> encodes a calcium-binding protein involved in exocytosis of neurotransmitters and neuropeptides. In line with <em>CAPDS</em> mRNA being mainly expressed in brain and pituitary (GTEx portal- see <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#S22">URLs</a>), immunoreactive CAPS-1 is localized in neural and various endocrine tissues<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R45">45</a>. In hippocampal synapses, CADPS regulates the pool of readily releasable vesicles at pre-synaptic terminals<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R46">46</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R47">47</a></td>
</tr>
<tr>
<td style="text-align:left"><em>KCNN2</em></td>
<td style="text-align:left">Chr5: 113,801,423 ASD locus (gene-wise analysis) Locus also significant in educational attainment<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R20">20</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R48">48</a>. KCNN2 synaptic levels are regulated by the E3 ubiquitin ligase UBE3A<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R49">49</a>, of which overexpression has been linked to ASD risk<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R49">49</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R50">50</a>.</td>
<td style="text-align:left">KCNN2 is a voltage-independent Ca2+-activated K+ channel that responds to changes in intracellular calcium concentration and couples calcium metabolism to potassium flux and membrane excitability. In CNS neurons, activation of KCNN2 modulates neuronal excitability by causing membrane hyperpolarization<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R51">51</a>. Hippocampal KCNN2 has roles in the formation of new memory<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R52">52</a>, encoding and consolidation of contextual fear<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R53">53</a>, and in drug-induced plasticity<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R54">54</a>.</td>
</tr>
<tr>
<td style="text-align:left"><em>KMT2E</em></td>
<td style="text-align:left">Chr7:104,744,219 ASD locus Locus also significant in schizophrenia<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R15">15</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R55">55</a> and in ASD-schizophrenia meta-analysis<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R5">5</a>. <em>KMT2E</em> de novo mutations are associated with ASD at FDR &lt; 0.1<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R56">56</a> A <em>KMT2E</em> credible SNP is a loss-of-function variant (<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#SD1">Supplementary Table 16</a>)</td>
<td style="text-align:left"><em>KMT2E</em> encodes Histone-lysine N-methyltransferase 2E and forms a family together with SETD5<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R57">57</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R58">58</a>. Evidence suggest that recognition of the histone H3K4me3 mark by the KMT2E PHD finger can facilitate the recruitment of KMT2E to transcription-active chromatin regions<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R59">59</a>,<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R60">60</a>. KMT2E has been implicated in chromatin regulation, control of cell cycle progression, and maintaining genomic stability<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R61">61</a>.</td>
</tr>
<tr>
<td style="text-align:left">MACROD2</td>
<td style="text-align:left">Chr20: 14836243 ASD locus Locus found significant in previous ASD GWAS<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R62">62</a> but not supported in larger study<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R63">63</a> <em>MACROD2</em> is the only protein-coding gene in the locus</td>
<td style="text-align:left">MACROD2 is a nuclear enzyme that binds to mono-ADP-ribosylated (MARylated) proteins and functions as an eraser of mono-ADP-ribosylation<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R64">64</a>. Intracellular MARylated histones and GSK3β are substrates of MACROD2, and the removal of MAR from GSK3β is responsible for reactivating of its kinase activity<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#R64">64</a>. This gene is expressed in lung and multiple regions of the brain. Low or no expression across most other tissue (GTEx portal- see <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/#S22">URLs</a>).</td>
</tr>
</tbody>
</table>
</div>
<blockquote>
<p>表2-选定的位点和候选者（按染色体排序）</p>
</blockquote>
<h3><span id="2-duo-xing-zhuang-gwas-yu-qi-ta-xing-zhuang-yi-yu-zheng-chong-du-yi-yu-zheng-jing-shen-fen-lie-zheng-zhu-yi-li-que-xian-duo-dong-zhang-ai-de-xiang-guan-fen-xi-fang-fa-ld-hub-yu-ld-score-regression">2.多性状 GWAS 与其他性状【抑郁症、重度抑郁症、精神分裂症、注意力缺陷/多动障碍】的相关分析（方法：LD hub与LD score regression）</span><a href="#2-duo-xing-zhuang-gwas-yu-qi-ta-xing-zhuang-yi-yu-zheng-chong-du-yi-yu-zheng-jing-shen-fen-lie-zheng-zhu-yi-li-que-xian-duo-dong-zhang-ai-de-xiang-guan-fen-xi-fang-fa-ld-hub-yu-ld-score-regression" class="header-anchor">#</a></h3><p>为了研究ASD和其他表型之间的遗传重叠程度，我们使用<strong>双变量LD评分回归法 LD score regression</strong>估计了与广泛的精神病和其他医学疾病、障碍和特征的遗传相关性（图2，补充表5）。发现几个性状有明显的相关性，包括精神分裂症（rG=0.211，P=1.03×10-5）和认知能力的测量，特别是教育程度（rG=0.199，P=2.56×10-9），表明与这些表型有大量的基因重叠，并证实了以前的报告。与以前的报告相比，我们发现与重度抑郁症有强烈的、高度显著的相关性（rG=0.412，P=1.40×10-25），我们报告与多动症有突出的重叠（rG=0.360，P=1.24×10-12）。此外，我们证实了以前基于ASD样本的一个子集所报道的非ASD人群样本中，8岁时与社会交流困难的遗传相关性（rG = 0.375，P = 0.0028）。</p>
<blockquote>
<p><strong>以下文献主要涉及与ASD表型存在大量基因重叠的其他广泛精神病和/或其他医学疾病、障碍和特征的遗传相关性。它们包括：抑郁症、重度抑郁症、精神分裂症、注意力缺陷/多动障碍；以及受教育程度、主观幸福感、睡眠时间型、自我报告的疲劳程度等。</strong></p>
<ul>
<li>Schizophrenia Working Group of the Psychiatric Genomics Consortium. Biological insights from 108 schizophrenia-associated genetic loci. <em>Nature</em> 511, 421–427 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4112379/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25056061">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Biological+insights+from+108+schizophrenia-associated+genetic+loci&amp;volume=511&amp;publication_year=2014&amp;pages=421-427&amp;pmid=25056061&amp;">Google Scholar</a>]</li>
<li>Zheng J et al. Ld hub: a centralized database and web interface to perform ld score regression that maximizes the potential of summary level gwas data for snp heritability and genetic correlation analysis. <em>Bioinformatics</em> 33, 272–279 (2017). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5542030/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27663502">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Bioinformatics&amp;title=Ld+hub:+a+centralized+database+and+web+interface+to+perform+ld+score+regression+that+maximizes+the+potential+of+summary+level+gwas+data+for+snp+heritability+and+genetic+correlation+analysis&amp;author=J+Zheng&amp;volume=33&amp;publication_year=2017&amp;pages=272-279&amp;pmid=27663502&amp;">Google Scholar</a>]</li>
<li>Okbay A et al. Genome-wide association study identifies 74 loci associated with educational attainment. <em>Nature</em> 533, 539–542 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4883595/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27225129">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Genome-wide+association+study+identifies+74+loci+associated+with+educational+attainment&amp;author=A+Okbay&amp;volume=533&amp;publication_year=2016&amp;pages=539-542&amp;pmid=27225129&amp;">Google Scholar</a>]</li>
<li>Anney RJL et al. Meta-analysis of gwas of over 16,000 individuals with autism spectrum disorder highlights a novel locus at 10q24.32 and a significant overlap with schizophrenia. <em>Molecular Autism</em> 8, 21 (2017). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5441062/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/28540026">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Molecular+Autism&amp;title=Meta-analysis+of+gwas+of+over+16,000+individuals+with+autism+spectrum+disorder+highlights+a+novel+locus+at+10q24.32+and+a+significant+overlap+with+schizophrenia&amp;author=RJL+Anney&amp;volume=8&amp;publication_year=2017&amp;pages=21&amp;pmid=28540026&amp;">Google Scholar</a>]</li>
<li>Bulik-Sullivan B et al. An atlas of genetic correlations across human diseases and traits. <em>Nature Genetics</em> 47, 1061–4036 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4797329/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/26414676">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=An+atlas+of+genetic+correlations+across+human+diseases+and+traits&amp;author=B+Bulik-Sullivan&amp;volume=47&amp;publication_year=2015&amp;pages=1061-4036&amp;pmid=26192917&amp;">Google Scholar</a>]</li>
<li>Okbay A et al. Genetic variants associated with subjective well-being, depressive symptoms, and neuroticism identified through genome-wide analyses. <em>Nature genetics</em> 48, 624–633 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4884152/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27089181">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+genetics&amp;title=Genetic+variants+associated+with+subjective+well-being,+depressive+symptoms,+and+neuroticism+identified+through+genome-wide+analyses&amp;author=A+Okbay&amp;volume=48&amp;publication_year=2016&amp;pages=624-633&amp;pmid=27089181&amp;">Google Scholar</a>]</li>
<li>Clarke T-K et al. Common polygenic risk for autism spectrum disorder (asd) is associated with cognitive ability in the general population. <em>Mol Psychiatry</em> 21 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4759203/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25754080">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Mol+Psychiatry&amp;title=Common+polygenic+risk+for+autism+spectrum+disorder+(asd">Google Scholar</a>+is+associated+with+cognitive+ability+in+the+general+population&amp;author=T-K+Clarke&amp;volume=21&amp;publication_year=2015&amp;)]</li>
<li>Cross-Disorder Group of the Psychiatric Genomics Consortium et al. Genetic relationship between five psychiatric disorders estimated from genome-wide snps. <em>Nature genetics</em> 45, 984–994 (2013). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3800159/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/23933821">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+genetics&amp;title=Genetic+relationship+between+five+psychiatric+disorders+estimated+from+genome-wide+snps&amp;volume=45&amp;publication_year=2013&amp;pages=984-994&amp;pmid=23933821&amp;">Google Scholar</a>]</li>
<li>Wray NR et al. Genome-wide association analyses identify 44 risk variants and refine the genetic architecture of major depressive disorder. <em>Nature Genetics</em> 50, 668–681 (2018). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5934326/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/29700475">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Genome-wide+association+analyses+identify+44+risk+variants+and+refine+the+genetic+architecture+of+major+depressive+disorder&amp;author=NR+Wray&amp;volume=50&amp;publication_year=2018&amp;pages=668-681&amp;pmid=29700475&amp;">Google Scholar</a>]</li>
<li>Demontis D et al. Discovery of the first genome-wide significant risk loci for adhd. <em>Nature Genetics</em> (2018). [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Discovery+of+the+first+genome-wide+significant+risk+loci+for+adhd&amp;author=D+Demontis&amp;publication_year=2018&amp;">Google Scholar</a>]</li>
<li>Pourcain BS et al. Asd and schizophrenia show distinct developmental profiles in common genetic overlap with population-based social communication difficulties, <em>Molecular Psychiatry</em> 23 (2017). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5382976/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/28044064">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Molecular+Psychiatry&amp;title=Asd+and+schizophrenia+show+distinct+developmental+profiles+in+common+genetic+overlap+with+population-based+social+communication+difficulties&amp;author=BS+Pourcain&amp;volume=23&amp;publication_year=2017&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p><img src="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/bin/nihms-1015648-f0002.jpg" srcset="/img/loading.gif" alt="An external file that holds a picture, illustration, etc. Object name is nihms-1015648-f0002.jpg"></p>
<blockquote>
<p>图2-<strong>与其他性状的遗传相关性</strong></p>
<p>ASD（N = 46,350）和其他性状之间的显著遗传相关性，经过Bonferroni校正，测试了<strong>LDhub</strong>上的234个性状，增加了一些新的表型。由LDSC19进行估计和测试。这里的结果对应于以下GWAS分析。IQ（N = 78,308），教育程度（N = 328,917），大学（N = 111,114），自我报告的疲劳（N = 108,976），神经质（N = 170,911），主观幸福感（N = 298,420）。精神分裂症（N=82,315）、重度抑郁症（N=480,359）、抑郁症状（N=161,460）、注意缺陷/多动障碍（ADHD）（N=53,293）和时间型（N=128,266）。</p>
<p>该分析的全部结果见补充表5。</p>
</blockquote>
<h4><span id="matg-fang-fa-gu-ji-asd-yu-qi-ta-biao-xing-guan-lian-xing-generalizes-the-standard-inverse-variance-weighted-meta-analysis-for-multiple-phenotypes-gai-gua-liao-biao-zhun-de-fan-fang-chai-jia-quan-yuan-fen-xi-gua-yong-yu-duo-chong-biao-xing">MATG方法估计ASD与其他表性关联性：generalizes the standard inverse-variance weighted meta-analysis for multiple phenotypes（概括了标准的反方差加权元分析，适用于多种表型）</span><a href="#matg-fang-fa-gu-ji-asd-yu-qi-ta-biao-xing-guan-lian-xing-generalizes-the-standard-inverse-variance-weighted-meta-analysis-for-multiple-phenotypes-gai-gua-liao-biao-zhun-de-fan-fang-chai-jia-quan-yuan-fen-xi-gua-yong-yu-duo-chong-biao-xing" class="header-anchor">#</a></h4><p>为了利用这些观察结果来发现可能在ASD和这些其他性状之间共享的位点，我们选择了三个特别有力量的、在遗传上相关的表型。它们是精神分裂症（N = 79,641）、重度抑郁症（N = 424,015）和教育程度（N = 328,917）。我们利用最近推出的<strong>MTAG方法</strong>，简单地说，它概括了标准的反方差加权元分析，适用于多种表型。在这种情况下，<strong><u>MTAG利用了这样一个事实，即鉴于ASD和第二个性状之间的整体遗传相关性，通过适当利用第二个性状的关联信息，可以改善与ASD相关的效应大小估计和证据</u></strong>。这三个与ASD相关的MTAG扫描的结果与主要的ASD扫描（以及相互之间）是相关的，但考虑到对三个扫描的探索，我们利用一个更保守的阈值1.67×10-8来宣布这些次级扫描的显著性，估计最大错误发现率（MaxFDR）为0.021。除了上面定义的几个ASD命中的更有力的证据外，另外七个区域的变体也达到了全基因组意义，包括三个与教育程度共享的位点和四个与严重抑郁症共享的位点（表1b，方框1，补充表6，补充图49-55）。我们注意到，在这7个例子中，ASD的效应大小估计强于次级性状，该结果不是这些其他扫描中最强信号的特征（补充表7-9）（事实上，这7个中的3个在次级性状中不显著，构成潜在的新发现）。此外，我们以MTAG为基准，运行了两个非常大的遗传性状（身高，N = 252,288，和体重指数（BMI），N = 322,154），与ASD没有预期的联系，没有显著的位点被添加到只有ASD的显著关联列表中。</p>
<blockquote>
<ul>
<li>Turley P et al. Multi-trait analysis of genome-wide association summary statistics using mtag. <em>Nature Genetics</em> 50, 229–237 (2018). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5805593/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/29292387">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Multi-trait+analysis+of+genome-wide+association+summary+statistics+using+mtag&amp;author=P+Turley&amp;volume=50&amp;publication_year=2018&amp;pages=229-237&amp;pmid=29292387&amp;">Google Scholar</a>]</li>
</ul>
<blockquote>
<p>其他与ASD并无显著联系的较大遗传性状（身高和BMI）</p>
<ul>
<li>Wood AR et al. Defining the role of common variation in the genomic and biological architecture of adult human height. <em>Nature Genetics</em> 46, 1173–1186 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4250049/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25282103">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Defining+the+role+of+common+variation+in+the+genomic+and+biological+architecture+of+adult+human+height&amp;author=AR+Wood&amp;volume=46&amp;publication_year=2014&amp;pages=1173-1186&amp;pmid=25282103&amp;">Google Scholar</a>]</li>
<li>Locke AE et al. Genetic studies of body mass index yield new insights for obesity biology. <em>Nature</em> 518, 197–206 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4382211/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25673413">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Genetic+studies+of+body+mass+index+yield+new+insights+for+obesity+biology&amp;author=AE+Locke&amp;volume=518&amp;publication_year=2015&amp;pages=197-206&amp;pmid=25673413&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<h3><span id="3-ji-yin-yu-ji-yin-ji-fen-xi">3.基因与基因集分析</span><a href="#3-ji-yin-yu-ji-yin-ji-fen-xi" class="header-anchor">#</a></h3><p>接下来，研究者使用MAGMA对主要ASD元分析进行了基于基因的关联分析，测试一个基因座内所有标记的联合关联（跨越基因组中的所有蛋白编码基因）。该分析确定了15个超过显著性阈值的基因（补充表10）。正如预期的那样，这些基因大部分位于GWAS中确定的全基因组范围内的重要基因座内，但有七个基因位于另外四个基因座内，包括KCNN2、MMP12、NTM和17号染色体上的一个基因簇（KANSLl、WNT3、MAPT和CRHRl）（补充图57-71）。特别是KCNN2是强烈相关的（P = 1.02 × 10-9），甚至远远超过了单变量的统计阈值，并被列入表1的描述中。</p>
<blockquote>
<p><strong>magma</strong></p>
<blockquote>
<ul>
<li>de Leeuw CA, Mooij JM, Heskes T &amp; Posthuma D Magma: generalized gene-set analysis of gwas data. <em>PLoS Comput Biol</em> 11, e1004219 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4401657/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25885710">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=PLoS+Comput+Biol&amp;title=Magma:+generalized+gene-set+analysis+of+gwas+data&amp;author=CA+de+Leeuw&amp;author=JM+Mooij&amp;author=T+Heskes&amp;author=D+Posthuma&amp;volume=11&amp;publication_year=2015&amp;pages=e1004219&amp;pmid=25885710&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<p>使用来自人类新皮层转录组数据的基因共表达模块（M13、M16和M17）和功能缺失不耐受基因（pLI&gt;0.9）进行富集分析，这些基因以前在神经发育障碍中显示出富集的证据，对后者（P = 0.014）和M16（P = 0.050）只产生名义意义（补充表11）。在Sanders等人的研究或罕见变异中牵涉到ASD的基因只显示出名义上的显著富集（P=0.063），而在SPARK联盟的策划基因列表中的富集是显著的（P=0.0034）。同样，对MsigDB的分子功能的Gene Ontology集的分析显示，在Bonferroni校正多重测试后没有显著的集（补充表12）。</p>
<blockquote>
<p><strong>human neocortex transcriptomic data</strong></p>
<blockquote>
<ul>
<li>Parikshak NN et al. Integrative functional genomic analyses implicate specific molecular pathways and circuits in autism. <em>Cell</em> 155, 1008–1021 (2013). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3934107/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/24267887">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Cell&amp;title=Integrative+functional+genomic+analyses+implicate+specific+molecular+pathways+and+circuits+in+autism&amp;author=NN+Parikshak&amp;volume=155&amp;publication_year=2013&amp;pages=1008-1021&amp;pmid=24267887&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p><strong>loss-of-function intolerant genes</strong></p>
<blockquote>
<ul>
<li>Samocha KE et al. A framework for the interpretation of de novo mutation in human disease. <em>Nat Genet</em> 46, 944–950 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25086666">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nat+Genet&amp;title=A+framework+for+the+interpretation+of+de+novo+mutation+in+human+disease&amp;author=KE+Samocha&amp;volume=46&amp;publication_year=2014&amp;pages=944-950&amp;pmid=25086666&amp;">Google Scholar</a>]</li>
<li>Lek M et al. Analysis of protein-coding genetic variation in 60,706 humans. <em>Nature</em> 536, 285–291 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5018207/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27535533">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Analysis+of+protein-coding+genetic+variation+in+60,706+humans&amp;author=M+Lek&amp;volume=536&amp;publication_year=2016&amp;pages=285-291&amp;pmid=27535533&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p><strong>evidence of enrichment in neurodevelopmental disorders</strong></p>
<blockquote>
<ul>
<li>Demontis D et al. Discovery of the first genome-wide significant risk loci for adhd. <em>Nature Genetics</em> (2018). [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Discovery+of+the+first+genome-wide+significant+risk+loci+for+adhd&amp;author=D+Demontis&amp;publication_year=2018&amp;">Google Scholar</a>]</li>
<li>Parikshak NN et al. Integrative functional genomic analyses implicate specific molecular pathways and circuits in autism. <em>Cell</em> 155, 1008–1021 (2013). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3934107/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/24267887">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Cell&amp;title=Integrative+functional+genomic+analyses+implicate+specific+molecular+pathways+and+circuits+in+autism&amp;author=NN+Parikshak&amp;volume=155&amp;publication_year=2013&amp;pages=1008-1021&amp;pmid=24267887&amp;">Google Scholar</a>]</li>
<li>Pardiñas A et al. Common schizophrenia alleles are enriched in mutation-intolerant genes and in regions under strong background selection. <em>Nature Genetics</em> 50, 381–389 (2018). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5918692/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/29483656">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Common+schizophrenia+alleles+are+enriched+in+mutation-intolerant+genes+and+in+regions+under+strong+background+selection&amp;author=A+Pardiñas&amp;volume=50&amp;publication_year=2018&amp;pages=381-389&amp;pmid=29483656&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p><strong>Genes implicated in ASD by studies or rare variants in Sanders et al.</strong></p>
<blockquote>
<ul>
<li>Sanders S et al. Insights into autism spectrum disorder genomic architecture and biology from 71 risk loci. <em>Neuron</em> 87, 1215–1233 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4624267/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/26402605">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Neuron&amp;title=Insights+into+autism+spectrum+disorder+genomic+architecture+and+biology+from+71+risk+loci&amp;author=S+Sanders&amp;volume=87&amp;publication_year=2015&amp;pages=1215-1233&amp;pmid=26402605&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p><strong>enrichment in the curated gene list from the SPARK consortium</strong></p>
<blockquote>
<ul>
<li>SPARK Consortium. Spark: A us cohort of 50,000 families to accelerate autism research. <em>Neuron</em> 97, 488–493. [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7444276/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/29420931">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Neuron&amp;title=Spark:+A+us+cohort+of+50,000+families+to+accelerate+autism+research&amp;volume=97&amp;pages=488-493&amp;pmid=29420931&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p><strong>analysis of Gene Ontology sets for molecular function from MsigDB</strong></p>
<blockquote>
<ul>
<li>Ashburner M et al. Gene ontology: tool for the unification of biology. the gene ontology consortium. <em>Nature genetics</em> 25, 25–29 (2000). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3037419/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/10802651">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+genetics&amp;title=Gene+ontology:+tool+for+the+unification+of+biology.+the+gene+ontology+consortium&amp;author=M+Ashburner&amp;volume=25&amp;publication_year=2000&amp;pages=25-29&amp;pmid=10802651&amp;">Google Scholar</a>]</li>
<li>Consortium, G. O. Gene ontology consortium: going forward. <em>Nucleic acids research</em> 43, D1049–D1056 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4383973/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25428369">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nucleic+acids+research&amp;title=Gene+ontology+consortium:+going+forward&amp;volume=43&amp;publication_year=2015&amp;pages=D1049-D1056&amp;pmid=25428369&amp;">Google Scholar</a>]</li>
<li>Subramanian A et al. Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. <em>PNAS</em> 102, 15545–15550 (2005). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/16199517">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=PNAS&amp;title=Gene+set+enrichment+analysis:+a+knowledge-based+approach+for+interpreting+genome-wide+expression+profiles&amp;author=A+Subramanian&amp;volume=102&amp;publication_year=2005&amp;pages=15545-15550&amp;pmid=16199517&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<h3><span id="4-duo-ji-yin-gou-zhu-de-pou-xi">4.多基因构筑的剖析</span><a href="#4-duo-ji-yin-gou-zhu-de-pou-xi" class="header-anchor">#</a></h3><h4><span id="4-1-zai-8-ge-gu-du-zheng-biao-xing-zhong-zhi-you-ren-zhi-biao-xing-shou-jiao-yu-cheng-du-yu-iq-chu-xian-liao-qiang-lie-de-yi-zhi-xing-you-zhi-li-zhang-ai-de-dui-zhao-zu-de-fen-zui-di-qi-ci-shi-you-zhi-li-zhang-ai-de-asd-bing-li-wu-zhi-li-zhang-ai-de-asd-bing-li-de-fen-you-ming-xian-gao-yu-qi-ta-zu">4.1 在8个孤独症表型中，只有认知表型（受教育程度与IQ）出现了强烈的异质性-有智力障碍的对照组得分最低，其次是有智力障碍的ASD病例，无智力障碍的ASD病例得分又明显高于其他组</span><a href="#4-1-zai-8-ge-gu-du-zheng-biao-xing-zhong-zhi-you-ren-zhi-biao-xing-shou-jiao-yu-cheng-du-yu-iq-chu-xian-liao-qiang-lie-de-yi-zhi-xing-you-zhi-li-zhang-ai-de-dui-zhao-zu-de-fen-zui-di-qi-ci-shi-you-zhi-li-zhang-ai-de-asd-bing-li-wu-zhi-li-zhang-ai-de-asd-bing-li-de-fen-you-ming-xian-gao-yu-qi-ta-zu" class="header-anchor">#</a></h4><p>由于ASD是一种高度异质性的疾病，我们探讨了iPSYCH样本中<strong><script type="math/tex">h_2^G</script></strong>在不同表型亚类中的分区情况，并使用<strong>GCTA</strong>估计这些群体之间的遗传相关性。我们研究了有无智力障碍（ID，N=1,873）和ICD-10诊断亚类的病例：儿童自闭症（F84.0，N=3,310）、非典型自闭症（F84.1，N=1,607）、阿斯伯格综合症（F84. 5，N = 4,622），和其他/不明确的广泛性发育障碍（PDD，F84.8-9，N = 5,795），在做配对比较时减少到不重叠的组（见补充表13）。虽然所有亚组之间的成对遗传相关性一直很高（在所有比较中，95%CIs包括1），但阿斯伯格综合症的<script type="math/tex">h_2^G</script>（<script type="math/tex">h_2^G</script>=0.097，SE=0. 001的<script type="math/tex">h_2^G</script>是儿童自闭症（<script type="math/tex">h_2^G</script>=0.049，SE=0.009，P=0.001）和其他/未指明PDD组（<script type="math/tex">h_2^G</script>=0.045，SE=0.008，P=0.001）的两倍（补充表14和15，补充图82和83）。同样，无ID的ASD的<script type="math/tex">h_2^G</script>（<script type="math/tex">h_2^G</script>=0.086，SE=0.005）被发现比有ID的病例（<script type="math/tex">h_2^G</script>=0.029，SE=0.013，P=0.015）高三倍。</p>
<blockquote>
<p><strong>GCTA</strong></p>
<blockquote>
<ul>
<li>Yang J, Lee SH, Goddard ME &amp; Visscher PM Gcta: a tool for genome-wide complex trait analysis. <em>Am J Hum Genet</em> 88, 76–82 (2011). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3014363/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/21167468">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Am+J+Hum+Genet&amp;title=Gcta:+a+tool+for+genome-wide+complex+trait+analysis&amp;author=J+Yang&amp;author=SH+Lee&amp;author=ME+Goddard&amp;author=PM+Visscher&amp;volume=88&amp;publication_year=2011&amp;pages=76-82&amp;pmid=21167468&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<p>为了进一步研究各亚型之间明显的多基因异质性，我们调查了在不同表型上训练的PRS如何在不同的ASD亚组中分布。我们关注与ASD有强烈遗传相关性的表型（如受教育程度），但也包括与ASD相关性很小或没有相关性的性状（如BMI）作为阴性对照。在这项分析中，我们对ASD亚群的归一化分数进行回归，同时在多变量回归中包括批次和主成分的协变量。在我们评估的八个表型中，<strong>只有认知表型表现出强烈的异质性（教育程度P = 1.8 × 10-8，IQ P = 3.7 × 10-9）（补充图84）</strong>。有趣的是，<u><strong>所有有无智力障碍的病例对照组在两种认知表型上都表现出明显不同的负荷：有智力障碍的对照组得分最低，其次是有智力障碍的ASD病例，无智力障碍的ASD病例得分又明显高于其他组（教育程度P=2.6×10-12，智商P=8.2×10-12）。</strong></u></p>
<blockquote>
<p>教育程度</p>
<blockquote>
<ul>
<li>Okbay A et al. Genome-wide association study identifies 74 loci associated with educational attainment. <em>Nature</em> 533, 539–542 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4883595/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27225129">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Genome-wide+association+study+identifies+74+loci+associated+with+educational+attainment&amp;author=A+Okbay&amp;volume=533&amp;publication_year=2016&amp;pages=539-542&amp;pmid=27225129&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>IQ</p>
<blockquote>
<ul>
<li>Sniekers S et al. Genome-wide association meta-analysis of 78,308 individuals identifies new loci and genes influencing human intelligence. <em>Nature genetics</em> 49, 1546–1718 (2017). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/28951626">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+genetics&amp;title=Genome-wide+association+meta-analysis+of+78,308+individuals+identifies+new+loci+and+genes+influencing+human+intelligence&amp;author=S+Sniekers&amp;volume=49&amp;publication_year=2017&amp;pages=1546-1718&amp;pmid=28825728&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<h4><span id="4-2-shou-jiao-yu-cheng-du-yu-a-si-bo-ge-he-er-tong-gu-du-zheng-gong-xiang-xiang-si-de-yi-chuan-jie-gou-er-qie-zai-asd-shou-shi-zhe-zhong-guan-cha-dao-de-yu-jiao-yu-cheng-du-zheng-xiang-guan-de-deng-wei-ji-yin-guo-duo-jin-xian-yu-a-si-bo-ge-zheng-he-er-tong-zi-bi-zheng">4.2 受教育程度与阿斯伯格和儿童孤独症共享相似的遗传结构，而且在ASD受试者中观察到的与教育程度正相关的等位基因过多（仅限于阿斯伯格症和儿童自闭症）</span><a href="#4-2-shou-jiao-yu-cheng-du-yu-a-si-bo-ge-he-er-tong-gu-du-zheng-gong-xiang-xiang-si-de-yi-chuan-jie-gou-er-qie-zai-asd-shou-shi-zhe-zhong-guan-cha-dao-de-yu-jiao-yu-cheng-du-zheng-xiang-guan-de-deng-wei-ji-yin-guo-duo-jin-xian-yu-a-si-bo-ge-zheng-he-er-tong-zi-bi-zheng" class="header-anchor">#</a></h4><p>关于从ASD亚型中分层构建的诊断亚型（补充表13），还是认知表型在各诊断类别中显示出最强的异质性（教育程度P=2.6×10-11，智商P=3. 4 × 10-8），而神经质（P = 0.0015）、时间型chronotype（P = 0.011）和主观幸福感（P = 0.029）显示出较弱但名义上显著的异质性，而精神分裂症（SCZ）、重度抑郁症（MD）和BMI24在各组间不显著（P &gt; 0.19）（图3）。只有在排除有智力障碍的受试者时，这一模式才稍有减弱（补充图85）。<strong><u>就神经质而言，非典型和其他/非特定PDD病例的PRS明显高于儿童自闭症和阿斯伯格症，P = 0.00013。考虑到每个亚类与每个表型的遗传重叠，强烈拒绝了各亚类同质性的假设（P = 1.6 × 10-11），从而确定这些亚类在遗传结构上确实有差异。</u></strong></p>
<blockquote>
<p>chronotype 时型</p>
<blockquote>
<p>可被理解为睡眠时间类型</p>
</blockquote>
</blockquote>
<p><img src="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/bin/nihms-1015648-f0003.jpg" srcset="/img/loading.gif" alt="An external file that holds a picture, illustration, etc. Object name is nihms-1015648-f0003.jpg"></p>
<blockquote>
<p>图3-<strong>在8个不同表型的不同 ASD 子群体中分析 PRS 负荷</strong></p>
<p>缩写概要：精神分裂症（SCZ），严重抑郁症（MD），教育程度（Edu），人类智力（IQ），主观幸福感（SWB），睡眠时间型，神经质和体重指数（BMI）。</p>
<p>条形图显示了13,076个病例和22,664个对照组的8个归一化分数对不同ASD亚型的多变量回归系数（coefficients from multivariate multivariable regression），并调整了批次和主要成分。这些亚型是分层定义的儿童自闭症（hCHA，N = 3 310）、非典型自闭症（hATA，N = 1 494）、阿斯伯格症（hAsp，N = 4 417）和集合的广泛性障碍发展组（hPDM，N = 3 855）。</p>
<p>主观幸福感、时间型和BMI的得分方向已被调换，以改善图形显示。补充图85中可以看到排除了智障受试者的相应图表，补充图84中可以看到将智力障碍作为一个亚型的图表。对内部训练的ASD评分应用同样的程序，除了预期的ID组（P=0.00027），没有显示系统的异质性（P=0.068）（补充图88）。</p>
<p>使用Pillai检验的线性假设。</p>
</blockquote>
<p>以教育程度为重点，特别是阿斯伯格综合症（P = 2.0 × 10-17）和儿童自闭症（P = 1.5 × 10-5），但其他/未指明的PDD组（P = 0.36）或非典型自闭症（P = 0.13）的PRS显著丰富（图3）。排除有智力障碍的个体后，这种情况只略有改变，非典型自闭症变得名义上显著（P = 0.020）（补充图85）。这些结果表明，<u><strong>受教育程度的遗传结构确实与ASD共享，但在整个疾病谱中程度不一</strong>。我们发现，<strong>在ASD受试者中观察到的与教育程度正相关的等位基因过多，仅限于阿斯伯格症和儿童自闭症，在非典型自闭症和其他/未指明的PDD中均未见此现象。</strong></u></p>
<blockquote>
<ul>
<li>Robinson EB et al. Genetic risk for autism spectrum disorders and neuropsychiatric variation in the general population. <em>Nature Genetics</em> 48, 552–555 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4986048/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/26998691">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Genetic+risk+for+autism+spectrum+disorders+and+neuropsychiatric+variation+in+the+general+population&amp;author=EB+Robinson&amp;volume=48&amp;publication_year=2016&amp;pages=552-555&amp;pmid=26998691&amp;">Google Scholar</a>]</li>
<li>Weiner D et al. Polygenic transmission disequilibrium confirms that common and rare variation act additively to create risk for autism spectrum disorders, <em>Nature Genetic</em> 49 (2017). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5552240/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/28504703">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetic&amp;title=Polygenic+transmission+disequilibrium+confirms+that+common+and+rare+variation+act+additively+to+create+risk+for+autism+spectrum+disorders&amp;author=D+Weiner&amp;volume=49&amp;publication_year=2017&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<h4><span id="4-3-prs-fen-xi-jie-guo-ge-ti-de-asd-feng-xian-yi-ji-liang-yi-lai-de-fang-shi-qu-jue-yu-shu-qian-ge-chang-jian-bian-yi-ti-de-duo-ji-yin-fu-dan-shui-ping-zhe-ke-yi-tong-guo-zeng-jia-asd-xiang-guan-xing-zhuang-de-snp-quan-chong-er-de-dao-jia-qiang">4.3 PRS分析结果-个体的ASD风险以剂量依赖的方式取决于数千个常见变异体的多基因负担水平，这可以通过增加ASD相关性状的SNP权重而得到加强</span><a href="#4-3-prs-fen-xi-jie-guo-ge-ti-de-asd-feng-xian-yi-ji-liang-yi-lai-de-fang-shi-qu-jue-yu-shu-qian-ge-chang-jian-bian-yi-ti-de-duo-ji-yin-fu-dan-shui-ping-zhe-ke-yi-tong-guo-zeng-jia-asd-xiang-guan-xing-zhuang-de-snp-quan-chong-er-de-dao-jia-qiang" class="header-anchor">#</a></h4><p>最后，研究者使用iPSYCH-PGC联合样本中的五组不同的目标和训练样本评估了ASD PRS的预测能力。观察到的PRS解释的平均方差（Nagelkerke’s R2）为2.45%（P = 5.58 × 10-140），基于PRS的集合病例-对照几率OR = 1.33（CI.95% 1.30-1.36）（补充图89和91）。将目标样本分为PRS十等分组，发现OR随着PRS的增加而增加。相对于最低的十分位数，最高PRS的受试者的OR增加到OR=2.80（CI.95% 2.53-3.10）（图4a和补充图92）。利用相关的表型，试图改善对ASD的预测，研究者产生了一个多表型PRS，作为表型特定PRS的加权和（见方法）。正如预期的那样，Nagelkerkes的R2在每个PRS中都有所增加，在全模型中达到最大值3.77%（P=2.03×10-215），在集合分析中，最高分位数的OR=3.57（CI.95% 3.22-3.96）（图4b及补充图93和94）。<strong><u>这些结果表明，个体的ASD风险以剂量依赖的方式取决于数千个常见变异体的多基因负担水平，这可以通过增加ASD相关性状的SNP权重而得到加强。</u></strong></p>
<p><img src="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/bin/nihms-1015648-f0004.jpg" srcset="/img/loading.gif" alt="An external file that holds a picture, illustration, etc. Object name is nihms-1015648-f0004.jpg"></p>
<blockquote>
<p>图4-十分位数图<strong>Decile plots</strong></p>
<p>(13,076个病例和22,664个对照组的每个十等分内的PRS的概率(OR)：</p>
<p>a.十等分图，内部训练的ASD评分的95%-CI(P值阈值为0.1)。</p>
<p> b.十等分图，从a组的ASD评分开始，依次加上主要抑郁症、主观幸福感、精神分裂症、教育程度和睡眠时间型的评分，加权的PRS之和。</p>
<p>在所有情况下，所用分数的P值阈值是具有最高的Nagelkerke’s R2的那个。补充图92和94显示了用于创建这些综合结果的跨留一组的稳定性。</p>
</blockquote>
<h3><span id="5-gong-neng-zhu-shi">5.功能注释</span><a href="#5-gong-neng-zhu-shi" class="header-anchor">#</a></h3><h4><span id="5-1-zai-bao-shou-de-dna-qu-yu-he-h3k4me1-zu-dan-bai-biao-ji-yi-ji-zuo-wei-yi-ge-qun-ti-zai-zhong-shu-shen-jing-xi-tong-cns-xi-bao-lei-xing-zhong-biao-da-de-ji-yin-zhong-yi-chuan-xing-ming-xian-fu-ji-dan-ge-xi-bao-zu-zhi-de-zeng-qiang-zi-xiang-guan-biao-ji-h3k4me1-zai-da-nao-he-shen-jing-yuan-xi-bao-xi-zhong-ming-xian-fu-ji">5.1 在保守的DNA区域和H3K4me1组蛋白标记，以及作为一个群体在中枢神经系统（CNS）细胞类型中表达的基因中，遗传性明显富集+单个细胞/组织的增强子相关标记H3K4me1在大脑和神经元细胞系中明显富集</span><a href="#5-1-zai-bao-shou-de-dna-qu-yu-he-h3k4me1-zu-dan-bai-biao-ji-yi-ji-zuo-wei-yi-ge-qun-ti-zai-zhong-shu-shen-jing-xi-tong-cns-xi-bao-lei-xing-zhong-biao-da-de-ji-yin-zhong-yi-chuan-xing-ming-xian-fu-ji-dan-ge-xi-bao-zu-zhi-de-zeng-qiang-zi-xiang-guan-biao-ji-h3k4me1-zai-da-nao-he-shen-jing-yuan-xi-bao-xi-zhong-ming-xian-fu-ji" class="header-anchor">#</a></h4><p>为了获得GWAS结果的可能生物学基础的信息，研究者进行了一些分析。首先，研究了ASD h2G在功能基因组类别（functional genomic categories）上的划分，以及使用分层LD得分回归stratified LD score regression的细胞类型特定的调控元素。这项分析发现，<strong>在保守的DNA区域和H3K4me1组蛋白标记，以及作为一个群体在中枢神经系统（CNS）细胞类型中表达的基因中，遗传性明显富集</strong>（补充图95和96），这与精神分裂症、严重抑郁症和躁郁症的观察结果一致。分析<strong>单个细胞/组织的增强子相关标记H3K4me1，研究者发现在大脑和神经元细胞系中明显富集</strong>（补充图97）。<strong>在发育中的大脑、胚胎基质、皮层衍生的神经球和胚胎干细胞（ESC）衍生的神经元中观察到最高的富集度</strong>，这与ASD是一种主要<strong>起源于产前的神经发育障碍</strong>相一致，这一点得到了罕见新变体分析数据的支持。</p>
<blockquote>
<p>stratified LD score regression 分层LD回归</p>
<blockquote>
<ul>
<li>Finucane HK et al. Partitioning heritability by functional annotation using genome-wide association summary statistics. <em>Nature Genetics</em> 47, 1228–1235 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4626285/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/26414678">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Partitioning+heritability+by+functional+annotation+using+genome-wide+association+summary+statistics&amp;author=HK+Finucane&amp;volume=47&amp;publication_year=2015&amp;pages=1228-1235&amp;pmid=26414678&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>conserved DNA regions and H3K4me1 histone marks</p>
<blockquote>
<ul>
<li>Shlyueva D, Stampfel G &amp; Stark A Transcriptional enhancers: from properties to genome-wide predictions. <em>Nature Reviews Genetics</em> 15, 272–286 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/24614317">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Reviews+Genetics&amp;title=Transcriptional+enhancers:+from+properties+to+genome-wide+predictions&amp;author=D+Shlyueva&amp;author=G+Stampfel&amp;author=A+Stark&amp;volume=15&amp;publication_year=2014&amp;pages=272-286&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>data from analysis of rare de novo variants</p>
<blockquote>
<ul>
<li>Won H et al. Chromosome conformation elucidates regulatory relationships in developing human brain. <em>Nature</em> 538, 523–527 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5358922/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27760116">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Chromosome+conformation+elucidates+regulatory+relationships+in+developing+human+brain&amp;author=H+Won&amp;volume=538&amp;publication_year=2016&amp;pages=523-527&amp;pmid=27760116&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<h4><span id="5-2-hi-c-shu-ju-fen-xi-fa-xian-asd-de-gong-tong-bian-yi-wei-yu-yu-yu-ce-zai-ren-lei-pi-zhi-fa-yu-zhong-huo-yue-de-diao-kong-yuan-jian-gao-du-fu-ji-de-qu-yu-fen-xi-ta-men-de-ping-jun-biao-da-gui-ji-guan-cha-dao-que-ding-de-asd-hou-xuan-ji-yin-zai-tai-er-pi-zhi-fa-yu-guo-cheng-zhong-biao-xian-chu-zui-gao-de-biao-da-liang">5.2 Hi-C数据分析发现ASD的共同变异位于与预测在人类皮质发育中活跃的调控元件高度富集的区域+分析它们的平均表达轨迹，观察到确定的ASD候选基因在胎儿皮质发育过程中表现出最高的表达量</span><a href="#5-2-hi-c-shu-ju-fen-xi-fa-xian-asd-de-gong-tong-bian-yi-wei-yu-yu-yu-ce-zai-ren-lei-pi-zhi-fa-yu-zhong-huo-yue-de-diao-kong-yuan-jian-gao-du-fu-ji-de-qu-yu-fen-xi-ta-men-de-ping-jun-biao-da-gui-ji-guan-cha-dao-que-ding-de-asd-hou-xuan-ji-yin-zai-tai-er-pi-zhi-fa-yu-guo-cheng-zhong-biao-xian-chu-zui-gao-de-biao-da-liang" class="header-anchor">#</a></h4><p><u><strong>ASD的共同变异位于与预测在人类皮质发育中活跃的调控元件高度富集的区域</strong></u>（补充图95-97）。由于大多数基因调控事件都是通过染色体循环在一定距离内发生的，我们利用发育中的胎儿大脑中生殖区（GZ）和有丝分裂后区皮质板（CP）的Hi-C数据来确定这些变异的潜在目标基因。我们对28个基因座进行了精细测绘，以确定含有可能的因果遗传风险的可信变异集（见方法）。可信的SNPs在胎儿大脑中明显富集了增强器标记（补充图98），再次证实了这些SNPs在大脑发育过程中可能的调节作用。</p>
<blockquote>
<p>利用Hi-C数据确定这些变异的潜在靶基因</p>
<blockquote>
<ul>
<li>Won H et al. Chromosome conformation elucidates regulatory relationships in developing human brain. <em>Nature</em> 538, 523–527 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5358922/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27760116">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Chromosome+conformation+elucidates+regulatory+relationships+in+developing+human+brain&amp;author=H+Won&amp;volume=538&amp;publication_year=2016&amp;pages=523-527&amp;pmid=27760116&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>利用多重关联信号发现可信变异集</p>
<blockquote>
<ul>
<li>Hormozdiari F, Kostem E, Kang EY, Pasaniuc B &amp; Eskin E Identifying causal variants at loci with multiple signals of association. <em>Genetics</em> 198, 497–508 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4196608/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25104515">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Genetics&amp;title=Identifying+causal+variants+at+loci+with+multiple+signals+of+association&amp;author=F+Hormozdiari&amp;author=E+Kostem&amp;author=EY+Kang&amp;author=B+Pasaniuc&amp;author=E+Eskin&amp;volume=198&amp;publication_year=2014&amp;pages=497-508&amp;pmid=25104515&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<p>根据Hi-C的位置或物理接触的证据，380个可信的SNPs（28个位点）可以分配给95个基因（40个蛋白编码），包括分配给9个基因的启动子内的39个SNPs，以及8个基因的蛋白编码序列内的16个SNPs（补充表16，补充图98）。Hi-C确定了86个基因，这些基因在大脑发育过程中与CP或GZ的可信SNP相互作用。在这些基因中，有34个与CP和GZ中的可信SNPs相互作用，这代表了一个高置信度的基因列表。值得注意的例子在图5中说明，并在方框1中强调。<strong>通过分析它们的平均表达轨迹，我们观察到确定的ASD候选基因（补充表16）在胎儿皮质发育过程中表现出最高的表达量，这与发育中的大脑中的调节元素的遗传性富集是一致的</strong>（图5e-g）。有趣的是，<u>ASD的常见和罕见变异都优先影响在皮质发生期表达的基因，突出了遗传风险在这一特定发育时期的潜在时空趋同，尽管该疾病具有深刻的遗传异质性</u>。</p>
<p><img src="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454898/bin/nihms-1015648-f0005.jpg" srcset="/img/loading.gif" alt="An external file that holds a picture, illustration, etc. Object name is nihms-1015648-f0005.jpg"></p>
<blockquote>
<p>图5-<strong>染色质相互作用鉴定 ASD 位点的可能靶基因</strong></p>
<p>a-d： 可信SNPs与1Mb侧翼区域的染色质相互作用图，提供了与可信SNPs物理相互作用的推测候选基因。基因模型基于Gencode v19，推测的目标基因用红色标记；可信SNP的基因组坐标被标记为GWAS；-log10(P-value)，SNP和每个10kbbin之间相互作用的显著性，灰色虚线为FDR=0.01（单边意义检验，计算为在由染色体和距离匹配的拟合Weibull分布下观察到较高接触频率的概率）；皮质板（CP）和生发区（GZ）的拓扑相关域（TAD）边界。 </p>
<p>e-f：ASD候选基因的发育表达轨迹显示在产前时期表达最高。显著性通过t检验（产前和产后样本的N=410和453）。箱形图显示中位数、四分位数范围（IQR），胡须加上IQR到第一和第三四分位数（e和g）。LOESS平滑曲线与实际数据点绘制（f）</p>
<p>g：与其他脑区相比，ASD候选基因在发育中的皮层高度表达。单向方差分析和事后Tukey检验，FDR校正。(产前/产后皮层、海马、杏仁核、纹状体、丘脑和小脑的N=410/453、39/36、33/37、48/34、37/36、32/39）。</p>
</blockquote>
<h2><span id="tao-lun">讨论</span><a href="#tao-lun" class="header-anchor">#</a></h2><p>尽管在过去的30-40年里，ASD的高遗传性已被认可，并且仍然是任何复杂疾病中最高的遗传性之一，这导致了更广泛的表型，其特征是超过1%的人口。虽然早期的GWAS允许估计常见的多基因变异应该解释ASD遗传性的很大一部分，但个别重要的位点仍然难以找到。这被怀疑是由于样本量有限，因为对精神分裂症的研究—具有类似的患病率、遗传性和降低的适应性—和重度抑郁症的研究只有在采用比ASD大五到十倍的样本量时才取得惊人的结果。这项研究最终证实了这一预期，明确显示了重大的 “命中”。</p>
<h3><span id="1-asd-yu-ling-wai-san-lei-biao-xing-ju-you-xiang-dang-cheng-du-de-yi-chuan-xiang-guan-xing-jing-shen-fen-lie-zheng-chong-du-yi-yu-zheng-he-shou-jiao-yu-cheng-du">1.ASD与另外三类表型具有相当程度的遗传相关性：精神分裂症、重度抑郁症和受教育程度</span><a href="#1-asd-yu-ling-wai-san-lei-biao-xing-ju-you-xiang-dang-cheng-du-de-yi-chuan-xiang-guan-xing-jing-shen-fen-lie-zheng-chong-du-yi-yu-zheng-he-shou-jiao-yu-cheng-du" class="header-anchor">#</a></h3><p>在此，研究者利用独特的丹麦资源，结合早期PGC数据的结果，报告了第一个与ASD相关的常见风险变异—比之前最大的发现样本增加了两倍多。其中，仅在ASD中就定义了5个位点，并利用三个相关表型（<strong>精神分裂症、抑郁症和教育程度</strong>）的GWAS结果和最近引入的分析方法MTAG，在更严格的门槛下提出了另外7个位点。全基因组LD得分回归分析以及即使在仅在ASD中定义的位点中，在这些其他性状扫描中也有额外的证据，这都表明ASD的多基因结构与成人精神疾病的风险和较高的教育程度和智力有明显的共享。应该注意的是，MTAG分析是以三个配对分析的方式进行的。这样我们就避免了如果我们一次运行三个或四个相关的表型可能出现的复杂的相互作用。事实上，尽管次要的汇总统计数字来自于大型的、高能的研究，但研究者得到的是对这些统计数字贡献的相对适度的权重，<u><strong>因为遗传相关性是适度的。最大的权重是精神分裂症的0.27，其次是重度抑郁症的0.24，以及教育程度的0.11</strong></u>。此外，估计最坏情况下的FDR只有0.021，比ASD GWAS单独的FDR只提高了0.001。因此，MTAG发现的所有位点都具有可接受的确定性，并且仅ASD就有大量的贡献（表1a, bb和补充表6）。研究者的期望是，随着样本量的大幅增加，大多数或所有这样的位点将可能在未来的仅有ASD的GWAS中被发现—然而，鉴于这些方法是多么新，这些特定变体的精确表型后果有待于所有这些性状GWAS的扩展。</p>
<h3><span id="2-dui-asd-yang-ben-hua-fen-wei-bu-tong-zi-lei-qun-guan-cha-dao-yu-qi-ta-xing-zhuang-de-yi-chuan-chong-die-de-qiang-lie-yi-zhi-xing-yu-jiao-gao-zhi-li-he-jiao-yu-cheng-du-xiang-guan-de-deng-wei-ji-yin-de-guo-liang-zhi-zai-jiao-gao-gong-neng-lei-bie-te-bie-shi-a-si-bo-ge-zong-he-zheng-he-mei-you-he-bing-zhi-li-zhang-ai-de-ge-ti-zhong-guan-cha-dao-qi-ta-wei-ming-que-de-pdd-he-fei-dian-xing-zi-bi-zheng-de-shen-jing-zhi-de-prs-ye-ming-xian-gao-yu-er-tong-zi-bi-zheng-he-a-si-bo-ge-zheng">2.对ASD样本划分为不同子类群，观察到与其他形状的遗传重叠的强烈异质性：与较高智力和教育程度相关的等位基因的过量只在较高功能类别（特别是阿斯伯格综合征和没有合并智力障碍的个体）中观察到，其他/未明确的PDD和非典型自闭症的神经质的PRS也明显高于儿童自闭症和阿斯伯格症</span><a href="#2-dui-asd-yang-ben-hua-fen-wei-bu-tong-zi-lei-qun-guan-cha-dao-yu-qi-ta-xing-zhuang-de-yi-chuan-chong-die-de-qiang-lie-yi-zhi-xing-yu-jiao-gao-zhi-li-he-jiao-yu-cheng-du-xiang-guan-de-deng-wei-ji-yin-de-guo-liang-zhi-zai-jiao-gao-gong-neng-lei-bie-te-bie-shi-a-si-bo-ge-zong-he-zheng-he-mei-you-he-bing-zhi-li-zhang-ai-de-ge-ti-zhong-guan-cha-dao-qi-ta-wei-ming-que-de-pdd-he-fei-dian-xing-zi-bi-zheng-de-shen-jing-zhi-de-prs-ye-ming-xian-gao-yu-er-tong-zi-bi-zheng-he-a-si-bo-ge-zheng" class="header-anchor">#</a></h3><p>在大多数GWAS研究中，几乎没有证据表明不同表型亚组之间存在关联的异质性。然而，在这项研究中，<strong><u>当我们把ASD样本分成不同的子群时，我们看到了与其他性状的遗传重叠的强烈异质性</u></strong>。特别是，<strong>与较高智力和教育程度相关的等位基因的过量只在较高功能类别（特别是阿斯伯格综合征和没有合并智力障碍的个体）中观察到，而在其他/未明确的PDD和智力障碍类别中没有观察到</strong>。这让人联想到，而且从逻辑上讲，<u>自发突变在后面这些类别中的作用要大得多，特别是在已知对智力障碍/发育迟缓的队列有更大影响的基因中</u>。有趣的是，<u>其他/未明确的PDD和非典型自闭症的神经质的PRS也明显高于儿童自闭症和阿斯伯格症</u>。观察到的这些不同的富集图谱为ASD亚型之间的异质性和质量不同的遗传结构提供了证据，这应该为旨在确定ASD病因和疾病机制的未来研究提供参考。</p>
<blockquote>
<p>自发突变对智力障碍/发育迟缓的影响</p>
<blockquote>
<ul>
<li>Kosmicki JA et al. Refining the role of de novo protein-truncating variants in neurodevelopmental disorders by using population reference samples. <em>Nature genetics</em> 49, 504–510 (2017). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5496244/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/28191890">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+genetics&amp;title=Refining+the+role+of+de+novo+protein-truncating+variants+in+neurodevelopmental+disorders+by+using+population+reference+samples&amp;author=JA+Kosmicki&amp;volume=49&amp;publication_year=2017&amp;pages=504-510&amp;pmid=28191890&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<h3><span id="3-you-zhi-li-zhang-ai-he-wu-zhi-li-zhang-ai-de-asd-bing-li-zhi-jian-gu-ji-de-snp-yi-chuan-xing-cun-zai-hen-da-chai-yi-er-a-si-bo-ge-zheng-de-yi-chuan-xing-zui-gao-zhe-wei-chang-qi-de-guan-cha-ti-gong-liao-yi-chuan-zheng-ju-pu-tong-bian-yi-ti-zai-gao-gong-neng-asd-bing-li-zhong-de-gong-xian-ke-neng-geng-wei-tu-chu">3.有智力障碍和无智力障碍的ASD病例之间估计的SNP遗传性存在很大差异，而阿斯伯格症的遗传性最高，这为长期的观察提供了遗传证据；普通变异体在高功能ASD病例中的贡献可能更为突出。</span><a href="#3-you-zhi-li-zhang-ai-he-wu-zhi-li-zhang-ai-de-asd-bing-li-zhi-jian-gu-ji-de-snp-yi-chuan-xing-cun-zai-hen-da-chai-yi-er-a-si-bo-ge-zheng-de-yi-chuan-xing-zui-gao-zhe-wei-chang-qi-de-guan-cha-ti-gong-liao-yi-chuan-zheng-ju-pu-tong-bian-yi-ti-zai-gao-gong-neng-asd-bing-li-zhong-de-gong-xian-ke-neng-geng-wei-tu-chu" class="header-anchor">#</a></h3><p><u><strong>有智力障碍和无智力障碍的ASD病例之间估计的SNP遗传性存在很大差异，而阿斯伯格症的遗传性最高，这为长期的观察提供了遗传证据</strong></u>。特别是，这与以下观察相吻合：<strong>在有智力障碍的ASD病例中，与没有合并智力障碍的病例相比，更经常观察到新的变体；智商与精神障碍家族史呈正相关；严重智力障碍（包括许多赋予ASD高风险的综合征）的遗传性远低于轻度智力障碍、一般智力和ASD的观察。因此，我们的数据表明，普通变异体在高功能ASD病例中的贡献可能更为突出，如阿斯伯格综合征</strong>，这也许并不令人惊讶。</p>
<blockquote>
<p>智力与精神障碍家族史呈正相关</p>
<blockquote>
<ul>
<li>Robinson EB et al. Autism spectrum disorder severity reflects the average contribution of de novo and familial influences. <em>PNAS</em> 111, 15161–15165 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4210299/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25288738">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=PNAS&amp;title=Autism+spectrum+disorder+severity+reflects+the+average+contribution+of+de+novo+and+familial+influences&amp;author=EB+Robinson&amp;volume=111&amp;publication_year=2014&amp;pages=15161-15165&amp;pmid=25288738&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>轻度智力障碍的遗传性</p>
<blockquote>
<ul>
<li>Reichenberg A et al. Discontinuity in the genetic and environmental causes of the intellectual disability spectrum. <em>PNAS</em> 113, 1098–1103 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4743770/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/26711998">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=PNAS&amp;title=Discontinuity+in+the+genetic+and+environmental+causes+of+the+intellectual+disability+spectrum&amp;author=A+Reichenberg&amp;volume=113&amp;publication_year=2016&amp;pages=1098-1103&amp;pmid=26711998&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>一般智力的遗传性</p>
<blockquote>
<ul>
<li>Polderman TJC et al. Meta-analysis of the heritability of human traits based on fifty years of twin studies. <em>Nature genetics</em> 47, 702–709 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25985137">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+genetics&amp;title=Meta-analysis+of+the+heritability+of+human+traits+based+on+fifty+years+of+twin+studies&amp;author=TJC+Polderman&amp;volume=47&amp;publication_year=2015&amp;pages=702-709&amp;pmid=25985137&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<h3><span id="4-hi-c-fen-xi-he-da-nao-zhuan-lu-zu-shu-ju-yi-ji-dan-ji-yin-zuo-de-fen-xi-qiang-diao-liao-yu-da-nao-fa-yu-he-shen-jing-yuan-gong-neng-xiang-guan-de-guo-cheng-de-can-yu-tong-shi-zai-dui-xin-de-he-han-jian-bian-yi-ti-de-yan-jiu-zhong-que-ding-liao-yi-xie-ji-yin-yu-asd-feng-xian-you-guan">4.Hi-C分析和大脑转录组数据以及单基因座的分析强调了与大脑发育和神经元功能相关的过程的参与；同时在对新的和罕见变异体的研究中，确定了一些基因与ASD风险有关。</span><a href="#4-hi-c-fen-xi-he-da-nao-zhuan-lu-zu-shu-ju-yi-ji-dan-ji-yin-zuo-de-fen-xi-qiang-diao-liao-yu-da-nao-fa-yu-he-shen-jing-yuan-gong-neng-xiang-guan-de-guo-cheng-de-can-yu-tong-shi-zai-dui-xin-de-he-han-jian-bian-yi-ti-de-yan-jiu-zhong-que-ding-liao-yi-xie-ji-yin-yu-asd-feng-xian-you-guan" class="header-anchor">#</a></h3><p>研究者用补充的功能基因组学数据进一步探讨了这些结果的功能意义，包括<u><strong>胎儿大脑的Hi-C分析和大脑转录组数据</strong></u>。全基因组范围的分析（分区$h^2_G$（补充图95-97）和脑转录组富集（图5e-g））以及单个基因座的分析（图5a-d，方框1）强调了<u>与大脑发育和神经元功能相关的过程的参与</u>。值得注意的是，<u><strong>在对新的和罕见的变异体的研究中，一些位于已确定的基因与ASD风险有关（方框1，补充表4），包括PTBP2、CADPS和KMT2E，这些基因在Hi-C分析中被发现与可信的SNPs相互作用（PTBP2、CADPS）或包含一个功能丧失的可信SNP（KMT2E）</strong></u>。有趣的是，在自闭症病例中发现了CADPS的姐妹基因CADPS2的异常剪接，其功能几乎相同，而且Cadps2基因敲除小鼠表现出行为异常，与自闭症的转化相关。PTBP2编码一个神经元剪接因子，在被诊断为ASD的人的大脑中已经发现了替代剪接的改变。</p>
<blockquote>
<p>Cadps2基因敲除小鼠表现出行为异常，与自闭症的转化相关</p>
<blockquote>
<ul>
<li>Sadakata T et al. Autistic-like phenotypes in cadps2-knockout mice and aberrant cadps2 splicing in autistic patients. <em>The Journal of clinical investigation</em> 117, 931–943 (2007). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1821065/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/17380209">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=The+Journal+of+clinical+investigation&amp;title=Autistic-like+phenotypes+in+cadps2-knockout+mice+and+aberrant+cadps2+splicing+in+autistic+patients&amp;author=T+Sadakata&amp;volume=117&amp;publication_year=2007&amp;pages=931-943&amp;pmid=17380209&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>PTBP2编码一个神经元剪接因子，在被诊断为ASD的人的大脑中已经发现了替代剪接的改变</p>
<blockquote>
<ul>
<li>Parikshak NN et al. Genome-wide changes in lncrna, splicing, and regional gene expression patterns in autism. <em>Nature</em> 540, 423–427 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7102905/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27919067">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Genome-wide+changes+in+lncrna,+splicing,+and+regional+gene+expression+patterns+in+autism&amp;author=NN+Parikshak&amp;volume=540&amp;publication_year=2016&amp;pages=423-427&amp;pmid=27919067&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<p>总之，研究者已经建立了第一套强有力的ASD常见变异关联，并开始奠定基础，通过这些基础，ASD的生物学和相关表型将不可避免地被更好地阐明。</p>
<h2><span id="fang-fa">方法</span><a href="#fang-fa" class="header-anchor">#</a></h2><h3><span id="1-ipsych-yang-ben">1.iPSYCH样本</span><a href="#1-ipsych-yang-ben" class="header-anchor">#</a></h3><p>iPSYCH ASD样本是基于人口的病例队列样本的一部分，从基线队列中提取，包括1981年5月1日至2005年12月31日期间在丹麦出生的所有儿童。符合条件的是已知母亲所生且在一岁生日时居住在丹麦的单胞胎。病例从丹麦精神病学中央研究登记册（DPCRR）12中确定，其中包括所有在丹麦精神病院（从1969年起）以及精神病门诊（从1995年起）接受治疗的人的数据。病例在2013年或更早被精神科医生根据ICD10诊断为ASD，包括诊断为儿童自闭症（ICD10代码F84.0）、非典型自闭症（F84.1）、阿斯伯格综合症（F84.5）、其他广泛性发育障碍（F84.8）和广泛性发育障碍，未指明（F84.9）。作为对照，我们从符合条件的儿童集合中随机抽取样本，不包括到2013年有ASD诊断的儿童。</p>
<blockquote>
<ul>
<li>Pedersen CB et al. The ipsych2012 case-cohort sample: new directions for unravelling the genetic and environmental architecture of severe mental disorders. <em>Molecular Psychiatry</em> 23 (2017). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5754466/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/28924187">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Molecular+Psychiatry&amp;title=The+ipsych2012+case-cohort+sample:+new+directions+for+unravelling+the+genetic+and+environmental+architecture+of+severe+mental+disorders&amp;author=CB+Pedersen&amp;volume=23&amp;publication_year=2017&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>这些样本使用唯一的个人识别号码与Statens Serum Institute (SSI)的丹麦新生儿筛查生物库(DNSB)相联系，在那里从Guthrie卡中提取DNA，并按照以前的描述进行全基因组三重扩增。根据制造商的说明，使用Illumina公司的PsychChip阵列（CA，San Diego，USA）在哈佛大学和麻省理工学院的Broad研究所（Cambridge，MA，USA）进行基因分型。小等位基因频率（MAF）&gt;0.01的标记的基因型调用是通过合并GenCall和Birdseed的调用集进行的，而不太频繁的变体则用zCall100调用。基因分型和数据处理分23次进行。</p>
<blockquote>
<p>DNA was extracted from Guthrie cards and whole genome amplified in triplicates as described previously</p>
<blockquote>
<ul>
<li><p>Hollegaard M et al. Robustness of genome-wide scanning using archived dried blood spot samples as a dna source. <em>BMC Genet</em> 12, 58 (2011). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3142526/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/21726430">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=BMC+Genet&amp;title=Robustness+of+genome-wide+scanning+using+archived+dried+blood+spot+samples+as+a+dna+source&amp;author=M+Hollegaard&amp;volume=12&amp;publication_year=2011&amp;pages=58&amp;pmid=21726430&amp;">Google Scholar</a>]</p>
</li>
<li><p>Børglum AD et al. Genome-wide study of association and interaction with maternal cytomegalovirus infection identifies new schizophrenia loci. <em>Molecular Psychiatry</em> 19, 325–333 (2014). Published online 29 January 2013. [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3932405/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/23358160">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Molecular+Psychiatry&amp;title=Genome-wide+study+of+association+and+interaction+with+maternal+cytomegalovirus+infection+identifies+new+schizophrenia+loci&amp;author=AD+Børglum&amp;volume=19&amp;publication_year=2014&amp;pages=325-333&amp;pmid=23358160&amp;">Google Scholar</a>]</p>
</li>
</ul>
</blockquote>
<p>GenCall</p>
<blockquote>
<ul>
<li>Illumina gencall data analysis software <em>Tech. Rep</em>, Illumina, Inc. (2005). [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?title=Tech.+Rep&amp;publication_year=2005&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>Birdseed</p>
<blockquote>
<ul>
<li>Korn JM et al. Integrated genotype calling and association analysis of snps, common copy number polymorphisms and rare cnvs. <em>Nature genetics</em> 40, 1253–1260 (2008). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2756534/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/18776909">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+genetics&amp;title=Integrated+genotype+calling+and+association+analysis+of+snps,+common+copy+number+polymorphisms+and+rare+cnvs&amp;author=JM+Korn&amp;volume=40&amp;publication_year=2008&amp;pages=1253-1260&amp;pmid=18776909&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<p>所有 iPSYCH 样本的分析以及与 PGC 样本的联合分析都在丹麦安全的国家基因组计算机高性能集群上进行。</p>
<h3><span id="2-psychiatric-genomic-consortium-pgc-yang-ben">2.Psychiatric Genomic Consortium (PGC)样本</span><a href="#2-psychiatric-genomic-consortium-pgc-yang-ben" class="header-anchor">#</a></h3><p>简而言之，五个队列向样本提供了基因型（N表示有基因型的三人组的数量）。Geschwind自闭症卓越中心【The Geschwind Autism Center of Excellence】（ACE；N=391）、自闭症基因组计划（AGP；N=2,272）、自闭症遗传资源交流中心（AGRE；N=974）、NIMH资料库、蒙特利尔/波士顿收集（MONBOS；N=1,396）和西蒙斯单倍体收集（SSC；N=2,231）。三者作为病例和假性对照进行分析。该样本的详细描述可在PGC网站上找到，更多的细节可在Anney等人的文章中找到。对PGC基因型的分析是在荷兰高性能计算中心SURFsara的计算机集群LISA上进行的。</p>
<blockquote>
<p>the Autism Genome Project</p>
<blockquote>
<ul>
<li>Anney R et al. A genome-wide scan for common alleles affecting risk for autism. <em>Hum Mol Genet</em> 19, 4072–4082 (2010). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2947401/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/20663923">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Hum+Mol+Genet&amp;title=A+genome-wide+scan+for+common+alleles+affecting+risk+for+autism&amp;author=R+Anney&amp;volume=19&amp;publication_year=2010&amp;pages=4072-4082&amp;pmid=20663923&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>the Autism Genetic Resource Exchange</p>
<blockquote>
<ul>
<li>Lajonchere CM &amp; Consortium, A. Changing the landscape of autism research: the autism genetic resource exchange. <em>Neuron</em> 68, 187–191 (2010). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3004528/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/20955925">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Neuron&amp;title=Changing+the+landscape+of+autism+research:+the+autism+genetic+resource+exchange&amp;author=CM+Lajonchere&amp;volume=68&amp;publication_year=2010&amp;pages=187-191&amp;pmid=20955925&amp;">Google Scholar</a>]</li>
<li>Geschwind DH et al. The autism genetic resource exchange: a resource for the study of autism and related neuropsychiatric conditions. <em>American journal of human genetics</em> 69, 463–466 (2001). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1235320/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/11452364">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=American+journal+of+human+genetics&amp;title=The+autism+genetic+resource+exchange:+a+resource+for+the+study+of+autism+and+related+neuropsychiatric+conditions&amp;author=DH+Geschwind&amp;volume=69&amp;publication_year=2001&amp;pages=463-466&amp;pmid=11452364&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>the Montreal/Boston Collection</p>
<blockquote>
<ul>
<li>Gauthier J et al. Autism spectrum disorders associated with x chromosome markers in french-canadian males. <em>Molecular psychiatry</em> 11, 206–213 (2006). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/16261168">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Molecular+psychiatry&amp;title=Autism+spectrum+disorders+associated+with+x+chromosome+markers+in+french-canadian+males&amp;author=J+Gauthier&amp;volume=11&amp;publication_year=2006&amp;pages=206-213&amp;pmid=16261168&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>the Simons Simplex Collection</p>
<blockquote>
<ul>
<li>Fischbach GD &amp; Lord C The simons simplex collection: a resource for identification of autism genetic risk factors. <em>Neuron</em> 68, 192–195 (2010). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/20955926">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Neuron&amp;title=The+simons+simplex+collection:+a+resource+for+identification+of+autism+genetic+risk+factors&amp;author=GD+Fischbach&amp;author=C+Lord&amp;volume=68&amp;publication_year=2010&amp;pages=192-195&amp;pmid=20955926&amp;">Google Scholar</a>]</li>
<li>Chaste P et al. A genome-wide association study of autism using the simons simplex collection: Does reducing phenotypic heterogeneity in autism increase genetic homogeneity? <em>Biol Psychiatry</em> 77, 775–784 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4379124/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25534755">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Biol+Psychiatry&amp;title=A+genome-wide+association+study+of+autism+using+the+simons+simplex+collection:+Does+reducing+phenotypic+heterogeneity+in+autism+increase+genetic+homogeneity?&amp;author=P+Chaste&amp;volume=77&amp;publication_year=2015&amp;pages=775-784&amp;pmid=25534755&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<p>作为p值小于10-6的位点的后续工作，研究者要求对北欧和东欧血统的五个样本进行查询，这些样本共有2,119个病例和142,379个对照。BUPGEN（挪威：164例和656例对照），PAGES（瑞典：926例和3841例对照，不属于上述PGC样本），芬兰自闭症病例对照研究（芬兰：159例和526例对照），deCODE（冰岛574例和136968例对照；东欧：296例和388例对照）。详见补充说明。</p>
<h3><span id="3-tong-ji-fen-xi">3.统计分析</span><a href="#3-tong-ji-fen-xi" class="header-anchor">#</a></h3><p>除非另有说明，所有的统计检验都是双面two.sided的。软件版本等可以在生命科学报告摘要(Life Sciences Reporting Summary)中找到。</p>
<h4><span id="3-1-gwas-fen-xi">3.1 GWAS 分析</span><a href="#3-1-gwas-fen-xi" class="header-anchor">#</a></h4><p>Ricopili是由精神病基因组学联盟（PGC）统计分析组开发的pipeline，用于质量控制、推断、原理成分分析（PCA）和主要关联分析。详见补充说明。在iPSYCH的情况下，数据在23个基因分型批次中被分别处理，在PGC样本中，每个研究都被分别处理。用SHAPEIT进行分型，用IMPUTE2进行推算，以1000基因组计划第3109期（1kGP3）的单倍型作为参考。</p>
<blockquote>
<p>Ricopili</p>
<blockquote>
<ul>
<li>Schizophrenia Working Group of the Psychiatric Genomics Consortium. Biological insights from 108 schizophrenia-associated genetic loci. <em>Nature</em> 511, 421–427 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4112379/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25056061">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Biological+insights+from+108+schizophrenia-associated+genetic+loci&amp;volume=511&amp;publication_year=2014&amp;pages=421-427&amp;pmid=25056061&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>IMPUTE2</p>
<blockquote>
<ul>
<li>Howie B, Fuchsberger C, Stephens M, Marchini J &amp; Abecasis GR Fast and accurate genotype imputation in genome-wide association studies through pre-phasing. <em>Nat Genet</em> 44, 955–959 (2012). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3696580/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/22820512">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nat+Genet&amp;title=Fast+and+accurate+genotype+imputation+in+genome-wide+association+studies+through+pre-phasing&amp;author=B+Howie&amp;author=C+Fuchsberger&amp;author=M+Stephens&amp;author=J+Marchini&amp;author=GR+Abecasis&amp;volume=44&amp;publication_year=2012&amp;pages=955-959&amp;pmid=22820512&amp;">Google Scholar</a>]</li>
<li>Howie BN, Donnelly P &amp; Marchini J A flexible and accurate genotype imputation method for the next generation of genome-wide association studies. <em>PLoS Genet</em> 5, e1000529 (2009). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2689936/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/19543373">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=PLoS+Genet&amp;title=A+flexible+and+accurate+genotype+imputation+method+for+the+next+generation+of+genome-wide+association+studies&amp;author=BN+Howie&amp;author=P+Donnelly&amp;author=J+Marchini&amp;volume=5&amp;publication_year=2009&amp;pages=e1000529&amp;pmid=19543373&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>SHAPEIT</p>
<blockquote>
<ul>
<li>Delaneau O, Marchini J &amp; Zagury J-F A linear complexity phasing method for thousands of genomes. <em>Nature methods</em> 9, 179–181 (2011). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/22138821">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+methods&amp;title=A+linear+complexity+phasing+method+for+thousands+of+genomes&amp;author=O+Delaneau&amp;author=J+Marchini&amp;author=J-F+Zagury&amp;volume=9&amp;publication_year=2011&amp;pages=179-181&amp;pmid=22138821&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<p>在排除了高连锁不平衡（LD）的区域后，基因型被修剪成一组大约30k的标记物。详见补充说明。使用PLINK的身份状态分析，确定πˆ&gt;0.2的受试者对，随机排除每对受试者中的一个受试者（倾向于保留病例）。使用smartPCA进行PCA。在iPSYCH中，欧洲血统的子样本被选为PC1-3空间中的一个椭圆体，其中心和比例使用子样本的平均值和8个标准差，这些子样本的父母和祖父母都已知出生在丹麦（N=31,500）。在PGC样本中，欧洲（CEU）子集的选择采用欧几里得距离测量法，由前三个主成分中的每一个解释的方差加权。与CEU和意大利托斯卡尼（TSI）HapMap参考人群的距离超过10个标准差的个体被排除。对剩余的13,076个病例和22,664个对照进行了二次PCA，为关联分析提供协变量。iPSYCH样本的数据生成流程中的受试者数量见补充表1。</p>
<blockquote>
<p>high linkage disequilibrium (LD)</p>
<blockquote>
<ul>
<li>Price AL et al. Long-range ld can confound genome scans in admixed populations. <em>Am J Hum Genet</em> 83, 132–5; author reply 135–9 (2008). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2443852/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/18606306">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Am+J+Hum+Genet&amp;title=Long-range+ld+can+confound+genome+scans+in+admixed+populations&amp;author=AL+Price&amp;volume=83&amp;publication_year=2008&amp;pages=132-5&amp;pmid=18606306&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>PLINK</p>
<blockquote>
<ul>
<li>Chang CC et al. Second-generation plink: rising to the challenge of larger and richer datasets. <em>GigaScience</em> 4, 7 (2015). Original DateCompleted: 20150227. [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4342193/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25722852">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=GigaScience&amp;title=Second-generation+plink:+rising+to+the+challenge+of+larger+and+richer+datasets&amp;author=CC+Chang&amp;volume=4&amp;publication_year=2015&amp;pages=7&amp;pmid=25722852&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>smartPCA</p>
<blockquote>
<ul>
<li><p>Patterson N, Price AL &amp; Reich D Population structure and eigenanalysis. <em>PLoS Genet</em> 2, e190 (2006). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1713260/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/17194218">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=PLoS+Genet&amp;title=Population+structure+and+eigenanalysis&amp;author=N+Patterson&amp;author=AL+Price&amp;author=D+Reich&amp;volume=2&amp;publication_year=2006&amp;pages=e190&amp;pmid=17194218&amp;">Google Scholar</a>]</p>
</li>
<li><p>Price AL et al. Principal components analysis corrects for stratification in genome-wide association studies. <em>Nat Genet</em> 38, 904–909 (2006). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/16862161">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nat+Genet&amp;title=Principal+components+analysis+corrects+for+stratification+in+genome-wide+association+studies&amp;author=AL+Price&amp;volume=38&amp;publication_year=2006&amp;pages=904-909&amp;pmid=16862161&amp;">Google Scholar</a>]</p>
</li>
</ul>
</blockquote>
</blockquote>
<p>关联分析是通过应用PLINK 1.9对归因的剂量数据（归因概率P(A1A2)+2P(A1A1)之和）进行的。在iPSYCH中，我们将前四个主成分（PC）作为协变量，以及除此之外的任何PC，与样本中的ASD显著相关，而来自PGC三人组的病例-伪对照不需要PC协变量。通过使用METAL（2010年7月版）采用反方差加权固定效应模型，对iPSYCH和iPSYCH与PGC的综合结果进行了批量分析和研究分析。对X染色体的男性和女性分别进行分析，然后一起进行荟萃分析。随后，我们应用质量过滤器，只允许归因信息得分≥0.7、MAF≥0.01和有效样本量（见补充说明）至少为研究最大值70%的标记。检验统计量的偏差可归因于隐性亲缘关系和种群分层而非多基因性的程度是由LD评分回归19（LDSC）中的截距来衡量的，即(intercept-1)和(mean(χ2)-1)的比率。</p>
<blockquote>
<p>METAL</p>
<blockquote>
<ul>
<li>Willer CJ, Li Y &amp; Abecasis GR Metal: fast and efficient meta-analysis of genomewide association scans. <em>Bioinformatics (Oxford, England)</em> 26, 2190–2191 (2010). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2922887/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/20616382">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Bioinformatics+(Oxford,+England">Google Scholar</a>&amp;title=Metal:+fast+and+efficient+meta-analysis+of+genomewide+association+scans&amp;author=CJ+Willer&amp;author=Y+Li&amp;author=GR+Abecasis&amp;volume=26&amp;publication_year=2010&amp;pages=2190-2191&amp;)]</li>
</ul>
</blockquote>
<p>an inverse variance weighted fixed effect model</p>
<blockquote>
<ul>
<li>Begum F, Ghosh D, Tseng GC &amp; Feingold E Comprehensive literature review and statistical considerations for gwas meta-analysis. <em>Nucleic acids research</em> 40, 3777–3784 (2012). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3351172/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/22241776">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nucleic+acids+research&amp;title=Comprehensive+literature+review+and+statistical+considerations+for+gwas+meta-analysis&amp;author=F+Begum&amp;author=D+Ghosh&amp;author=GC+Tseng&amp;author=E+Feingold&amp;volume=40&amp;publication_year=2012&amp;pages=3777-3784&amp;pmid=22241776&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>LD score regression (LDSC)</p>
<blockquote>
<ul>
<li>Bulik-Sullivan BK et al. Ld score regression distinguishes confounding from polygenicity in genome-wide association studies. <em>Nat Genet</em> 47, 291–295 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495769/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25642630">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nat+Genet&amp;title=Ld+score+regression+distinguishes+confounding+from+polygenicity+in+genome-wide+association+studies&amp;author=BK+Bulik-Sullivan&amp;volume=47&amp;publication_year=2015&amp;pages=291-295&amp;pmid=25642630&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<p>MTAG被应用于标准设置。iPSYCH-PGC荟萃分析的汇总统计与主要抑郁症（不包括丹麦取样，但包括23andMe的汇总统计；111,902例，312,113例对照，平均χ2=1. 477），精神分裂症（也不包括丹麦样本；34,129例，45,512例对照，平均χ2=1.804）和教育程度（328,917例，平均χ2=1.648）。这些研究比ASD扫描具有更强的统计能力，但在MTAG的背景下，遗传相关性不大，所以在MTAG分析中赋予次要表型的权重仍然比较低（不高于0.27）。最大的FDR是按照MTAG论文中的建议估计的。详见补充说明。结果是成团的，我们通过选择那些在iPSYCH-PGC荟萃分析或有随访样本的荟萃分析中显著性为5×10-8或在三个MTAG分析中任何一个显著性为1.67×l0-8的位点来突出感兴趣的位点。在为不同的分析创建曼哈顿图时，由这五项分析中每个标记的最小p值组成的综合GWAS被用作背景，显示了最大限度的实现和个别分析对其的贡献。</p>
<blockquote>
<p>MTAG</p>
<blockquote>
<ul>
<li>Turley P et al. Multi-trait analysis of genome-wide association summary statistics using mtag. <em>Nature Genetics</em> 50, 229–237 (2018). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5805593/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/29292387">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Multi-trait+analysis+of+genome-wide+association+summary+statistics+using+mtag&amp;author=P+Turley&amp;volume=50&amp;publication_year=2018&amp;pages=229-237&amp;pmid=29292387&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<h4><span id="3-2-ji-yu-ji-yin-de-guan-lian-he-ji-yin-ji-fen-xi">3.2 基于基因的关联和基因集分析</span><a href="#3-2-ji-yu-ji-yin-de-guan-lian-he-ji-yin-ji-fen-xi" class="header-anchor">#</a></h4><p>MAGMA 1.0675被应用于ASD GWAS的汇总统计，以测试基于基因的关联。使用NCBI 37.3基因定义并将分析限制在位于转录区的SNP上，用-log(SNP p-value)之和作为测试统计量来测试平均SNP关联。所得的基于基因的p值被进一步用于MAGMA的竞争性基因组富集分析。其中一项分析探索了来自Parikshak等人2013年的M13、M16和M17候选基因组、来自外显子聚集联盟数据的受限、功能缺失不耐受基因（pLI&gt;0.9）（详见补充说明），以及Sanders等人的自闭症罕见变异研究中发现的基因组和SPARK联盟的策划基因列表。另一个是对MsigDB 6.083中的Gene Ontology集81,82的分子功能的不可知分析。我们只分析了广泛的MHC区域（hg19:chr6:25-35M）以外的基因，并且只包括有10-1000个基因的基因集。来自Sanders等人和SPARK的基因集只包括MHC中的一个基因，为了尽可能真实地反映该基因集，我们免除了MHC的排除。对所有具有明显富集的基因集进行了检查，以确保该信号不是由一个或几个具有密切LD的多个基因的相关基因组驱动的。</p>
<blockquote>
<p>MAGMA</p>
<blockquote>
<ul>
<li>de Leeuw CA, Mooij JM, Heskes T &amp; Posthuma D Magma: generalized gene-set analysis of gwas data. <em>PLoS Comput Biol</em> 11, e1004219 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4401657/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25885710">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=PLoS+Comput+Biol&amp;title=Magma:+generalized+gene-set+analysis+of+gwas+data&amp;author=CA+de+Leeuw&amp;author=JM+Mooij&amp;author=T+Heskes&amp;author=D+Posthuma&amp;volume=11&amp;publication_year=2015&amp;pages=e1004219&amp;pmid=25885710&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>One analysis explored the candidate sets M13, M16 and M17 from Parikshak et al. 2013</p>
<blockquote>
<ul>
<li>Parikshak NN et al. Integrative functional genomic analyses implicate specific molecular pathways and circuits in autism. <em>Cell</em> 155, 1008–1021 (2013). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3934107/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/24267887">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Cell&amp;title=Integrative+functional+genomic+analyses+implicate+specific+molecular+pathways+and+circuits+in+autism&amp;author=NN+Parikshak&amp;volume=155&amp;publication_year=2013&amp;pages=1008-1021&amp;pmid=24267887&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p> loss-of-function intolerant genes (pLI &gt; 0.9)</p>
<blockquote>
<ul>
<li><p>Samocha KE et al. A framework for the interpretation of de novo mutation in human disease. <em>Nat Genet</em> 46, 944–950 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25086666">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nat+Genet&amp;title=A+framework+for+the+interpretation+of+de+novo+mutation+in+human+disease&amp;author=KE+Samocha&amp;volume=46&amp;publication_year=2014&amp;pages=944-950&amp;pmid=25086666&amp;">Google Scholar</a>]</p>
</li>
<li><p>Lek M et al. Analysis of protein-coding genetic variation in 60,706 humans. <em>Nature</em> 536, 285–291 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5018207/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27535533">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Analysis+of+protein-coding+genetic+variation+in+60,706+humans&amp;author=M+Lek&amp;volume=536&amp;publication_year=2016&amp;pages=285-291&amp;pmid=27535533&amp;">Google Scholar</a>]</p>
</li>
</ul>
</blockquote>
<p>Sanders等人的自闭症罕见变异研究中发现的基因组和SPARK联盟的策划基因列表</p>
<blockquote>
<ul>
<li><p>Sanders S et al. Insights into autism spectrum disorder genomic architecture and biology from 71 risk loci. <em>Neuron</em> 87, 1215–1233 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4624267/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/26402605">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Neuron&amp;title=Insights+into+autism+spectrum+disorder+genomic+architecture+and+biology+from+71+risk+loci&amp;author=S+Sanders&amp;volume=87&amp;publication_year=2015&amp;pages=1215-1233&amp;pmid=26402605&amp;">Google Scholar</a>]</p>
</li>
<li><p>SPARK Consortium. Spark: A us cohort of 50,000 families to accelerate autism research. <em>Neuron</em> 97, 488–493. [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7444276/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/29420931">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Neuron&amp;title=Spark:+A+us+cohort+of+50,000+families+to+accelerate+autism+research&amp;volume=97&amp;pages=488-493&amp;pmid=29420931&amp;">Google Scholar</a>]</p>
</li>
</ul>
</blockquote>
<p>对MsigDB 6.0中的Gene Ontology集的分子功能的不可知分析</p>
<blockquote>
<ul>
<li>Ashburner M et al. Gene ontology: tool for the unification of biology. the gene ontology consortium. <em>Nature genetics</em> 25, 25–29 (2000). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3037419/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/10802651">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+genetics&amp;title=Gene+ontology:+tool+for+the+unification+of+biology.+the+gene+ontology+consortium&amp;author=M+Ashburner&amp;volume=25&amp;publication_year=2000&amp;pages=25-29&amp;pmid=10802651&amp;">Google Scholar</a>]</li>
<li>Consortium, G. O. Gene ontology consortium: going forward. <em>Nucleic acids research</em> 43, D1049–D1056 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4383973/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25428369">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nucleic+acids+research&amp;title=Gene+ontology+consortium:+going+forward&amp;volume=43&amp;publication_year=2015&amp;pages=D1049-D1056&amp;pmid=25428369&amp;">Google Scholar</a>]</li>
<li>Subramanian A et al. Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. <em>PNAS</em> 102, 15545–15550 (2005). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/16199517">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=PNAS&amp;title=Gene+set+enrichment+analysis:+a+knowledge-based+approach+for+interpreting+genome-wide+expression+profiles&amp;author=A+Subramanian&amp;volume=102&amp;publication_year=2005&amp;pages=15545-15550&amp;pmid=16199517&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<h4><span id="3-3-snp-yi-chuan-li-heritability">3.3 SNP遗传力（heritability）</span><a href="#3-3-snp-yi-chuan-li-heritability" class="header-anchor">#</a></h4><p>SNP遗传性，h^2_G，对ASD GWAS全样本使用LDSC估计，对LDSC来说太小的子样本使用GCTA。对于LDSC，我们使用了基于1000基因组计划的欧洲血统样本的预先计算的LD分数，仅限于HapMap3 SNPs。具有标准LDSC过滤的汇总统计被回归到这些分数上。对于责任规模的估计，我们使用丹麦的人口流行率为1.22%。由于缺乏适当的亚型患病率估计，我们根据病例样本的构成对全谱系患病率进行了调整。</p>
<p>对于对LDSC来说太小的子样本，采用了GCTA的GREML方法。在去除INDELs的最佳猜测基因型（基因型概率&gt;0.8，缺失率<0.01，maf>0.05）上，对关联样本（即欧洲血统的受试者，πˆ≤0.2）进行了遗传相关度矩阵（GRM），为所有成对的个体组合提供了相关度估计。估算SNPs解释的表型变异（REML），包括PC 1-4作为连续协变量，以及任何其他名义上与表型显著相关的PC，以及作为分类指标协变量的批次。测试非重叠组的等效遗传性是通过保留对照组和随机分配不同病例标签的互换试验（有1000次互换）完成的。</0.01，maf></p>
<p>按照Finucane等的做法，我们对功能注释的SNPs和位于细胞类型特异性调控元件的SNPs进行了遗传性的富集分析。首先使用与Finucane等人相同的24个重叠的功能注释（从53个剥离），我们将ASD GWAS汇总统计中的χ2回归到从上述网站下载的细胞类型特异性LD分数上，基线分数、回归权重和等位基因频率基于欧洲血统1000基因组计划数据。一个类别的富集度被定义为该类别中SNP遗传性的比例除以该类别中SNP的比例。我们仍然遵循Finucane等人的观点，使用220个细胞类型的特定注释进行了类似的分析，这些注释分为10个重叠的组。除此之外，我们还根据来自Roadmap Epigenomics Mapping Consortium的H3K4Me1推算缺口峰数据的注释进行了分析；更具体的是排除了广泛的MHC-区域（Chr6:25-35MB）的信息。</p>
<blockquote>
<p>LDSC</p>
<blockquote>
<ul>
<li>Bulik-Sullivan BK et al. Ld score regression distinguishes confounding from polygenicity in genome-wide association studies. <em>Nat Genet</em> 47, 291–295 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495769/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25642630">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nat+Genet&amp;title=Ld+score+regression+distinguishes+confounding+from+polygenicity+in+genome-wide+association+studies&amp;author=BK+Bulik-Sullivan&amp;volume=47&amp;publication_year=2015&amp;pages=291-295&amp;pmid=25642630&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>GCTA</p>
<blockquote>
<ul>
<li>Yang J, Lee SH, Goddard ME &amp; Visscher PM Gcta: a tool for genome-wide complex trait analysis. <em>Am J Hum Genet</em> 88, 76–82 (2011). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3014363/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/21167468">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Am+J+Hum+Genet&amp;title=Gcta:+a+tool+for+genome-wide+complex+trait+analysis&amp;author=J+Yang&amp;author=SH+Lee&amp;author=ME+Goddard&amp;author=PM+Visscher&amp;volume=88&amp;publication_year=2011&amp;pages=76-82&amp;pmid=21167468&amp;">Google Scholar</a>]</li>
<li>Lee SH, Wray NR, Goddard ME &amp; Visscher PM Estimating missing heritability for disease from genome-wide association studies. <em>Am J Hum Genet</em> 88, 294–305 (2011). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3059431/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/21376301">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Am+J+Hum+Genet&amp;title=Estimating+missing+heritability+for+disease+from+genome-wide+association+studies&amp;author=SH+Lee&amp;author=NR+Wray&amp;author=ME+Goddard&amp;author=PM+Visscher&amp;volume=88&amp;publication_year=2011&amp;pages=294-305&amp;pmid=21376301&amp;">Google Scholar</a>]</li>
<li>Yang J et al. Common snps explain a large proportion of the heritability for human height. <em>Nat Genet</em> 42, 565–569 (2010). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3232052/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/20562875">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nat+Genet&amp;title=Common+snps+explain+a+large+proportion+of+the+heritability+for+human+height&amp;author=J+Yang&amp;volume=42&amp;publication_year=2010&amp;pages=565-569&amp;pmid=20562875&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>1000基因组计划</p>
<blockquote>
<ul>
<li>1000 Genomes Project Consortium et al. An integrated map of genetic variation from 1,092 human genomes. <em>Nature</em> 491, 56–65 (2012). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3498066/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/23128226">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=An+integrated+map+of+genetic+variation+from+1,092+human+genomes&amp;volume=491&amp;publication_year=2012&amp;pages=56-65&amp;pmid=23128226&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>HapMap3</p>
<blockquote>
<ul>
<li>Consortium, I. H.. et al. Integrating common and rare genetic variation in diverse human populations. <em>Nature</em> 467, 52–58 (2010). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3173859/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/20811451">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Integrating+common+and+rare+genetic+variation+in+diverse+human+populations&amp;volume=467&amp;publication_year=2010&amp;pages=52-58&amp;pmid=20811451&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>Roadmap Epigenomics Mapping Consortium</p>
<blockquote>
<ul>
<li>Consortium, R. E. et al. Integrative analysis of 111 reference human epigenomes. <em>Nature</em> 518, 317–330 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4530010/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25693563">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Integrative+analysis+of+111+reference+human+epigenomes&amp;volume=518&amp;publication_year=2015&amp;pages=317-330&amp;pmid=25693563&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<h4><span id="3-3-yi-chuan-xiang-guan-xing-genetic-correlation">3.3 遗传相关性/genetic correlation</span><a href="#3-3-yi-chuan-xiang-guan-xing-genetic-correlation" class="header-anchor">#</a></h4><p>对于主要的ASD样本，SNP的相关性，rG是用LDSC估计的，对于ASD亚型和亚组的分析，样本量普遍较小，我们使用GCTA。在这两种情况下，我们遵循上述相同的程序。对于除少数表型外的所有表型，LDSC的相关估计是通过上传到LD hub来实现的，与所有一起234个表型进行比较。</p>
<blockquote>
<p>D hub</p>
<blockquote>
<ul>
<li>Zheng J et al. Ld hub: a centralized database and web interface to perform ld score regression that maximizes the potential of summary level gwas data for snp heritability and genetic correlation analysis. <em>Bioinformatics</em> 33, 272–279 (2017). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5542030/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27663502">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Bioinformatics&amp;title=Ld+hub:+a+centralized+database+and+web+interface+to+perform+ld+score+regression+that+maximizes+the+potential+of+summary+level+gwas+data+for+snp+heritability+and+genetic+correlation+analysis&amp;author=J+Zheng&amp;volume=33&amp;publication_year=2017&amp;pages=272-279&amp;pmid=27663502&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<h4><span id="3-4-duo-ji-yin-feng-xian-ping-fen-polygenic-risk-score">3.4 多基因风险评分/polygenic risk score</span><a href="#3-4-duo-ji-yin-feng-xian-ping-fen-polygenic-risk-score" class="header-anchor">#</a></h4><p>对于多基因风险评分（PRS），我们应用标准的Ricopili参数对汇总统计数字进行了分组（详见补充说明）。为了避免潜在的链式冲突，我们排除了所有不由Ricopili生成的摘要统计的含糊标记，使用相同的归因参考。PRS是在默认的P值阈值（5e-8、1e-6、1e-4、0.001、0.01、0.05、0.1、0.2、0.5和1）下产生的，是ASD GWAS样本中等位基因剂量的加权和。对训练集中遵守p值阈值的标记进行求和，并以训练集中估计的标记的加性规模效应度量（log(OR)或β）进行称重。分数在分析前被归一化。</p>
<p>我们使用Nagelkerke的$R^2$和几率图以及分数十位数的置信区间来评估预测能力。$R^2$和优势比都是在回归分析中估计的，包括相关的PC和基因分型波的指标变量。</p>
<p>由于在iPSYCH和PGC之外缺乏大量的ASD样本，我们以下述方式在内部训练了一套ASD的PRS。我们将样本分为五个子样本，这些子样本的大小大致相同，尊重分批划分。然后，我们运行了五个GWAS，从训练集中依次剔除每个组，并将这些结果与PGC的结果进行了元分析。这就为五个子样本中的每个子样本产生了一套对其补充训练的PRS。在分析之前，每个分数都在其定义的组别上被归一化。我们评估了每个组和整个样本的预测能力。</p>
<p>为了利用与其他表型的遗传重叠来改善预测，我们通过在内部训练的ASD评分中加入其他高度相关的表型的PRS，以加权的方式创建了一系列新的PRS。详见补充信息。</p>
<p>为了分析ASD亚型与PRS的关系，我们以下列方式定义了一套分层的表型。第一个层次的亚型是儿童自闭症，层次的非典型自闭症被定义为所有患有非典型自闭症且没有儿童自闭症诊断的人，层次的阿斯伯格症是所有患有阿斯伯格症诊断且既没有儿童自闭症也没有非典型自闭症的人。最后，我们将其他广泛性发育障碍和广泛性发育障碍，未指明的，归入广泛性发育障碍混合型，其分级版本包括每个人都有这样的诊断而没有前面的诊断（补充表13）。我们通过对亚型上的分数做多变量回归，同时在线性回归中调整相关的PC和波段指标变量，研究了一些表现型与ASD的高rG（以及一些低rG作为阴性对照）在不同ASD亚型上的PRS分布。详见补充说明。</p>
<h4><span id="3-5-hi-c-fen-xi">3.5 Hi-C分析</span><a href="#3-5-hi-c-fen-xi" class="header-anchor">#</a></h4><p>Hi-C数据产生于两个主要的皮层：主要包含有丝分裂活跃的神经祖细胞的生发区（GZ），以及主要由有丝分裂后的神经元组成的皮层和皮层下板（CP）。我们首先使用CAVIAR从ASD GWAS中确定的排名靠前的位点中得出一组可信的SNPs（推定的因果SNPs）。将显示出最强关联的30个基因座与Hi-C参考数据相交，得到28个基因座进行分析。为了测试可信的SNPs是否在胎儿大脑的活跃标记中富集，我们采用了以前描述的GREAT。可信的SNPs被细分为没有已知功能的SNPs（未注释的）和有功能注释的SNPs（基因启动子中的SNPs和引起非同义变异的SNPs）（补充图98）。然后，我们将未加注释的可信SNPs与胎儿皮质发育过程中的染色质接触图谱整合在一起，定义了与基因间或音内SNPs物理相互作用的基因（补充图98）。</p>
<blockquote>
<p>CAVIAR</p>
<blockquote>
<ul>
<li>Hormozdiari F, Kostem E, Kang EY, Pasaniuc B &amp; Eskin E Identifying causal variants at loci with multiple signals of association. <em>Genetics</em> 198, 497–508 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4196608/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25104515">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Genetics&amp;title=Identifying+causal+variants+at+loci+with+multiple+signals+of+association&amp;author=F+Hormozdiari&amp;author=E+Kostem&amp;author=EY+Kang&amp;author=B+Pasaniuc&amp;author=E+Eskin&amp;volume=198&amp;publication_year=2014&amp;pages=497-508&amp;pmid=25104515&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>胎儿大脑的活跃标记</p>
<blockquote>
<ul>
<li>Consortium, R. E. et al. Integrative analysis of 111 reference human epigenomes. <em>Nature</em> 518, 317–330 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4530010/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25693563">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Integrative+analysis+of+111+reference+human+epigenomes&amp;volume=518&amp;publication_year=2015&amp;pages=317-330&amp;pmid=25693563&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>GREAT</p>
<blockquote>
<ul>
<li><p>Won H et al. Chromosome conformation elucidates regulatory relationships in developing human brain. <em>Nature</em> 538, 523–527 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5358922/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27760116">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Chromosome+conformation+elucidates+regulatory+relationships+in+developing+human+brain&amp;author=H+Won&amp;volume=538&amp;publication_year=2016&amp;pages=523-527&amp;pmid=27760116&amp;">Google Scholar</a>]</p>
</li>
<li><p>McLean CY et al. Great improves functional interpretation of cis-regulatory regions. <em>Nature biotechnology</em> 28, 495–501 (2010). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4840234/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/20436461">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+biotechnology&amp;title=Great+improves+functional+interpretation+of+cis-regulatory+regions&amp;author=CY+McLean&amp;volume=28&amp;publication_year=2010&amp;pages=495-501&amp;">Google Scholar</a>]</p>
</li>
</ul>
</blockquote>
</blockquote>
<p>人脑的时空转录组图谱是从Kang等人那里获得的122。我们使用了多个脑区的转录组图谱，其发育历时跨越了产前（受孕后6-37周，PCW）和产后（4个月-42年）时期。使用R中的scale(center=T, scale=F)+1函数对每个样本的表达值进行对数转换并以平均表达水平为中心。</p>
<blockquote>
<p>人脑时空转录组</p>
<blockquote>
<ul>
<li>Kang HJ et al. Spatio-temporal transcriptome of the human brain. <em>Nature</em> 478, 483–489 (2011). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3566780/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/22031440">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Spatio-temporal+transcriptome+of+the+human+brain&amp;author=HJ+Kang&amp;volume=478&amp;publication_year=2011&amp;pages=483-489&amp;pmid=22031440&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
</blockquote>
<h2><span id="zi-yuan">资源</span><a href="#zi-yuan" class="header-anchor">#</a></h2><p>The GenomeDK high performance-computing cluster in Denmark, <a target="_blank" rel="noopener" href="https://genome.au.dk/">https://genome.au.dk</a>; </p>
<p>the iPSYCH project, <a target="_blank" rel="noopener" href="http://ipsych.au.dk/">http://ipsych.au.dk</a>, </p>
<p>the iPSYCH download page, <a target="_blank" rel="noopener" href="http://ipsych.au.dk/downloads/">http://ipsych.au.dk/downloads/</a>; </p>
<p>the NIMH Repository, <a target="_blank" rel="noopener" href="https://www.nimhgenetics.org/available_data/autism/">https://www.nimhgenetics.org/available_data/autism/</a>; </p>
<p>the PGC download site, <a target="_blank" rel="noopener" href="https://www.med.unc.edu/pgc/results-and-downloads">https://www.med.unc.edu/pgc/results-and-downloads</a>; </p>
<blockquote>
<p>为了从 PGC 样本和 iPSYCH 样本中获得基因型，研究人员应分别与 PIs 主管 Mark j. Daly 和 Anders d. Børglum 联系。</p>
</blockquote>
<p>the LISA cluster at SURFsara, <a target="_blank" rel="noopener" href="https://userinfo.surfsara.nl/systems/lisa">https://userinfo.surfsara.nl/systems/lisa</a>; </p>
<p>plink 1.9, <a target="_blank" rel="noopener" href="http://www.cog-genomics.org/plink/1.9/">www.cog-genomics.org/plink/1.9/</a>; </p>
<p>LDSC and associated files, <a target="_blank" rel="noopener" href="https://github.com/bulik/ldsc">https://github.com/bulik/ldsc</a>; </p>
<p>LD hub, <a target="_blank" rel="noopener" href="http://ldsc.broadinstitute.org/ldhub/">http://ldsc.broadinstitute.org/ldhub/</a>; </p>
<p>GTExportal, <a target="_blank" rel="noopener" href="https://gtexportal.org/home/">https://gtexportal.org/home/</a></p>
<h2><span id="gao-yin-yong-wen-xian">高引用文献</span><a href="#gao-yin-yong-wen-xian" class="header-anchor">#</a></h2><p>【以往GWAS】</p>
<ul>
<li>Anney RJL et al. Meta-analysis of gwas of over 16,000 individuals with autism spectrum disorder highlights a novel locus at 10q24.32 and a significant overlap with schizophrenia. <em>Molecular Autism</em> 8, 21 (2017). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5441062/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/28540026">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Molecular+Autism&amp;title=Meta-analysis+of+gwas+of+over+16,000+individuals+with+autism+spectrum+disorder+highlights+a+novel+locus+at+10q24.32+and+a+significant+overlap+with+schizophrenia&amp;author=RJL+Anney&amp;volume=8&amp;publication_year=2017&amp;pages=21&amp;pmid=28540026&amp;">Google Scholar</a>]</li>
<li>Ma D et al. A genome-wide association study of autism reveals a common novel risk locus at 5p14.1. <em>Annals of human genetics</em> 73, 263–273 (2009). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2918410/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/19456320">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Annals+of+human+genetics&amp;title=A+genome-wide+association+study+of+autism+reveals+a+common+novel+risk+locus+at+5p14.1&amp;author=D+Ma&amp;volume=73&amp;publication_year=2009&amp;pages=263-273&amp;pmid=19456320&amp;">Google Scholar</a>]</li>
<li>Ma D et al. A genome-wide association study of autism reveals a common novel risk locus at 5p14.1. <em>Annals of human genetics</em> 73, 263–273 (2009). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2918410/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/19456320">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Annals+of+human+genetics&amp;title=A+genome-wide+association+study+of+autism+reveals+a+common+novel+risk+locus+at+5p14.1&amp;author=D+Ma&amp;volume=73&amp;publication_year=2009&amp;pages=263-273&amp;pmid=19456320&amp;">Google Scholar</a>]</li>
<li>Anney R et al. Individual common variants exert weak effects on the risk for autism spectrum disorderspi. <em>Hum Mol Genet</em> 21, 4781–4792 (2012). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3471395/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/22843504">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Hum+Mol+Genet&amp;title=Individual+common+variants+exert+weak+effects+on+the+risk+for+autism+spectrum+disorderspi&amp;author=R+Anney&amp;volume=21&amp;publication_year=2012&amp;pages=4781-4792&amp;pmid=22843504&amp;">Google Scholar</a>]</li>
</ul>
<p>【其他类型精神疾病】</p>
<p>精神分裂症</p>
<blockquote>
<ul>
<li>Schizophrenia Working Group of the Psychiatric Genomics Consortium. Biological insights from 108 schizophrenia-associated genetic loci. <em>Nature</em> 511, 421–427 (2014). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4112379/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25056061">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Biological+insights+from+108+schizophrenia-associated+genetic+loci&amp;volume=511&amp;publication_year=2014&amp;pages=421-427&amp;pmid=25056061&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>严重抑郁症</p>
<blockquote>
<ul>
<li>Wray NR et al. Genome-wide association analyses identify 44 risk variants and refine the genetic architecture of major depressive disorder. <em>Nature Genetics</em> 50, 668–681 (2018). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5934326/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/29700475">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=Genome-wide+association+analyses+identify+44+risk+variants+and+refine+the+genetic+architecture+of+major+depressive+disorder&amp;author=NR+Wray&amp;volume=50&amp;publication_year=2018&amp;pages=668-681&amp;pmid=29700475&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>躁郁症</p>
<blockquote>
<ul>
<li>Bulik-Sullivan B et al. An atlas of genetic correlations across human diseases and traits. <em>Nature Genetics</em> 47, 1061–4036 (2015). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4797329/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/26414676">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature+Genetics&amp;title=An+atlas+of+genetic+correlations+across+human+diseases+and+traits&amp;author=B+Bulik-Sullivan&amp;volume=47&amp;publication_year=2015&amp;pages=1061-4036&amp;pmid=26192917&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>
<p>受教育程度</p>
<blockquote>
<ul>
<li>Okbay A et al. Genome-wide association study identifies 74 loci associated with educational attainment. <em>Nature</em> 533, 539–542 (2016). [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4883595/">PMC free article</a>] [<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/27225129">PubMed</a>] [<a target="_blank" rel="noopener" href="https://scholar.google.com/scholar_lookup?journal=Nature&amp;title=Genome-wide+association+study+identifies+74+loci+associated+with+educational+attainment&amp;author=A+Okbay&amp;volume=533&amp;publication_year=2016&amp;pages=539-542&amp;pmid=27225129&amp;">Google Scholar</a>]</li>
</ul>
</blockquote>

            </div>
            <hr>
            <div>
              <div class="post-metas mb-3">
                
                  <div class="post-meta mr-3">
                    <i class="iconfont icon-category"></i>
                    
                      <a class="hover-with-bg" href="/categories/LITERATURE-READING/">LITERATURE READING</a>
                    
                  </div>
                
                
                  <div class="post-meta">
                    <i class="iconfont icon-tags"></i>
                    
                      <a class="hover-with-bg" href="/tags/GWAS/">GWAS</a>
                    
                      <a class="hover-with-bg" href="/tags/ASD/">ASD</a>
                    
                  </div>
                
              </div>
              
                <p class="note note-warning">
                  
                    本博客所有文章除特别声明外，均采用 <a target="_blank" href="https://creativecommons.org/licenses/by-sa/4.0/deed.zh" rel="nofollow noopener noopener">CC BY-SA 4.0 协议</a> ，转载请注明出处！
                  
                </p>
              
              
                <div class="post-prevnext">
                  <article class="post-prev col-6">
                    
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2021/12/08/20211208Pan02%E8%8D%AF%E6%95%88%E6%96%B9%E6%A1%88%E5%B0%8F%E9%BC%A0%E7%BB%99%E8%8D%AF%E9%85%8D%E7%BD%AE%E8%A1%A8/">
                        <span class="hidden-mobile">2021.12.08-Pan02药效方案-小鼠给药配置表</span>
                        <span class="visible-mobile">下一篇</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
              <!-- Comments -->
              <article class="comments" id="comments">
                
                  
                
                
  <div class="disqus" style="width:100%">
    <div id="disqus_thread"></div>
    <script>
      Fluid.utils.waitElementVisible('disqus_thread', function() {
        Fluid.utils.createCssLink('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqusjs.css');
        Fluid.utils.createScript('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqus.js', function () {
          new DisqusJS({
            shortname: 'fluid',
            apikey: ''
          });
        });
      });
    </script>
    <noscript>Please enable JavaScript to view the
      <a target="_blank" href="https://disqus.com/?ref_noscript" rel="nofollow noopener noopener">comments powered by Disqus.</a>
    </noscript>
  </div>


              </article>
            
          </article>
        </div>
      </div>
    </div>
    
      <div class="d-none d-lg-block col-lg-2 toc-container" id="toc-ctn">
        <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;目录</p>
  <div class="toc-body" id="toc-body"></div>
</div>

      </div>
    
  </div>
</div>

<!-- Custom -->


    

    
      <a id="scroll-top-button" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">搜索</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v"
                 for="local-search-input">关键词</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>
    

    
  </main>

  <footer class="text-center mt-5 py-3">
  <div class="footer-content">
     <a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a> <i class="iconfont icon-love"></i> <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener"><span>Fluid</span></a> 
  </div>
  

  

  
</footer>

<!-- SCRIPTS -->

  <script  src="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" ></script>
<script  src="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/js/bootstrap.min.js" ></script>
<script  src="/js/debouncer.js" ></script>
<script  src="/js/events.js" ></script>
<script  src="/js/plugins.js" ></script>

<!-- Plugins -->


  
    <script  src="/js/lazyload.js" ></script>
  



  



  <script  src="https://cdn.jsdelivr.net/npm/tocbot@4.12.0/dist/tocbot.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/anchor-js@4.3.0/anchor.min.js" ></script>



  <script defer src="https://cdn.jsdelivr.net/npm/clipboard@2.0.6/dist/clipboard.min.js" ></script>






  <script  src="https://cdn.jsdelivr.net/npm/typed.js@2.0.11/lib/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var title = document.getElementById('subtitle').title;
      
      typing(title)
      
    })(window, document);
  </script>



  <script  src="/js/local-search.js" ></script>
  <script>
    (function () {
      var path = "/local-search.xml";
      $('#local-search-input').on('click', function() {
        searchFunc(path, 'local-search-input', 'local-search-result');
      });
      $('#modalSearch').on('shown.bs.modal', function() {
        $('#local-search-input').focus();
      });
    })()
  </script>





  

  
    <!-- MathJax -->
    <script>
      MathJax = {
        tex: {
          inlineMath: [['$', '$'], ['\\(', '\\)']]
        },
        options: {
          renderActions: {
            findScript: [10, doc => {
              document.querySelectorAll('script[type^="math/tex"]').forEach(node => {
                const display = !!node.type.match(/; *mode=display/);
                const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display);
                const text = document.createTextNode('');
                node.parentNode.replaceChild(text, node);
                math.start = { node: text, delim: '', n: 0 };
                math.end = { node: text, delim: '', n: 0 };
                doc.math.push(math);
              });
            }, '', false],
            insertedScript: [200, () => {
              document.querySelectorAll('mjx-container').forEach(node => {
                let target = node.parentNode;
                if (target.nodeName.toLowerCase() === 'li') {
                  target.parentNode.classList.add('has-jax');
                }
              });
            }, '', false]
          }
        }
      };
    </script>

    <script async src="https://cdn.jsdelivr.net/npm/mathjax@3.1.2/es5/tex-svg.js" ></script>

  











<!-- 主题的启动项 保持在最底部 -->
<script  src="/js/boot.js" ></script>



</body>
</html>
