

<!DOCTYPE html>
<html lang="zh-CN" data-default-color-scheme=&#34;auto&#34;>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/favicon.png">
  <link rel="icon" href="/img/favicon.png">
  <meta name="viewport"
        content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  
    <meta name="description" content="">
  
  <meta name="author" content="John Doe">
  <meta name="keywords" content="">
  
  <title>2021-10-14 GWAS PRS TEST - Hexo</title>

  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" />


  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/github-markdown-css@4.0.0/github-markdown.min.css" />
  <link  rel="stylesheet" href="/lib/hint/hint.min.css" />

  
    
    
      
      <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@10.4.0/styles/github-gist.min.css" />
    
  

  
    <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.css" />
  



<!-- 主题依赖的图标库，不要自行修改 -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_ba1fz6golrf.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_kmeydafke9r.css">


<link  rel="stylesheet" href="/css/main.css" />

<!-- 自定义样式保持在最底部 -->


  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    var CONFIG = {"hostname":"example.com","root":"/","version":"1.8.8","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"right","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"copy_btn":true,"image_zoom":{"enable":true},"toc":{"enable":true,"headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"onlypost":false},"web_analytics":{"enable":false,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":null,"app_key":null,"server_url":null}}};
  </script>
  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
<meta name="generator" content="Hexo 5.4.0"></head>


<body>
  <header style="height: 70vh;">
    <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand"
       href="/">&nbsp;<strong>Liu Keqin</strong>&nbsp;</a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/">
                <i class="iconfont icon-home-fill"></i>
                首页
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/">
                <i class="iconfont icon-archive-fill"></i>
                归档
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/">
                <i class="iconfont icon-category-fill"></i>
                分类
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/">
                <i class="iconfont icon-tags-fill"></i>
                标签
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/">
                <i class="iconfont icon-user-fill"></i>
                关于
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" data-toggle="modal" data-target="#modalSearch">&nbsp;<i
                class="iconfont icon-search"></i>&nbsp;</a>
          </li>
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" href="javascript:">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

    <div class="banner" id="banner" parallax=true
         style="background: url('/img/default.png') no-repeat center center;
           background-size: cover;">
      <div class="full-bg-img">
        <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
          <div class="page-header text-center fade-in-up">
            <span class="h2" id="subtitle" title="2021-10-14 GWAS PRS TEST">
              
            </span>

            
              <div class="mt-3">
  
  
    <span class="post-meta">
      <i class="iconfont icon-date-fill" aria-hidden="true"></i>
      <time datetime="2021-10-14 17:18" pubdate>
        2021年10月14日 下午
      </time>
    </span>
  
</div>

<div class="mt-1">
  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-chart"></i>
      7.5k 字
    </span>
  

  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-clock-fill"></i>
      
      
      117
       分钟
    </span>
  

  
  
</div>

            
          </div>

          
        </div>
      </div>
    </div>
  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="d-none d-lg-block col-lg-2"></div>
    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div class="py-5" id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">2021-10-14 GWAS PRS TEST</h1>
            
            <div class="markdown-body">
              <h1><span id="2021-10-14-gwas-prs-test">2021-10-14 GWAS PRS TEST</span><a href="#2021-10-14-gwas-prs-test" class="header-anchor">#</a></h1><p><strong>运行环境：</strong></p>
<p>MacOS Catalina 10.15.7</p>
<p><strong>软件及网页工具：</strong></p>
<p>PRSice_mac (<a target="_blank" rel="noopener" href="https://choishingwan.github.io/PRS-Tutorial/base/">https://choishingwan.github.io/PRS-Tutorial/base/</a>)</p>
<p>PLINK v1.90b4 64-bit (20 Mar 2017)</p>
<p>LDpred2</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs R">install.packages(<span class="hljs-string">&quot;bigsnpr&quot;</span>)<br>library(bigsnpr)<br></code></pre></td></tr></table></figure>
<p>lassosum（未安装，failed）</p>
<h2><span id="1-qc-of-base-data">1.QC of base data</span><a href="#1-qc-of-base-data" class="header-anchor">#</a></h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">gunzip -c Height.gwas.txt.gz | head<br></code></pre></td></tr></table></figure>
<p>根据PRSice的tutorial，输入文件（GWAS statistic summary）应包含以下列：</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>CHR</th>
<th>BP</th>
<th>SNP</th>
<th>A1</th>
<th>A2</th>
<th>N</th>
<th>SE</th>
<th>P</th>
<th>OR</th>
<th>INFO</th>
<th>MAF</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>756604</td>
<td>rs3131962</td>
<td>A</td>
<td>G</td>
<td>388028</td>
<td>0.00301666</td>
<td>0.483171</td>
<td>0.997886915712657</td>
<td>0.890557941364774</td>
<td>0.369389592764921</td>
</tr>
<tr>
<td>1</td>
<td>768448</td>
<td>rs12562034</td>
<td>A</td>
<td>G</td>
<td>388028</td>
<td>0.00329472</td>
<td>0.834808</td>
<td>1.00068731609353</td>
<td>0.895893511351165</td>
<td>0.336845754096289</td>
</tr>
<tr>
<td>1</td>
<td>779322</td>
<td>rs4040617</td>
<td>G</td>
<td>A</td>
<td>388028</td>
<td>0.00303344</td>
<td>0.42897</td>
<td>0.997603556067569</td>
<td>0.897508290615237</td>
<td>0.377368010940814</td>
</tr>
</tbody>
</table>
</div>
<ul>
<li><strong>CHR</strong>: The chromosome in which the SNP resides</li>
<li><strong>BP</strong>: Chromosomal co-ordinate of the SNP</li>
<li><strong>SNP</strong>: SNP ID, usually in the form of rs-ID</li>
<li><strong>A1</strong>: The effect allele of the SNP</li>
<li><strong>A2</strong>: The non-effect allele of the SNP</li>
<li><strong>N</strong>: Number of samples used to obtain the effect size estimate</li>
<li><strong>SE</strong>: The standard error (SE) of the effect size esimate</li>
<li><strong>P</strong>: The P-value of association between the SNP genotypes and the base phenotype</li>
<li><strong>OR</strong>: The effect size estimate of the SNP, if the outcome is binary/case-control. If the outcome is continuous or treated as continuous then this will usually be BETA</li>
<li><strong>INFO</strong>: The imputation information score</li>
<li><strong>MAF</strong>: The minor allele frequency (MAF) of the SNP</li>
</ul>
<p>查看当前使用的来自PGC数据库：<a target="_blank" rel="noopener" href="https://www.med.unc.edu/pgc/download-results/的数据集，该数据包含以下列：">https://www.med.unc.edu/pgc/download-results/的数据集，该数据包含以下列：</a></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">##   CHR        SNP        BP A1 A2  INFO      OR     SE        P</span><br><span class="hljs-comment">## 1   8 rs62513865 101592213  T  C 0.949 1.00652 0.0270 0.808600</span><br><span class="hljs-comment">## 2   8 rs79643588 106973048  A  G 0.997 1.01786 0.0240 0.460600</span><br><span class="hljs-comment">## 3   8 rs17396518 108690829  T  G 0.987 0.96127 0.0140 0.004647</span><br><span class="hljs-comment">## 4   8   rs983166 108681675  A  C 0.998 0.97990 0.0139 0.145200</span><br><span class="hljs-comment">## 5   8 rs28842593 103044620  T  C 0.857 0.99591 0.0203 0.841500</span><br><span class="hljs-comment">## 6   8 rs35107696 109712249  A AT 0.999 1.01308 0.0165 0.430200</span><br></code></pre></td></tr></table></figure>
<h3><span id="1-1-qc-checklist-base-data-heritability-check">1.1 QC checklist: Base data-Heritability check</span><a href="#1-1-qc-checklist-base-data-heritability-check" class="header-anchor">#</a></h3><p>遗传力应＞0.05</p>
<h3><span id="1-2-qc-checklist-base-data-effect-allele">1.2 QC checklist: Base data-Effect allele</span><a href="#1-2-qc-checklist-base-data-effect-allele" class="header-anchor">#</a></h3><p>必须了解哪个等位基因是效应等位基因，哪个是无效等位基因，才能确保 PRS 的结果是正确的。</p>
<blockquote>
<p>一些 GWAS 结果文件没有明确哪个等位基因是有效等位基因，哪个是无效等位基因。如果在计算生产者责任计划时作出不正确的假设，那么生产者责任计划对目标数据的影响将是错误的。</p>
</blockquote>
<h3><span id="1-3-qc-checklist-base-data-genome-build">1.3 QC checklist: Base data-Genome build</span><a href="#1-3-qc-checklist-base-data-genome-build" class="header-anchor">#</a></h3><p>概要统计应该与我们将要使用的目标数据基因组构建相同。你必须检查你的基础数据和目标数据是否在同一个基因组构建上，如果他们不是，那么就使用一个工具，比如 LiftOver，使构建在数据集上保持一致。</p>
<h3><span id="1-4-qc-checklist-base-data-standard-gwas-qc">1.4 QC checklist: Base data-Standard GWAS QC</span><a href="#1-4-qc-checklist-base-data-standard-gwas-qc" class="header-anchor">#</a></h3><p>正如文中所描述的那样，基础数据和目标数据都应该受到 GWAS 中执行的标准严格的质量控制步骤的约束。如果基础数据是从公共来源获得的汇总统计数据，那么你能够在这些数据上执行的典型 QC 步骤是根据 INFO 得分和 MAF 过滤 SNPs。由于较低的统计能力(和较高的基因分型错误概率) ，较低的微小等位基因频率(MAF)或估算信息得分(INFO)的 SNPs 更有可能产生假阳性结果。</p>
<p>因此，低 MAF 和 INFO 的 SNPs 在执行下游分析之前通常会被去除。我们建议移除 MAF &lt; 1% 和 INFO &lt; 0.8的 SNPs (如果灵敏度检查显示可靠的结果，那么在非常大的基本样本量情况下，这些阈值可以降低)。这些 SNP 过滤器可以通过以下代码实现:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs bash">gunzip -c Height.gwas.txt.gz |<br>awk <span class="hljs-string">&#x27;NR==1 || ($11 &gt; 0.01) &amp;&amp; ($10 &gt; 0.8) &#123;print&#125;&#x27;</span> |<br>gzip  &gt; Height.gz<br><span class="hljs-comment">#$11表示第11列MAF</span><br><span class="hljs-comment">#实例命令</span><br></code></pre></td></tr></table></figure>
<p>PGC数据集以做过移除，故不需要这一步。</p>
<h3><span id="1-5-qc-checklist-base-data-mismatching-snps">1.5 QC checklist: Base data-Mismatching SNPs</span><a href="#1-5-qc-checklist-base-data-mismatching-snps" class="header-anchor">#</a></h3><p>基准数据和目标数据中报告的等位基因不匹配的SNP可以通过将等位基因与目标数据中的互补等位基因进行 “链式翻转 “来解决，如基准数据中的A/C和目标数据中的G/T的SNP，或者不可解决，如基准中的C/G和目标中的C/T的SNP。大多数多基因评分软件对可解决的SNP自动进行链式翻转，并删除不可解决的不匹配的SNP。</p>
<p>因为我们需要目标数据来知道哪些 SNPs 具有不匹配的等位基因，所以我们将在目标数据中执行这种链翻转。</p>
<h3><span id="1-6-qc-checklist-base-data-duplicate-snps">1.6 QC checklist: Base data-Duplicate SNPs</span><a href="#1-6-qc-checklist-base-data-duplicate-snps" class="header-anchor">#</a></h3><p>如果在生成基础数据时发生了错误，那么基础数据文件中可能有重复的SNPs。大多数PRS软件不允许在基础数据输入中出现重复的SNPs，因此应使用下面的命令将其删除。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs bash">gunzip -c Height.gwas.txt.gz |<br>awk <span class="hljs-string">&#x27;&#123;seen[$3]++; if(seen[$3]==1)&#123; print&#125;&#125;&#x27;</span> |<br>gzip - &gt; Height.nodup.gz<br><span class="hljs-comment">#$3表示第三列SNP</span><br><span class="hljs-comment">#示例命令</span><br></code></pre></td></tr></table></figure>
<p>本实验中需要的命令为：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs bash">gzip iPSYCH_PGC_ASD_Nov2017 &gt; iPSYCH_PGC_ASD_Nov2017_20211014.gz<br><br>gunzip -c iPSYCH_PGC_ASD_Nov2017_20211014.gz |<br>awk <span class="hljs-string">&#x27;&#123;seen[$2]++; if(seen[$2]==1)&#123; print&#125;&#125;&#x27;</span> |<br>gzip - &gt; iPSYCH_PGC_ASD_Nov2017_20211014.gz<br><span class="hljs-comment">#$2表示第2列SNP</span><br></code></pre></td></tr></table></figure>
<h3><span id="1-7-qc-checklist-base-data-ambiguous-snps">1.7 QC checklist: Base data-Ambiguous SNPs</span><a href="#1-7-qc-checklist-base-data-ambiguous-snps" class="header-anchor">#</a></h3><p>如果碱基和目标数据是使用不同的基因分型芯片产生的，并且用于其中一个的染色体链（+/-）是未知的，那么就不可能在不同的数据集中配对模糊的SNP的等位基因（即那些具有互补等位基因的SNP，要么是C/G或A/T SNP），因为它将不知道碱基和目标数据是否指的是同一个等位基因。虽然可以用等位基因频率来推断哪些等位基因在同一条链上，但对于MAF接近50%的SNP或当碱基和目标数据来自不同人群时，其准确性可能很低。因此，我们建议删除所有模糊的SNPs，以避免引入这种潜在的系统误差来源。</p>
<p>模糊的SNP可以在基础数据中被删除，那么在随后的分析中就不会有这样的SNP，因为分析只对基础数据和目标数据之间重叠的SNP进行。</p>
<p>可用以下代码保留无歧义的SNP：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs bash">gunzip -c Height.nodup.gz |<br>awk <span class="hljs-string">&#x27;!( ($4==&quot;A&quot; &amp;&amp; $5==&quot;T&quot;) ||</span><br><span class="hljs-string">        ($4==&quot;T&quot; &amp;&amp; $5==&quot;A&quot;) || </span><br><span class="hljs-string">        ($4==&quot;G&quot; &amp;&amp; $5==&quot;C&quot;) || </span><br><span class="hljs-string">        ($4==&quot;C&quot; &amp;&amp; $5==&quot;G&quot;)) &#123;print&#125;&#x27;</span> |<br>    gzip &gt; Height.QC.gz<br><span class="hljs-comment">#示例代码</span><br></code></pre></td></tr></table></figure>
<p>本实验中需要的命令为</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs bash">gunzip -c iPSYCH_PGC_ASD_Nov2017_20211014.gz |<br>awk <span class="hljs-string">&#x27;!( ($4==&quot;A&quot; &amp;&amp; $5==&quot;T&quot;) || </span><br><span class="hljs-string">        ($4==&quot;T&quot; &amp;&amp; $5==&quot;A&quot;) || </span><br><span class="hljs-string">        ($4==&quot;G&quot; &amp;&amp; $5==&quot;C&quot;) || </span><br><span class="hljs-string">        ($4==&quot;C&quot; &amp;&amp; $5==&quot;G&quot;)) &#123;print&#125;&#x27;</span> |<br>    gzip &gt; iPSYCH_PGC_ASD_Nov2017_20211014.gz<br></code></pre></td></tr></table></figure>
<h3><span id="1-8-qc-checklist-base-data-sex-chromosomes">1.8 QC checklist: Base data-Sex chromosomes</span><a href="#1-8-qc-checklist-base-data-sex-chromosomes" class="header-anchor">#</a></h3><p>有时会出现样本标签错误，这可能导致无效的结果。误标样本的一个迹象是报告的性别与性染色体所显示的性别不同。虽然这可能是由于性别和性别认同的不同，但它也可能反映了样本的误标或误报，因此，生物性别和报告性别不匹配的个体通常被删除。见目标数据部分，其中进行了性别检查。</p>
<h3><span id="1-9-qc-checklist-base-data-sample-overlap">1.9 QC checklist: Base data-Sample overlap</span><a href="#1-9-qc-checklist-base-data-sample-overlap" class="header-anchor">#</a></h3><p>由于目标数据是模拟的，所以这里的基础数据和目标数据之间没有重叠的样本。</p>
<h3><span id="1-10-qc-checklist-base-data-relatedness">1.10 QC checklist: Base data-Relatedness</span><a href="#1-10-qc-checklist-base-data-relatedness" class="header-anchor">#</a></h3><p>基础数据和目标数据内部和之间的密切相关的个体可能导致重叠的结果，限制了结果的普遍性(参见论文的相关章节)。目标数据中的相关性在目标数据部分中进行测试。</p>
<h2><span id="2-qc-of-target-data">2. QC of Target Data</span><a href="#2-qc-of-target-data" class="header-anchor">#</a></h2><h3><span id="2-1-obtaining-the-target-data">2.1 Obtaining the target data</span><a href="#2-1-obtaining-the-target-data" class="header-anchor">#</a></h3><p>目标数据由单个级别的基因型-表型数据组成，通常在您的实验室/部门/协作中生成。在本教程中，我们使用千人基因组计划的欧洲样本模拟了一些基因型-表型数据。你可在此下载有关资料。</p>
<blockquote>
<p> 目标数据——包含来自一个样本的个体的基因型和通常的表型(s) ，与 GWAS 样本无关。</p>
<p>（需要Plink）</p>
</blockquote>
<p>解压</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">unzip EUR.zip<br></code></pre></td></tr></table></figure>
<h3><span id="2-2-qc-checklist-target-data">2.2 QC checklist: Target data</span><a href="#2-2-qc-checklist-target-data" class="header-anchor">#</a></h3><p>以下是包含目标数据的质量控制检查表的质量控制步骤。</p>
<h3><span id="2-3-qc-checklist-sample-size">2.3 QC checklist: Sample size</span><a href="#2-3-qc-checklist-sample-size" class="header-anchor">#</a></h3><p>我们建议使用者只对至少100个人的目标数据进行PRS分析。这里我们的目标数据的样本量是503人。</p>
<h3><span id="2-4-qc-checklist-file-transfer">2.4 QC checklist: File transfer</span><a href="#2-4-qc-checklist-file-transfer" class="header-anchor">#</a></h3><p>通常情况下，我们不需要下载和传输目标数据文件，因为它通常是在本地生成的。然而，该文件应该包含一个md5sum代码，以防我们将数据文件发送给合作者，他们可能想确认该文件在传输过程中没有变化。</p>
<p>每个目标文件的 md5sum 代码：</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>File</th>
<th>md5sum</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>EUR.bed</strong></td>
<td>98bcef133f683b1272d3ea5f97742e0e</td>
</tr>
<tr>
<td><strong>EUR.bim</strong></td>
<td>6b286002904880055a9c94e01522f059</td>
</tr>
<tr>
<td><strong>EUR.cov</strong></td>
<td>85ed18288c708e095385418517e9c3bd</td>
</tr>
<tr>
<td><strong>EUR.fam</strong></td>
<td>e7b856f0c7bcaffc8405926d08386e97</td>
</tr>
<tr>
<td><strong>EUR.height</strong></td>
<td>dd445ce969a81cded20da5c88b82d4df</td>
</tr>
</tbody>
</table>
</div>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">md5sum EUR.zip<br></code></pre></td></tr></table></figure>
<p>如果文件是完整的，那么 md5sum 生成一个字符串，在这种情况下应该是和上表相符。如果生成了不同的字符串，则文件已损坏。</p>
<h3><span id="2-5-qc-checklist-genome-build">2.5 QC checklist: Genome build</span><a href="#2-5-qc-checklist-genome-build" class="header-anchor">#</a></h3><p>如基础数据部分所述，我们的基础和目标数据的基因组构建是相同的，因为它应该是相同的。</p>
<h3><span id="2-6-qc-checklist-standard-gwas-qc">2.6 QC checklist: Standard GWAS QC</span><a href="#2-6-qc-checklist-standard-gwas-qc" class="header-anchor">#</a></h3><p>目标数据的质量必须至少控制在 GWAS 研究中实施的标准之下，例如去除基因分型率低的 SNPs，低次等位基因频率，脱离 Hardy-Weinberg 平衡，去除基因分型率低的个体。</p>
<p>下面的 plink 命令将其中一些 QC 指标应用于目标数据:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs bash">plink \<br>    --bfile EUR \<br>    --maf 0.01 \<br>    --hwe 1e-6 \<br>    --geno 0.01 \<br>    --mind 0.01 \<br>    --write-snplist \<br>    --make-just-fam \<br>    --out EUR.QC<br></code></pre></td></tr></table></figure>
<p>return</p>
<figure class="highlight vhdl"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br></pre></td><td class="code"><pre><code class="hljs vhdl">PLINK v1.<span class="hljs-number">90</span>b4 <span class="hljs-number">64</span>-<span class="hljs-built_in">bit</span> (<span class="hljs-number">20</span> Mar <span class="hljs-number">2017</span>)             www.cog-genomics.org/plink/<span class="hljs-number">1.9</span>/<br>(C) <span class="hljs-number">2005</span>-<span class="hljs-number">2017</span> Shaun Purcell, Christopher Chang   GNU General Public License v3<br>Logging <span class="hljs-keyword">to</span> EUR.QC.log.<br>Options <span class="hljs-keyword">in</span> effect:<br>  <span class="hljs-comment">--bfile EUR</span><br>  <span class="hljs-comment">--geno 0.01</span><br>  <span class="hljs-comment">--hwe 1e-6</span><br>  <span class="hljs-comment">--maf 0.01</span><br>  <span class="hljs-comment">--make-just-fam</span><br>  <span class="hljs-comment">--mind 0.01</span><br>  <span class="hljs-comment">--out EUR.QC</span><br>  <span class="hljs-comment">--write-snplist</span><br><br><span class="hljs-number">8192</span> MB RAM detected; reserving <span class="hljs-number">4096</span> MB <span class="hljs-keyword">for</span> main workspace.<br><span class="hljs-number">551892</span> variants loaded from .bim <span class="hljs-keyword">file</span>.<br><span class="hljs-number">503</span> people (<span class="hljs-number">240</span> males, <span class="hljs-number">263</span> females) loaded from .fam.<br><span class="hljs-number">14</span> people removed due <span class="hljs-keyword">to</span> missing genotype data (<span class="hljs-comment">--mind).</span><br>IDs written <span class="hljs-keyword">to</span> EUR.QC.irem .<br>Using <span class="hljs-number">1</span> thread (no multithreaded calculations invoked).<br>Before main variant filters, <span class="hljs-number">489</span> founders <span class="hljs-keyword">and</span> <span class="hljs-number">0</span> nonfounders present.<br>Calculating allele frequencies... done.<br><span class="hljs-literal">Warning</span>: <span class="hljs-number">1806</span> het. haploid genotypes present (see EUR.QC.hh ); many commands<br>treat these as missing.<br>Total genotyping rate <span class="hljs-keyword">in</span> remaining samples <span class="hljs-keyword">is</span> <span class="hljs-number">0.999816</span>.<br><span class="hljs-number">5353</span> variants removed due <span class="hljs-keyword">to</span> missing genotype data (<span class="hljs-comment">--geno).</span><br><span class="hljs-literal">Warning</span>: <span class="hljs-comment">--hwe observation counts vary by more than 10%, due to the X</span><br>chromosome.  You may want <span class="hljs-keyword">to</span> <span class="hljs-keyword">use</span> a less stringent <span class="hljs-comment">--hwe p-value threshold for X</span><br>chromosome variants.<br><span class="hljs-comment">--hwe: 944 variants removed due to Hardy-Weinberg exact test.</span><br><span class="hljs-number">5061</span> variants removed due <span class="hljs-keyword">to</span> minor allele threshold(s)<br>(<span class="hljs-comment">--maf/--max-maf/--mac/--max-mac).</span><br><span class="hljs-number">540534</span> variants <span class="hljs-keyword">and</span> <span class="hljs-number">489</span> people pass filters <span class="hljs-keyword">and</span> QC.<br><span class="hljs-literal">Note</span>: No phenotypes present.<br>List <span class="hljs-keyword">of</span> variant IDs written <span class="hljs-keyword">to</span> EUR.QC.snplist .<br><span class="hljs-comment">--make-just-fam to EUR.QC.fam ... done.</span><br></code></pre></td></tr></table></figure>
<p>每个参数对应于以下内容：</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Value</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>bfile</td>
<td>EUR</td>
<td>Informs <code>plink</code> that the input genotype files should have a prefix of <code>EUR</code></td>
</tr>
<tr>
<td>maf</td>
<td>0.01</td>
<td>Removes all SNPs with minor allele frequency less than 0.01. Genotyping errors typically have a larger influence on SNPs with low MAF. Studies with large sample sizes could apply a lower MAF threshold</td>
</tr>
<tr>
<td>hwe</td>
<td>1e-6</td>
<td>Removes SNPs with low P-value from the Hardy-Weinberg Equilibrium Fisher’s exact or chi-squared test. SNPs with significant P-values from the HWE test are more likely affected by genotyping error or the effects of natural selection. Filtering should be performed on the control samples to avoid filtering SNPs that are causal (under selection in cases). When phenotype information is included, plink will automatically perform the filtering in the controls.</td>
</tr>
<tr>
<td>geno</td>
<td>0.01</td>
<td>Excludes SNPs that are missing in a high fraction of subjects. A two-stage filtering process is usually performed (see <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6001694/">Marees et al</a>).</td>
</tr>
<tr>
<td>mind</td>
<td>0.01</td>
<td>Excludes individuals who have a high rate of genotype missingness, since this may indicate problems in the DNA sample or processing. (see <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6001694/">Marees et al</a> for more details).</td>
</tr>
<tr>
<td>make-just-fam</td>
<td>-</td>
<td>Informs <code>plink</code> to only generate the QC’ed sample name to avoid generating the .bed file.</td>
</tr>
<tr>
<td>write-snplist</td>
<td>-</td>
<td>Informs <code>plink</code> to only generate the QC’ed SNP list to avoid generating the .bed file.</td>
</tr>
<tr>
<td>out</td>
<td>EUR.QC</td>
<td>Informs <code>plink</code> that all output should have a prefix of <code>EUR.QC</code></td>
</tr>
</tbody>
</table>
</div>
<blockquote>
<p>通常，我们可以使用新的样本列表生成一个新的基因型文件。但是，这会占用大量的存储空间。使用<code>plink</code>‘s <code>--extract</code>, <code>--exclude</code>, <code>--keep</code>, <code>--remove</code>, <code>--make-just-fam</code> and <code>--write-snplist</code> functions,我们可以单独处理示例和 SNPs 列表，而无需复制基因型文件，从而减少存储空间的使用。</p>
</blockquote>
<p>个体中非常高或低的杂合率可能是由于 DNA 污染或高度近交所致。因此，具有极端杂合性的样品通常会在下游分析之前被移除。</p>
<p>首先，我们执行修剪以去除高度相关的 SNPs:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs bash">plink \<br>    --bfile EUR \<br>    --keep EUR.QC.fam \<br>    --extract EUR.QC.snplist \<br>    --indep-pairwise 200 50 0.25 \<br>    --out EUR.QC<br></code></pre></td></tr></table></figure>
<p>每个参数对应于以下内容：</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Value</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>bfile</td>
<td>EUR</td>
<td>Informs <code>plink</code> that the input genotype files should have a prefix of <code>EUR</code></td>
</tr>
<tr>
<td>keep</td>
<td>EUR.QC.fam</td>
<td>Informs <code>plink</code> that we only want to use samples in <code>EUR.QC.fam</code> in the analysis</td>
</tr>
<tr>
<td>extract</td>
<td>EUR.QC.snplist</td>
<td>Informs <code>plink</code> that we only want to use SNPs in <code>EUR.QC.snplist</code> in the analysis</td>
</tr>
<tr>
<td>indep-pairwise</td>
<td>200 50 0.25</td>
<td>Informs <code>plink</code> that we wish to perform pruning with a window size of 200 variants, sliding across the genome with step size of 50 variants at a time, and filter out any SNPs with LD r2r2 higher than 0.25</td>
</tr>
<tr>
<td>out</td>
<td>EUR.QC</td>
<td>Informs <code>plink</code> that all output should have a prefix of <code>EUR.QC</code></td>
</tr>
</tbody>
</table>
</div>
<p>这将生成两个文件1) EUR.QC.prune.in 和2) EUR.QC.prune.out。</p>
<p>return</p>
<figure class="highlight mercury"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br></pre></td><td class="code"><pre><code class="hljs mercury">PLINK v1.<span class="hljs-number">90</span>b4 <span class="hljs-number">64</span>-bit (<span class="hljs-number">20</span> Mar <span class="hljs-number">2017</span>)             www.cog-genomics.org/plink/<span class="hljs-number">1.9</span>/<br>(C) <span class="hljs-number">2005</span>-<span class="hljs-number">2017</span> Shaun Purcell, Christopher Chang   GNU General Public License v3<br>Logging to EUR.QC.log.<br>Options in effect:<br>  --bfile EUR<br>  --extract EUR.QC.snplist<br>  --indep-pairwise <span class="hljs-number">200</span> <span class="hljs-number">50</span> <span class="hljs-number">0.25</span><br>  --keep EUR.QC.fam<br>  --out EUR.QC<br><br><span class="hljs-number">8192</span> MB RAM detected; reserving <span class="hljs-number">4096</span> MB for main workspace.<br><span class="hljs-number">551892</span> variants loaded from .bim file.<br><span class="hljs-number">503</span> people (<span class="hljs-number">240</span> males, <span class="hljs-number">263</span> females) loaded from .fam.<br>--extract: <span class="hljs-number">540534</span> variants remaining.<br>--keep: <span class="hljs-number">489</span> people remaining.<br>Using <span class="hljs-number">1</span> thread (no multithreaded calculations invoked).<br>Before main variant filters, <span class="hljs-number">489</span> founders and <span class="hljs-number">0</span> nonfounders present.<br>Calculating allele frequencies... done.<br>Warning: <span class="hljs-number">1273</span> het. haploid genotypes present (see EUR.QC.hh ); many commands<br>treat these as missing.<br>Total genotyping rate in remaining samples <span class="hljs-keyword">is</span> <span class="hljs-number">0.999919</span>.<br><span class="hljs-number">540534</span> variants and <span class="hljs-number">489</span> people pass filters and QC.<br>Note: No phenotypes present.<br>Pruned <span class="hljs-number">20214</span> variants from chromosome <span class="hljs-number">1</span>, leaving <span class="hljs-number">21127</span>.<br>Pruned <span class="hljs-number">20984</span> variants from chromosome <span class="hljs-number">2</span>, leaving <span class="hljs-number">20879</span>.<br>Pruned <span class="hljs-number">17463</span> variants from chromosome <span class="hljs-number">3</span>, leaving <span class="hljs-number">17812</span>.<br>Pruned <span class="hljs-number">16344</span> variants from chromosome <span class="hljs-number">4</span>, leaving <span class="hljs-number">16690</span>.<br>Pruned <span class="hljs-number">15027</span> variants from chromosome <span class="hljs-number">5</span>, leaving <span class="hljs-number">15981</span>.<br>Pruned <span class="hljs-number">20731</span> variants from chromosome <span class="hljs-number">6</span>, leaving <span class="hljs-number">15475</span>.<br>Pruned <span class="hljs-number">14046</span> variants from chromosome <span class="hljs-number">7</span>, leaving <span class="hljs-number">14606</span>.<br>Pruned <span class="hljs-number">13289</span> variants from chromosome <span class="hljs-number">8</span>, leaving <span class="hljs-number">13667</span>.<br>Pruned <span class="hljs-number">10723</span> variants from chromosome <span class="hljs-number">9</span>, leaving <span class="hljs-number">11951</span>.<br>Pruned <span class="hljs-number">12722</span> variants from chromosome <span class="hljs-number">10</span>, leaving <span class="hljs-number">13383</span>.<br>Pruned <span class="hljs-number">13053</span> variants from chromosome <span class="hljs-number">11</span>, leaving <span class="hljs-number">12666</span>.<br>Pruned <span class="hljs-number">11654</span> variants from chromosome <span class="hljs-number">12</span>, leaving <span class="hljs-number">12500</span>.<br>Pruned <span class="hljs-number">8429</span> variants from chromosome <span class="hljs-number">13</span>, leaving <span class="hljs-number">9609</span>.<br>Pruned <span class="hljs-number">8072</span> variants from chromosome <span class="hljs-number">14</span>, leaving <span class="hljs-number">8791</span>.<br>Pruned <span class="hljs-number">7978</span> variants from chromosome <span class="hljs-number">15</span>, leaving <span class="hljs-number">8470</span>.<br>Pruned <span class="hljs-number">8808</span> variants from chromosome <span class="hljs-number">16</span>, leaving <span class="hljs-number">9500</span>.<br>Pruned <span class="hljs-number">8032</span> variants from chromosome <span class="hljs-number">17</span>, leaving <span class="hljs-number">8724</span>.<br>Pruned <span class="hljs-number">7204</span> variants from chromosome <span class="hljs-number">18</span>, leaving <span class="hljs-number">8560</span>.<br>Pruned <span class="hljs-number">7014</span> variants from chromosome <span class="hljs-number">19</span>, leaving <span class="hljs-number">6726</span>.<br>Pruned <span class="hljs-number">6341</span> variants from chromosome <span class="hljs-number">20</span>, leaving <span class="hljs-number">7508</span>.<br>Pruned <span class="hljs-number">3629</span> variants from chromosome <span class="hljs-number">21</span>, leaving <span class="hljs-number">4280</span>.<br>Pruned <span class="hljs-number">4085</span> variants from chromosome <span class="hljs-number">22</span>, leaving <span class="hljs-number">4323</span>.<br>Pruned <span class="hljs-number">16235</span> variants from chromosome <span class="hljs-number">23</span>, leaving <span class="hljs-number">5229</span>.<br>Pruning complete.  <span class="hljs-number">272077</span> of <span class="hljs-number">540534</span> variants removed.<br>Marker lists written to EUR.QC.prune.in and EUR.QC.prune.out .<br></code></pre></td></tr></table></figure>
<p>杂合性率可以通过 plink 计算出来:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs bash">plink \<br>    --bfile EUR \<br>    --extract EUR.QC.prune.in \<br>    --keep EUR.QC.fam \<br>    --het \<br>    --out EUR.QC<br></code></pre></td></tr></table></figure>
<p>return</p>
<figure class="highlight reasonml"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><code class="hljs reasonml">PLINK v1.<span class="hljs-number">90</span>b4 <span class="hljs-number">64</span>-bit (<span class="hljs-number">20</span> Mar <span class="hljs-number">2017</span>)             www.cog-genomics.org/plink/<span class="hljs-number">1.9</span>/<br>(C) <span class="hljs-number">2005</span>-<span class="hljs-number">2017</span> Shaun Purcell, Christopher Chang   GNU General Public License v3<br>Logging <span class="hljs-keyword">to</span> <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>log.<br>Options <span class="hljs-keyword">in</span> effect:<br>  --bfile EUR<br>  --extract <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>prune.<span class="hljs-keyword">in</span><br>  --het<br>  --keep <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>fam<br>  --out EUR.QC<br><br><span class="hljs-number">8192</span> MB RAM detected; reserving <span class="hljs-number">4096</span> MB <span class="hljs-keyword">for</span> main workspace.<br><span class="hljs-number">551892</span> variants loaded from .bim file.<br><span class="hljs-number">503</span> people (<span class="hljs-number">240</span> males, <span class="hljs-number">263</span> females) loaded from .fam.<br>--extract: <span class="hljs-number">268457</span> variants remaining.<br>--keep: <span class="hljs-number">489</span> people remaining.<br>Using <span class="hljs-number">1</span> thread (no multithreaded calculations invoked).<br>Before main variant filters, <span class="hljs-number">489</span> founders <span class="hljs-keyword">and</span> <span class="hljs-number">0</span> nonfounders present.<br>Calculating allele frequencies... <span class="hljs-keyword">done</span>.<br>Warning: <span class="hljs-number">558</span> het. haploid genotypes present (see <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>hh ); many commands<br>treat these <span class="hljs-keyword">as</span> missing.<br>Total genotyping rate <span class="hljs-keyword">in</span> remaining samples is <span class="hljs-number">0.999958</span>.<br><span class="hljs-number">268457</span> variants <span class="hljs-keyword">and</span> <span class="hljs-number">489</span> people pass filters <span class="hljs-keyword">and</span> QC.<br>Note: No phenotypes present.<br>--het: <span class="hljs-number">263228</span> variants scanned, report written <span class="hljs-keyword">to</span> <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>het .<br></code></pre></td></tr></table></figure>
<p> 这将生成 EUR.QC.het 文件，其中包含用于评估杂合性的 f 系数估计。我们将从平均值中删除 f 系数大于3的个体，这可以使用下面的 r 命令执行(假设你已经下载了 r，那么你可以在终端中键入 r 来打开一个 r 会话) :</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs r">dat &lt;- read.table(<span class="hljs-string">&quot;EUR.QC.het&quot;</span>, header=<span class="hljs-built_in">T</span>) <span class="hljs-comment"># Read in the EUR.het file, specify it has header</span><br>m &lt;- mean(dat$<span class="hljs-built_in">F</span>) <span class="hljs-comment"># Calculate the mean  </span><br>s &lt;- sd(dat$<span class="hljs-built_in">F</span>) <span class="hljs-comment"># Calculate the SD</span><br>valid &lt;- subset(dat, <span class="hljs-built_in">F</span> &lt;= m+<span class="hljs-number">3</span>*s &amp; <span class="hljs-built_in">F</span> &gt;= m-<span class="hljs-number">3</span>*s) <span class="hljs-comment"># Get any samples with F coefficient within 3 SD of the population mean</span><br>write.table(valid[,<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">2</span>)], <span class="hljs-string">&quot;EUR.valid.sample&quot;</span>, <span class="hljs-built_in">quote</span>=<span class="hljs-built_in">F</span>, row.names=<span class="hljs-built_in">F</span>) <span class="hljs-comment"># print FID and IID for valid samples</span><br>q() <span class="hljs-comment"># exit R</span><br></code></pre></td></tr></table></figure>
<h3><span id="2-7-qc-checklist-ambiguous-snps">2.7 QC checklist: Ambiguous SNPs</span><a href="#2-7-qc-checklist-ambiguous-snps" class="header-anchor">#</a></h3><p>这些将在基础数据中被删除。</p>
<h3><span id="2-8-qc-checklist-mismatching-snps">2.8 QC checklist: Mismatching SNPs</span><a href="#2-8-qc-checklist-mismatching-snps" class="header-anchor">#</a></h3><p>在基础数据和目标数据中报告的等位基因不匹配的 SNP，可以通过将等位基因翻转到它们的互补等位基因来解决，例如目标数据，例如基础数据中有 a/c 的 SNP 和目标数据中的 g/t。这可以通过以下步骤来实现:</p>
<blockquote>
<p>大多数 PRS 软件将自动执行链切换，因此这一步骤通常是不需要的。</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#将 bim 文件、汇总统计和 QC SNP 列表加载到 r 中</span><br><br><span class="hljs-comment">#有data.table和magritr包</span><br><span class="hljs-comment"># magrittr allow us to do piping, which help to reduce the </span><br><span class="hljs-comment"># amount of intermediate data types</span><br>library(data.table)<br>library(magrittr)<br>setwd(<span class="hljs-string">&quot;/Volumes/My Passport for Mac/data/EUR&quot;</span>)<br><span class="hljs-comment"># Read in bim file </span><br>bim &lt;- fread(<span class="hljs-string">&quot;EUR.bim&quot;</span>) %&gt;%<br>    <span class="hljs-comment"># Note: . represents the output from previous step</span><br>    <span class="hljs-comment"># The syntax here means, setnames of the data read from</span><br>    <span class="hljs-comment"># the bim file, and replace the original column names by </span><br>    <span class="hljs-comment"># the new names</span><br>    setnames(., colnames(.), <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;CHR&quot;</span>, <span class="hljs-string">&quot;SNP&quot;</span>, <span class="hljs-string">&quot;CM&quot;</span>, <span class="hljs-string">&quot;BP&quot;</span>, <span class="hljs-string">&quot;B.A1&quot;</span>, <span class="hljs-string">&quot;B.A2&quot;</span>)) %&gt;%<br>    <span class="hljs-comment"># And immediately change the alleles to upper cases</span><br>    .[,<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;B.A1&quot;</span>,<span class="hljs-string">&quot;B.A2&quot;</span>):=<span class="hljs-built_in">list</span>(toupper(B.A1), toupper(B.A2))]<br><span class="hljs-comment"># Read in summary statistic data (require data.table v1.12.0+)</span><br>height &lt;- fread(<span class="hljs-string">&quot;/Volumes/My\ Passport\ for\ Mac/data/14671989/iPSYCH_PGC_ASD_Nov2017.gz&quot;</span>) %&gt;%<br>    <span class="hljs-comment"># And immediately change the alleles to upper cases</span><br>    .[,<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;A1&quot;</span>,<span class="hljs-string">&quot;A2&quot;</span>):=<span class="hljs-built_in">list</span>(toupper(A1), toupper(A2))]<br><span class="hljs-comment"># Read in QCed SNPs</span><br>qc &lt;- fread(<span class="hljs-string">&quot;EUR.QC.snplist&quot;</span>, header=<span class="hljs-built_in">F</span>)<br><br><span class="hljs-comment"># 识别需要翻转 dna 链的 SNPs</span><br><span class="hljs-comment"># Merge summary statistic with target</span><br>info &lt;- merge(bim, height, by=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;SNP&quot;</span>, <span class="hljs-string">&quot;CHR&quot;</span>, <span class="hljs-string">&quot;BP&quot;</span>)) %&gt;%<br>    <span class="hljs-comment"># And filter out QCed SNPs</span><br>    .[SNP %in% qc[,V1]]<br><br><span class="hljs-comment"># Function for calculating the complementary allele</span><br>complement &lt;- <span class="hljs-keyword">function</span>(x)&#123;<br>    <span class="hljs-built_in">switch</span> (x,<br>        <span class="hljs-string">&quot;A&quot;</span> = <span class="hljs-string">&quot;T&quot;</span>,<br>        <span class="hljs-string">&quot;C&quot;</span> = <span class="hljs-string">&quot;G&quot;</span>,<br>        <span class="hljs-string">&quot;T&quot;</span> = <span class="hljs-string">&quot;A&quot;</span>,<br>        <span class="hljs-string">&quot;G&quot;</span> = <span class="hljs-string">&quot;C&quot;</span>,<br>        <span class="hljs-built_in">return</span>(<span class="hljs-literal">NA</span>)<br>    )<br>&#125; <br><span class="hljs-comment"># Get SNPs that have the same alleles across base and target</span><br>info.match &lt;- info[A1 == B.A1 &amp; A2 == B.A2, SNP]<br><span class="hljs-comment"># Identify SNPs that are complementary between base and target</span><br>com.snps &lt;- info[sapply(B.A1, complement) == A1 &amp;<br>                    sapply(B.A2, complement) == A2, SNP]<br><span class="hljs-comment"># Now update the bim file</span><br>bim[SNP %in% com.snps, <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;B.A1&quot;</span>, <span class="hljs-string">&quot;B.A2&quot;</span>) :=<br>        <span class="hljs-built_in">list</span>(sapply(B.A1, complement),<br>            sapply(B.A2, complement))]<br><br><span class="hljs-comment">#识别需要在目标中重新编码的 SNPs (以确保在目标数据中编码的等位基因是基础摘要统计中的有效等位基因)</span><br><span class="hljs-comment"># identify SNPs that need recoding</span><br>recode.snps &lt;- info[B.A1==A2 &amp; B.A2==A1, SNP]<br><span class="hljs-comment"># Update the bim file</span><br>bim[SNP %in% recode.snps, <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;B.A1&quot;</span>, <span class="hljs-string">&quot;B.A2&quot;</span>) :=<br>        <span class="hljs-built_in">list</span>(B.A2, B.A1)]<br><br><span class="hljs-comment"># identify SNPs that need recoding &amp; complement</span><br>com.recode &lt;- info[sapply(B.A1, complement) == A2 &amp;<br>                    sapply(B.A2, complement) == A1, SNP]<br><span class="hljs-comment"># Now update the bim file</span><br>bim[SNP %in% com.recode, <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;B.A1&quot;</span>, <span class="hljs-string">&quot;B.A2&quot;</span>) :=<br>        <span class="hljs-built_in">list</span>(sapply(B.A2, complement),<br>            sapply(B.A1, complement))]<br><span class="hljs-comment"># Write the updated bim file</span><br>fwrite(bim[,<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;SNP&quot;</span>, <span class="hljs-string">&quot;B.A1&quot;</span>)], <span class="hljs-string">&quot;EUR.a1&quot;</span>, col.names=<span class="hljs-built_in">F</span>, sep=<span class="hljs-string">&quot;\t&quot;</span>)<br><br><span class="hljs-comment"># 确定在基因组和目标上有不同等位基因的 SNPs (通常是由于基因组构建或 Indel 的不同)</span><br>mismatch &lt;- bim[!(SNP %in% info.match |<br>                    SNP %in% com.snps |<br>                    SNP %in% recode.snps |<br>                    SNP %in% com.recode), SNP]<br>write.table(mismatch, <span class="hljs-string">&quot;EUR.mismatch&quot;</span>, <span class="hljs-built_in">quote</span>=<span class="hljs-built_in">F</span>, row.names=<span class="hljs-built_in">F</span>, col.names=<span class="hljs-built_in">F</span>)<br>q() <span class="hljs-comment"># exit R</span><br></code></pre></td></tr></table></figure>
<h3><span id="2-9-qc-checklist-duplicate-snps">2.9 QC checklist: Duplicate SNPs</span><a href="#2-9-qc-checklist-duplicate-snps" class="header-anchor">#</a></h3><p>确保删除目标数据中任何重复的 SNPs (这些目标数据是模拟的，因此不包括重复的 SNPs)。</p>
<h3><span id="2-10-qc-checklist-sex-chromosomes">2.10 QC checklist: Sex chromosomes</span><a href="#2-10-qc-checklist-sex-chromosomes" class="header-anchor">#</a></h3><p>有时样品标签会出现错误，可能导致无效的结果。标签错误的一个迹象是报告的性别和性染色体之间的差异。虽然这可能是由于性别和性别认同的差异，但也可能反映出样本标签错误或误报，因此，生物性别和报告性别不匹配的个人通常被剔除。性别检查可以在 PLINK 中进行，如果个体的 x 染色体纯合度估计值(f 统计量) &lt; 0.2，如果估计值大于0.8，个体被称为女性。</p>
<p>在执行性别检查之前，应该进行修剪。然后可以很容易地使用 plink 进行性别检查。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs bash">plink \<br>    --bfile EUR \<br>    --extract EUR.QC.prune.in \<br>    --keep EUR.valid.sample \<br>    --check-sex \<br>    --out EUR.QC<br></code></pre></td></tr></table></figure>
<p>return</p>
<figure class="highlight reasonml"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><code class="hljs reasonml">PLINK v1.<span class="hljs-number">90</span>b4 <span class="hljs-number">64</span>-bit (<span class="hljs-number">20</span> Mar <span class="hljs-number">2017</span>)             www.cog-genomics.org/plink/<span class="hljs-number">1.9</span>/<br>(C) <span class="hljs-number">2005</span>-<span class="hljs-number">2017</span> Shaun Purcell, Christopher Chang   GNU General Public License v3<br>Logging <span class="hljs-keyword">to</span> <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>log.<br>Options <span class="hljs-keyword">in</span> effect:<br>  --bfile EUR<br>  --check-sex<br>  --extract <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>prune.<span class="hljs-keyword">in</span><br>  --keep <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span></span>valid.sample<br>  --out EUR.QC<br><br><span class="hljs-number">8192</span> MB RAM detected; reserving <span class="hljs-number">4096</span> MB <span class="hljs-keyword">for</span> main workspace.<br><span class="hljs-number">551892</span> variants loaded from .bim file.<br><span class="hljs-number">503</span> people (<span class="hljs-number">240</span> males, <span class="hljs-number">263</span> females) loaded from .fam.<br>--extract: <span class="hljs-number">268457</span> variants remaining.<br>--keep: <span class="hljs-number">487</span> people remaining.<br>Using <span class="hljs-number">1</span> thread (no multithreaded calculations invoked).<br>Before main variant filters, <span class="hljs-number">487</span> founders <span class="hljs-keyword">and</span> <span class="hljs-number">0</span> nonfounders present.<br>Calculating allele frequencies... <span class="hljs-keyword">done</span>.<br>Warning: <span class="hljs-number">558</span> het. haploid genotypes present (see <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>hh ); many commands<br>treat these <span class="hljs-keyword">as</span> missing.<br>Total genotyping rate <span class="hljs-keyword">in</span> remaining samples is <span class="hljs-number">0.999958</span>.<br><span class="hljs-number">268457</span> variants <span class="hljs-keyword">and</span> <span class="hljs-number">487</span> people pass filters <span class="hljs-keyword">and</span> QC.<br>Note: No phenotypes present.<br>--check-sex: <span class="hljs-number">5229</span> Xchr <span class="hljs-keyword">and</span> <span class="hljs-number">0</span> Ychr variant(s) scanned, <span class="hljs-number">4</span> problems detected.<br>Report written <span class="hljs-keyword">to</span> <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>sexcheck .<br></code></pre></td></tr></table></figure>
<p>这将生成一个名为 EUR.QC.sexcheck 的文件，其中包含每个人的 f- 统计信息。如果 f 统计值大于0.8，个体通常被称为生物学上的男性，如果 f &lt; 0.2，则被称为生物学上的女性。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs R">library(data.table)<br><span class="hljs-comment"># Read in file</span><br>valid &lt;- fread(<span class="hljs-string">&quot;EUR.valid.sample&quot;</span>)<br>dat &lt;- fread(<span class="hljs-string">&quot;EUR.QC.sexcheck&quot;</span>)[FID%in%valid$FID]<br>fwrite(dat[STATUS==<span class="hljs-string">&quot;OK&quot;</span>,<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;FID&quot;</span>,<span class="hljs-string">&quot;IID&quot;</span>)], <span class="hljs-string">&quot;EUR.QC.valid&quot;</span>, sep=<span class="hljs-string">&quot;\t&quot;</span>) <br>q() <span class="hljs-comment"># exit R</span><br></code></pre></td></tr></table></figure>
<h3><span id="2-11-qc-checklist-sample-overlap">2.11 QC checklist: Sample overlap</span><a href="#2-11-qc-checklist-sample-overlap" class="header-anchor">#</a></h3><p>由于目标数据是模拟的，所以这里的基础数据和目标数据之间没有重叠的样本(参见文件的相关部分讨论避免样本重叠的重要性)。</p>
<h3><span id="2-12-qc-checklist-relatedness">2.12 QC checklist: Relatedness</span><a href="#2-12-qc-checklist-relatedness" class="header-anchor">#</a></h3><p>目标数据中密切相关的个体可能导致过度拟合的结果，限制了结果的通用性。</p>
<p>在计算相关性之前，应该执行修剪。样本中有一个一级或二级亲属的个体(^ π &gt; 0.125)可以通过以下命令删除:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs bash">plink \<br>    --bfile EUR \<br>    --extract EUR.QC.prune.in \<br>    --keep EUR.QC.valid \<br>    --rel-cutoff 0.125 \<br>    --out EUR.QC<br></code></pre></td></tr></table></figure>
<p>return</p>
<figure class="highlight applescript"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><code class="hljs applescript">PLINK v1<span class="hljs-number">.90</span>b4 <span class="hljs-number">64</span>-bit (<span class="hljs-number">20</span> Mar <span class="hljs-number">2017</span>)             www.cog-genomics.org/plink/<span class="hljs-number">1.9</span>/<br>(C) <span class="hljs-number">2005</span><span class="hljs-number">-2017</span> Shaun Purcell, Christopher Chang   GNU General Public License v3<br>Logging <span class="hljs-keyword">to</span> EUR.QC.<span class="hljs-built_in">log</span>.<br>Options <span class="hljs-keyword">in</span> effect:<br>  <span class="hljs-comment">--bfile EUR</span><br>  <span class="hljs-comment">--extract EUR.QC.prune.in</span><br>  <span class="hljs-comment">--keep EUR.QC.valid</span><br>  <span class="hljs-comment">--out EUR.QC</span><br>  <span class="hljs-comment">--rel-cutoff 0.125</span><br><br><span class="hljs-number">8192</span> MB RAM detected; reserving <span class="hljs-number">4096</span> MB <span class="hljs-keyword">for</span> main workspace.<br><span class="hljs-number">551892</span> variants loaded <span class="hljs-keyword">from</span> .bim <span class="hljs-built_in">file</span>.<br><span class="hljs-number">503</span> people (<span class="hljs-number">240</span> males, <span class="hljs-number">263</span> females) loaded <span class="hljs-keyword">from</span> .fam.<br><span class="hljs-comment">--extract: 268457 variants remaining.</span><br><span class="hljs-comment">--keep: 483 people remaining.</span><br>Using up <span class="hljs-keyword">to</span> <span class="hljs-number">4</span> threads (change this <span class="hljs-keyword">with</span> <span class="hljs-comment">--threads).</span><br>Before main variant filters, <span class="hljs-number">483</span> founders <span class="hljs-keyword">and</span> <span class="hljs-number">0</span> nonfounders present.<br>Calculating allele frequencies... done.<br>Total genotyping rate <span class="hljs-keyword">in</span> remaining samples <span class="hljs-keyword">is</span> <span class="hljs-number">0.999957</span>.<br><span class="hljs-number">268457</span> variants <span class="hljs-keyword">and</span> <span class="hljs-number">483</span> people pass filters <span class="hljs-keyword">and</span> QC (<span class="hljs-keyword">before</span> <span class="hljs-comment">--rel-cutoff).</span><br>Note: No phenotypes present.<br>Excluding <span class="hljs-number">5229</span> variants <span class="hljs-keyword">on</span> non-autosomes <span class="hljs-keyword">from</span> relationship matrix calc.<br>Relationship matrix calculation complete.<br><span class="hljs-number">0</span> people excluded <span class="hljs-keyword">by</span> <span class="hljs-comment">--rel-cutoff.</span><br>Remaining sample IDs written <span class="hljs-keyword">to</span> EUR.QC.rel.<span class="hljs-built_in">id</span> .<br></code></pre></td></tr></table></figure>
<h3><span id="2-13-qc-checklist-generate-final-qc-ed-target-data-file">2.13 QC checklist: Generate final QC’ed target data file</span><a href="#2-13-qc-checklist-generate-final-qc-ed-target-data-file" class="header-anchor">#</a></h3><p>在执行完整的分析之后，您可以使用以下命令生成一个 QC’ed 数据集:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><code class="hljs bash">plink \<br>    --bfile EUR \<br>    --make-bed \<br>    --keep EUR.QC.rel.id \<br>    --out EUR.QC \<br>    --extract EUR.QC.snplist \<br>    --exclude EUR.mismatch \<br>    --a1-allele EUR.a1<br></code></pre></td></tr></table></figure>
<p>每个参数对应于以下内容</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Value</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>bfile</td>
<td>EUR</td>
<td>Informs <code>plink</code> that the input genotype files should have a prefix of <code>EUR</code></td>
</tr>
<tr>
<td>keep</td>
<td>EUR.QC.rel.id</td>
<td>Informs <code>plink</code> that we only want to keep samples in <code>EUR.QC.rel.id</code></td>
</tr>
<tr>
<td>extract</td>
<td>EUR.QC.snplist</td>
<td>Informs <code>plink</code> that we only want to use SNPs in <code>EUR.QC.snplist</code> in the analysis</td>
</tr>
<tr>
<td>exclude</td>
<td>EUR.mismatch</td>
<td>Informs <code>plink</code> that we wish to remove any SNPs in <code>EUR.mismatch</code></td>
</tr>
<tr>
<td>a1-allele</td>
<td>EUR.a1</td>
<td>Fix all A1 alleles to those specified in <code>EUR.a1</code></td>
</tr>
<tr>
<td>out</td>
<td>EUR.QC</td>
<td>Informs <code>plink</code> that all output should have a prefix of <code>EUR.QC</code></td>
</tr>
</tbody>
</table>
</div>
<p>return</p>
<figure class="highlight reasonml"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><code class="hljs reasonml">PLINK v1.<span class="hljs-number">90</span>b4 <span class="hljs-number">64</span>-bit (<span class="hljs-number">20</span> Mar <span class="hljs-number">2017</span>)             www.cog-genomics.org/plink/<span class="hljs-number">1.9</span>/<br>(C) <span class="hljs-number">2005</span>-<span class="hljs-number">2017</span> Shaun Purcell, Christopher Chang   GNU General Public License v3<br>Logging <span class="hljs-keyword">to</span> <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>log.<br>Options <span class="hljs-keyword">in</span> effect:<br>  --a1-allele <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span></span>a1<br>  --bfile EUR<br>  --exclude <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span></span>mismatch<br>  --extract <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>snplist<br>  --keep <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>rel.id<br>  --make-bed<br>  --out EUR.QC<br><br><span class="hljs-number">8192</span> MB RAM detected; reserving <span class="hljs-number">4096</span> MB <span class="hljs-keyword">for</span> main workspace.<br><span class="hljs-number">551892</span> variants loaded from .bim file.<br><span class="hljs-number">503</span> people (<span class="hljs-number">240</span> males, <span class="hljs-number">263</span> females) loaded from .fam.<br>--extract: <span class="hljs-number">540534</span> variants remaining.<br>--exclude: <span class="hljs-number">465549</span> variants remaining.<br>--keep: <span class="hljs-number">483</span> people remaining.<br>Using <span class="hljs-number">1</span> thread (no multithreaded calculations invoked).<br>Before main variant filters, <span class="hljs-number">483</span> founders <span class="hljs-keyword">and</span> <span class="hljs-number">0</span> nonfounders present.<br>Calculating allele frequencies... <span class="hljs-keyword">done</span>.<br>--a1-allele: <span class="hljs-number">465549</span> assignments made.<br><span class="hljs-number">465549</span> variants <span class="hljs-keyword">and</span> <span class="hljs-number">483</span> people pass filters <span class="hljs-keyword">and</span> QC.<br>Note: No phenotypes present.<br>--make-bed <span class="hljs-keyword">to</span> <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>bed + <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>bim + <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">EUR</span>.</span><span class="hljs-module"><span class="hljs-identifier">QC</span>.</span></span>fam<span class="hljs-operator"> ... </span><span class="hljs-keyword">done</span>.<br></code></pre></td></tr></table></figure>
<h2><span id="3-calculating-prs">3. calculating PRS</span><a href="#3-calculating-prs" class="header-anchor">#</a></h2><h2><span id="3-1-plink">3.1 plink</span><a href="#3-1-plink" class="header-anchor">#</a></h2><h4><span id="3-1-1-required-data">3.1.1 Required Data</span><a href="#3-1-1-required-data" class="header-anchor">#</a></h4><p>在之前，我们已经生成了以下文件</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>File Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Height.QC.gz</strong></td>
<td>The post-QCed summary statistic</td>
</tr>
<tr>
<td><strong>EUR.QC.bed</strong></td>
<td>The genotype file after performing some basic filtering</td>
</tr>
<tr>
<td><strong>EUR.QC.bim</strong></td>
<td>This file contains the SNPs that passed the basic filtering</td>
</tr>
<tr>
<td><strong>EUR.QC.fam</strong></td>
<td>This file contains the samples that passed the basic filtering</td>
</tr>
<tr>
<td><strong>EUR.height</strong></td>
<td>This file contains the phenotype of the samples</td>
</tr>
<tr>
<td><strong>EUR.cov</strong></td>
<td>This file contains the covariates of the samples</td>
</tr>
</tbody>
</table>
</div>
<h4><span id="3-1-2-update-effect-size">3.1.2 Update Effect Size</span><a href="#3-1-2-update-effect-size" class="header-anchor">#</a></h4><p>当效应大小与疾病风险有关，因此以优势比(OR)而不是 BETA (连续性状)给出时，则 PRS 被计算为 OR 的乘积。为了简化这个计算，我们使用 OR 的自然对数，这样可以用求和来计算 PRS (之后可以反向转换)。我们可以用 r 得到转换后的汇总统计量:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs R">library(data.table)<br>dat &lt;- fread(<span class="hljs-string">&quot;Height.QC.gz&quot;</span>)<br>fwrite(dat[,BETA:=<span class="hljs-built_in">log</span>(OR)], <span class="hljs-string">&quot;Height.QC.Transformed&quot;</span>, sep=<span class="hljs-string">&quot;\t&quot;</span>)<br></code></pre></td></tr></table></figure>
<h4><span id="3-1-3-clumping">3.1.3 Clumping</span><a href="#3-1-3-clumping" class="header-anchor">#</a></h4><p>连接不平衡，相当于整个基因组中遗传变体基因型之间的相关性，使得识别因果独立遗传变体的贡献极具挑战性。捕捉合适水平的因果信号的方法之一是进行聚类，即删除SNP，只保留弱相关的SNP，但优先保留与所研究表型最相关的SNP。可以使用plink中的以下命令进行聚类。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs bash">plink \<br>    --bfile EUR.QC \<br>    --clump-p1 1 \<br>    --clump-r2 0.1 \<br>    --clump-kb 250 \<br>    --clump Height.QC.Transformed \<br>    --clump-snp-field SNP \<br>    --clump-field P \<br>    --out EUR<br></code></pre></td></tr></table></figure>
<p>相应的参数见下表：</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Value</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>clump-p1</td>
<td>1</td>
<td>P-value threshold for a SNP to be included as an index SNP. 1 is selected such that all SNPs are include for clumping</td>
</tr>
<tr>
<td>clump-r2</td>
<td>0.1</td>
<td>SNPs having r2r2 higher than 0.1 with the index SNPs will be removed</td>
</tr>
<tr>
<td>clump-kb</td>
<td>250</td>
<td>SNPs within 250k of the index SNP are considered for clumping</td>
</tr>
<tr>
<td>clump</td>
<td>Height.QC.Transformed</td>
<td>Base data (summary statistic) file containing the P-value information</td>
</tr>
<tr>
<td>clump-snp-field</td>
<td>SNP</td>
<td>Specifies that the column <code>SNP</code> contains the SNP IDs</td>
</tr>
<tr>
<td>clump-field</td>
<td>P</td>
<td>Specifies that the column <code>P</code> contains the P-value information</td>
</tr>
</tbody>
</table>
</div>
<p>return</p>
<figure class="highlight livecodeserver"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><code class="hljs livecodeserver">PLINK v1<span class="hljs-number">.90</span>b4 <span class="hljs-number">64</span>-bit (<span class="hljs-number">20</span> Mar <span class="hljs-number">2017</span>)             www.cog-genomics.org/plink/<span class="hljs-number">1.9</span>/<br>(C) <span class="hljs-number">2005</span><span class="hljs-number">-2017</span> Shaun Purcell, Christopher Chang   GNU General Public License v3<br>Logging <span class="hljs-built_in">to</span> EUR.<span class="hljs-built_in">log</span>.<br>Options <span class="hljs-keyword">in</span> effect:<br>  <span class="hljs-comment">--bfile EUR.QC</span><br>  <span class="hljs-comment">--clump Height.QC.Transformed</span><br>  <span class="hljs-comment">--clump-field P</span><br>  <span class="hljs-comment">--clump-kb 250</span><br>  <span class="hljs-comment">--clump-p1 1</span><br>  <span class="hljs-comment">--clump-r2 0.1</span><br>  <span class="hljs-comment">--clump-snp-field SNP</span><br>  <span class="hljs-comment">--out EUR</span><br><br><span class="hljs-number">8192</span> MB RAM detected; reserving <span class="hljs-number">4096</span> MB <span class="hljs-keyword">for</span> main workspace.<br><span class="hljs-number">465549</span> variants loaded <span class="hljs-built_in">from</span> .bim <span class="hljs-built_in">file</span>.<br><span class="hljs-number">483</span> people (<span class="hljs-number">232</span> males, <span class="hljs-number">251</span> females) loaded <span class="hljs-built_in">from</span> .fam.<br>Using <span class="hljs-number">1</span> thread (no multithreaded calculations invoked).<br>Before main variant filters, <span class="hljs-number">483</span> founders <span class="hljs-keyword">and</span> <span class="hljs-number">0</span> nonfounders present.<br>Calculating allele frequencies... done.<br><span class="hljs-number">465549</span> variants <span class="hljs-keyword">and</span> <span class="hljs-number">483</span> people pass filters <span class="hljs-keyword">and</span> QC.<br>Note: No phenotypes present.<br>Warning: <span class="hljs-string">&#x27;rs1076829&#x27;</span> is missing <span class="hljs-built_in">from</span> <span class="hljs-keyword">the</span> main dataset, <span class="hljs-keyword">and</span> is <span class="hljs-keyword">a</span> top variant.<br>Warning: <span class="hljs-string">&#x27;rs3129818&#x27;</span> is missing <span class="hljs-built_in">from</span> <span class="hljs-keyword">the</span> main dataset, <span class="hljs-keyword">and</span> is <span class="hljs-keyword">a</span> top variant.<br>Warning: <span class="hljs-string">&#x27;rs3118359&#x27;</span> is missing <span class="hljs-built_in">from</span> <span class="hljs-keyword">the</span> main dataset, <span class="hljs-keyword">and</span> is <span class="hljs-keyword">a</span> top variant.<br><span class="hljs-number">59884</span> more top variant IDs missing; see <span class="hljs-built_in">log</span> <span class="hljs-built_in">file</span>.<br><span class="hljs-comment">--clump: 164619 clumps formed from 439730 top variants.</span><br>Results written <span class="hljs-built_in">to</span> EUR.clumped .<br></code></pre></td></tr></table></figure>
<p>由-clump 计算的 r2 值是基于最大似然单倍型频率估计</p>
<p>这将生成 eur.clandped，包含执行成丛后的索引 SNPs。我们可以通过执行以下命令提取索引 SNP ID:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">awk <span class="hljs-string">&#x27;NR!=1&#123;print $3&#125;&#x27;</span> EUR.clumped &gt;  EUR.valid.snp<br></code></pre></td></tr></table></figure>
<blockquote>
<p>如果你的目标数据很小(例如 n &lt; 500) ，你可以使用千人基因组计划样本来计算 LD。确保使用最接近反映基本样本的总体。</p>
</blockquote>
<h4><span id="3-1-4-generate-prs">3.1.4 Generate PRS</span><a href="#3-1-4-generate-prs" class="header-anchor">#</a></h4><p>plink提供了两个有效的函数 —score 和 —q-score-range 用于计算PRS得分</p>
<p>为此，我们需要以下文件：</p>
<p>基础数据文件 Height.QC.Transformed</p>
<p>一个包含SNP ID和它们相应的P值的文件（$3是因为SNP ID位于第三列；$8是因为P值位于第八列)</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">awk <span class="hljs-string">&#x27;&#123;print $3,$8&#125;&#x27;</span> Height.QC.Transformed &gt; SNP.pvalue<br></code></pre></td></tr></table></figure>
<p>一个包含将SNPs纳入PRS的不同P值阈值的文件。这里计算与几个阈值相对应的PRS，以示说明。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;0.001 0 0.001&quot;</span> &gt; range_list <br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;0.05 0 0.05&quot;</span> &gt;&gt; range_list<br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;0.1 0 0.1&quot;</span> &gt;&gt; range_list<br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;0.2 0 0.2&quot;</span> &gt;&gt; range_list<br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;0.3 0 0.3&quot;</span> &gt;&gt; range_list<br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;0.4 0 0.4&quot;</span> &gt;&gt; range_list<br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;0.5 0 0.5&quot;</span> &gt;&gt; range_list<br></code></pre></td></tr></table></figure>
<p>然后我们可以使用以下命令计算PRS:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs bash">plink \<br>    --bfile EUR.QC \<br>    --score Height.QC.Transformed 3 4 12 header \<br>    --q-score-range range_list SNP.pvalue \<br>    --extract EUR.valid.snp \<br>    --out EUR<br></code></pre></td></tr></table></figure>
<p>参数如下：</p>
<div class="table-container">
<table>
<thead>
<tr>
<th>Paramter</th>
<th>Value</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>score</td>
<td>Height.QC.Transformed 3 4 12 header</td>
<td>We read from the <strong>Height.QC.Transformed</strong> file, assuming that the <code>3</code>st column is the SNP ID; <code>4</code>th column is the effective allele information; the <code>12</code>th column is the effect size estimate; and that the file contains a <code>header</code></td>
</tr>
<tr>
<td>q-score-range</td>
<td>range_list SNP.pvalue</td>
<td>We want to calculate PRS based on the thresholds defined in <strong>range_list</strong>, where the threshold values (P-values) were stored in <strong>SNP.pvalue</strong></td>
</tr>
</tbody>
</table>
</div>
<p>这将会产生7个文件：</p>
<ol>
<li>EUR.0.5.profile</li>
<li>EUR.0.4.profile</li>
<li>EUR.0.3.profile</li>
<li>EUR.0.2.profile</li>
<li>EUR.0.1.profile</li>
<li>EUR.0.05.profile</li>
<li>EUR.0.001.profile</li>
</ol>
<p><img src="/Users/keqinliu/Library/Application Support/typora-user-images/截屏2021-10-15 上午11.01.09.png" srcset="/img/loading.gif" alt="截屏2021-10-15 上午11.01.09"></p>
<h4><span id="3-1-5-accounting-for-population-stratification">3.1.5 Accounting for Population Stratification</span><a href="#3-1-5-accounting-for-population-stratification" class="header-anchor">#</a></h4><p>人口结构是 GWAS 中混杂的主要来源，通常由主成分(pc)作为协变量来解释。我们可以将pc纳入我们的 PRS 分析，以考虑人口分层。</p>
<p>我们可以使用 plink 计算 pc:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-comment"># First, we need to perform prunning</span><br>plink \<br>    --bfile EUR.QC \<br>    --indep-pairwise 200 50 0.25 \<br>    --out EUR<br><span class="hljs-comment"># Then we calculate the first 6 PCs</span><br>plink \<br>    --bfile EUR.QC \<br>    --extract EUR.prune.in \<br>    --pca 6 \<br>    --out EUR<br></code></pre></td></tr></table></figure>
<p>在这里 pc 已经被存储在 EUR.eigenvec 文件中，并且可以用作回归模型中的协变量来解释种群分层。</p>
<h4><span id="3-1-6-finding-the-best-fit-prs">3.1.6 Finding the “best-fit” PRS</span><a href="#3-1-6-finding-the-best-fit-prs" class="header-anchor">#</a></h4><p>在C+T方法下提供 “最佳拟合 “PRS的P值阈值通常是未知的。为了接近 “最佳拟合 “的PRS，我们可以在一系列P值阈值下计算的PRS之间进行回归，然后选择能够解释最高表型变异的PRS（请参见我们论文中关于过拟合问题的第4.6节）。这可以用R语言实现，方法如下。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><code class="hljs R">library(data.table)<br>library(magrittr)<br>p.threshold &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-number">0.001</span>,<span class="hljs-number">0.05</span>,<span class="hljs-number">0.1</span>,<span class="hljs-number">0.2</span>,<span class="hljs-number">0.3</span>,<span class="hljs-number">0.4</span>,<span class="hljs-number">0.5</span>)<br>phenotype &lt;- fread(<span class="hljs-string">&quot;EUR.height&quot;</span>)<br>pcs &lt;- fread(<span class="hljs-string">&quot;EUR.eigenvec&quot;</span>, header=<span class="hljs-built_in">F</span>) %&gt;%<br>    setnames(., colnames(.), <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;FID&quot;</span>, <span class="hljs-string">&quot;IID&quot;</span>, paste0(<span class="hljs-string">&quot;PC&quot;</span>,<span class="hljs-number">1</span>:<span class="hljs-number">6</span>)) )<br>covariate &lt;- fread(<span class="hljs-string">&quot;EUR.cov&quot;</span>)<br>pheno &lt;- merge(phenotype, covariate) %&gt;%<br>        merge(., pcs)<br>null.r2 &lt;- summary(lm(Height~., data=pheno[,-<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;FID&quot;</span>, <span class="hljs-string">&quot;IID&quot;</span>)]))$r.squared<br>prs.result &lt;- <span class="hljs-literal">NULL</span><br><span class="hljs-keyword">for</span>(i <span class="hljs-keyword">in</span> p.threshold)&#123;<br>    pheno.prs &lt;- paste0(<span class="hljs-string">&quot;EUR.&quot;</span>, i, <span class="hljs-string">&quot;.profile&quot;</span>) %&gt;%<br>        fread(.) %&gt;%<br>        .[,<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;FID&quot;</span>, <span class="hljs-string">&quot;IID&quot;</span>, <span class="hljs-string">&quot;SCORE&quot;</span>)] %&gt;%<br>        merge(., pheno, by=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;FID&quot;</span>, <span class="hljs-string">&quot;IID&quot;</span>))<br><br>    model &lt;- lm(Height~., data=pheno.prs[,-<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;FID&quot;</span>,<span class="hljs-string">&quot;IID&quot;</span>)]) %&gt;%<br>            summary<br>    model.r2 &lt;- model$r.squared<br>    prs.r2 &lt;- model.r2-null.r2<br>    prs.coef &lt;- model$coeff[<span class="hljs-string">&quot;SCORE&quot;</span>,]<br>    prs.result %&lt;&gt;% rbind(.,<br>        data.frame(Threshold=i, R2=prs.r2, <br>                    P=<span class="hljs-built_in">as.numeric</span>(prs.coef[<span class="hljs-number">4</span>]), <br>                    BETA=<span class="hljs-built_in">as.numeric</span>(prs.coef[<span class="hljs-number">1</span>]),<br>                    SE=<span class="hljs-built_in">as.numeric</span>(prs.coef[<span class="hljs-number">2</span>])))<br>&#125;<br>print(prs.result[which.max(prs.result$R2),])<br><br><span class="hljs-comment">##   Threshold        R2            P    BETA       SE</span><br><span class="hljs-comment">## 5       0.3 0.1594805 4.312232e-25 40639.7 3700.694</span><br></code></pre></td></tr></table></figure>
<p>因此，能够产生最佳拟合PRS的P值是：0.3</p>
<p>最佳拟合的PRS可以解释的phenotypic variation是：0.1594805</p>

            </div>
            <hr>
            <div>
              <div class="post-metas mb-3">
                
                  <div class="post-meta mr-3">
                    <i class="iconfont icon-category"></i>
                    
                      <a class="hover-with-bg" href="/categories/BIOINFO-EXPERIMENT-RECORDS/">BIOINFO EXPERIMENT RECORDS</a>
                    
                  </div>
                
                
                  <div class="post-meta">
                    <i class="iconfont icon-tags"></i>
                    
                      <a class="hover-with-bg" href="/tags/GWAS/">GWAS</a>
                    
                      <a class="hover-with-bg" href="/tags/PRS/">PRS</a>
                    
                  </div>
                
              </div>
              
                <p class="note note-warning">
                  
                    本博客所有文章除特别声明外，均采用 <a target="_blank" href="https://creativecommons.org/licenses/by-sa/4.0/deed.zh" rel="nofollow noopener noopener">CC BY-SA 4.0 协议</a> ，转载请注明出处！
                  
                </p>
              
              
                <div class="post-prevnext">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/2021/10/29/20211029%E7%A0%94%E7%A9%B6%E7%94%9F%E5%BC%80%E9%A2%98%E8%AE%BA%E6%96%87/">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">2021.10.29 SIBS研究生开题文件</span>
                        <span class="visible-mobile">上一篇</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2021/09/27/20210927asd/">
                        <span class="hidden-mobile">2021.9.27实验记录iPSYCH_PGC_ASD_Nov2017</span>
                        <span class="visible-mobile">下一篇</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
              <!-- Comments -->
              <article class="comments" id="comments">
                
                  
                
                
  <div class="disqus" style="width:100%">
    <div id="disqus_thread"></div>
    <script>
      Fluid.utils.waitElementVisible('disqus_thread', function() {
        Fluid.utils.createCssLink('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqusjs.css');
        Fluid.utils.createScript('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqus.js', function () {
          new DisqusJS({
            shortname: 'fluid',
            apikey: ''
          });
        });
      });
    </script>
    <noscript>Please enable JavaScript to view the
      <a target="_blank" href="https://disqus.com/?ref_noscript" rel="nofollow noopener noopener">comments powered by Disqus.</a>
    </noscript>
  </div>


              </article>
            
          </article>
        </div>
      </div>
    </div>
    
      <div class="d-none d-lg-block col-lg-2 toc-container" id="toc-ctn">
        <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;目录</p>
  <div class="toc-body" id="toc-body"></div>
</div>

      </div>
    
  </div>
</div>

<!-- Custom -->


    

    
      <a id="scroll-top-button" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">搜索</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v"
                 for="local-search-input">关键词</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>
    

    
  </main>

  <footer class="text-center mt-5 py-3">
  <div class="footer-content">
     <a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a> <i class="iconfont icon-love"></i> <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener"><span>Fluid</span></a> 
  </div>
  

  

  
</footer>

<!-- SCRIPTS -->

  <script  src="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" ></script>
<script  src="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/js/bootstrap.min.js" ></script>
<script  src="/js/debouncer.js" ></script>
<script  src="/js/events.js" ></script>
<script  src="/js/plugins.js" ></script>

<!-- Plugins -->


  
    <script  src="/js/lazyload.js" ></script>
  



  



  <script  src="https://cdn.jsdelivr.net/npm/tocbot@4.12.0/dist/tocbot.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/anchor-js@4.3.0/anchor.min.js" ></script>



  <script defer src="https://cdn.jsdelivr.net/npm/clipboard@2.0.6/dist/clipboard.min.js" ></script>






  <script  src="https://cdn.jsdelivr.net/npm/typed.js@2.0.11/lib/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var title = document.getElementById('subtitle').title;
      
      typing(title)
      
    })(window, document);
  </script>



  <script  src="/js/local-search.js" ></script>
  <script>
    (function () {
      var path = "/local-search.xml";
      $('#local-search-input').on('click', function() {
        searchFunc(path, 'local-search-input', 'local-search-result');
      });
      $('#modalSearch').on('shown.bs.modal', function() {
        $('#local-search-input').focus();
      });
    })()
  </script>





  

  
    <!-- MathJax -->
    <script>
      MathJax = {
        tex: {
          inlineMath: [['$', '$'], ['\\(', '\\)']]
        },
        options: {
          renderActions: {
            findScript: [10, doc => {
              document.querySelectorAll('script[type^="math/tex"]').forEach(node => {
                const display = !!node.type.match(/; *mode=display/);
                const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display);
                const text = document.createTextNode('');
                node.parentNode.replaceChild(text, node);
                math.start = { node: text, delim: '', n: 0 };
                math.end = { node: text, delim: '', n: 0 };
                doc.math.push(math);
              });
            }, '', false],
            insertedScript: [200, () => {
              document.querySelectorAll('mjx-container').forEach(node => {
                let target = node.parentNode;
                if (target.nodeName.toLowerCase() === 'li') {
                  target.parentNode.classList.add('has-jax');
                }
              });
            }, '', false]
          }
        }
      };
    </script>

    <script async src="https://cdn.jsdelivr.net/npm/mathjax@3.1.2/es5/tex-svg.js" ></script>

  











<!-- 主题的启动项 保持在最底部 -->
<script  src="/js/boot.js" ></script>



</body>
</html>
