

<!DOCTYPE html>
<html lang="zh-CN" data-default-color-scheme=&#34;auto&#34;>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/favicon.png">
  <link rel="icon" href="/img/favicon.png">
  <meta name="viewport"
        content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  
    <meta name="description" content="">
  
  <meta name="author" content="John Doe">
  <meta name="keywords" content="">
  
  <title>2021.9.10-9.16实验记录:GWAS analysis in R - Hexo</title>

  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" />


  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/github-markdown-css@4.0.0/github-markdown.min.css" />
  <link  rel="stylesheet" href="/lib/hint/hint.min.css" />

  
    
    
      
      <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@10.4.0/styles/github-gist.min.css" />
    
  

  
    <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.css" />
  



<!-- 主题依赖的图标库，不要自行修改 -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_ba1fz6golrf.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_kmeydafke9r.css">


<link  rel="stylesheet" href="/css/main.css" />

<!-- 自定义样式保持在最底部 -->


  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    var CONFIG = {"hostname":"example.com","root":"/","version":"1.8.8","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"right","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"copy_btn":true,"image_zoom":{"enable":true},"toc":{"enable":true,"headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"onlypost":false},"web_analytics":{"enable":false,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":null,"app_key":null,"server_url":null}}};
  </script>
  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
<meta name="generator" content="Hexo 5.4.0"></head>


<body>
  <header style="height: 70vh;">
    <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand"
       href="/">&nbsp;<strong>Liu Keqin</strong>&nbsp;</a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/">
                <i class="iconfont icon-home-fill"></i>
                首页
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/">
                <i class="iconfont icon-archive-fill"></i>
                归档
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/">
                <i class="iconfont icon-category-fill"></i>
                分类
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/">
                <i class="iconfont icon-tags-fill"></i>
                标签
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/">
                <i class="iconfont icon-user-fill"></i>
                关于
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" data-toggle="modal" data-target="#modalSearch">&nbsp;<i
                class="iconfont icon-search"></i>&nbsp;</a>
          </li>
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" href="javascript:">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

    <div class="banner" id="banner" parallax=true
         style="background: url('/img/default.png') no-repeat center center;
           background-size: cover;">
      <div class="full-bg-img">
        <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
          <div class="page-header text-center fade-in-up">
            <span class="h2" id="subtitle" title="2021.9.10-9.16实验记录:GWAS analysis in R">
              
            </span>

            
              <div class="mt-3">
  
  
    <span class="post-meta">
      <i class="iconfont icon-date-fill" aria-hidden="true"></i>
      <time datetime="2021-09-16 17:18" pubdate>
        2021年9月16日 下午
      </time>
    </span>
  
</div>

<div class="mt-1">
  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-chart"></i>
      13.8k 字
    </span>
  

  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-clock-fill"></i>
      
      
      215
       分钟
    </span>
  

  
  
</div>

            
          </div>

          
        </div>
      </div>
    </div>
  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="d-none d-lg-block col-lg-2"></div>
    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div class="py-5" id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">2021.9.10-9.16实验记录:GWAS analysis in R</h1>
            
            <div class="markdown-body">
              <h1 id="2021-9-10-9-16实验记录-GWAS-analysis-in-R"><a href="#2021-9-10-9-16实验记录-GWAS-analysis-in-R" class="headerlink" title="2021.9.10-9.16实验记录: GWAS analysis in R"></a>2021.9.10-9.16实验记录: GWAS analysis in R</h1><h2 id="R代码教程"><a href="#R代码教程" class="headerlink" title="R代码教程"></a>R代码教程</h2><p>全基因组关联分析策略通常包括四个广泛定义的部分。(i) 数据预处理data pre‐processing；(ii) 新数据生成new data generation；(iii) 统计分析statistical analysis；和(iv) 分析后询问post‐analytic interrogation。这些调查的一个主要目标是识别和描述SNPs与疾病进展或疾病结果措施之间的关联。</p>
<p>根据这四个阶段产生的十个步骤如下：</p>
<ol>
<li>将数据读入R，创建R对象 reading data into R to create an R object</li>
<li>SNP级过滤（第一部分）SNP‐level filtering (part 1)</li>
<li>样本级过滤 sample‐level filtering</li>
<li>SNP级过滤（第二部分）SNP‐level filtering (part 2)</li>
<li>主成分分析（PCA）principal component analysis (PCA)</li>
<li>非类型基因型的归因 imputation of non‐typed genotypes</li>
<li>类型SNP的关联分析 association analysis of typed SNPs</li>
<li>归因数据的关联分析 association analysis of imputed data</li>
<li>归因和类型SNP结果的整合 integration of imputed and typed SNP results</li>
<li>关联结果的可视化和质量控制 visualization and quality control of association findings</li>
</ol>
<p><img src="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5019244/bin/SIM-34-3769-g001.jpg" srcset="/img/loading.gif" alt="An external file that holds a picture, illustration, etc. Object name is SIM-34-3769-g001.jpg"></p>
<blockquote>
<p>Substructure 亚结构，也称为群体混合和群体分层，是指群体遗传历史(例如，迁徙、选择和/或种族融合)导致群体内存在遗传多样性(例如，不同等位基因频率)。</p>
</blockquote>
<h3 id="数据预处理data-pre‐processing"><a href="#数据预处理data-pre‐processing" class="headerlink" title="数据预处理data pre‐processing"></a>数据预处理data pre‐processing</h3><p>在本例中，样本使用Affymetrix 6.0 GeneChip进行基因分型，并以.CEL格式提供给用户。</p>
<blockquote>
<p>基于期望最大化类型算法的Birdseed调用算法被应用于生成每个样本在每个SNP上的基因型和置信度分数。</p>
<p>反过来，PERL和unix脚本被用来将这些转换成.ped和.map文件。</p>
<p>注意，虽然R可以读取.ped和.map文件，但通常最好先将这两个文件转换为.bim、.bed和.fam文件。将文本文件转化成二进制文件，可以大大缩减文件大小。</p>
</blockquote>
<p>本教程主要利用了几个R包（来自Bioconductor）</p>
<p>snpStats</p>
<blockquote>
<p>具有以各种格式读取基因型数据并进行质量控制、估算和关联分析的功能</p>
</blockquote>
<p>SNPRelate</p>
<blockquote>
<p>包括样本级质量控制和计算效率高的主成分(PC)计算的功能</p>
</blockquote>
<p>其他软件包包括数据可视化/代码的功能(ggplot2、 LDheatmap、 postgwas)、数据操作(plyr)和并行处理(doparlel) ，以及它们的依赖关系。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><code class="hljs R">setwd(<span class="hljs-string">&quot;/Users/keqinliu/Downloads/GWASTutorial_Files&quot;</span>) <span class="hljs-comment">#先设置工作目录</span><br>getwd() <span class="hljs-comment">#检查工作目录</span><br><span class="hljs-comment">## [1] &quot;/Users/keqinliu/Downloads/GWASTutorial_Files&quot;</span><br><br>BiocManager::install(<span class="hljs-string">&quot;snpStats&quot;</span>) <span class="hljs-comment">#安装snpStats</span><br><span class="hljs-comment">#注意，R版本＞3.5，应使用BiocManager进行安装</span><br><span class="hljs-comment">#R版本＜3.5，则可使用bioconductor安装</span><br>BiocManager::install(<span class="hljs-string">&quot;SNPRelate&quot;</span>)<br>BiocManager::install(<span class="hljs-string">&quot;rtracklayer&quot;</span>)<br>BiocManager::install(<span class="hljs-string">&quot;biomaRt&quot;</span>)<br>install.packages(<span class="hljs-string">&quot;GenABEL&quot;</span>, <span class="hljs-string">&quot;LDheatmap&quot;</span>, <span class="hljs-string">&quot;doParallel&quot;</span>, <span class="hljs-string">&quot;coin&quot;</span>, <span class="hljs-string">&quot;igraph&quot;</span>) <span class="hljs-comment">#安装所需的R包</span><br>library(devtools)<br><br>data.dir &lt;- <span class="hljs-string">&quot;/Users/keqinliu/Downloads/GWAS&quot;</span><br>out.dir &lt;- data.dir <span class="hljs-comment">#设置不同的输出文件目录，以防与工作目录冲突</span><br><br><span class="hljs-comment">#以下代码用于设定输入文件</span><br>gwas.fn &lt;- lapply(<span class="hljs-built_in">c</span>(bed=<span class="hljs-string">&#x27;bed&#x27;</span>, bim=<span class="hljs-string">&quot;bim&quot;</span>, fam=<span class="hljs-string">&quot;fam&quot;</span>, gds=<span class="hljs-string">&quot;gds&quot;</span>),<span class="hljs-keyword">function</span>(n) sprintf(<span class="hljs-string">&quot;%s/GWASTutorial.%s&quot;</span>, data.dir, n))<br>clinical.fn &lt;- sprintf(<span class="hljs-string">&quot;%s/GWASTutorial_clinical.csv&quot;</span>, data.dir) <span class="hljs-comment">#sprintf函数用于把格式化的数据写入某个字符串缓冲区。</span><br>onethou.fn &lt;- lapply(<span class="hljs-built_in">c</span>(info=<span class="hljs-string">&quot;info&quot;</span>, ped=<span class="hljs-string">&quot;ped&quot;</span>), <span class="hljs-keyword">function</span>(n) sprintf(<span class="hljs-string">&quot;%s/chr16_1000g_CEU.%s&quot;</span>, data.dir, n))<br>protein.coding.coords.fnamo &lt;- sprintf(<span class="hljs-string">&quot;%s/ProCodgene_doords.csv&quot;</span>, out.dir)<br>                     <br><span class="hljs-comment">#以下代码用于设定输出文件</span><br>gwaa.fname &lt;- sprintf(<span class="hljs-string">&quot;%s/GWAStutorialout.txt&quot;</span>, out.dir)<br>gwaa.unadj.fname &lt;- sprintf(<span class="hljs-string">&quot;%sGWAStutorialoutUnadj.txt&quot;</span>, out.dir)<br>input.out.fname &lt;- sprintf(<span class="hljs-string">&quot;%s/GWAStutorial_inputationOut.csv&quot;</span>, out.dir)<br>CEPT.fname &lt;- sprintf(<span class="hljs-string">&quot;%s/CEPT_GWASour.csv&quot;</span>, out.dir)    <br><span class="hljs-comment">#gwaa.fname等value都是文件目录（地址）                     </span><br>                    <br></code></pre></td></tr></table></figure>
<h4 id="在R中读取和格式化数据"><a href="#在R中读取和格式化数据" class="headerlink" title="在R中读取和格式化数据"></a>在R中读取和格式化数据</h4><p>为了阅读.fam .bim和.bed 文件，可以使用 Bioconductor snpStats 包中的 read.plink () 函数。</p>
<blockquote>
<p>结果列表的基因型位置包含一个 SnpMatrix 对象，它是基因型数据的矩阵，每个 SNP 有一列，每个研究参与者有一行。</p>
<p>临床数据(gwastufactorial clinical. csv)可以用我们熟悉的方式读取，即read.csv()。</p>
<p>最后，我们在这个阶段的数据子集只包括在基因型和表型文件中都有数据的个体。</p>
</blockquote>
<p>一旦我们读取了基因型和临床信息，我们就可以开始 GWA 数据预处理的下一步了。这涉及到两个过滤数据的阶段，分别在 SNP 和样本级别。</p>
<p>不过，在不同的情况中，分析的顺序可能会有所不同，这取决于是进行单一GWA分析(如本文所述) ，还是分析人员准备将结果纳入需要跨多项研究协调数据的更大的meta‐analysis。在后一种情况下，以下过滤步骤可能在分析之后被排除或集中执行，因为总结级数据是跨研究合并的。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br></pre></td><td class="code"><pre><code class="hljs r"><span class="hljs-comment">#reading data into r</span><br>library(snpStats)<br><span class="hljs-comment">#read in PLINK files to create list</span><br>geno &lt;- read.plink(gwas.fn$bed, gwas.fn$bim, gwas.fn$fam, na.strings = (<span class="hljs-string">&quot;-9&quot;</span>))<br><br><span class="hljs-comment">#-step1-</span><br><span class="hljs-comment">#reading data into r</span><br><br><span class="hljs-comment">#-step1a-</span><br>library(snpStats)<br><span class="hljs-comment">#read in PLINK files to create list</span><br>geno &lt;- read.plink(gwas.fn$bed, gwas.fn$bim, gwas.fn$fam, na.strings = (<span class="hljs-string">&quot;-9&quot;</span>))<br><br><span class="hljs-comment">#-step1b-</span><br><span class="hljs-comment">#obtain the genotype snpmatrix object from generated list</span><br>genotype &lt;- geno$genotypes<br>print(genotype)<br><span class="hljs-comment">## A SnpMatrix with  1401 rows and  861473 columns</span><br><span class="hljs-comment">## Row names:  10002 ... 11596 </span><br><span class="hljs-comment">## Col names:  rs10458597 ... rs5970564</span><br><span class="hljs-comment">#obtain teh snp info table from the list</span><br>genobim &lt;- geno$map<br>colnames(genobim) &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;chr&quot;</span>, <span class="hljs-string">&quot;SNP&quot;</span>, <span class="hljs-string">&quot;gen.dist&quot;</span>, <span class="hljs-string">&quot;position&quot;</span>, <span class="hljs-string">&quot;A1&quot;</span>, <span class="hljs-string">&quot;A2&quot;</span>)<br>print(head(genobim))<br><span class="hljs-comment">##            chr        SNP gen.dist position   A1 A2</span><br><span class="hljs-comment">## rs10458597   1 rs10458597        0   564621 &lt;NA&gt;  C</span><br><span class="hljs-comment">## rs12565286   1 rs12565286        0   721290    G  C</span><br><span class="hljs-comment">## rs12082473   1 rs12082473        0   740857    T  C</span><br><span class="hljs-comment">## rs3094315    1  rs3094315        0   752566    C  T</span><br><span class="hljs-comment">## rs2286139    1  rs2286139        0   761732    C  T</span><br><span class="hljs-comment">## rs11240776   1 rs11240776        0   765269    G  A</span><br><br><span class="hljs-comment">#remove raw file to open up memory</span><br>rm(geno)<br><br><span class="hljs-comment">#-step1c-</span><br><span class="hljs-comment">#read in clinical file</span><br>clinical &lt;- read.csv(clinical.fn, colClasses = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;character&quot;</span>, <span class="hljs-string">&quot;factor&quot;</span>, <span class="hljs-string">&quot;factor&quot;</span>, <span class="hljs-built_in">rep</span>(<span class="hljs-string">&quot;numeric&quot;</span>,<span class="hljs-number">4</span>)))<br>rownames(clinical) &lt;- clinical$FamID<br>print(head(clinical))<br><span class="hljs-comment">##       FamID CAD sex age  tg hdl ldl</span><br><span class="hljs-comment">## 10002 10002   1   1  60  NA  NA  NA</span><br><span class="hljs-comment">## 10004 10004   1   2  50  55  23  75</span><br><span class="hljs-comment">## 10005 10005   1   1  55 105  37  69</span><br><span class="hljs-comment">## 10007 10007   1   1  52 314  54 108</span><br><span class="hljs-comment">## 10008 10008   1   1  58 161  40  94</span><br><span class="hljs-comment">## 10009 10009   1   1  59 171  46  92</span><br><br><span class="hljs-comment">#-step1d-</span><br><span class="hljs-comment">#subset genotype for individuals with clinical data</span><br>genotype &lt;- genotype[clinical$FamID, ]<br>print(genotypes)<br><span class="hljs-comment">## A SnpMatrix with  1401 rows and  861473 columns</span><br><span class="hljs-comment">## Row names:  10002 ... 11596 </span><br><span class="hljs-comment">## Col names:  rs10458597 ... rs5970564 </span><br><span class="hljs-comment">#在提供的数据示例中，基因型信息可用于n=1401个有可用表型数据的861473个类型SNPs</span><br></code></pre></td></tr></table></figure>
<h4 id="单核苷酸多态性级过滤-part1"><a href="#单核苷酸多态性级过滤-part1" class="headerlink" title="单核苷酸多态性级过滤(part1)"></a>单核苷酸多态性级过滤(part1)</h4><p>这一步涉及删除(也称为“过滤”)不包括分析的 SNPs。这种排除的原因包括大量的缺失数据、低变异性和基因分型错误。通常，首先执行基于大量缺失数据和较低可变性的 SNP 级过滤。然后是样本级过滤 ，最后是基于可能的基因分型错误的 SNP 级过滤</p>
<blockquote>
<p>这样做的理由是，样本级的关联性和亚结构都会影响用于根据基因分型错误过滤SNP的Hardy-Weinberg平衡（HWE）标准。重复SNP和样本级过滤的迭代程序也很常见，直到没有额外的样本被删除。</p>
<p>然而，在这个教程中，没有样本被过滤，认为这个循环是不必要的。</p>
</blockquote>
<h5 id="SNP‐level-filtering-call-rate"><a href="#SNP‐level-filtering-call-rate" class="headerlink" title="SNP‐level filtering: call rate"></a>SNP‐level filtering: call rate</h5><p>call rate被定义为研究中相应SNP信息没有缺失的个体比例。</p>
<blockquote>
<p>在本教程中，我们使用95%的调用率进行过滤，这意味着我们保留数据缺失少于5%的SNP。在较小的样本环境中，可以采用更严格的切点（例如，小于5%）。</p>
</blockquote>
<h5 id="SNP‐level-filtering-minor-allele-frequency-MAF"><a href="#SNP‐level-filtering-minor-allele-frequency-MAF" class="headerlink" title="SNP‐level filtering: minor allele frequency (MAF)"></a>SNP‐level filtering: minor allele frequency (MAF)</h5><p>在研究参与者中，如果某个SNP有很大程度的同质性，通常会导致推断该SNP与所研究性状之间的统计学意义的力量不足。当我们有一个非常小的MAF，使大多数个体有两个主要等位基因的拷贝时，就会发生这种情况。这里，我们删除了MAF小于1%的SNP（MAF &lt;0.01）。在某些情况下，特别是小样本环境下，采用5%的切割点。</p>
<blockquote>
<p>这是说，比如，99%以上的研究者都有一个相同的SNP，那么这个SNP就失去了统计学上的意义（即在参与者中，这个SNP有很大程度上的同质性）。</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br></pre></td><td class="code"><pre><code class="hljs r"><span class="hljs-comment">#使用以下脚本同时对call rate和MAF进行过滤</span><br><span class="hljs-comment">#-step2-</span><br><span class="hljs-comment">#snp-level filtering</span><br><br><span class="hljs-comment">#-step2a-</span><br><span class="hljs-comment">#create snp summary statistics (MAF, call rate, etc.)</span><br>snpsum.col &lt;- col.summary(genotype)<br>print(head(snpsum.col))<br><br><span class="hljs-comment">#-step2b-</span><br><span class="hljs-comment">#setting thresholds</span><br>call &lt;- 0.95<br>minor &lt;- 0.01<br><br><span class="hljs-comment">#filter on MAF and call rate</span><br>use &lt;- with(snpsum.col, (!<span class="hljs-built_in">is.na</span>(MAF) &amp; MAF &gt; minor) &amp; Call.rate &gt;= <span class="hljs-built_in">call</span>)<br>use[<span class="hljs-built_in">is.na</span>(use)] &lt;- <span class="hljs-literal">FALSE</span> <span class="hljs-comment">#remove NA&#x27;s as well</span><br><br>cat(ncol(genotype)-<span class="hljs-built_in">sum</span>(use), <span class="hljs-string">&quot;SNPS WILL BE REMOVED DUE TO LOW MAF OR CALL RATE.\n&quot;</span>)<br><br><span class="hljs-comment">#subset genotype and snp summary data for snps that pass call rate and MAF critoria</span><br>genotype &lt;- genotype[,use]<br><span class="hljs-comment">## 出现报错</span><br><span class="hljs-comment">## Error in .local(x, i, j, ..., drop) : </span><br><span class="hljs-comment">##   logical column selection vector is wrong length</span><br><span class="hljs-built_in">sum</span>(use)<br><span class="hljs-comment">## [1] 0</span><br>ncol(genotype)<br><span class="hljs-comment">## [1] 861473</span><br><span class="hljs-comment">#显然，设置的threshold根本没起作用，但不明白原因。</span><br>use[<span class="hljs-built_in">is.na</span>(use)] &lt;- <span class="hljs-literal">FALSE</span> <span class="hljs-comment">#remove NA&#x27;s as well</span><br>cat(ncol(genotype)-<span class="hljs-built_in">sum</span>(use), <span class="hljs-string">&quot;SNPS WILL BE REMOVED DUE TO LOW MAF OR CALL RATE.\n&quot;</span>)<br><span class="hljs-comment">## 861473 SNPS WILL BE REMOVED DUE TO LOW MAF OR CALL RATE.</span><br><br><span class="hljs-comment">## 解决方案如下：it worked！</span><br><span class="hljs-comment">#filter on MAF and call rate</span><br>use &lt;- with(snpsum.col, (!<span class="hljs-built_in">is.na</span>(MAF) &amp; MAF &gt; <span class="hljs-number">0.01</span>) &amp; Call.rate &gt;= <span class="hljs-number">0.95</span>) <span class="hljs-comment">#直接把threshold改成了数值</span><br>use[<span class="hljs-built_in">is.na</span>(use)] &lt;- <span class="hljs-literal">FALSE</span> <span class="hljs-comment">#remove NA&#x27;s as well</span><br><br><span class="hljs-built_in">sum</span>(use)<br><span class="hljs-comment">## [1] 658186</span><br><br>cat(ncol(genotype)-<span class="hljs-built_in">sum</span>(use), <span class="hljs-string">&quot;SNPS WILL BE REMOVED DUE TO LOW MAF OR CALL RATE.\n&quot;</span>)<br><span class="hljs-comment">## 203287 SNPS WILL BE REMOVED DUE TO LOW MAF OR CALL RATE.</span><br><br><span class="hljs-comment">#subset genotype and snp summary data for snps that pass call rate and MAF critoria</span><br>genotype &lt;- genotype[,use]<br>snpsum.col &lt;- snpsum.col[use,]<br>print(genotype)<br><span class="hljs-comment">## A SnpMatrix with  1401 rows and  658186 columns</span><br><span class="hljs-comment">## Row names:  10002 ... 11596 </span><br><span class="hljs-comment">## Col names:  rs12565286 ... rs5970564    </span><br></code></pre></td></tr></table></figure>
<h4 id="样本级别过滤"><a href="#样本级别过滤" class="headerlink" title="样本级别过滤"></a>样本级别过滤</h4><p>数据预处理的第三阶段涉及过滤样本，也就是说，删除我们选择的被排除在分析之外的个体。样本级过滤的标准通常基于缺失数据、样本污染、相关性(用于基于人口的调查)以及种族、民族或性别的模糊性或不一致性。</p>
<h5 id="Sample‐level-filtering-call-rate"><a href="#Sample‐level-filtering-call-rate" class="headerlink" title="Sample‐level filtering: call rate"></a>Sample‐level filtering: call rate</h5><p>与基于call rate的SNP级过滤类似，我们排除那些在超过预先定义的百分比的分型SNP中缺失基因型数据的个体。这个跨SNP的缺失比例被称为sample call rate，我们采用95%的阈值。也就是说，缺失基因型数据超过5%的类型SNPs的个体将被删除。一个新的减维SnpMatrix基因型对象被创建，它包含了这个filter。</p>
<h5 id="Sample‐level-filtering-heterozygosity"><a href="#Sample‐level-filtering-heterozygosity" class="headerlink" title="Sample‐level filtering: heterozygosity"></a>Sample‐level filtering: heterozygosity</h5><p>杂合度是指在一个个体内，在一个给定的SNP上存在两个等位基因中的每一个。这在HWE下预计会以 $2<em>p</em>(1-p)$ 的概率出现，其中p是该SNP的显性等位基因频率（假设是双等位SNP）。一个人体内跨类型SNP的杂合度过高可能是样本质量差的表现，而杂合度不足则表明该人有近亲繁殖或其他亚结构。因此，近亲繁殖系数 $|F|=（1-O/E）&gt;0.10$ 的样本被剔除，其中O和E分别是个体内杂合SNP的观察数和预期数。请注意，我们是根据每个个体的观察到的SNP来计算该个体的预期计数的。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step3-</span><br><span class="hljs-comment">#sample-level filtering</span><br><br><span class="hljs-comment">#-step3a-</span><br>library(SNPRelate) <span class="hljs-comment">#estimating LD, relatedness, PCA</span><br>library(plyr)<br><br><span class="hljs-comment">#create sample statistic (call rate, heterozygosity)</span><br>snpsum.row &lt;- row.summary(genotype)<br><br><span class="hljs-comment">#add the f stat (inbreeding coefficient) to snpsum.row</span><br>MAF &lt;- snpsum.col$MAF<br>callmatrix &lt;- !<span class="hljs-built_in">is.na</span>(genotype)<br>hetExp &lt;- callmatrix %*% (<span class="hljs-number">2</span>*MAF*(<span class="hljs-number">1</span>-MAF))<br>hetObs &lt;- with(snpsum.row, Heterozygosity*(ncol(genotype))*Call.rate)<br>snpsum.row$hetF &lt;- 1-(hetObs/hetExp)<br><br>head(snpsum.row)<br><span class="hljs-comment">##       Call.rate Certain.calls Heterozygosity          hetF</span><br><span class="hljs-comment">## 10002 0.9826554             1      0.3289825 -0.0247708291</span><br><span class="hljs-comment">## 10004 0.9891581             1      0.3242931 -0.0103236529</span><br><span class="hljs-comment">## 10005 0.9918427             1      0.3231825 -0.0062550972</span><br><span class="hljs-comment">## 10007 0.9861027             1      0.3241469 -0.0098475016</span><br><span class="hljs-comment">## 10008 0.9823333             1      0.3228218 -0.0075941985</span><br><span class="hljs-comment">## 10009 0.9913034             1      0.3213658 -0.0002633189</span><br><br><span class="hljs-comment">#-step3b-</span><br><span class="hljs-comment">#setting thresholds</span><br>sampcall &lt;- 0.95 <span class="hljs-comment">#sample call rate cut-off</span><br>hetcutoff &lt;- 0.1 <span class="hljs-comment">#inbreeding coefficient cut-off</span><br><br>sampleuse &lt;- with(snpsum.row, !<span class="hljs-built_in">is.na</span>(Call.rate) &amp; Call.rate &gt; <span class="hljs-number">0.95</span> &amp; <span class="hljs-built_in">abs</span>(hetF) &lt;= <span class="hljs-number">0.1</span>)<br>sampleuse[<span class="hljs-built_in">is.na</span>(sampleuse)] &lt;- <span class="hljs-literal">FALSE</span> <span class="hljs-comment">#remove NA&#x27;s as well</span><br>cat(nrow(genotype)-<span class="hljs-built_in">sum</span>(sampleuse), <span class="hljs-string">&quot;subject will be removed due to low sample call rate or inbredding coefficient.\n&quot;</span>)<br><span class="hljs-comment">## 0 subject will be removed due to low sample call rate or inbredding coefficient.</span><br><br><span class="hljs-comment">#subset genotype and clinical data for subjects who pass call rate and heterozygosity crtieria</span><br>genotype &lt;- genotype[sampleuse, ]<br>clinical &lt;- clinical[rownames(genotype), ]<br></code></pre></td></tr></table></figure>
<p>由于提供的 PennCATH 数据是预过滤的，因此不会根据近亲繁殖系数 | f | &gt; 0.10来过滤额外的样本。</p>
<h5 id="Sample‐level-filtering-cryptic-relatedness-duplicates-and-gender-identity"><a href="#Sample‐level-filtering-cryptic-relatedness-duplicates-and-gender-identity" class="headerlink" title="Sample‐level filtering: cryptic relatedness, duplicates, and gender identity."></a>Sample‐level filtering: cryptic relatedness, duplicates, and gender identity.</h5><p>样本级过滤：隐性关联性、重复和性别确认。</p>
<p>基于人群的队列研究通常仅限于无亲缘关系的个体，而后面分型SNP的关联分析中描述的广义线性建模方法假定了个体间的独立性。但是，在复杂疾病的区域性队列研究（如基于医院的队列研究）中，来自同一家庭的个体可能会被无意中招募到。衡量一对样本之间的亲缘关系（或重复）的常用方法是基于血缘关系的认同（IBD）。IBD的亲缘关系系数大于0.10，可能表明有亲缘关系、重复或样本混合。通常情况下，具有较低基因型调用率的相关对的个体被移除。我们注意到，在这个阶段也可以检查性别确认，以确认自我报告的性别与观察到的X和Y染色体一致。</p>
<blockquote>
<p>但是，在本例所提供的数据例子中，性染色体不可用，因此，没有提供关于性别认同的过滤例子。</p>
</blockquote>
<p>我们首先应用linkage disequilibrium (LD)修剪，使用0.2的阈值，这消除了数据中很大程度的冗余，减少了染色体假象chromosomal artifacts的影响。这种降维步骤通常在IBD分析和PCA之前应用，在后面的文本中应用于ancestry filtering，并导致大量的计算节省。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step3c-</span><br><span class="hljs-comment">#checking for relatedness</span><br>ld.thresh &lt;- 0.2 <span class="hljs-comment">#LD cut-off</span><br>kin.thresh &lt;- 0.1 <span class="hljs-comment">#kinship cut-off</span><br><span class="hljs-comment">#create gds file, required for SNPRelate functions</span><br>snpgdsBED2GDS(gwas.fn$bed, gwas.fn$fam, gwas.fn$bim, gwas.fn$gds)<br><span class="hljs-comment">## Start file conversion from PLINK BED to SNP GDS ...</span><br><span class="hljs-comment">##     BED file: &#x27;/Users/keqinliu/Downloads/GWAS/GWASTutorial.bed&#x27;</span><br><span class="hljs-comment">##         SNP-major mode (Sample X SNP), 288.4M</span><br><span class="hljs-comment">##     FAM file: &#x27;/Users/keqinliu/Downloads/GWAS/GWASTutorial.fam&#x27;</span><br><span class="hljs-comment">##     BIM file: &#x27;/Users/keqinliu/Downloads/GWAS/GWASTutorial.bim&#x27;</span><br><span class="hljs-comment">## Mon Sep 13 15:43:32 2021     (store sample id, snp id, position, and chromosome)</span><br><span class="hljs-comment">##     start writing: 1401 samples, 861473 SNPs ...</span><br><span class="hljs-comment">## [==================================================] 100%, completed, 8s  </span><br><span class="hljs-comment">## Mon Sep 13 15:43:40 2021 	Done.</span><br><span class="hljs-comment">## Optimize the access efficiency ...</span><br><span class="hljs-comment">## Clean up the fragments of GDS file:</span><br><span class="hljs-comment">##     open the file &#x27;/Users/keqinliu/Downloads/GWAS/GWASTutorial.gds&#x27; (292.4M)</span><br><span class="hljs-comment">##     # of fragments: 39</span><br><span class="hljs-comment">##     save to &#x27;/Users/keqinliu/Downloads/GWAS/GWASTutorial.gds.tmp&#x27;</span><br><span class="hljs-comment">##     rename &#x27;/Users/keqinliu/Downloads/GWAS/GWASTutorial.gds.tmp&#x27; (292.4M, reduced: 252B)</span><br><span class="hljs-comment">##     # of fragments: 18</span><br>genofile &lt;- openfn.gds(gwas.fn$gds, readonly = <span class="hljs-literal">FALSE</span>)<br><br><span class="hljs-comment">#automatically added &quot;-1&quot; sample suffixes are removed</span><br>gds.ids &lt;- read.gdsn(index.gdsn(genofile, <span class="hljs-string">&quot;sample.id&quot;</span>))<br>gds.ids &lt;- sub(<span class="hljs-string">&quot;-1&quot;</span>, <span class="hljs-string">&quot;&quot;</span>, gds.ids)<br>add.gdsn(genofile, <span class="hljs-string">&quot;sample.id&quot;</span>, gds.ids, replace = <span class="hljs-literal">TRUE</span>)<br><br><span class="hljs-comment">#prune SNPs for IBD analysis</span><br>set.seed(<span class="hljs-number">1000</span>)<br><br>geno.sample.ids &lt;- rownames(genotype)<br>snpSVB &lt;- snpgdsLDpruning(genofile, ld.threshold = <span class="hljs-number">0.2</span>, sample.id = geno.sample.ids, <span class="hljs-comment">#only analyze teh filtered samples</span><br>                          snp.id = colnames(genotype)) <span class="hljs-comment">#only analyze the filtered SNPs</span><br><span class="hljs-comment">## Hint: it is suggested to call `snpgdsOpen&#x27; to open a SNP GDS file instead of `openfn.gds&#x27;.</span><br><span class="hljs-comment">## SNP pruning based on LD:</span><br><span class="hljs-comment">## Excluding 203,287 SNPs (non-autosomes or non-selection)</span><br><span class="hljs-comment">## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span><br><span class="hljs-comment">##     # of samples: 1,401</span><br><span class="hljs-comment">##     # of SNPs: 658,186</span><br><span class="hljs-comment">##     using 1 thread</span><br><span class="hljs-comment">##     sliding window: 500,000 basepairs, Inf SNPs</span><br><span class="hljs-comment">##     |LD| threshold: 0.2</span><br><span class="hljs-comment">##     method: composite</span><br><span class="hljs-comment">## Chromosome 1: 8.24%, 5,856/71,038</span><br><span class="hljs-comment">## Chromosome 3: 8.16%, 4,943/60,565</span><br><span class="hljs-comment">## Chromosome 6: 8.05%, 4,362/54,176</span><br><span class="hljs-comment">## Chromosome 12: 8.60%, 3,621/42,124</span><br><span class="hljs-comment">## Chromosome 21: 9.44%, 1,176/12,463</span><br><span class="hljs-comment">## Chromosome 2: 7.69%, 5,669/73,717</span><br><span class="hljs-comment">## Chromosome 4: 8.27%, 4,603/55,675</span><br><span class="hljs-comment">## Chromosome 7: 8.47%, 3,928/46,391</span><br><span class="hljs-comment">## Chromosome 11: 7.89%, 3,490/44,213</span><br><span class="hljs-comment">## Chromosome 10: 8.01%, 3,838/47,930</span><br><span class="hljs-comment">## Chromosome 8: 7.69%, 3,716/48,299</span><br><span class="hljs-comment">## Chromosome 5: 8.10%, 4,549/56,178</span><br><span class="hljs-comment">## Chromosome 14: 8.80%, 2,470/28,054</span><br><span class="hljs-comment">## Chromosome 9: 8.23%, 3,385/41,110</span><br><span class="hljs-comment">## Chromosome 17: 11.13%, 2,220/19,939</span><br><span class="hljs-comment">## Chromosome 13: 8.40%, 2,878/34,262</span><br><span class="hljs-comment">## Chromosome 20: 9.39%, 2,137/22,753</span><br><span class="hljs-comment">## Chromosome 15: 9.23%, 2,391/25,900</span><br><span class="hljs-comment">## Chromosome 16: 9.17%, 2,529/27,591</span><br><span class="hljs-comment">## Chromosome 18: 9.00%, 2,361/26,231</span><br><span class="hljs-comment">## Chromosome 19: 12.99%, 1,492/11,482</span><br><span class="hljs-comment">## Chromosome 22: 11.21%, 1,276/11,382</span><br><span class="hljs-comment">## 72,890 markers are selected in total.</span><br>snpset.ibd &lt;- unlist(snpSVB, use.names = <span class="hljs-literal">FALSE</span>)<br>cat(<span class="hljs-built_in">length</span>(snpset.ibd), <span class="hljs-string">&quot;will be used in IBD analysis\n&quot;</span>)<br><span class="hljs-comment">## 72890 will be used in IBD analysis</span><br><span class="hljs-comment">#与tutorial中的 expect 72812 SNPs 相互冲突</span><br></code></pre></td></tr></table></figure>
<p>这使得SNP的数量从第2步结束时的658,186个减少到72890个。接下来，我们计算成对的IBD距离来搜索样本的关联性。采用的策略是迭代删除配对亲缘关系系数大于0.1的最高数量的主体。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step3d-</span><br><span class="hljs-comment">#find IBD coefficients using method of moments procedure. include pairwise kinship</span><br>ibd &lt;- snpgdsIBDMoM(genofile, kinship = <span class="hljs-literal">TRUE</span>, sample.id = geno.sample.ids, snp.id = snpset.ibd, num.thread = <span class="hljs-number">1</span>)<br><span class="hljs-comment">## Hint: it is suggested to call `snpgdsOpen&#x27; to open a SNP GDS file instead of `openfn.gds&#x27;.</span><br><span class="hljs-comment">## IBD analysis (PLINK method of moment) on genotypes:</span><br><span class="hljs-comment">## Excluding 788,583 SNPs (non-autosomes or non-selection)</span><br><span class="hljs-comment">## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span><br><span class="hljs-comment">##     # of samples: 1,401</span><br><span class="hljs-comment">##     # of SNPs: 72,890</span><br><span class="hljs-comment">##     using 1 thread</span><br><span class="hljs-comment">## PLINK IBD:    the sum of all selected genotypes (0,1,2) = 32728566</span><br><span class="hljs-comment">## Mon Sep 13 16:05:05 2021    (internal increment: 8064)</span><br><span class="hljs-comment">## [==================================================] 100%, completed, 7s  </span><br><span class="hljs-comment">## Mon Sep 13 16:05:13 2021    Done.</span><br>ibdcoeff &lt;- snpgdsIBDSelection(ibd) <span class="hljs-comment">#pairwise sample comparison</span><br>head(ibdcoeff)<br><span class="hljs-comment">##     ID1   ID2        k0         k1    kinship</span><br><span class="hljs-comment">## 1 10002 10004 0.9005063 0.09949372 0.02487343</span><br><span class="hljs-comment">## 2 10002 10005 0.9128056 0.08719439 0.02179860</span><br><span class="hljs-comment">## 3 10002 10007 0.9107539 0.08924614 0.02231153</span><br><span class="hljs-comment">## 4 10002 10008 0.9205765 0.07942348 0.01985587</span><br><span class="hljs-comment">## 5 10002 10009 0.9528404 0.04715958 0.01178989</span><br><span class="hljs-comment">## 6 10002 10010 0.8986051 0.10139486 0.02534872</span><br><br><span class="hljs-comment">#-step3e#</span><br><span class="hljs-comment">#check if there are any candidates for relatedness</span><br>ibdcoeff &lt;- ibdcoeff[ibdcoeff$kinship &gt;= <span class="hljs-number">0.1</span>]<br><br><span class="hljs-comment">#iteratively remove samples with high kinship </span><br>related.samples &lt;- <span class="hljs-literal">NULL</span><br><span class="hljs-keyword">while</span> (nrow(ibdcoeff)&gt;<span class="hljs-number">0</span>) &#123;<br>  <span class="hljs-comment">#count the number of occurrences of each and take the top one</span><br>  sample.counts &lt;- arrange(count(<span class="hljs-built_in">c</span>(ibdcoeff$ID1, ibdcoeff$ID2)), -freq)<br>  rm.sample &lt;- sample.counts[<span class="hljs-number">1</span>, <span class="hljs-string">&#x27;x&#x27;</span>]<br>  cat(<span class="hljs-string">&quot;removeing sample&quot;</span>, <span class="hljs-built_in">as.character</span>(rm.sample), <span class="hljs-string">&#x27;too close related to&#x27;</span>, sample.counts[<span class="hljs-number">1</span>,<span class="hljs-string">&#x27;freq&#x27;</span>],<span class="hljs-string">&#x27;other samples.\n&#x27;</span>)<br>  <span class="hljs-comment">#remove from ibdcoeff and add to list</span><br>  ibdcoeff &lt;- ibdcoeff[ibdcoeff$ID1 !=rm.sample &amp; ibdcoeff$ID2 != rm.sample,]<br>  related.samples &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-built_in">as.character</span>(rm.sample),related.samples)<br>&#125;<br><span class="hljs-comment">## removeing sample  too close related to NA other samples.</span><br><br><span class="hljs-comment">#filter genotype and clinical to include only unrelated samples</span><br>genotype &lt;- genotype[ !(rownames(genotype) %in% related.samples), ]<br>clinical &lt;- clinical[ !(clinical$FamID %in% related.samples), ]<br>geno.sample.ids &lt;- rownames(genotype)<br>cat(<span class="hljs-built_in">length</span>(related.samples), <span class="hljs-string">&quot;similar samples removed due to correlation coefficient &gt;= 0.1&quot;</span>, <span class="hljs-string">&quot;\n&quot;</span> )<br><span class="hljs-comment">## 0 similar samples removed due to correlation coefficient &gt;= 0.1 </span><br>print(genotype)<br><span class="hljs-comment">## A SnpMatrix with  1401 rows and  658186 columns</span><br><span class="hljs-comment">## Row names:  10002 ... 11596 </span><br><span class="hljs-comment">## Col names:  rs12565286 ... rs5970564 </span><br></code></pre></td></tr></table></figure>
<p>在本示例中，没有任何样本是基于IBD亲属系数&gt;0.10而被过滤掉的。</p>
<h5 id="Sample‐level-filtering-ancestry"><a href="#Sample‐level-filtering-ancestry" class="headerlink" title="Sample‐level filtering: ancestry."></a>Sample‐level filtering: ancestry.</h5><p>PCA是一种根据观察到的基因构成将个体可视化并分类到祖先群体的方法。我们这样做有两个原因：</p>
<p>首先，自我报告的种族和民族可能与完全基于遗传信息的个体集群不同，</p>
<p>其次，出现一个似乎不属于种族/民族集群的个体可能暗示着样本级错误。</p>
<p>我们使用LD修剪后的72,812个SNPs子集（步骤3-c）作为PCA的输入。第一阶段LD剪枝的另一种策略是 “HapMap rooted “分析，它包括首先在一个reference panel上进行PCA，例如HapMap或1000 genome，然后将研究样本投射到所产生的空间。这种方法在此不作介绍，但可以用Kinship-based INference for Gwas（KING）软件的现有功能来实现。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step3f-</span><br><span class="hljs-comment">#checking for ancestry</span><br><span class="hljs-comment">#find PCA matrix</span><br>pca &lt;- snpgdsPCA(genofile, sample.id = geno.sample.ids, snp.id = snpset.ibd, num.thread = <span class="hljs-number">1</span>)<br><span class="hljs-comment">#creating data frame of first two principal components</span><br>pctab &lt;- data.frame(sample.id = pca$sample.id,<br>                    PC1 = pca$eigenvect[,<span class="hljs-number">1</span>], <span class="hljs-comment">#the first eigenvector</span><br>                    PC2 = pca$eigenvect[,<span class="hljs-number">2</span>], <span class="hljs-comment">#the second eigenvector</span><br>                    stringsAsFactors = <span class="hljs-literal">FALSE</span>)<br><span class="hljs-comment">#plot the first two principle components</span><br>plot(pctab$PC2, pctab$PC1, xlab=<span class="hljs-string">&quot;principal component 2&quot;</span>, ylab=<span class="hljs-string">&quot;principal component 1&quot;</span>, main = <span class="hljs-string">&quot;ancestry plot&quot;</span>)<br><br></code></pre></td></tr></table></figure>
<p><img src="pcaplot.png" srcset="/img/loading.gif" alt="pcaplot"></p>
<h4 id="单核苷酸多态性级过滤-part-2"><a href="#单核苷酸多态性级过滤-part-2" class="headerlink" title="单核苷酸多态性级过滤(part 2)"></a>单核苷酸多态性级过滤(part 2)</h4><h5 id="SNP‐level-filtering-HWE"><a href="#SNP‐level-filtering-HWE" class="headerlink" title="SNP‐level filtering: HWE."></a>SNP‐level filtering: HWE.</h5><p>违反HWE可以表明存在种群亚结构或发生基因分型错误。虽然它们并不总是可以区分的，但通常的做法是假设有基因分型错误，并删除违反HWE的SNP。如果有病例对照状态，我们将这种过滤限制在对对照组的分析上，因为病例中的违反可能是一种关联的迹象。在一个给定的SNP上，通常使用观察到的基因型和预期的基因型之间的 $X^2$ 拟合度测试来衡量对HWE的偏离。我们删除那些HWE测试统计量在对照组中的相应P值小于 $1×10^{-6}$ 的SNPs。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step4-</span><br><span class="hljs-comment">#hardy-weinberg snp filtering on cad controls</span><br>hardy &lt;- 10^-<span class="hljs-number">6</span> <span class="hljs-comment">#HWE cut-off</span><br>CADcontrols &lt;- clinical[ clinical$CAD==<span class="hljs-number">0</span>, <span class="hljs-string">&#x27;FamID&#x27;</span> ]<br>snpsum.colCont &lt;- col.summary(genotype[CADcontrols,])<br>HWEuse &lt;- with(snpsum.colCont, !<span class="hljs-built_in">is.na</span>(z.HWE) &amp; (<span class="hljs-built_in">abs</span>(z.HWE) &lt; <span class="hljs-built_in">abs</span>(qnorm(<span class="hljs-number">0.5</span>*<span class="hljs-number">10</span>^-<span class="hljs-number">6</span>))))<br>rm(snpsum.colCont)<br><br>HWEuse[<span class="hljs-built_in">is.na</span>(HWEuse)] &lt;- <span class="hljs-literal">FALSE</span> <span class="hljs-comment">#remove NA&#x27;s as well</span><br>cat(ncol(genotype)-<span class="hljs-built_in">sum</span>(HWEuse), <span class="hljs-string">&quot;SNP will be removed due to high HWE.\n&quot;</span>)<br><br><span class="hljs-comment">#subset genotype and SNP summary data for SNPs that pass HWE criteria</span><br>genotype &lt;- genotype[,HWEuse]<br>print(genotype)<br><span class="hljs-comment">## A SnpMatrix with  1401 rows and  656890 columns</span><br><span class="hljs-comment">## Row names:  10002 ... 11596 </span><br><span class="hljs-comment">## Col names:  rs12565286 ... rs28729663 </span><br></code></pre></td></tr></table></figure>
<p>在 CAD 控制中，我们根据 HWE $p &lt; 1 × 10^{-6}$ 筛选出额外的1,296个 SNPs。结果是656,890个单核苷酸多态性可以在关联分析中考虑。</p>
<h3 id="新数据生成"><a href="#新数据生成" class="headerlink" title="新数据生成"></a>新数据生成</h3><p>在完成SNP和样本级的过滤后，我们在进行统计分析之前产生两种新的数据：</p>
<p>第一种是PC，旨在捕捉潜在的人口亚结构的信息，这些信息通常在自我报告的种族和民族变量中是不存在的。</p>
<blockquote>
<p>这种数据用于进行主成分分析，创建捕获种群子结构的主要组成部分。</p>
</blockquote>
<p>第二种是未定型的SNP的基因型，可能与结果有功能关系，因此为识别关联提供额外的力量。</p>
<h4 id="主成分分析（PCA）principal-component-analysis-PCA"><a href="#主成分分析（PCA）principal-component-analysis-PCA" class="headerlink" title="主成分分析（PCA）principal component analysis (PCA)"></a>主成分分析（PCA）principal component analysis (PCA)</h4><p>Creating principal components for capturing population‐substructure 创建捕获种群子结构的主要组成部分</p>
<p>亚结构，也被称为种群混杂和种群分层，指的是在一个明显的同质种群中存在遗传多样性（如不同的等位基因频率），这是由于种群遗传历史（如迁移、选择和/或种族融合）造成的。基于观察到的基因型数据的PC可以捕捉到关于亚结构的信息，并且可以直接使用SNPRelate软件包中的snpgdsPCA()函数根据全基因型数据生成。</p>
<p>值得注意的是，我们在PCA之前再次应用LD修剪。通常情况下，前10个PC被认为是可能的混杂因素。这个数字是常规的，但也是任意的，一种选择是根据它们所解释的变异性的预设比例来选择PC的数量。λ统计量通常用于评估是否有必要纳入PC。这将在步骤10（量化-量化（Q-Q）图和λ-统计量）中进一步描述。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step5-</span><br><span class="hljs-comment">#generating principal components for modeling</span><br><br><span class="hljs-comment">#-step5a-</span><br><span class="hljs-comment">#set LD threshold to 0.2</span><br>ld.thresh &lt;- 0.2<br><br>set.seed(<span class="hljs-number">1000</span>)<br>geno.sample.ids &lt;- rownames(genotype)<br>snpSVB &lt;- snpgdsLDpruning(genofile, ld.threshold = <span class="hljs-number">0.2</span>,<br>                          sample.id = geno.sample.ids, <span class="hljs-comment">#only analyze the filtered samples</span><br>                          snp.id = colnames(genotype)) <span class="hljs-comment">#only analyze the filtered SNPs</span><br>snpset.pca &lt;- unlist(snpSVB, use.names = <span class="hljs-literal">FALSE</span>)<br>cat(<span class="hljs-built_in">length</span>(snpset.pca),<span class="hljs-string">&quot;\n&quot;</span>)<br><span class="hljs-comment">## Hint: it is suggested to call `snpgdsOpen&#x27; to open a SNP GDS file instead of `openfn.gds&#x27;.</span><br><span class="hljs-comment">## SNP pruning based on LD:</span><br><span class="hljs-comment">## Excluding 204,583 SNPs (non-autosomes or non-selection)</span><br><span class="hljs-comment">## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span><br><span class="hljs-comment">##     # of samples: 1,401</span><br><span class="hljs-comment">##     # of SNPs: 656,890</span><br><span class="hljs-comment">##     using 1 thread</span><br><span class="hljs-comment">##     sliding window: 500,000 basepairs, Inf SNPs</span><br><span class="hljs-comment">##     |LD| threshold: 0.2</span><br><span class="hljs-comment">##     method: composite</span><br><span class="hljs-comment">## Chromosome 1: 8.21%, 5,834/71,038</span><br><span class="hljs-comment">## Chromosome 3: 8.14%, 4,933/60,565</span><br><span class="hljs-comment">## Chromosome 6: 8.03%, 4,350/54,176</span><br><span class="hljs-comment">## Chromosome 12: 8.57%, 3,609/42,124</span><br><span class="hljs-comment">## Chromosome 21: 9.44%, 1,177/12,463</span><br><span class="hljs-comment">## Chromosome 2: 7.68%, 5,658/73,717</span><br><span class="hljs-comment">## Chromosome 4: 8.24%, 4,590/55,675</span><br><span class="hljs-comment">## Chromosome 7: 8.45%, 3,922/46,391</span><br><span class="hljs-comment">## Chromosome 11: 7.88%, 3,485/44,213</span><br><span class="hljs-comment">## Chromosome 10: 7.96%, 3,817/47,930</span><br><span class="hljs-comment">## Chromosome 8: 7.67%, 3,704/48,299</span><br><span class="hljs-comment">## Chromosome 5: 8.06%, 4,526/56,178</span><br><span class="hljs-comment">## Chromosome 14: 8.78%, 2,464/28,054</span><br><span class="hljs-comment">## Chromosome 9: 8.20%, 3,370/41,110</span><br><span class="hljs-comment">## Chromosome 17: 11.12%, 2,217/19,939</span><br><span class="hljs-comment">## Chromosome 13: 8.34%, 2,857/34,262</span><br><span class="hljs-comment">## Chromosome 20: 9.38%, 2,135/22,753</span><br><span class="hljs-comment">## Chromosome 15: 9.20%, 2,384/25,900</span><br><span class="hljs-comment">## Chromosome 16: 9.13%, 2,519/27,591</span><br><span class="hljs-comment">## Chromosome 18: 8.97%, 2,353/26,231</span><br><span class="hljs-comment">## Chromosome 19: 12.97%, 1,489/11,482</span><br><span class="hljs-comment">## Chromosome 22: 11.18%, 1,272/11,382</span><br><span class="hljs-comment">## 72,665 markers are selected in total.</span><br>cat(<span class="hljs-built_in">length</span>(snpset.pca),<span class="hljs-string">&quot;\n&quot;</span>)<br><span class="hljs-comment">## 72665</span><br>pca &lt;- snpgdsPCA(genofile, sample.id = geno.sample.ids, snp.id = snpset.pca, num.thread = <span class="hljs-number">1</span>)<br><span class="hljs-comment">## Hint: it is suggested to call `snpgdsOpen&#x27; to open a SNP GDS file instead of `openfn.gds&#x27;.</span><br><span class="hljs-comment">## Principal Component Analysis (PCA) on genotypes:</span><br><span class="hljs-comment">## Excluding 788,808 SNPs (non-autosomes or non-selection)</span><br><span class="hljs-comment">## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)</span><br><span class="hljs-comment">##     # of samples: 1,401</span><br><span class="hljs-comment">##     # of SNPs: 72,665</span><br><span class="hljs-comment">##     using 1 thread</span><br><span class="hljs-comment">##     # of principal components: 32</span><br><span class="hljs-comment">## PCA:    the sum of all selected genotypes (0,1,2) = 32694833</span><br><span class="hljs-comment">## CPU capabilities: Double-Precision SSE2</span><br><span class="hljs-comment">## Tue Sep 14 13:32:27 2021    (internal increment: 252)</span><br><span class="hljs-comment">## [==================================================] 100%, completed, 43s </span><br><span class="hljs-comment">## Tue Sep 14 13:33:10 2021    Begin (eigenvalues and eigenvectors)</span><br><span class="hljs-comment">## Tue Sep 14 13:33:11 2021    Done.</span><br><br><span class="hljs-comment"># find and record first 10 principal components</span><br><span class="hljs-comment"># pcs will be a N:10 matrix </span><br><span class="hljs-comment"># each column is a principal component</span><br>pcs &lt;- data.frame(FamID = pca$sample.id, pca$eigenvect[,<span class="hljs-number">1</span>:<span class="hljs-number">10</span>],<br>                  stringsAsFactors = <span class="hljs-literal">FALSE</span>)<br>colnames(pcs)[<span class="hljs-number">2</span>:<span class="hljs-number">11</span>] &lt;- paste(<span class="hljs-string">&quot;pc&quot;</span>,<span class="hljs-number">1</span>:<span class="hljs-number">10</span>,sep = <span class="hljs-string">&quot;&quot;</span>)<br>print(head(pcs))<br><span class="hljs-comment">##   FamID          pc1           pc2          pc3           pc4           pc5           pc6           pc7          pc8</span><br><span class="hljs-comment">## 1 10002  0.008136946  0.0102555005  0.004035296 -0.0067555328  0.0290486205 -0.0158332357 -0.0368010948  0.007497037</span><br><span class="hljs-comment">## 2 10004 -0.011771047 -0.0080186632  0.004526211  0.0137606581 -0.0005964856  0.0322921253 -0.0004926945 -0.001072583</span><br><span class="hljs-comment">## 3 10005 -0.016963522 -0.0051956716 -0.013847881  0.0112937916 -0.0330688171 -0.0326298549 -0.0113624994 -0.001566384</span><br><span class="hljs-comment">## 4 10007 -0.010673757  0.0037745524  0.004574423 -0.0015019307 -0.0127123211 -0.0007371223 -0.0048545424 -0.013640426</span><br><span class="hljs-comment">## 5 10008 -0.016487939  0.0012844871 -0.019463117 -0.0051653724  0.0074000952 -0.0024935274  0.0022416080  0.041584582</span><br><span class="hljs-comment">## 6 10009 -0.014734448  0.0007935566 -0.017633485 -0.0005575061 -0.0547999107 -0.0288053889 -0.0075189053  0.016484556</span><br><span class="hljs-comment">##            pc9          pc10</span><br><span class="hljs-comment">## 1 -0.006501613  0.0003016164</span><br><span class="hljs-comment">## 2  0.034174191 -0.0185128130</span><br><span class="hljs-comment">## 3  0.005833437 -0.0020015591</span><br><span class="hljs-comment">## 4 -0.015690254 -0.0190696938</span><br><span class="hljs-comment">## 5  0.027821854 -0.0104951963</span><br><span class="hljs-comment">## 6  0.028223932 -0.0144727186</span><br><br></code></pre></td></tr></table></figure>
<h4 id="非类型基因型的归因-imputation-of-non‐typed-genotypes"><a href="#非类型基因型的归因-imputation-of-non‐typed-genotypes" class="headerlink" title="非类型基因型的归因 imputation of non‐typed genotypes"></a>非类型基因型的归因 imputation of non‐typed genotypes</h4><p>Imputing non‐typed single nucleotide polymorphisms using 1000 Genomes data </p>
<p>使用芯片阵列技术测量的类型SNPs通常捕获了大约100万个多态性，这些多态性在一般人群中至少有1%的变化。更为普遍的是，由于功能（因果）变体可能没有被测量到，所以人们对分析非类型SNPs的基因型与疾病结果的关联感兴趣。利用广泛的关于参考单倍型及其LD结构的外部资源，如HapMap和1000基因组数据，我们可以估算出未测量的基因型数据。三个精心描述并推荐的用于SNP水平归因的独立软件包是IMPUTE2、MACH和BEAGLE。归入的基因型可以报告为 “最佳猜测 “的基因型，或报告为基因组上某一位置的每个基因型的后验概率。重要的是，这种估计过程中的不确定性需要在关联分析中加以考虑，因此，我们从此区分了基因型和估算的数据。后面的步骤8中描述了专门考虑归入的SNP数据的不确定性的方法。</p>
<p>在归纳之后，要进行质量控制步骤，以过滤具有高度不确定性的归纳数据。常见的不确定性的衡量标准是信息含量和$R^2$ 。我们将$R^2$ 的阈值定为0.7，以纳入关联分析，在这种情况下，$R^2$ 是将每个归入的SNP回归到区域类型的SNP的线性模型的关联值。这在snpStats软件包的文档中有进一步描述。此外，我们在这个阶段排除了MAF（分配最高后验概率基因型后）小于0.01的SNPs。为了说明问题，我们使用R软件包snpStats中的snp.imputation()和impute.snps()函数，对同一染色体（16号染色体）上的1000个基因组SNP的有限集合进行归因，这些SNP在GWA关联分析（步骤7）中被确定为全基因组显著的基因分型。在实践中，通常在所有的染色体上进行归因，从而产生多达1250万个分型和归因的SNPs，可以对其进行关联分析。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br></pre></td><td class="code"><pre><code class="hljs r"><span class="hljs-comment">#-step6-</span><br><span class="hljs-comment">#genotype imputation</span><br><br><span class="hljs-comment">#-step6a-</span><br><span class="hljs-comment">#read in 1000g data for given chromosome 16</span><br>thougeno &lt;- read.pedfile(onethou.fn$ped, snps = onethou.fn$info, which = <span class="hljs-number">1</span>)<br><br><span class="hljs-comment"># obtain genotype data for given chromosome</span><br>genoMatrix &lt;- thougeno$genotypes<br><br><span class="hljs-comment"># obtain the chromosome position for each SNP</span><br>support &lt;- thougeno$map<br>colnames(support) &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;SNP&quot;</span>, <span class="hljs-string">&quot;position&quot;</span>, <span class="hljs-string">&quot;A1&quot;</span>, <span class="hljs-string">&quot;A2&quot;</span>)<br>head(support)<br><span class="hljs-comment">##           SNP position A1 A2</span><br><span class="hljs-comment">## 1 rs140769322    60180  3  2</span><br><span class="hljs-comment">## 2 rs188810967    60288  2  1</span><br><span class="hljs-comment">## 3  rs76368850    60291  2  4</span><br><span class="hljs-comment">## 4 rs185537431    60778  3  1</span><br><span class="hljs-comment">## 5 rs542544747    60842  2  1</span><br><span class="hljs-comment">## 6   rs4021615    61349  1  3</span><br><br><span class="hljs-comment">#imputation of non-typed 1000g SNPs</span><br>presSnps &lt;- colnames(genotype)<br><br><span class="hljs-comment">#subset for SNPs on given chromosome</span><br>presDatChr &lt;- genobim[genobim$SNP %in% presSnps &amp; genobim$chr==<span class="hljs-number">16</span>, ]<br>targetSnps &lt;- presDatChr$SNP<br><br><span class="hljs-comment">#subset 1000g data for our SNPs</span><br><span class="hljs-comment">#&quot;missing&quot; and &quot;present&quot; are snpMatrix objects needed for imputation rules</span><br>is.present &lt;- colnames(genoMatrix) %in% targetSnps<br>missing &lt;- genoMatrix[,!is.present]<br>print(<span class="hljs-built_in">missing</span>)<br><span class="hljs-comment">## A SnpMatrix with  99 rows and  377819 columns</span><br><span class="hljs-comment">## Row names:  CEU_1 ... CEU_99 </span><br><span class="hljs-comment">## Col names:  rs140769322 ... rs111706106 </span><br><br>present &lt;- genoMatrix[,is.present]<br>print(present)<br><span class="hljs-comment">## A SnpMatrix with  99 rows and  20632 columns</span><br><span class="hljs-comment">## Row names:  CEU_1 ... CEU_99 </span><br><span class="hljs-comment">## Col names:  rs41340949 ... rs4785775</span><br><br><span class="hljs-comment">#obtain position of SNPs to be used for imputation rules</span><br>pos.pres &lt;- support$position[is.present]<br>pos.mis &lt;- support$position[!is.present]<br><br><span class="hljs-comment">#calculate and store imputation rules using snp.imputation()</span><br>rules &lt;- snp.imputation(present, <span class="hljs-built_in">missing</span>, pos.pres, pos.mis)<br><span class="hljs-comment">## SNPs tagged by a single SNP: 82119</span><br><span class="hljs-comment">## SNPs tagged by multiple tag haplotypes (saturated model): 115769</span><br></code></pre></td></tr></table></figure>
<p>然后去掉failed imputation，输入相关不确定度高的SNPs，并且输入低估计MAF的SNPs。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step6b-</span><br><span class="hljs-comment">#remove failed imputations</span><br>rules &lt;- rules[can.impute(rules)]<br>cat(<span class="hljs-string">&quot;imputation rules for&quot;</span>, <span class="hljs-built_in">length</span>(rules), <span class="hljs-string">&quot;SNPs were estimated\n&quot;</span>) <br><span class="hljs-comment">## imputation rules for 197888 SNPs were estimated</span><br><br><span class="hljs-comment">#quality control for imputation certainty and MAF</span><br><span class="hljs-comment">#set thresholds</span><br>r2threshold &lt;- 0.7<br>minor &lt;- 0.01<br><br><span class="hljs-comment">#filter on imputation certainty and MAF</span><br>rules &lt;- rules[imputation.r2(rules) &gt;= <span class="hljs-number">0.7</span>]<br>cat(<span class="hljs-built_in">length</span>(rules), <span class="hljs-string">&quot;imputation rules remain after uncertain imputation were removed\n&quot;</span>)<br><span class="hljs-comment">## 162565 imputation rules remain after uncertain imputation were removed</span><br><br>rules &lt;- rules[imputation.maf(rules) &gt;= <span class="hljs-number">0.01</span>]<br>cat(<span class="hljs-built_in">length</span>(rules), <span class="hljs-string">&quot;imputation rules remain after MAF filtering\n&quot;</span>)<br><span class="hljs-comment">## 162565 imputation rules remain after MAF filtering</span><br><br><span class="hljs-comment">#obtain posterior expectation of genotypes of imputed snps</span><br>target &lt;- genotype[,targetSnps]<br>imputed &lt;- impute.snps(rules, target, <span class="hljs-built_in">as.numeric</span> = <span class="hljs-literal">FALSE</span>)<br>print(imputed)<br><span class="hljs-comment">## A SnpMatrix with  1401 rows and  162565 columns</span><br><span class="hljs-comment">## Row names:  10002 ... 11596 </span><br><span class="hljs-comment">## Col names:  rs560777354;rs80001234 ... rs62053708 </span><br></code></pre></td></tr></table></figure>
<p>这一步的分析结果：在16号染色体上的162565个基因组设定的SNPs被带到步骤8中进行关联分析。</p>
<p>需要强调的是，在关联分析（association analysis）的背景下，需要考虑归属的不确定性（the uncertainty in imputation），因此这些SNPs与下一步中分析的类型SNPs是分开考虑的。</p>
<h3 id="全基因组关联分析-Genome-wide-association-analysis"><a href="#全基因组关联分析-Genome-wide-association-analysis" class="headerlink" title="全基因组关联分析 Genome-wide association analysis"></a>全基因组关联分析 Genome-wide association analysis</h3><h4 id="类型SNP的关联分析-association-analysis-of-typed-SNPs"><a href="#类型SNP的关联分析-association-analysis-of-typed-SNPs" class="headerlink" title="类型SNP的关联分析 association analysis of typed SNPs"></a>类型SNP的关联分析 association analysis of typed SNPs</h4><p>关联分析通常涉及将每个SNP单独回归到一个给定的性状上，并根据病人层面的临床、人口和环境因素进行调整。假设每个SNP的基础遗传模型（如显性、隐性或加性）将影响结果；然而，由于大量的SNP和与结果的关系通常不确定，通常会选择一个单一的加性模型。在这种情况下，如所提供的代码所示，每个SNP被表示为相应的小等位基因数（0、1或2）。</p>
<p>值得注意的是，基于替代模型（如显性或隐性）的SNP变量的编码是直接的，所描述的关联分析也是如此。在实践中，Bonferonni校正的全基因组显著性阈值为 $5×10^{-8}$，用于控制家族性的错误率。这个临界值是基于研究，表明整个基因组大约有100万个独立的SNPs，所以不管调查中实际的 typed 或 imputed 的SNPs数量如何，都要应用。</p>
<p>在我们的数据例子中，我们使用反正态转换的高密度脂蛋白胆固醇作为响应，调整了年龄、性别和前10个PC。高密度脂蛋白胆固醇是一个与心血管疾病相关的复杂特征，年龄和性别是其既定的风险因素。这两个协变量和任意选择的10个PC是心血管疾病性状关联研究的惯例。重要的是，在任何模型拟合过程中，必须评估模型假设的适当性，特别是所研究性状的正态性。对高密度脂蛋白胆固醇直方图的目视检查显示了一些极端值，因此，选择了反正态转换。其他转换，如对数转换，也可能是合理的，并具有保持观测值之间相对距离的优势。我们在本教程中不强调这一点，因为可以应用标准的统计建模实践。下面的代码为分析准备了表型数据。</p>
<blockquote>
<p>本教程需要GenABEL包，但该包已被从CRAN上移除，故只能使用下载压缩包然后本地安装的方法。</p>
<p>步骤如下：</p>
<ol>
<li><p>在terminal中</p>
 <figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs shell">curl -O https://cran.r-project.org/src/contrib/Archive/GenABEL/GenABEL_1.8-0.tar.gz<br>curl -O https://cran.r-project.org/src/contrib/Archive/GenABEL.data/GenABEL.data_1.0.0.tar.gz<br></code></pre></td></tr></table></figure>
</li>
<li><p>RStudio - Tools - Install Packages - Install from: Package Archive File</p>
</li>
</ol>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step7-</span><br><span class="hljs-comment">#association analysis of typed SNPs</span><br>library(GenABEL)<br><br><span class="hljs-comment">#-step7a-</span><br><span class="hljs-comment">#merge clinical data and principal components to create phenotype table</span><br>phenoSub &lt;- merge(clinical, pcs) <span class="hljs-comment"># data.frame =&gt; [FamID CAD sex age hdl pc1 pc2 ... pc10]</span><br><br><span class="hljs-comment">#we will do a rank-based inverse normal transformation of hdl</span><br>phenoSub$phenotype &lt;- rntransform(phenoSub$hdl, family = gaussian)<br><br><span class="hljs-comment">#show that the assumption of normality met after transformation</span><br>par(mfrow=<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">2</span>))<br>hist(phenoSub$hdl, main=<span class="hljs-string">&quot;histogram of HDL&quot;</span>, xlab = <span class="hljs-string">&quot;HDL&quot;</span>)<br>hist(phenoSub$phenotype, main=<span class="hljs-string">&quot;histogram of tranformed HDL&quot;</span>, xlab = <span class="hljs-string">&quot;transformed HDL&quot;</span>)<br></code></pre></td></tr></table></figure>
<p><img src="hist.png" srcset="/img/loading.gif" alt="hist"></p>
<blockquote>
<p>转换为了正态分布</p>
</blockquote>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#remove unnecessary columns from table</span><br>phenoSub$hdl &lt;- <span class="hljs-literal">NULL</span><br>phenoSub$ldl &lt;- <span class="hljs-literal">NULL</span><br>phenoSub$tg &lt;- <span class="hljs-literal">NULL</span><br>phenoSub$CAD &lt;- <span class="hljs-literal">NULL</span><br><br><span class="hljs-comment">#remove columns to match names necessary for GWAS() function</span><br>phenoSub &lt;- rename(phenoSub, replace = <span class="hljs-built_in">c</span>(FamID=<span class="hljs-string">&quot;id&quot;</span>))<br><br><span class="hljs-comment">#include only subjects with hdl data</span><br>phenoSub &lt;- phenoSub[!<span class="hljs-built_in">is.na</span>(phenoSub$phenotype),]<br><br>print(head(phenoSub))<br><span class="hljs-comment">##      id sex age         pc1           pc2          pc3           pc4           pc5           pc6           pc7</span><br><span class="hljs-comment">## 2 10004   2  50 -0.01177105 -0.0080186632  0.004526211  0.0137606581 -0.0005964856  0.0322921253 -0.0004926945</span><br><span class="hljs-comment">## 3 10005   1  55 -0.01696352 -0.0051956716 -0.013847881  0.0112937916 -0.0330688171 -0.0326298549 -0.0113624994</span><br><span class="hljs-comment">## 4 10007   1  52 -0.01067376  0.0037745524  0.004574423 -0.0015019307 -0.0127123211 -0.0007371223 -0.0048545424</span><br><span class="hljs-comment">## 5 10008   1  58 -0.01648794  0.0012844871 -0.019463117 -0.0051653724  0.0074000952 -0.0024935274  0.0022416080</span><br><span class="hljs-comment">## 6 10009   1  59 -0.01473445  0.0007935566 -0.017633485 -0.0005575061 -0.0547999107 -0.0288053889 -0.0075189053</span><br><span class="hljs-comment">## 7 10010   1  54 -0.01427468  0.0052373331 -0.014937383  0.0313489032  0.0104569709  0.0104441393 -0.0111072206</span><br><span class="hljs-comment">##            pc8          pc9         pc10  phenotype</span><br><span class="hljs-comment">## 2 -0.001072583  0.034174191 -0.018512813 -2.2877117</span><br><span class="hljs-comment">## 3 -0.001566384  0.005833437 -0.002001559 -0.4749316</span><br><span class="hljs-comment">## 4 -0.013640426 -0.015690254 -0.019069694  0.8855512</span><br><span class="hljs-comment">## 5  0.041584582  0.027821854 -0.010495196 -0.1644639</span><br><span class="hljs-comment">## 6  0.016484556  0.028223932 -0.014472719  0.3938940</span><br><span class="hljs-comment">## 7 -0.014092489 -0.020411854  0.017169084  1.7109552</span><br></code></pre></td></tr></table></figure>
<p>由于需要拟合的模型较多，建议并行运行GWA分析，即在几个核心中同时进行。每个核心对SNP的一个子集进行GWA分析，当所有核心的计算完成后，输出被返回到原来的顺序。</p>
<p>在接下来的文字中，我们描述了一种在MAC、Unix和Windows操作系统上并行运行分析的跨平台方法。detectCores()函数可以用来确定可用的工作者数量。为了并行地运行分析，我们使用doParallel包中的dopar()函数，表示工作器的数量。doPar()的输出是一个ascii文本文件。这些都包含在我们开发的GWAA()函数中，可在本稿件的补充信息B中找到。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step7b-</span><br><span class="hljs-comment">#run GWAS analysis (using parallel processing)</span><br>library(plyr)<br>source(<span class="hljs-string">&quot;GWAA.R&quot;</span>)<br></code></pre></td></tr></table></figure>
<p>出现报错，显示无法加载函数。</p>
<p>解决方案：使用全局自定义函数。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># ---- gwaa ----</span><br><br><span class="hljs-comment"># Genome-wide Association Analysis</span><br><span class="hljs-comment"># Parallel implementation of linear model fitting on each SNP</span><br><br>GWAA &lt;- <span class="hljs-keyword">function</span>(genodata=genotypes,  phenodata=phenotypes, family = gaussian, filename=<span class="hljs-literal">NULL</span>,<br>                 append=<span class="hljs-literal">FALSE</span>, workers=getOption(<span class="hljs-string">&quot;mc.cores&quot;</span>,<span class="hljs-number">2L</span>), flip=<span class="hljs-literal">TRUE</span>,<br>                 select.snps=<span class="hljs-literal">NULL</span>, hosts=<span class="hljs-literal">NULL</span>, nSplits=<span class="hljs-number">10</span>)<br>&#123;<br>    <span class="hljs-keyword">if</span> (!require(doParallel)) &#123; stop(<span class="hljs-string">&quot;Missing doParallel package&quot;</span>) &#125;<br>    <br>    <span class="hljs-comment">#Check that a filename was specified</span><br>    <span class="hljs-keyword">if</span>(<span class="hljs-built_in">is.null</span>(filename)) stop(<span class="hljs-string">&quot;Must specify a filename for output.&quot;</span>)<br>    <br>    <span class="hljs-comment">#Check that the genotype data is of class &#x27;SnpMatrix&#x27;</span><br>    <span class="hljs-keyword">if</span>( <span class="hljs-built_in">class</span>(genodata)!=<span class="hljs-string">&quot;SnpMatrix&quot;</span>) stop(<span class="hljs-string">&quot;Genotype data must of class &#x27;SnpMatrix&#x27;.&quot;</span>)<br>    <br>    <span class="hljs-comment">#Check that there is a variable named &#x27;phenotype&#x27; in phenodata table</span><br>    <span class="hljs-keyword">if</span>( !<span class="hljs-string">&quot;phenotype&quot;</span> %in% colnames(phenodata))  stop(<span class="hljs-string">&quot;Phenotype data must have column named &#x27;phenotype&#x27;&quot;</span>)<br>    <br>    <span class="hljs-comment">#Check that there is a variable named &#x27;id&#x27; in phenodata table</span><br>    <span class="hljs-keyword">if</span>( !<span class="hljs-string">&quot;id&quot;</span> %in% colnames(phenodata)) stop(<span class="hljs-string">&quot;Phenotype data must have column named &#x27;id&#x27;.&quot;</span>)<br>    <br>    <span class="hljs-comment">#If a vector of SNPs is given, subset genotype data for these SNPs</span><br>    <span class="hljs-keyword">if</span>(!<span class="hljs-built_in">is.null</span>(select.snps)) genodata&lt;-genodata[,which(colnames(genodata)%in%select.snps)]<br>    <br>    <span class="hljs-comment">#Check that there are still SNPs in &#x27;SnpMatrix&#x27; object</span><br>    <span class="hljs-keyword">if</span>(ncol(genodata)==<span class="hljs-number">0</span>) stop(<span class="hljs-string">&quot;There are no SNPs in the &#x27;SnpMatrix&#x27; object.&quot;</span>)<br>    <br>    <span class="hljs-comment">#Print the number of SNPs to be checked</span><br>    cat(paste(ncol(genodata), <span class="hljs-string">&quot; SNPs included in analysis.\n&quot;</span>))<br>    <br>    <span class="hljs-comment">#If append=FALSE than we will overwrite file with column names</span><br>    <span class="hljs-keyword">if</span>(!isTRUE(append)) &#123;<br>        columns&lt;-<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;SNP&quot;</span>, <span class="hljs-string">&quot;Estimate&quot;</span>, <span class="hljs-string">&quot;Std.Error&quot;</span>, <span class="hljs-string">&quot;t-value&quot;</span>, <span class="hljs-string">&quot;p-value&quot;</span>)<br>        write.table(t(columns), filename, row.names=<span class="hljs-literal">FALSE</span>, col.names=<span class="hljs-literal">FALSE</span>, <span class="hljs-built_in">quote</span>=<span class="hljs-literal">FALSE</span>)<br>    &#125;<br><br>    <span class="hljs-comment"># Check sample counts</span><br>    <span class="hljs-keyword">if</span> (nrow(phenodata) != nrow(genodata)) &#123;<br>        warning(<span class="hljs-string">&quot;Number of samples mismatch.  Using subset found in phenodata.&quot;</span>)<br>    &#125;<br><br>    <span class="hljs-comment"># Order genodata rows to be the same as phenodata</span><br>    genodata &lt;- genodata[phenodata$id,]<br><br>    cat(nrow(genodata), <span class="hljs-string">&quot;samples included in analysis.\n&quot;</span>)<br><br>    <span class="hljs-comment"># Change which allele is counted (major or minor)</span><br>    flip.matrix&lt;-<span class="hljs-keyword">function</span>(x) &#123;<br>        zero2 &lt;- which(x==<span class="hljs-number">0</span>)<br>        two0 &lt;- which(x==<span class="hljs-number">2</span>)<br>        x[zero2] &lt;- <span class="hljs-number">2</span><br>        x[two0] &lt;- <span class="hljs-number">0</span><br>        <span class="hljs-built_in">return</span>(x)<br>    &#125;<br><br>    nSNPs &lt;- ncol(genodata)<br>    genosplit &lt;- <span class="hljs-built_in">ceiling</span>(nSNPs/nSplits) <span class="hljs-comment"># number of SNPs in each subset</span><br><br>    snp.start &lt;- seq(<span class="hljs-number">1</span>, nSNPs, genosplit) <span class="hljs-comment"># index of first SNP in group</span><br>    snp.stop &lt;- pmin(snp.start+genosplit-<span class="hljs-number">1</span>, nSNPs) <span class="hljs-comment"># index of last SNP in group</span><br><br>    <span class="hljs-keyword">if</span> (<span class="hljs-built_in">is.null</span>(hosts)) &#123;<br>        <span class="hljs-comment"># On Unix this will use fork and mclapply.  On Windows it</span><br>        <span class="hljs-comment"># will create multiple processes on localhost.</span><br>        cl &lt;- makeCluster(workers)<br>    &#125; <span class="hljs-keyword">else</span> &#123;<br>        <span class="hljs-comment"># The listed hosts must be accessible by the current user using</span><br>        <span class="hljs-comment"># password-less ssh with R installed on all hosts, all </span><br>        <span class="hljs-comment"># packages installed, and &quot;rscript&quot; is in the default PATH.</span><br>        <span class="hljs-comment"># See docs for makeCluster() for more information.</span><br>        cl &lt;- makeCluster(hosts, <span class="hljs-string">&quot;PSOCK&quot;</span>)<br>    &#125;<br>    show(cl)                            <span class="hljs-comment"># report number of workers and type of parallel implementation</span><br>    registerDoParallel(cl)<br><br>    foreach (part=<span class="hljs-number">1</span>:nSplits) %do% &#123;<br>        <span class="hljs-comment"># Returns a standar matrix of the alleles encoded as 0, 1 or 2</span><br>        genoNum &lt;- as(genodata[,snp.start[part]:snp.stop[part]], <span class="hljs-string">&quot;numeric&quot;</span>)<br><br>        <span class="hljs-comment"># Flip the numeric values of genotypes to count minor allele</span><br>        <span class="hljs-keyword">if</span> (isTRUE(flip)) genoNum &lt;- flip.matrix(genoNum)<br><br>        <span class="hljs-comment"># For each SNP, concatenate the genotype column to the</span><br>        <span class="hljs-comment"># phenodata and fit a generalized linear model</span><br>        rsVec &lt;- colnames(genoNum)<br>        res &lt;- foreach(snp.name=rsVec, .combine=<span class="hljs-string">&#x27;rbind&#x27;</span>) %dopar% &#123;<br>            a &lt;- summary(glm(phenotype~ . - id, family=family, data=cbind(phenodata, snp=genoNum[,snp.name])))<br>            a$coefficients[<span class="hljs-string">&#x27;snp&#x27;</span>,]<br>        &#125;<br><br>        <span class="hljs-comment"># write results so far to a file</span><br>        write.table(cbind(rsVec,res), filename, append=<span class="hljs-literal">TRUE</span>, <span class="hljs-built_in">quote</span>=<span class="hljs-literal">FALSE</span>, col.names=<span class="hljs-literal">FALSE</span>, row.names=<span class="hljs-literal">FALSE</span>)<br><br>        cat(sprintf(<span class="hljs-string">&quot;GWAS SNPs %s-%s (%s%% finished)\n&quot;</span>, snp.start[part], snp.stop[part], <span class="hljs-number">100</span>*part/nSplits))<br>    &#125;<br><br>    stopCluster(cl)<br>  <br>    <span class="hljs-built_in">return</span>(print(<span class="hljs-string">&quot;Done.&quot;</span>))<br>&#125;<br><span class="hljs-comment">#自定义GWAA函数</span><br><br><span class="hljs-comment">#note: this function writes a file, but does not produce a r project</span><br>start &lt;- Sys.time()<br>GWAA(genodata=genotype, phenodata=phenoSub, filename=gwaa.fname)<br><span class="hljs-comment">## [1] &quot;Done.&quot;</span><br><span class="hljs-comment">## Warning message:</span><br><span class="hljs-comment">## In GWAA(genodata = genotype, phenodata = phenoSub, filename = gwaa.fname) :</span><br><span class="hljs-comment">##   Number of samples mismatch.  Using subset found in phenodata.</span><br><br>end &lt;- Sys.time()<br>print(end-start)<br><span class="hljs-comment">## Time difference of 2.043743 hours</span><br></code></pre></td></tr></table></figure>
<h4 id="归因数据的关联分析-association-analysis-of-imputed-data"><a href="#归因数据的关联分析-association-analysis-of-imputed-data" class="headerlink" title="归因数据的关联分析 association analysis of imputed data"></a>归因数据的关联分析 association analysis of imputed data</h4><p>应用snpStats使用相应的后验概率进行估算 SNPs 的关联分析</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step8-</span><br><span class="hljs-comment">#association analysis of imputed SNPs</span><br><br><span class="hljs-comment">#-step8a-</span><br><span class="hljs-comment">#carry out associate testing for imputed SNPs using snp.rhs.tests()</span><br>rownames(phenoSub) &lt;- phenoSub$id<br>imp &lt;- snp.rhs.tests(phenotype ~ sex + age + pc1 + pc2 + pc3 + pc4 + pc5 + pc6 + pc7 + pc8 + pc9 + pc10,<br>                     family = <span class="hljs-string">&quot;Gaussian&quot;</span>,<br>                     data = phenoSub,<br>                     snp.data = target,<br>                     rules = rules)<br><br><span class="hljs-comment">#obtain p values for imputed SNPs by calling methods on the returned GlmTests object</span><br>results &lt;- data.frame(SNP = imp@snp.names, p.value = p.value(imp), stringsAsFactors = <span class="hljs-literal">FALSE</span>)<br>results &lt;- results[!<span class="hljs-built_in">is.na</span>(results$p.value),]<br><br><span class="hljs-comment">#write a file containing the results</span><br>write.csv(results, input.out.fname, row.names = <span class="hljs-literal">FALSE</span>)<br><br><span class="hljs-comment">#merge imputation testing results with support to obtain coordinates</span><br>imputeOut &lt;- merge(results, support[, <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;SNP&quot;</span>, <span class="hljs-string">&quot;position&quot;</span>)])<br>imputeOut$chr &lt;- 16<br>imputeOut$type &lt;- <span class="hljs-string">&quot;imputed&quot;</span><br><br><span class="hljs-comment">#find the -log_10 of the p-value</span><br>imputeOut$Neg_logP &lt;- -log10(imputeOut$p.value)<br><br><span class="hljs-comment">#order by p-value</span><br>imputeOut &lt;- arrange(imputeOut, p.value)<br>print(head(imputeOut))<br><span class="hljs-comment">##          SNP      p.value position chr    type Neg_logP</span><br><span class="hljs-comment">## 1  rs1532624 1.036455e-07 57005479  16 imputed 6.984449</span><br><span class="hljs-comment">## 2  rs7205804 1.036455e-07 57004889  16 imputed 6.984449</span><br><span class="hljs-comment">## 3 rs12446515 1.746000e-07 56987015  16 imputed 6.757956</span><br><span class="hljs-comment">## 4 rs17231506 1.746000e-07 56994528  16 imputed 6.757956</span><br><span class="hljs-comment">## 5   rs173539 1.746000e-07 56988044  16 imputed 6.757956</span><br><span class="hljs-comment">## 6   rs183130 1.746000e-07 56991363  16 imputed 6.757956</span><br></code></pre></td></tr></table></figure>
<p>现在，我们在16号染色体上总共确定了22个推算的SNPs，这些SNPs在 $5×10^{-6}$ 的暗示性关联阈值下是显著的。</p>
<p>接下来，我们只选择CETP区域内（±5Kb）的SNPs进行报告。在此，我们使用自开发的map2gene()函数，，它可以确定属于指定基因区域的SNPs集合。这个函数使用基于基因组参考联盟GRCh37（hg19）的基因坐标，在文件ProCodgene_coords.csv中提供。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step8b-</span><br><span class="hljs-comment"># ---- map2gene ----</span><br><span class="hljs-comment"># Returns the subset of SNPs that are within extend.boundary of gene</span><br><span class="hljs-comment"># using the coords table of gene locations.</span><br>map2gene &lt;- <span class="hljs-keyword">function</span>(gene, coords, SNPs, extend.boundary = <span class="hljs-number">5000</span>) &#123;<br>  coordsSub &lt;- coords[coords$gene == gene,] <span class="hljs-comment">#Subset coordinate file for spcified gene</span><br>  <br>  coordsSub$start &lt;- coordsSub$start - extend.boundary <span class="hljs-comment"># Extend gene boundaries</span><br>  coordsSub$stop &lt;- coordsSub$stop + extend.boundary<br>  <br>  SNPsub &lt;- SNPs[SNPs$position &gt;= coordsSub$start &amp; SNPs$position &lt;= coordsSub$stop &amp;<br>                   SNPs$chr == coordsSub$chr,] <span class="hljs-comment">#Subset for SNPs in gene</span><br>  <br>  <span class="hljs-built_in">return</span>(data.frame(SNPsub, gene = gene, stringsAsFactors = <span class="hljs-literal">FALSE</span>))<br>&#125;<br><br><span class="hljs-comment">#read in file containing protein coding genes coords</span><br>genes &lt;- read.csv(protein.coding.coords.fnamo, stringsAsFactors = <span class="hljs-literal">FALSE</span>)<br><br><span class="hljs-comment">#subset for CETP SNPs</span><br>impCETP &lt;- map2gene(<span class="hljs-string">&quot;CETP&quot;</span>, coords = genes, SNPs = imputeOut)<br><br><span class="hljs-comment">#filter only the imputed CETP SNP genotypes</span><br>impCETPgeno &lt;- imputed[,impCETP$SNP]<br></code></pre></td></tr></table></figure>
<p>在这个阶段，我们将70个归纳的SNPs映射到CETP区域，其中16个在 $5×10^{-6}$ 的提示性关联阈值上是显著的。</p>
<h3 id="Post‐analytic-visualization-and-genomic-interrogation-后分析可视化和基因组审查"><a href="#Post‐analytic-visualization-and-genomic-interrogation-后分析可视化和基因组审查" class="headerlink" title="Post‐analytic visualization and genomic interrogation 后分析可视化和基因组审查"></a>Post‐analytic visualization and genomic interrogation 后分析可视化和基因组审查</h3><h4 id="归因和类型SNP结果的整合-integration-of-imputed-and-typed-SNP-results"><a href="#归因和类型SNP结果的整合-integration-of-imputed-and-typed-SNP-results" class="headerlink" title="归因和类型SNP结果的整合 integration of imputed and typed SNP results"></a>归因和类型SNP结果的整合 integration of imputed and typed SNP results</h4><p>在这个阶段，将SNPs归属到基因座上并报告染色体和碱基对的位置，也被称为坐标或位置，也是很常见的。</p>
<p>值得注意的是，SNP的坐标取决于基因组的构建，在我们的数据例子中，我们使用基因组参考联盟GRCh37（hg19）构建。典型的结果展示包括基因和基因座名称；SNP名称；染色体编号；碱基对位置，根据指定的构建；来自模型拟合程序的系数估计（或赔率）；相应的标准误差；和相关的P值。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><code class="hljs r"><span class="hljs-comment">#-step9-</span><br><span class="hljs-comment">#data integration</span><br><br><span class="hljs-comment">#-step9a-</span><br><span class="hljs-comment">#read in GWAS output that was produced by GWAA function</span><br>GWASout &lt;- read.table(gwaa.fname, header = <span class="hljs-literal">TRUE</span>, colClasses = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;character&quot;</span>, <span class="hljs-built_in">rep</span>(<span class="hljs-string">&quot;numeric&quot;</span>,<span class="hljs-number">4</span>)))<br><br><span class="hljs-comment">#find the -log_10 of the p-value</span><br>GWASout$Neg_logP &lt;- -log10(GWASout$p.value)<br><br><span class="hljs-comment">#merge output with genobim by SNP name to add position and chromosome number</span><br>GWASout &lt;- merge(GWASout, genobim[,<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;SNP&quot;</span>,<span class="hljs-string">&quot;chr&quot;</span>,<span class="hljs-string">&quot;position&quot;</span>)])<br><br><span class="hljs-comment">#order SNPs by significance</span><br>GWASout &lt;- arrange(GWASout, -Neg_logP)<br>print(head(GWASout))<br><span class="hljs-comment">##          SNP   Estimate  Std.Error   t.value      p.value Neg_logP chr  position</span><br><span class="hljs-comment">## 1  rs1532625  0.2015758 0.03748045  5.378159 8.944627e-08 7.048438  16  57005301</span><br><span class="hljs-comment">## 2   rs247617  0.2100771 0.03979461  5.279034 1.524209e-07 6.816955  16  56990716</span><br><span class="hljs-comment">## 3 rs10945761  0.1868139 0.04087883  4.569942 5.344253e-06 5.272113   6 162065367</span><br><span class="hljs-comment">## 4  rs9647610  0.1840277 0.04066594  4.525353 6.586987e-06 5.181313   6 162066421</span><br><span class="hljs-comment">## 5  rs3803768 -0.3048526 0.06755966 -4.512346 6.993775e-06 5.155288  17  80872028</span><br><span class="hljs-comment">## 6  rs4821708 -0.1808045 0.04018791 -4.498978 7.456555e-06 5.127462  22  38164106</span><br></code></pre></td></tr></table></figure>
<p>现在还需要一个额外的步骤来结合估算的数据结果和类型的SNP结果。值得注意的是，基因型推算可以涉及推算所有的SNPs，包括未观察到的和已分型的SNPs。因此，分析者可以选择从归因结果中只选择未分型或未通过SNP级过滤的SNP。在我们的例子中（第6步），我们归集了未分型的SNP以及未通过SNP水平过滤的SNP（第2和第4步）。然后，这个组合中的GWA重要SNPs可以进一步可视化，并按照步骤10的描述进行询问。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step9b-</span><br><span class="hljs-comment">#combine typed and imputed</span><br>GWASout$type &lt;- <span class="hljs-string">&quot;typed&quot;</span><br><br>GWAScomb &lt;- rbind.fill(GWASout, imputeOut)<br>head(GWAScomb)<br><span class="hljs-comment">##          SNP   Estimate  Std.Error   t.value      p.value Neg_logP chr  position  type</span><br><span class="hljs-comment">## 1  rs1532625  0.2015758 0.03748045  5.378159 8.944627e-08 7.048438  16  57005301 typed</span><br><span class="hljs-comment">## 2   rs247617  0.2100771 0.03979461  5.279034 1.524209e-07 6.816955  16  56990716 typed</span><br><span class="hljs-comment">## 3 rs10945761  0.1868139 0.04087883  4.569942 5.344253e-06 5.272113   6 162065367 typed</span><br><span class="hljs-comment">## 4  rs9647610  0.1840277 0.04066594  4.525353 6.586987e-06 5.181313   6 162066421 typed</span><br><span class="hljs-comment">## 5  rs3803768 -0.3048526 0.06755966 -4.512346 6.993775e-06 5.155288  17  80872028 typed</span><br><span class="hljs-comment">## 6  rs4821708 -0.1808045 0.04018791 -4.498978 7.456555e-06 5.127462  22  38164106 typed</span><br><br>tail(GWAScomb)<br><span class="hljs-comment">##              SNP Estimate Std.Error t.value   p.value     Neg_logP chr position    type</span><br><span class="hljs-comment">## 818527 rs7201255       NA        NA      NA 0.9999695 1.325135e-05  16 53869223 imputed</span><br><span class="hljs-comment">## 818528 rs7203572       NA        NA      NA 0.9999695 1.325135e-05  16 53869366 imputed</span><br><span class="hljs-comment">## 818529  rs257500       NA        NA      NA 0.9999870 5.647027e-06  16 64971479 imputed</span><br><span class="hljs-comment">## 818530  rs369139       NA        NA      NA 0.9999870 5.647027e-06  16 64968347 imputed</span><br><span class="hljs-comment">## 818531   rs40327       NA        NA      NA 0.9999870 5.647027e-06  16 64978301 imputed</span><br><span class="hljs-comment">## 818532  rs466296       NA        NA      NA 0.9999870 5.647027e-06  16 64962442 imputed</span><br><br><span class="hljs-comment">#subset for CETP SNPs</span><br>typCETP &lt;- map2gene(<span class="hljs-string">&quot;CETP&quot;</span>, coords = genes, SNPs = GWASout)<br><br><span class="hljs-comment">#combine CETP SNPs from imputed and typed analysis</span><br>CETP &lt;- rbind.fill(typCETP, impCETP)[,<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;SNP&quot;</span>, <span class="hljs-string">&quot;p.value&quot;</span>, <span class="hljs-string">&quot;Neg_logP&quot;</span>, <span class="hljs-string">&quot;chr&quot;</span>, <span class="hljs-string">&quot;position&quot;</span>, <span class="hljs-string">&quot;type&quot;</span>, <span class="hljs-string">&quot;gene&quot;</span>)]<br>print(CETP)<br><span class="hljs-comment">##            SNP      p.value   Neg_logP chr position    type gene</span><br><span class="hljs-comment">## 1    rs1532625 8.944627e-08 7.04843778  16 57005301   typed CETP</span><br><span class="hljs-comment">## 2     rs289742 4.257667e-04 3.37082835  16 57017762   typed CETP</span><br><span class="hljs-comment">## 3     rs289715 4.740815e-03 2.32414700  16 57008508   typed CETP</span><br><span class="hljs-comment">## 4    rs6499863 1.394646e-02 1.85553612  16 56992017   typed CETP</span><br><span class="hljs-comment">## 5    rs4783962 1.011933e-01 0.99484809  16 56995038   typed CETP</span><br><span class="hljs-comment">## 6    rs1800777 1.083617e-01 0.96512430  16 57017319   typed CETP</span><br><span class="hljs-comment">## 7   rs12708980 6.331485e-01 0.19849440  16 57012379   typed CETP</span><br><span class="hljs-comment">## 8    rs1532624 1.036455e-07 6.98444942  16 57005479 imputed CETP</span><br><span class="hljs-comment">## 9    rs7205804 1.036455e-07 6.98444942  16 57004889 imputed CETP</span><br><span class="hljs-comment">## 10  rs17231506 1.746000e-07 6.75795566  16 56994528 imputed CETP</span><br><span class="hljs-comment">## 11    rs183130 1.746000e-07 6.75795566  16 56991363 imputed CETP</span><br><span class="hljs-comment">## 12   rs3764261 1.746000e-07 6.75795566  16 56993324 imputed CETP</span><br><span class="hljs-comment">## 13    rs821840 1.746000e-07 6.75795566  16 56993886 imputed CETP</span><br><span class="hljs-comment">## 14  rs11508026 1.174797e-06 5.93003733  16 56999328 imputed CETP</span><br><span class="hljs-comment">## 15  rs12444012 1.174797e-06 5.93003733  16 57001438 imputed CETP</span><br><span class="hljs-comment">## 16  rs12720926 1.174797e-06 5.93003733  16 56998918 imputed CETP</span><br><span class="hljs-comment">## 17   rs4784741 1.174797e-06 5.93003733  16 57001216 imputed CETP</span><br><span class="hljs-comment">## 18  rs34620476 1.177972e-06 5.92886509  16 56996649 imputed CETP</span><br><span class="hljs-comment">## 19    rs708272 1.177972e-06 5.92886509  16 56996288 imputed CETP</span><br><span class="hljs-comment">## 20    rs711752 1.177972e-06 5.92886509  16 56996211 imputed CETP</span><br><span class="hljs-comment">## 21  rs12720922 3.392213e-06 5.46951693  16 57000885 imputed CETP</span><br><span class="hljs-comment">## 22   rs8045855 3.392213e-06 5.46951693  16 57000696 imputed CETP</span><br><span class="hljs-comment">## 23  rs12149545 3.692106e-06 5.43272585  16 56993161 imputed CETP</span><br><span class="hljs-comment">## 24   rs1800775 1.423952e-05 4.84650452  16 56995236 imputed CETP</span><br><span class="hljs-comment">## 25   rs3816117 1.423952e-05 4.84650452  16 56996158 imputed CETP</span><br><span class="hljs-comment">## 26  rs11076175 1.431416e-05 4.84423402  16 57006378 imputed CETP</span><br><span class="hljs-comment">## 27   rs7499892 1.431416e-05 4.84423402  16 57006590 imputed CETP</span><br><span class="hljs-comment">## 28  rs11076176 1.100766e-04 3.95830512  16 57007446 imputed CETP</span><br><span class="hljs-comment">## 29    rs289714 1.123948e-04 3.94925380  16 57007451 imputed CETP</span><br><span class="hljs-comment">## 30    rs158478 1.958087e-04 3.70816791  16 57007734 imputed CETP</span><br><span class="hljs-comment">## 31   rs9939224 3.075278e-04 3.51211557  16 57002732 imputed CETP</span><br><span class="hljs-comment">## 32  rs12447620 4.344681e-04 3.36204212  16 57014319 imputed CETP</span><br><span class="hljs-comment">## 33    rs158480 4.344681e-04 3.36204212  16 57008227 imputed CETP</span><br><span class="hljs-comment">## 34    rs158617 4.344681e-04 3.36204212  16 57008287 imputed CETP</span><br><span class="hljs-comment">## 35  rs11076174 4.293928e-03 2.36714525  16 57003146 imputed CETP</span><br><span class="hljs-comment">## 36 rs112039804 4.742793e-03 2.32396582  16 57018856 imputed CETP</span><br><span class="hljs-comment">## 37  rs12708985 4.742793e-03 2.32396582  16 57014610 imputed CETP</span><br><span class="hljs-comment">## 38    rs736274 4.742793e-03 2.32396582  16 57009769 imputed CETP</span><br><span class="hljs-comment">## 39 rs201825234 1.403181e-02 1.85288617  16 56991948 imputed CETP</span><br><span class="hljs-comment">## 40   rs2115429 1.403181e-02 1.85288617  16 56992842 imputed CETP</span><br><span class="hljs-comment">## 41   rs6499861 1.403181e-02 1.85288617  16 56991495 imputed CETP</span><br><span class="hljs-comment">## 42   rs6499862 1.403181e-02 1.85288617  16 56991524 imputed CETP</span><br><span class="hljs-comment">## 43    rs158479 1.764815e-02 1.75330083  16 57008048 imputed CETP</span><br><span class="hljs-comment">## 44    rs289713 1.848673e-02 1.73313993  16 57006829 imputed CETP</span><br><span class="hljs-comment">## 45  rs12720918 2.690313e-02 1.57019722  16 56994212 imputed CETP</span><br><span class="hljs-comment">## 46  rs12920974 2.690313e-02 1.57019722  16 56993025 imputed CETP</span><br><span class="hljs-comment">## 47  rs36229787 3.284685e-02 1.48350624  16 56993897 imputed CETP</span><br><span class="hljs-comment">## 48    rs289712 4.630073e-02 1.33441214  16 57006305 imputed CETP</span><br><span class="hljs-comment">## 49    rs820299 5.313036e-02 1.27465724  16 57000284 imputed CETP</span><br><span class="hljs-comment">## 50  rs34946873 5.924969e-02 1.22731392  16 56991143 imputed CETP</span><br><span class="hljs-comment">## 51  rs12597002 6.482442e-02 1.18826138  16 57002404 imputed CETP</span><br><span class="hljs-comment">## 52  rs60545348 6.482442e-02 1.18826138  16 57001985 imputed CETP</span><br><span class="hljs-comment">## 53    rs708273 6.482442e-02 1.18826138  16 56999949 imputed CETP</span><br><span class="hljs-comment">## 54   rs4369653 6.674449e-02 1.17558455  16 56997551 imputed CETP</span><br><span class="hljs-comment">## 55   rs4587963 7.720580e-02 1.11235007  16 56997369 imputed CETP</span><br><span class="hljs-comment">## 56      rs5880 8.144506e-02 1.08913528  16 57015091 imputed CETP</span><br><span class="hljs-comment">## 57   rs1800776 9.403016e-02 1.02673283  16 56995234 imputed CETP</span><br><span class="hljs-comment">## 58    rs289746 9.558715e-02 1.01960051  16 57020205 imputed CETP</span><br><span class="hljs-comment">## 59  rs12447839 1.014589e-01 0.99370984  16 56993935 imputed CETP</span><br><span class="hljs-comment">## 60  rs12447924 1.014589e-01 0.99370984  16 56994192 imputed CETP</span><br><span class="hljs-comment">## 61    rs158477 1.602619e-01 0.79516962  16 57007610 imputed CETP</span><br><span class="hljs-comment">## 62  rs12708983 2.717574e-01 0.56581860  16 57014411 imputed CETP</span><br><span class="hljs-comment">## 63  rs66495554 2.827752e-01 0.54855862  16 57018636 imputed CETP</span><br><span class="hljs-comment">## 64  rs12720889 2.857284e-01 0.54404657  16 57012563 imputed CETP</span><br><span class="hljs-comment">## 65  rs12934552 3.188555e-01 0.49640603  16 57021433 imputed CETP</span><br><span class="hljs-comment">## 66  rs12708968 3.476125e-01 0.45890456  16 56994819 imputed CETP</span><br><span class="hljs-comment">## 67  rs17245715 3.476125e-01 0.45890456  16 56994990 imputed CETP</span><br><span class="hljs-comment">## 68   rs4783961 4.661962e-01 0.33143130  16 56994894 imputed CETP</span><br><span class="hljs-comment">## 69  rs12598522 4.853361e-01 0.31395741  16 57022352 imputed CETP</span><br><span class="hljs-comment">## 70  rs56315364 4.853361e-01 0.31395741  16 57021524 imputed CETP</span><br><span class="hljs-comment">## 71 rs117427818 5.101389e-01 0.29231153  16 57010486 imputed CETP</span><br><span class="hljs-comment">## 72  rs11860407 5.935059e-01 0.22657493  16 57010828 imputed CETP</span><br><span class="hljs-comment">## 73   rs2033254 5.935059e-01 0.22657493  16 57009985 imputed CETP</span><br><span class="hljs-comment">## 74   rs7405284 6.134481e-01 0.21222214  16 57001275 imputed CETP</span><br><span class="hljs-comment">## 75   rs1800774 6.200971e-01 0.20754027  16 57015545 imputed CETP</span><br><span class="hljs-comment">## 76  rs36229786 6.227677e-01 0.20567392  16 56993901 imputed CETP</span><br><span class="hljs-comment">## 77  rs12708974 8.568509e-01 0.06709476  16 57005550 imputed CETP</span><br></code></pre></td></tr></table></figure>
<h4 id="关联结果的可视化和质量控制-visualization-and-quality-control-of-association-findings"><a href="#关联结果的可视化和质量控制-visualization-and-quality-control-of-association-findings" class="headerlink" title="关联结果的可视化和质量控制 visualization and quality control of association findings"></a>关联结果的可视化和质量控制 visualization and quality control of association findings</h4><p>这里描述了三种可视化工具。除了这些可视化方法，使用其他基因模型进行关联分析也是一个有用的敏感度分析。</p>
<h5 id="Manhattan-plots-曼哈顿图"><a href="#Manhattan-plots-曼哈顿图" class="headerlink" title="Manhattan plots 曼哈顿图"></a>Manhattan plots 曼哈顿图</h5><p>曼哈顿图被用来按染色体位置直观地显示GWA的显著性水平。在曼哈顿图中，每个点对应于一个SNP。X轴代表基因坐标，显示的数字对应于染色体编号。y轴是对数p值的负值，所以大的值对应小的p值。实心横线表示Bonferonni校正的显著性阈值（$- log(5 × 10^{-8})$）。</p>
<p>虚线是一个不太严格的暗示性关联阈值（$- log(5 × 10^{-6})$），我们将其作为暗示性关联的指标，需要进一步验证。</p>
<p>通过对该图的目视检查，可以识别出P值相对较小的SNP，这些SNP位于P值相对较大且不显著的区域，表明有可能是虚假的发现。CETP区域的多个信号表明，这可能是一个真实的信号。该图是用我们开发的GWAS_Manhattan()生成的。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step10-</span><br><span class="hljs-comment">#visualizing and QC of GWA findings</span><br><span class="hljs-comment"># ---- manhattan ----</span><br><span class="hljs-comment"># Receives a data.frame of SNPs with Neg_logP, chr, position, and type.</span><br><span class="hljs-comment"># Plots Manhattan plot with significant SNPs highlighted.</span><br>GWAS_Manhattan &lt;- <span class="hljs-keyword">function</span>(GWAS, col.snps=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;black&quot;</span>,<span class="hljs-string">&quot;gray&quot;</span>),<br>                           col.detected=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;black&quot;</span>), col.imputed=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;blue&quot;</span>), col.text=<span class="hljs-string">&quot;black&quot;</span>,<br>                           title=<span class="hljs-string">&quot;GWAS Tutorial Manhattan Plot&quot;</span>, display.text=<span class="hljs-literal">TRUE</span>,<br>                           bonferroni.alpha=<span class="hljs-number">0.05</span>, bonferroni.adjustment=<span class="hljs-number">1000000</span>,<br>                           Lstringent.adjustment=<span class="hljs-number">10000</span>) &#123;<br>  <br>  bonferroni.thresh &lt;- -log10(bonferroni.alpha / bonferroni.adjustment)<br>  Lstringent.thresh &lt;- -log10(bonferroni.alpha / Lstringent.adjustment)<br>  xscale &lt;- 1000000<br>  <br>  manhat &lt;- GWAS[!grepl(<span class="hljs-string">&quot;[A-z]&quot;</span>,GWAS$chr),]<br>  <br>  <span class="hljs-comment">#sort the data by chromosome and then location</span><br>  manhat.ord &lt;- manhat[order(<span class="hljs-built_in">as.numeric</span>(manhat$chr),manhat$position),]<br>  manhat.ord &lt;- manhat.ord[!<span class="hljs-built_in">is.na</span>(manhat.ord$position),]<br>  <br>  <span class="hljs-comment">##Finding the maximum position for each chromosome</span><br>  max.pos &lt;- sapply(<span class="hljs-number">1</span>:<span class="hljs-number">21</span>, <span class="hljs-keyword">function</span>(i) &#123; <span class="hljs-built_in">max</span>(manhat.ord$position[manhat.ord$chr==i],<span class="hljs-number">0</span>) &#125;)<br>  max.pos2 &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>, <span class="hljs-built_in">cumsum</span>(max.pos))                  <br>  <br>  <span class="hljs-comment">#Add spacing between chromosomes</span><br>  max.pos2 &lt;- max.pos2 + <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>:<span class="hljs-number">21</span>) * xscale * <span class="hljs-number">10</span><br>  <br>  <span class="hljs-comment">#defining the positions of each snp in the plot</span><br>  manhat.ord$pos &lt;- manhat.ord$position + max.pos2[<span class="hljs-built_in">as.numeric</span>(manhat.ord$chr)]<br>  <br>  <span class="hljs-comment"># alternate coloring of chromosomes</span><br>  manhat.ord$col &lt;- col.snps[<span class="hljs-number">1</span> + <span class="hljs-built_in">as.numeric</span>(manhat.ord$chr) %% <span class="hljs-number">2</span>]<br>  <br>  <span class="hljs-comment"># draw the chromosome label roughly in the middle of each chromosome band</span><br>  text.pos &lt;- sapply(<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>:<span class="hljs-number">22</span>), <span class="hljs-keyword">function</span>(i) &#123; mean(manhat.ord$pos[manhat.ord$chr==i]) &#125;)<br>  <br>  <span class="hljs-comment"># Plot the data</span><br>  plot(manhat.ord$pos[manhat.ord$type==<span class="hljs-string">&quot;typed&quot;</span>]/xscale, manhat.ord$Neg_logP[manhat.ord$type==<span class="hljs-string">&quot;typed&quot;</span>],<br>       pch=<span class="hljs-number">20</span>, cex=<span class="hljs-number">.3</span>, col= manhat.ord$col[manhat.ord$type==<span class="hljs-string">&quot;typed&quot;</span>], xlab=<span class="hljs-literal">NA</span>,<br>       ylab=<span class="hljs-string">&quot;Negative Log P-value&quot;</span>, axes=<span class="hljs-built_in">F</span>, ylim=<span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>,<span class="hljs-built_in">max</span>(manhat$Neg_logP)+<span class="hljs-number">1</span>))<br>  <span class="hljs-comment">#Add x-label so that it is close to axis</span><br>  mtext(side = <span class="hljs-number">1</span>, <span class="hljs-string">&quot;Chromosome&quot;</span>, line = <span class="hljs-number">1.25</span>)<br>  <br>  points(manhat.ord$pos[manhat.ord$type==<span class="hljs-string">&quot;imputed&quot;</span>]/xscale, manhat.ord$Neg_logP[manhat.ord$type==<span class="hljs-string">&quot;imputed&quot;</span>],<br>         pch=<span class="hljs-number">20</span>, cex=<span class="hljs-number">.4</span>, col = col.imputed)<br>  <br>  points(manhat.ord$pos[manhat.ord$type==<span class="hljs-string">&quot;typed&quot;</span>]/xscale, manhat.ord$Neg_logP[manhat.ord$type==<span class="hljs-string">&quot;typed&quot;</span>],<br>         pch=<span class="hljs-number">20</span>, cex=<span class="hljs-number">.3</span>, col = manhat.ord$col[manhat.ord$type==<span class="hljs-string">&quot;typed&quot;</span>])<br>  <br>  axis(<span class="hljs-number">2</span>)<br>  abline(h=<span class="hljs-number">0</span>)<br>  <br>  SigNifSNPs &lt;- <span class="hljs-built_in">as.character</span>(GWAS[GWAS$Neg_logP &gt; Lstringent.thresh &amp; GWAS$type==<span class="hljs-string">&quot;typed&quot;</span>, <span class="hljs-string">&quot;SNP&quot;</span>])<br>  <br>  <span class="hljs-comment">#Add legend</span><br>  legend(<span class="hljs-string">&quot;topright&quot;</span>,<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;Bonferroni corrected threshold (p = 5E-8)&quot;</span>, <span class="hljs-string">&quot;Candidate threshold (p = 5E-6)&quot;</span>),<br>         border=<span class="hljs-string">&quot;black&quot;</span>, col=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;gray60&quot;</span>, <span class="hljs-string">&quot;gray60&quot;</span>), pch=<span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>, <span class="hljs-number">0</span>), lwd=<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">1</span>),<br>         lty=<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">2</span>), pt.cex=<span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>,<span class="hljs-number">0</span>), bty=<span class="hljs-string">&quot;o&quot;</span>, cex=<span class="hljs-number">0.6</span>)<br>  <br>  <span class="hljs-comment">#Add chromosome number</span><br>  text(text.pos/xscale, -<span class="hljs-number">.3</span>, seq(<span class="hljs-number">1</span>,<span class="hljs-number">22</span>,by=<span class="hljs-number">1</span>), xpd=<span class="hljs-literal">TRUE</span>, cex=<span class="hljs-number">.8</span>)<br>  <br>  <span class="hljs-comment">#Add bonferroni line</span><br>  abline(h=bonferroni.thresh, untf = <span class="hljs-literal">FALSE</span>, col = <span class="hljs-string">&quot;gray60&quot;</span>)<br>  <br>  <span class="hljs-comment">#Add &quot;less stringent&quot; line</span><br>  abline(h=Lstringent.thresh, untf = <span class="hljs-literal">FALSE</span>, col = <span class="hljs-string">&quot;gray60&quot;</span>, lty = <span class="hljs-number">2</span> )<br>  <br>  <span class="hljs-comment">#Plotting detected genes</span><br>  <span class="hljs-comment">#Were any genes detected?</span><br>  <span class="hljs-keyword">if</span> (<span class="hljs-built_in">length</span>(SigNifSNPs)&gt;<span class="hljs-number">0</span>)&#123;<br>    <br>    sig.snps &lt;- manhat.ord[,<span class="hljs-string">&#x27;SNP&#x27;</span>] %in% SigNifSNPs<br>    <br>    points(manhat.ord$pos[sig.snps]/xscale,<br>           manhat.ord$Neg_logP[sig.snps],<br>           pch=<span class="hljs-number">20</span>,col=col.detected, bg=col.detected,cex=<span class="hljs-number">0.5</span>)<br>    <br>    text(manhat.ord$pos[sig.snps]/xscale,<br>         manhat.ord$Neg_logP[sig.snps],<br>         <span class="hljs-built_in">as.character</span>(manhat.ord[sig.snps,<span class="hljs-number">1</span>]), col=col.text, offset=<span class="hljs-number">1</span>, adj=-<span class="hljs-number">.1</span>, cex=<span class="hljs-number">.5</span>)<br>  &#125;<br>&#125;<br>par(mfrow=<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">1</span>))<br><br><span class="hljs-comment">#-step10a-</span><br><span class="hljs-comment">#create manhattan plot</span><br>GWAS_Manhattan(GWAScomb)<br></code></pre></td></tr></table></figure>
<p><img src="manhattan_plot.png" srcset="/img/loading.gif" alt="manhattan_plot"></p>
<h5 id="Q–Q-plots-and-the-λ‐statistic-QQ图和λ统计量"><a href="#Q–Q-plots-and-the-λ‐statistic-QQ图和λ统计量" class="headerlink" title="Q–Q plots and the λ‐statistic QQ图和λ统计量"></a>Q–Q plots and the <em>λ</em>‐statistic QQ图和λ统计量</h5><p>Q-Q图用于可视化SNP水平测试统计量的预期分布和观察分布之间的关系。在考虑了建模框架中人口亚结构的潜在混杂因素后，分布的尾部被拉近到y=x线。如果该图中的数据从y = x线上移或下移，那么我们将考虑可能的系统性偏差。最后，右上角尾部与y = x线的轻微偏差粗略地表明，数据中存在着某种形式的关联。</p>
<p>偏离这条路线的程度由λ统计量来正式衡量，其中接近1的数值表明对潜在的子结构进行了适当的调整。虽然在对PC进行调整后，λ得到了改善（从λ=1.014到λ=1.0032），但由于这个PennCATH样本来自一个相对同质的人群，所以没有观察到数值上的巨大差异。一般来说，目标是使λ值接近于1；λ&gt;1.2表明有分层，通常在这种情况下会包括额外的PC，在某些情况下，该研究会被排除在随后的荟萃分析中。计算一个考虑到样本量的标准化λ，在不同研究的数值对比中特别有用，以便纳入荟萃分析。我们应用以下代码为未调整和调整模型生成标准化的λ，结果是未调整和调整模型的λ分别为1.0108和1.000632。对于二元性状，可以应用标准化方法。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step10b-</span><br><span class="hljs-comment">#return the GWAS using unadjusted model</span><br>phenoSub2 &lt;- phenoSub[,<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;id&quot;</span>, <span class="hljs-string">&quot;phenotype&quot;</span>)] <span class="hljs-comment">#remove all extra fators, leave only phenotype</span><br><br>GWAA(genodata = genotype, phenodata = phenoSub2, filename = gwaa.unadj.fname)<br>GWASoutUnadj &lt;- read.table(gwaa.unadj.fname, head=<span class="hljs-literal">TRUE</span>, colClasses = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;character&quot;</span>,<span class="hljs-built_in">rep</span>(<span class="hljs-string">&quot;numeric&quot;</span>,<span class="hljs-number">4</span>)))<br><br><span class="hljs-comment">#create QQ plots for adjusted and unadjusted model outputs</span><br>par(mfrow=<span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>,<span class="hljs-number">2</span>))<br>lambdaAdj &lt;- estlambda(GWASout$t.value^<span class="hljs-number">2</span>, plot = <span class="hljs-literal">TRUE</span>, method = <span class="hljs-string">&quot;median&quot;</span>)<br>lambdaUnadj &lt;- estlambda(GWASoutUnadj$t.value^<span class="hljs-number">2</span>, plot = <span class="hljs-literal">TRUE</span>, method = <span class="hljs-string">&quot;median&quot;</span>)<br>cat(sprintf(<span class="hljs-string">&quot;Unadjusted lambda: %s\nAdjusted lambda: %s\n&quot;</span>, lambdaUnadj$estimate, lambdaAdj$estimate))<br><span class="hljs-comment">## Unadjusted lambda: 1.01417377078806</span><br><span class="hljs-comment">## Adjusted lambda: 1.00056081659278</span><br><br><span class="hljs-comment">#-step10c-</span><br><span class="hljs-comment">#calculate standardized lambda</span><br>lambdaAdj_1000 &lt;- 1+(lambdaAdj$estimate-<span class="hljs-number">1</span>)/nrow(phenoSub)*<span class="hljs-number">1000</span><br>lambdaUnadj_1000 &lt;- 1+(lambdaUnadj$estimate-<span class="hljs-number">1</span>)/nrow(phenoSub)*<span class="hljs-number">1000</span><br>cat(sprintf(<span class="hljs-string">&quot;standardized unadjusted lambda: %s\nstandard adjusted lambda: %s\n&quot;</span>, lambdaUnadj_1000, lambdaAdj_1000))<br><span class="hljs-comment">## standardized unadjusted lambda: 1.0108279379588</span><br><span class="hljs-comment">## standard adjusted lambda: 1.0004284313161</span><br></code></pre></td></tr></table></figure>
<p><img src="qqplot.png" srcset="/img/loading.gif" alt="qqplot"></p>
<h5 id="Heatmaps-and-regional-association-plots-热图和区域关联图"><a href="#Heatmaps-and-regional-association-plots-热图和区域关联图" class="headerlink" title="Heatmaps and regional association plots 热图和区域关联图"></a>Heatmaps and regional association plots 热图和区域关联图</h5><p>热图和区域关联图通常在GWA分析的背景下使用，以可视化GWA显著SNPs和附近区域SNPs之间的LD模式，LD通常用 $r^2$ 或 $D′$ 来衡量，它们都是标量D的转化，定义为两个主要等位基因的联合概率与两个边际概率的乘积之间的差异，其中调整是基于等位基因频率。</p>
<p>阴影的程度表示LD的数量，因此较深的方块表示较高的LD。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#-step10d-</span><br>install.packages(<span class="hljs-string">&quot;LDheatmap&quot;</span>)<br>BiocManager::install(<span class="hljs-string">&quot;rtracklayer&quot;</span>)<br><br><span class="hljs-comment">#add &quot;rs247671&quot; to CETP</span><br>CETP &lt;- rbind.fill(GWASout[GWASout$SNP == <span class="hljs-string">&quot;rs247617&quot;</span>,],CETP)<br><br><span class="hljs-comment">#combine genotype and imputed genotypes for CETP region</span><br>subgen &lt;- cbind(genotype[,colnames(genotype) %in% CETP$SNP], impCETPgeno) <span class="hljs-comment">#CETP subsets from typed and imputed SNPs</span><br><br><span class="hljs-comment">#subset SNPs for only certain genptypes</span><br>certain &lt;- apply(as(subgen,<span class="hljs-string">&#x27;numeric&#x27;</span>), <span class="hljs-number">2</span>, <span class="hljs-keyword">function</span>(x) &#123;<span class="hljs-built_in">all</span>(x %in% <span class="hljs-built_in">c</span>(<span class="hljs-number">0</span>,<span class="hljs-number">1</span>,<span class="hljs-number">2</span>,<span class="hljs-literal">NA</span>))&#125;)<br>subgen &lt;- subgen[,certain]<br><br><span class="hljs-comment">#subset and order CETP SNPs by position</span><br>CETP &lt;- CETP[CETP$SNP %in% colnames(subgen),]<br>CETP &lt;- arrange(CETP, position)<br>subgen &lt;- subgen[, order(match(colnames(subgen),CETP$SNP)) ]<br><br><span class="hljs-comment"># Create LDheatmap</span><br>ld &lt;- ld(subgen, subgen, stats=<span class="hljs-string">&quot;R.squared&quot;</span>) <span class="hljs-comment"># Find LD map of CETP SNPs</span><br><br>ll &lt;- LDheatmap(ld, CETP$position, flip=<span class="hljs-literal">TRUE</span>, name=<span class="hljs-string">&quot;myLDgrob&quot;</span>, title=<span class="hljs-literal">NULL</span>)<br><br><span class="hljs-comment"># Add genes, recombination</span><br>llplusgenes &lt;- LDheatmap.addGenes(ll, chr = <span class="hljs-string">&quot;chr16&quot;</span>, genome = <span class="hljs-string">&quot;hg19&quot;</span>, genesLocation = <span class="hljs-number">0.01</span>)<br><br><span class="hljs-comment"># Add plot of -log(p)</span><br>library(ggplot2)<br>library(grid)<br><br>plot.new()<br>llQplot2&lt;-LDheatmap.addGrob(llplusgenes, rectGrob(gp = gpar(col = <span class="hljs-string">&quot;white&quot;</span>)),height = <span class="hljs-number">.34</span>)<br>pushViewport(viewport(x = <span class="hljs-number">0.483</span>, y= <span class="hljs-number">0.76</span>, width = <span class="hljs-number">.91</span> ,height = <span class="hljs-number">.4</span>))<br></code></pre></td></tr></table></figure>
<p><img src="ldheatmap.png" srcset="/img/loading.gif" alt="ldheatmap"></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs R">grid.draw(ggplotGrob(&#123;<br>  qplot(position, Neg_logP, data = CETP, xlab=<span class="hljs-string">&quot;&quot;</span>, ylab = <span class="hljs-string">&quot;Negative Log P-value&quot;</span>, xlim = <span class="hljs-built_in">range</span>(CETP$position),<br>        asp = <span class="hljs-number">1</span>/<span class="hljs-number">10</span>, color = factor(type), colour=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;#000000&quot;</span>, <span class="hljs-string">&quot;#D55E00&quot;</span>)) + <br>    theme(axis.text.x = element_blank(),<br>          axis.title.y = element_text(size = rel(<span class="hljs-number">0.75</span>)), legend.position = <span class="hljs-string">&quot;none&quot;</span>, <br>          panel.background = element_blank(), <br>          axis.line = element_line(colour = <span class="hljs-string">&quot;black&quot;</span>)) +<br>    scale_color_manual(values = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;red&quot;</span>, <span class="hljs-string">&quot;black&quot;</span>))<br>&#125;))<br></code></pre></td></tr></table></figure>
<p><img src="ldheatmap_new.png" srcset="/img/loading.gif" alt="ldheatmap_new"></p>
<h5 id="使用外部资源的额外数据查询Additional-data-interrogation-using-external-resources"><a href="#使用外部资源的额外数据查询Additional-data-interrogation-using-external-resources" class="headerlink" title="使用外部资源的额外数据查询Additional data interrogation using external resources"></a>使用外部资源的额外数据查询Additional data interrogation using external resources</h5><p>如果能提供研究结果的背景，报告SNP级别的关联分析结果会更有意义。例如，研究者可能想知道一个有统计学意义的SNP是否在一个蛋白质编码基因内、基因间、或接近与被调查疾病相关的特定组织或细胞类型中的调控元素（如甲基化标记）。</p>
<p>可能相关的外部数据类型见下表。</p>
<p>这些数据分为八大类，大致按照代表从DNA信息到调节到表达的过程的顺序。</p>
<ol>
<li>SNP；</li>
<li>基因元件；</li>
<li>染色质状态；</li>
<li>表观遗传标记；</li>
<li>转录因子结合；</li>
<li>RNA表达；</li>
<li>SNP-mRNA关联；</li>
<li>其他组学数据。</li>
</ol>
<p>我们强调，这个表格并不全面，而是提供了一个可利用的大量外部数据资源的一瞥。与列出的每个类别相关的数据都有广泛的来源。UCSC基因组浏览器提供了一套精心设计的综合生物信息学工具和数据库，包括许多来自补充信息C，表1中所列的资源，可以进一步审视GWA的结果。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># -step10d-</span><br><span class="hljs-comment"># Create regional association plot</span><br><span class="hljs-comment"># Create data.frame of most significant SNP only</span><br>install.packages(<span class="hljs-string">&quot;coin&quot;</span>)<br>install.packages(<span class="hljs-string">&quot;~/postgwas_1.11.tar.gz&quot;</span>, repos = <span class="hljs-literal">NULL</span>, type = <span class="hljs-string">&quot;source&quot;</span>)<br>library(postgwas)<br><br>snps&lt;-data.frame(SNP=<span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;rs1532625&quot;</span>))<br><br><span class="hljs-comment"># Change column names necessary to run regionalplot function</span><br>GWAScomb &lt;- rename(GWAScomb, <span class="hljs-built_in">c</span>(p.value=<span class="hljs-string">&quot;P&quot;</span>, chr=<span class="hljs-string">&quot;CHR&quot;</span>, position=<span class="hljs-string">&quot;BP&quot;</span>))<br><br><br><span class="hljs-comment"># Edit biomartConfigs so regionalplot function</span><br><span class="hljs-comment"># pulls from human genome build 37/hg19</span><br><br>myconfig &lt;- biomartConfigs$hsapiens<br>myconfig$hsapiens$gene$host &lt;- <span class="hljs-string">&quot;grch37.ensembl.org&quot;</span><br>myconfig$hsapiens$gene$mart &lt;- <span class="hljs-string">&quot;ENSEMBL_MART_ENSEMBL&quot;</span><br>myconfig$hsapiens$snp$host &lt;- <span class="hljs-string">&quot;grch37.ensembl.org&quot;</span><br>myconfig$hsapiens$snp$mart &lt;- <span class="hljs-string">&quot;ENSEMBL_MART_SNP&quot;</span><br><br><br><span class="hljs-comment"># Run regionalplot using HAPMAP data (pop = CEU)</span><br>regionalplot(snps, GWAScomb, biomart.config = myconfig, window.size = <span class="hljs-number">400000</span>, draw.snpname = data.frame(<br>  snps = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;rs1532625&quot;</span>, <span class="hljs-string">&quot;rs247617&quot;</span>), <br>  text = <span class="hljs-built_in">c</span>(<span class="hljs-string">&quot;rs1532625&quot;</span>, <span class="hljs-string">&quot;rs247617&quot;</span>),<br>  angle = <span class="hljs-built_in">c</span>(<span class="hljs-number">20</span>, <span class="hljs-number">160</span>),<br>  <span class="hljs-built_in">length</span> = <span class="hljs-built_in">c</span>(<span class="hljs-number">1</span>, <span class="hljs-number">1</span>), <br>  cex = <span class="hljs-built_in">c</span>(<span class="hljs-number">0.8</span>)<br>),<br>ld.options = <span class="hljs-built_in">list</span>(<br>  gts.source = <span class="hljs-number">2</span>, <br>  max.snps.per.window = <span class="hljs-number">2000</span>, <br>  rsquare.min = <span class="hljs-number">0.8</span>, <br>  show.rsquare.text = <span class="hljs-literal">FALSE</span><br>),<br>out.format = <span class="hljs-built_in">list</span>(file = <span class="hljs-literal">NULL</span>, panels.per.page = <span class="hljs-number">2</span>))<br><span class="hljs-comment">## Reading GWAS datasets ...</span><br><span class="hljs-comment">## PLINK format detected</span><br><span class="hljs-comment">## Assembling regions...</span><br><span class="hljs-comment">## Initializing biomart connection (SNPs)...</span><br><span class="hljs-comment">## Error in bmRequest(request = request, verbose = verbose) : </span><br><span class="hljs-comment">##   Not Found (HTTP 404).</span><br><span class="hljs-comment"># 尚未解决</span><br></code></pre></td></tr></table></figure>

            </div>
            <hr>
            <div>
              <div class="post-metas mb-3">
                
                  <div class="post-meta mr-3">
                    <i class="iconfont icon-category"></i>
                    
                      <a class="hover-with-bg" href="/categories/BIOINFO-EXPERIMENT-RECORDS/">BIOINFO EXPERIMENT RECORDS</a>
                    
                  </div>
                
                
                  <div class="post-meta">
                    <i class="iconfont icon-tags"></i>
                    
                      <a class="hover-with-bg" href="/tags/R/">R</a>
                    
                      <a class="hover-with-bg" href="/tags/GWAS/">GWAS</a>
                    
                  </div>
                
              </div>
              
                <p class="note note-warning">
                  
                    本博客所有文章除特别声明外，均采用 <a target="_blank" href="https://creativecommons.org/licenses/by-sa/4.0/deed.zh" rel="nofollow noopener noopener">CC BY-SA 4.0 协议</a> ，转载请注明出处！
                  
                </p>
              
              
                <div class="post-prevnext">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/2021/09/18/20210918MODERNAmRNA%E7%96%AB%E8%8B%97%E9%A1%B9%E7%9B%AE%E6%A6%82%E8%BF%B0/">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">MODERNA mRNA 疫苗项目概述</span>
                        <span class="visible-mobile">上一篇</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2021/09/16/hello-world/">
                        <span class="hidden-mobile">Hello World</span>
                        <span class="visible-mobile">下一篇</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
              <!-- Comments -->
              <article class="comments" id="comments">
                
                  
                
                
  <div class="disqus" style="width:100%">
    <div id="disqus_thread"></div>
    <script>
      Fluid.utils.waitElementVisible('disqus_thread', function() {
        Fluid.utils.createCssLink('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqusjs.css');
        Fluid.utils.createScript('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqus.js', function () {
          new DisqusJS({
            shortname: 'fluid',
            apikey: ''
          });
        });
      });
    </script>
    <noscript>Please enable JavaScript to view the
      <a target="_blank" href="https://disqus.com/?ref_noscript" rel="nofollow noopener noopener">comments powered by Disqus.</a>
    </noscript>
  </div>


              </article>
            
          </article>
        </div>
      </div>
    </div>
    
      <div class="d-none d-lg-block col-lg-2 toc-container" id="toc-ctn">
        <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;目录</p>
  <div class="toc-body" id="toc-body"></div>
</div>

      </div>
    
  </div>
</div>

<!-- Custom -->


    

    
      <a id="scroll-top-button" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">搜索</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v"
                 for="local-search-input">关键词</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>
    

    
  </main>

  <footer class="text-center mt-5 py-3">
  <div class="footer-content">
     <a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a> <i class="iconfont icon-love"></i> <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener"><span>Fluid</span></a> 
  </div>
  

  

  
</footer>

<!-- SCRIPTS -->

  <script  src="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" ></script>
<script  src="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/js/bootstrap.min.js" ></script>
<script  src="/js/debouncer.js" ></script>
<script  src="/js/events.js" ></script>
<script  src="/js/plugins.js" ></script>

<!-- Plugins -->


  
    <script  src="/js/lazyload.js" ></script>
  



  



  <script  src="https://cdn.jsdelivr.net/npm/tocbot@4.12.0/dist/tocbot.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/anchor-js@4.3.0/anchor.min.js" ></script>



  <script defer src="https://cdn.jsdelivr.net/npm/clipboard@2.0.6/dist/clipboard.min.js" ></script>






  <script  src="https://cdn.jsdelivr.net/npm/typed.js@2.0.11/lib/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var title = document.getElementById('subtitle').title;
      
      typing(title)
      
    })(window, document);
  </script>



  <script  src="/js/local-search.js" ></script>
  <script>
    (function () {
      var path = "/local-search.xml";
      $('#local-search-input').on('click', function() {
        searchFunc(path, 'local-search-input', 'local-search-result');
      });
      $('#modalSearch').on('shown.bs.modal', function() {
        $('#local-search-input').focus();
      });
    })()
  </script>















<!-- 主题的启动项 保持在最底部 -->
<script  src="/js/boot.js" ></script>



</body>
</html>
