

<!DOCTYPE html>
<html lang="zh-CN" data-default-color-scheme=&#34;auto&#34;>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/favicon.png">
  <link rel="icon" href="/img/favicon.png">
  <meta name="viewport"
        content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  
    <meta name="description" content="">
  
  <meta name="author" content="John Doe">
  <meta name="keywords" content="">
  
  <title>2021.1.20 human RME - Hexo</title>

  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" />


  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/github-markdown-css@4.0.0/github-markdown.min.css" />
  <link  rel="stylesheet" href="/lib/hint/hint.min.css" />

  
    
    
      
      <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@10.4.0/styles/github-gist.min.css" />
    
  

  
    <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.css" />
  



<!-- 主题依赖的图标库，不要自行修改 -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_ba1fz6golrf.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_kmeydafke9r.css">


<link  rel="stylesheet" href="/css/main.css" />

<!-- 自定义样式保持在最底部 -->


  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    var CONFIG = {"hostname":"example.com","root":"/","version":"1.8.8","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"right","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"copy_btn":true,"image_zoom":{"enable":true},"toc":{"enable":true,"headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"onlypost":false},"web_analytics":{"enable":false,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":null,"app_key":null,"server_url":null}}};
  </script>
  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
<meta name="generator" content="Hexo 5.4.0"></head>


<body>
  <header style="height: 70vh;">
    <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand"
       href="/">&nbsp;<strong>Liu Keqin</strong>&nbsp;</a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/">
                <i class="iconfont icon-home-fill"></i>
                首页
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/">
                <i class="iconfont icon-archive-fill"></i>
                归档
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/">
                <i class="iconfont icon-category-fill"></i>
                分类
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/">
                <i class="iconfont icon-tags-fill"></i>
                标签
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/">
                <i class="iconfont icon-user-fill"></i>
                关于
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" data-toggle="modal" data-target="#modalSearch">&nbsp;<i
                class="iconfont icon-search"></i>&nbsp;</a>
          </li>
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" href="javascript:">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

    <div class="banner" id="banner" parallax=true
         style="background: url('/img/default.png') no-repeat center center;
           background-size: cover;">
      <div class="full-bg-img">
        <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
          <div class="page-header text-center fade-in-up">
            <span class="h2" id="subtitle" title="2021.1.20 human RME">
              
            </span>

            
              <div class="mt-3">
  
  
    <span class="post-meta">
      <i class="iconfont icon-date-fill" aria-hidden="true"></i>
      <time datetime="2021-01-20 17:18" pubdate>
        2021年1月20日 下午
      </time>
    </span>
  
</div>

<div class="mt-1">
  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-chart"></i>
      22.9k 字
    </span>
  

  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-clock-fill"></i>
      
      
      397
       分钟
    </span>
  

  
  
</div>

            
          </div>

          
        </div>
      </div>
    </div>
  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="d-none d-lg-block col-lg-2"></div>
    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div class="py-5" id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">2021.1.20 human RME</h1>
            
            <div class="markdown-body">
              <h1 id="How-to-analyze-human-RME"><a href="#How-to-analyze-human-RME" class="headerlink" title="How to analyze human RME"></a>How to analyze human RME</h1><p><strong>pdf版获取</strong><br><br></p>


	<div class="row">
    <embed src="https://jinsyliu.github.io/pdf/20210119humanRME.pdf" width="100%" height="550" type="application/pdf">
	</div>



<br>

<h2 id="1-Background"><a href="#1-Background" class="headerlink" title="1. Background"></a>1. Background</h2><p>区分印记基因和单等位基因</p>
<h3 id="1-1-RME"><a href="#1-1-RME" class="headerlink" title="1.1 RME"></a>1.1 RME</h3><p>在二倍体生物体中，一般认为每个基因的两个等位基因都会在相似的时间和水平上表达。然而，一些基因可以优先表达或严格从一个单一的等位基因，这一过程被称为单等位基因表达。单等位基因表达可能是由于等位基因之间的 DNA 序列多态性，例如在增强子或启动子序列中的多态性可能影响基因转录的效率，或基因组大部分的拷贝数变异(copy-number variations，CNVs)。在没有DNA序列多态性的情况下，也会出现单等位基因表达，最为典型的例子是雌性哺乳动物的X染色体失活（XCI）过程，它保证了XX雌性和XY雄性之间的X连锁基因剂量补偿\citep{wutz2011gene}。另一个典型的单等位基因调控的程序化例子则是印记基因。</p>
<p>印记基因是一种亲本起源特异性的单等位基因表达。印迹基因的单等位基因表达受生殖系等位基因特异性甲基化(allele-specific methylation，ASM，传统上称为差异甲基化区域(differentially methylated region，DMR))控制\citep{tycko2010allele, lo2003allelic}。在这些生殖系遗传系(gASMs)中 ，一个亲本等位基因的印迹控制区的 CpGs 是甲基化的，而另一个等位基因的 CpGs 是非甲基化的。当用亚硫酸氢盐Sanger测序检测时，这些区域的 PCR 克隆有一半是高甲基化的，另一半是低甲基化的，因此呈现出双峰式甲基化模式。重要的是，这些 gASMs/DMR 在胚胎发育早期的整体去甲基化和再甲基化周期中被认为是稳定的\citep{bartolomei25mammalian}。一旦由于精子和/或卵母细胞的暴露，这些遗传性组织被改变，这种改变可以作为“表观遗传记忆”进入体细胞。这种表观遗传改变的水平或许能在个体中维持数十年。此前，在一项针对跨代表观遗传的研究中，研究者们发现，出生前即暴露于荷兰饥荒的个体在出生60年后，印记的 IGF2基因的 DNA 甲基化水平仍然低于未暴露的同性兄弟姐妹\citep{heijmans2008persistent}。目前发现，大约超过100个基因座受父母起源依赖性表达的影响。许多印迹基因在出生前和出生后的新陈代谢中都起着重要作用，它们的双等位基因表达可以导致严重的表型\citep{barlow2011genomic}。不过，在某些组织比如大脑中，处于胎儿出生后神经发育的需要，神经干细胞会发生印记基因的沉默释放（release from silencing）\citep{ferron2011postnatal}。</p>
<p>除了亲本起源特异性单等位基因表达（parent-of-origin-specific monoallelic expression），哺乳动物基因组有大量的基因显示随机单等位基因表达(random monoallelic expression，RME)\citep{lo2003allelic, deng2014single, gendrel2014developmental}。旧观点认为，常染色体上的非印迹基因要么双细胞表达，要么双等位基因抑制。单等位基因表达被认为只发生在位于印记位点或女性 X 染色体的基因（其中一个来自父系或母系的X染色体在女性中被随机灭活）\citep{tycko2010allele}。但是，最近的研究发现，非印迹的常染色体基因的单等位基因的表达似乎不是一个偶发现象，而是在人类和小鼠基因组中的一个保守特征\citep{lo2003allelic, deng2014single, gendrel2014developmental}。虽然RME现象似乎经常发生，但表观遗传学机制好像无法对其发生进行解释\citep{gendrel2014developmental}。</p>
<p>这些随机单等位基因表达一般发生在常染色体基因座上\citep{chess2012mechanisms}，包括大基因家族的重要成员。这些随机单等位基因以高度组织特异性的方式表达，一般参与感觉或免疫系统功能(如人类白细胞抗原HLA)，如淋巴细胞中的免疫球蛋白（immunoglobulin）和T细胞受体基因（T cell receptor genes）、嗅觉神经元中的气味剂受体（odorant receptor ，OR）基因、浦肯野神经元（Purkinje neurons）中的原珠蛋白（protocadherins，Pcdhs）等\citep{chess2012mechanisms}。这种单基因和单配位表达（monogenic and monoallelic expression）的现象（也称为等位基因排异（allelic exclusion））被认为对明确细胞身份和保证细胞多样性至关重要。事实上，研究者认为，等位基因排斥保证了每个 b 细胞或 t 细胞只产生一个单一的抗原受体，这种方式有助于免疫多样性\citep{cedar2008choreography}。等位基因排斥似乎代表了一种普遍的表观遗传现象。重要的是，单等位基因的表达在细胞谱系和个体之间存在差异。因此，RME 可能产生多样性的基因表达模式，对细胞的命运和生理有重要影响\citep{ohlsson2007widespread}。</p>
<p>单等位基因表达状态可以稳定地保持在几代细胞中。</p>
<p>单等位表达基因在发育过程中起着重要作用。此前的研究发现，许多常染色体基因的单等位基因表达发生在体内，在正常发育过程中，并趋向于高度组织特异性\citep{gendrel2014developmental}。</p>
<p>RME导致的表观遗传改变与许多人类疾病的发生密切相关。例如，表观遗传学研究表明，癌细胞具有全基因组低甲基化和区域特异性低甲基化的特点，而等位基因失衡(allelic imbalance，AI)与癌症发生风险上升相关。乳腺癌和卵巢癌散发病例中，由于启动子过度甲基化导致的 BRCA1表达缺失与两种癌症的发病相关\citep{esteller2000promoter, thrall2006brca1}。先前的一项研究报道，13个人类基因中有6个，包括 BRCA1和 p53，在两个等位基因中表达有显著差异，并且这种差异是通过孟德尔定律遗传的\citep{yan2002allelic}。</p>
<p>此外，人类常染色体显性遗传性疾病涉及单等位基因表达。</p>
<h2 id="How-to-analyze-ASE-in-RNA-seq"><a href="#How-to-analyze-ASE-in-RNA-seq" class="headerlink" title="How to analyze ASE in RNA-seq?"></a>How to analyze ASE in RNA-seq?</h2><p>\citep{borel2015biased}</p>
<blockquote>
<p>To investigate the extent of allele-specific transcription of autosomal human protein-coding genes, we used single-cell RNA sequencing (RNA-seq) technology to study 203 single cells from two different human primary fibroblast cell lines. </p>
<p>By analyzing informative single-nucleotide variants (SNVs), we determined the relative mRNA abundance of each of the two alleles. For most of the actively transcribed genes, our results revealed that one allele was predominantly detected in a single cell at a particular point in time, whereas the second allele was at low levels or undetectable. </p>
<p>We observed a stochastic process given that equal numbers of single cells expressed one or the other allele and a minority of single cells expressed both alleles. Interestingly, we detected only a few genes with an equal mRNA level from both alleles in all single cells. Detailed genomic characterization of these “single-cell biallelic” genes revealed that they express high levels of mRNA in a large number of cells. </p>
</blockquote>
<p>method</p>
<blockquote>
<ol>
<li><p>Single-Cell Capture</p>
</li>
<li><p>cDNA Synthesis and Pre-amplification of Single Cells</p>
</li>
<li><p>Total RNA Extraction from Bulk Cell Samples</p>
</li>
<li><p>mRNA-Seq Library Preparation</p>
</li>
<li><p>Whole-Genome Sequencing</p>
</li>
<li><p>Spike-In Experiment</p>
<p> 由于不同文库测序深度不同，比较前当然要进行均一化！用总reads进行均一化可能最简单，其基于以下两个基本假设：</p>
<ul>
<li>绝大多数的gene表达量不变；</li>
<li>高表达量的gene表达量不发生改变；</li>
</ul>
<p> 但在转录组中，通常一小部分极高丰度基因往往会贡献很多reads，如果这些“位高权重”的基因还是差异表达的，则会影响所有其它基因分配到的reads数，而且，两个样本总mRNA量完全相同的前提假设也过于理想了。那如何比较呢，各个方家使出浑身解数，有用中位数的，有用75分位数的，有用几何平均数的，有用TMM(trimmed mean of Mvalues)的等等，总之要<strong>找一个更稳定的参考</strong>值。</p>
<p> <em>House-keeping gene(s)</em></p>
<p> 矫正的思路很简单，就是在变化的样本中寻找不变的量</p>
<p> 那么在不同RNA-seq样本中，那些是不变的量呢？一个很容易想到的就是<strong>管家基因</strong> (House-keeping gene(s))</p>
<p> 使用Housekeeping gene的办法来进行相对定量，这种办法在一定程度上能够解决我们遇到的问题。但其实这种办法有一个<strong>非常强的先验假设</strong>：housekeeping gene的表达量不怎么发生变化。其实housekeeping gene list有几千个，这几千个基因有一定程度上的变化是有可能的。</p>
<p> An RNA spike-in is an RNA transcript of known sequence and quantity used to calibrate measurements in RNA hybridization assays, such as DNA microarray experiments, RT-qPCR, and RNA-Seq.</p>
<p> A spike-in is designed to bind to a DNA molecule with a matching sequence, known as a control probe. This process of specific binding is called hybridization. A known quantity of RNA spike-in is mixed with the experiment sample during preparation. The degree of hybridization between the spike-ins and the control probes is used to normalize the hybridization measurements of the sample RNA.</p>
<p> 在RNA-Seq建库的过程中掺入一些预先知道序列信息以及序列绝对数量的内参。这样在进行RNA-Seq测序的时候就可以通过不同样本之间内参（spike-in）的量来做一条标准曲线，就可以非常准确地对不同样本之间的表达量进行矫正。在这种操作下，可以一定程度上认为是一种绝对定量。（类似于housekeeping gene）</p>
<p> 举例说明：</p>
<p> 通过在样品制备过程中，混入指定数量的spike-in，我们就可以知道不同样本中的基因绝对比表达值。如等细胞数的样本A和样本B，在每个样本中，我加入了等量的spike-in。最后分析发现，spike-in占样本A的1%，但是占样本B的5%。这表明样本A的RNA表达量也许普遍比样本B的表达量高五倍左右。</p>
<p> <strong>ERCC control RNA</strong></p>
<p> ERCC = External RNA Controls Consortium</p>
<p> ERCC就是一个专门为了定制一套spike-in RNA而成立的组织，这个组织早在2003年的时候就已经宣告成立。主要的工作就是设计了一套非常好用的spike-in RNA，方便microarray，以及RNA-Seq进行内参定量。</p>
<p> 在RNA-Seq中增加ERCC的绝对量是可以获得FPKM的绝对量的增加，并且两者成非常好的线性关系。这也是我们能够对RNA-Seq样本进行掺入内参的一个基本前提。</p>
<p> R中用来处理带有ERCC spike-in的RNA-Seq数据的包：RUVSeq \citep{risso2014normalization}</p>
</li>
<li><p>RPSM Calculation</p>
<p> RPSM stands for reads at a single-nucleotide position per sequencing read length (in kb) and per million mapped reads. The formula for RPSM is (10^6 × A) / (B × C), where A is the number of mappable reads at a nucleotide position, B is the total number of mappable reads of the sample, and C is the sequencing read length (in kb; C = 0.199).</p>
</li>
<li><p>Read Mapping for RNA-Seq Samples</p>
</li>
<li><p>ASE analysis</p>
<p> ASE analysis was performed as in \citep{lappalainen2013transcriptome} In brief, we considered heterozygous sites obtained from whole-genome sequencing with DNA reads supporting both alleles. We used a minimum site quality call of 200. We excluded sites susceptible to allelic mapping bias, namely (1) sites with 50 bp mappability &lt; 1 according to the UCSC mappability track (implying that the 50 bp flanking region of the site is not unique in the genome), and (2) sites where overlapping simulated RNA-seq reads showed a &gt;5% mapping difference between those that carried the reference allele and those that carried the non-reference allele (see the methods in \citep{lappalainen2013transcriptome}).</p>
<p> In all analyses, we only used uniquely mapped RNA-seq reads (GEM mapping quality &gt; 150) and sites with base quality &gt; 20 and support from at least 16 reads. Using information from SAMtools (v.0.1.19) mpileup\citep{li2009sequence}, we obtained for each site and each sample the number of reads mapping in the reference, the number of alternative alleles, and the sum of both. Each site was then annotated with the overlapped genomic feature in GENCODE annotation v.15 or the novel exons from the de novo assemblies of each sample. For each site, the number of single cells (and non-single cells) where the site was assessed was also counted. The distribution of allelic ratios for all samples is reported in Figure S9.</p>
<p> 等位基因特异性表达不平衡（Allelic Expression Imbalance，AEI）可以作为表型用于寻找功能顺式作用多态性，同时可以作为转录区的分子标记来观察同一个体中不同等位基因的特异性表达差异。在杂合子个体中，分别来自父亲与母亲的等位基因，它们在同一细胞中，处于相同的外部环境中，在没有顺式作用多态性（或特定基因表观遗传修饰）的情况下，他们的表达量应该是一样的。与此相反，个体的顺式作用多态性会影响基因的表达及mRNA的加工，导致等位基因有不同的mRNA表达量水平，即AEI(等位基因表达不平衡)。它可以作为一个综合所有顺式作用因子的定量测量。另外，当目的SNP位点位于转录区时，可以直接在总RNA反转录cDNA中观察到SNP 不同等位基因的特异性表达差异，也就是用这个SNP作为Marker对两条不同allele分别进行表达定量。从而观察在同一杂合子个体中两种不同的等位基因型是否对表达水平造成了影响，也就是说排除了个体差异和环境影响之后是否存在由该位点不同基因型所造成的表达差异。该实验需要此SNP位点分型为杂合子的个体的基因组DNA和目的组织细胞抽提得到的总RNA（或者已经反转录好的cDNA）。实验中我们将用基因组DNA的两条allele作为1:1的校正内参，观察目的组织细胞中的RNA中两条allele的比例是否偏离1:1，最终判断是否存在AEI现象，进一步确定该位点与表达水平的关系。</p>
<p> （需要检测的对象：杂合子的个体的基因组DNA和目的组织细胞抽提得到的总RNA）</p>
</li>
<li><p>Gene Quantification and De Novo Assembly</p>
<p>We used the software Cufflinks (v.2.1.1)\citep{trapnell2012differential, trapnell2010transcript} with default parameters and GENCODE v.12 as a reference annotation\citep{harrow2012gencode}. On the basis of Cufflinks transcript (170,086) quantifications, we selected for further analysis single cells that passed the arbitrary threshold of 12,000 transcripts expressed at FPKM (fragments per kilobase of exon per million reads mapped) &gt; 0.3. We retained 163 UCF1014 single-cell samples expressing an average of 15,807 transcripts (the remaining samples expressed an average of 4,998 transcripts). Additionally, for each sample we performed de novo assembly to identify novel transcripts (Figure S4) without using the reference annotation. We then used the program cuffcompare to compare the assembled transcripts with the GENCODE reference annotation (v.15). Finally, for the four bulk RNA samples, we merged the four assemblies into a merged bulk RNA assembly. We compared each single-cell de novo assembly against the merged bulk RNA assembly to identify novel single-cell-specific transcripts. The program intersectBed from bedtools28 was used for this last comparison.</p>
</li>
</ol>
</blockquote>
<p>softwares（主要针对ASE分析）</p>
<blockquote>
<ol>
<li><p>SAMtools (v.0.1.19)</p>
</li>
<li><p>GENCODE annotation v.15</p>
</li>
<li><p>GENCODE v.12</p>
</li>
<li><p>Cufflinks(v. 2.1.1)</p>
</li>
<li><p>cuffcompare</p>
</li>
<li><p>GENCODE reference annotation (v.15).</p>
</li>
<li><p>intersectBed （from bedtools）</p>
</li>
<li><p>gemtools v. 1.6.2</p>
</li>
<li><p>RUVSeq(R) (处理spike-in内参) \citep{borel2015biased}</p>
</li>
<li><p>GATK ASEReadCounter \citep{mckenna2010genome}</p>
<p>如何利用SNP信息构建伪亲代基因组，然后把子代的测序信息mapping上去(用STAR）？</p>
<p>输入：BAM files (with proper headers) to be analyzed for ASE；A VCF file with specific sites to process</p>
<p>输出：A table of allele counts at the given sites. By default, it is formatted as a tab-delimited text file that is readable by R and compatible with <a target="_blank" rel="noopener" href="http://www.well.ox.ac.uk/~rivas/mamba/">Mamba</a>, a downstream tool developed for allele-specific expression analysis.</p>
</li>
<li><p>Bedtools</p>
</li>
<li><p>STAR</p>
</li>
<li><p>Mamba</p>
<p>a tool for further analysis data by GATK</p>
<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs awk">git clone https:<span class="hljs-regexp">//gi</span>t.code.sf.net<span class="hljs-regexp">/p/m</span>ambas/mambas mambas-mambas <span class="hljs-comment">#install</span><br></code></pre></td></tr></table></figure></li>
<li><p>MBASED(利用RNA-seq数据进行ASE检测，将多个单核苷酸变异位点的信息聚合在一起，以获得ASE的基因水平测量，即使在事先没有相位信息的情况下也可以进行) \citep{mayba2014mbased}</p>
</li>
<li><p>GeneiASE(仅使用 RNA-seq 数据，不需要已知或估计的单倍型，并且可以作为可下载的软件包使用。问题是，不能确定真正的单倍型，不能确定基因究竟是来自于父系等位基因还是母系等位基因的表达。) \citep{edsgard2016geneiase}</p>
</li>
<li><p>QuASAR(R包，当基因型数据无法获取时，可以用此方法。 i)从下一代测序读取基因分型，ii)对杂合子位点的等位基因不平衡进行推断。测序数据可以是 RNA-seq、 DNase-seq、 ATAC-seq 或任何其他类型的高通量测序数据。) \citep{harvey2015quasar}</p>
</li>
<li><p>ASEP（只用RNA-seq数据检测ASE）\citep{fan2020asep}</p>
</li>
<li><p>SCALE（需要提前确定来自父母系的等位基因，R中的输入数据是矩阵格式，分为母系基因的read count matrix和父系基因的read count matrix，输出是某个基因的父系与母系等位基因表达量/频率）\citep{borel2015biased}</p>
</li>
<li><p>phASER \citep{castel2016rare}</p>
<p>输入：VCF文件（VCF，或 Variant Call Format，它是一种标准化的文本文件格式，用于表示 SNP、 indel 和结构变化调用。VCF 规范过去由千人基因组计划卫生组织维护，但其管理和进一步开发已被基因组学与健康全球联盟的基因组数据工具包团队接管。）tabix index for the VCF，BAM format file containing RNA-seq reads，index for the BAM file。</p>
</li>
</ol>
</blockquote>
<p>eQTL analysis</p>
<blockquote>
<p>Expression quantitative trait loci (eQTL) 基因表达数量性状基因座</p>
<p>eQTL 研究成本高昂，需要大样本量和对每个样本进行全基因组分型</p>
<p>To date, most eQTL studies have considered the effects of genetic variation on expression within a single tissue (typically blood). </p>
<p>分析软件/程序：</p>
<p>MT-eQTL</p>
</blockquote>
<p>ASE analysis</p>
<blockquote>
<p>analysis of allele-specific expression (ASE) 分析等位基因特异性表达</p>
<p>通常是通过计算与杂合位点上每个等位基因相匹配的 RNA-seq 读数和检验1:1等位基因比率的零假设来实现的</p>
<p>原则上，当基因型信息不容易获得时，可以从 RNA-seq 读数直接推断。然而，在考虑基因型调用的不确定性的情况下，目前还没有联合推断基因型和进行 ASE 推断的方法。</p>
</blockquote>
<p>利用SNP来定位不同来源的等位基因 \citep{xie2019modeling}    BLMRM方法（R包）</p>
<blockquote>
<p>公牛的 DNA 经过了下一代测序(DNA-seq) ，以确定他的基因组和母牛的参考基因组之间的所有 SNPs。</p>
<p>然后应用基因组分析工具包(Genome Analysis Toolkit，GATK)\citep{mckenna2010genome}和 SAMtools \citep{li2011statistical}来调用 SNP，只使用两个管道所识别的 SNPs 来生成伪 基因组。</p>
<p>最后，利用 HISAT2 \citep{kim2015hisat}和 BWA \citep{li2009fast}将来自母牛 × 公牛 F1代的 RNA-seq 片段定位到二倍体基因组，并保留两种方法鉴定的变异体，以减少假阳性的可能影响。</p>
</blockquote>
<p>利用亲本特异性 SNPs 分析等位基因特异表达基因。\citep{ahn2019analysis}</p>
<blockquote>
<p>Method: 进行了父母基因组和后代转录组测序使用下一代测序。</p>
<p>随后，利用单核苷酸多态性(single nucleotide polymorphism，SNPs)的个体基因组定位和用于正反杂交的同一品种亲本的联合基因组定位，采用两种不同的方法对 snp 进行了基因组尺度识别。</p>
<p>利用亲本特异性 SNPs 分析等位基因特异表达基因。</p>
<p>Result: 由于测序结果的基因组覆盖率较低(约4 ×) ，大多数 SNPs 对于亲本谱系鉴定后代表达的等位基因没有信息价值，因此被排除在我们的分析之外。因此，包含336个基因的436个单核苷酸多态性可用于检测父系等位基因在后代中的不平衡表达。通过计算双亲等位基因在后代中的阅读比例，我们鉴定了7个表现等位基因偏向表达的基因(p &lt; 0.05) ，其中包括以前报道的3个基因和本研究中新发现的4个基因。</p>
<p>Read mapping: 下载该物种的参考基因组。使用 BWA MEM 包(版本0.7.17-r1188) \citep{li2009fast}将序列读取与引用对齐。SAM 文件使用 samtools 索引转换为 BAM 格式，并使用 samtools sort \citep{li2009sequence}进行排序。RNA-seq 读取到参考基因组的映射是使用带有默认选项和2-pass 模式的 STAR 包(版本2.5.3 a)进行的。</p>
<p>Variant calling and filtration： 使用 Picard 工具中的 markduplices (2.15.0版， <a target="_blank" rel="noopener" href="https://broadinstitute.github.io/Picard/)%E5%88%A0%E9%99%A4%E4%BA%86%E6%98%A0%E5%B0%84%E8%AF%BB%E5%8F%96%E7%9A%84%E5%89%AF%E6%9C%AC%E3%80%82%E5%AF%B9%E4%BA%8E">https://broadinstitute.github.io/Picard/)删除了映射读取的副本。对于</a> RNA-seq 读操作，read 组使用 GATK AddOrReplaceReadGroups 添加到映射读操作中，而使用 GATK Split NCigarReads \citep{poplin2017scaling}删除内含区域中映射的溢出读操作。基本质量的阅读是重新校准使用 GATK BaseRecalibrator (版本3.8)的全基因组测序和 RNA-seq 的结果，质量调整的阅读是使用 GATK PrintReads。父母基因组和后代转录组的变异体分别用 GATK 和 HaplotypeCaller 命名，而所谓的变异体用 GATK GenotypeGVCFs \citep{poplin2017scaling}进行联合基因分型。基因型变体用 NCBI dbSNP 150 \citep{sherry2001dbsnp}和 ENSEMBL 注释(发行版92)注释，使用 GATK VariantAnnotator 和 snpEff \citep{cingolani2012program}。随后，使用 GATK variantfilling 和 selectvariant 去除强链偏倚变异(Fisher strand &gt; 30)、低质量深度变异(&lt; 2)和35bp 窗口内3个或3个以上单核苷酸多单核苷酸多态性(SNP)变异体。我们还过滤了低深度变异(个体映射读深度[ DP ] &lt; 3，同一品种联合映射读深度 &lt; 6)。最后，利用 gatkselectvariant 和 snpSift \citep{ruden2012using}筛选出外显 snp。</p>
<p>Workflow:</p>
<p><img src="/Users/keqinliu/studying_document/bioinfo/human_RME/images/workflow.jpg" srcset="/img/loading.gif" alt="workflow"></p>
</blockquote>
<blockquote>
<p>母体读数的比例计算为1 - 父体读数的比例。在该研究中，采用的任意判断等位基因表达偏倚的标准对于任何给定的等位基因都是 &lt; 0.3或 &gt; 0.7。当比值在0.3和0.7之间时，则认为它是双等位基因表达。</p>
</blockquote>
<p>利用千人基因组计划鉴别等位基因表达\citep{jadhav2019rna}</p>
<blockquote>
<p>数据来源：65 trios from the HapMap / 1000 genomes projects with RNA-Seq data from lymphoblastoid cell lines (LCLs)，and 131 trios from the Genome-of-the-Netherlands （查资料发现好像是血液样本的WGS测序？）</p>
<p>目前思路：</p>
<p>先选用4个CHS家庭（一共8位父母的基因组测序数据low coverage，4个孩子的RNA-seq数据，其中两个男孩，两个女孩）试着分析</p>
</blockquote>
<p>family trio 1(SH007)</p>
<p>**Collection UUID:**qr1hi-4zz18-hxa1at658d85c1r（用于在run program时使用）</p>
<p>*Content address:*d41d8cd98f00b204e9800998ecf8427e+0</p>
<table>
<thead>
<tr>
<th>HG00421(father)  <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRR1606795">https://www.ebi.ac.uk/ena/browser/view/SRR1606795</a> (downloaded) (fastq-dump ing) 59520 (bam downloaded)</th>
</tr>
</thead>
<tbody><tr>
<td>HG00422(mother) <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRR1606537">https://www.ebi.ac.uk/ena/browser/view/SRR1606537</a> (downloaded)(fastq-dump ing) 59419 (bam downloaded)</td>
</tr>
<tr>
<td>HG00423(female child, Lymphoblastoid cell total RNA)  <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRX2432923">https://www.ebi.ac.uk/ena/browser/view/SRX2432923</a> （downloaded）(bam)</td>
</tr>
</tbody></table>
<p>Family trio 2(SH002)</p>
<table>
<thead>
<tr>
<th>HG00406(father) <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRR1602073">https://www.ebi.ac.uk/ena/browser/view/SRR1602073</a> (fastq-dump ing) 58908 (bam downloaded)</th>
</tr>
</thead>
<tbody><tr>
<td>HG00407(mother) <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRR1602075">https://www.ebi.ac.uk/ena/browser/view/SRR1602075</a>  (downloading) 42363 (bam downloaded)</td>
</tr>
<tr>
<td>HG00408(female child, Lymphoblastoid cell total RNA) <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRX2432921">https://www.ebi.ac.uk/ena/browser/view/SRX2432921</a> (bam)</td>
</tr>
</tbody></table>
<p>Family trio3(SH014)</p>
<table>
<thead>
<tr>
<th>HG00442(father) <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRR1607190">https://www.ebi.ac.uk/ena/browser/view/SRR1607190</a> (fastq-dump ing)18181  (downloaded)</th>
</tr>
</thead>
<tbody><tr>
<td>HG00443(mother) <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRR1606504">https://www.ebi.ac.uk/ena/browser/view/SRR1606504</a> (fastq-dump ing)65250 (downloading bam) 11075</td>
</tr>
<tr>
<td>HG00444(male child, Lymphoblastoid cell total RNA) <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRX2432926">https://www.ebi.ac.uk/ena/browser/view/SRX2432926</a> (bam)</td>
</tr>
</tbody></table>
<p>family trio 4(SH021)</p>
<table>
<thead>
<tr>
<th>HG00463(father) <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRR1596842">https://www.ebi.ac.uk/ena/browser/view/SRR1596842</a> (mapping) 49012 (downloaded)</th>
</tr>
</thead>
<tbody><tr>
<td>HG00464(mother) <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRR1596844">https://www.ebi.ac.uk/ena/browser/view/SRR1596844</a> (prefetching) 13266 (downloaded)</td>
</tr>
<tr>
<td>HG00465(male child, Lymphoblastoid cell total RNA) <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ena/browser/view/SRX2432931">https://www.ebi.ac.uk/ena/browser/view/SRX2432931</a> (bam)</td>
</tr>
</tbody></table>
<p>QC by fastqc</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs shell">fastqc /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH007/SRX2432923/SRR5117458/SRR5117458.fastq.gz --extract -o /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH007/fastqc_report <br><br>fastqc /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH007/SRX2432923/SRR5117459/SRR5117459.fastq.gz --extract -o /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH007/fastqc_report #failed<br><span class="hljs-meta">#</span><span class="bash"> <span class="hljs-keyword">in</span> SH007/fastqc_report</span><br></code></pre></td></tr></table></figure>

<p>GATK preprocess</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs shell">samtools faidx new_hg19.fa #generate .fai<br>samtools dict new_hg19.fa -o new_hg19.dict # generate .dict<br></code></pre></td></tr></table></figure>



<p>mapping by STAR/hisat2</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><code class="hljs shell"><span class="hljs-meta"> #</span><span class="bash">generate index</span><br> /Users/keqinliu/STAR-2.7.6a/bin/MacOSX_x86_64/STAR --runThreadN 6 --runMode genomeGenerate \<br> --genomeDir /Users/keqinliu/studying_document/bioinfo/human_RME/STAR/SH007 \<br> --genomeFastaFiles /Users/keqinliu/Downloads/hg38.fa \ <br> --sjdbGTFfile /Users/keqinliu/Downloads/human.gtf \<br> --sjdbOverhang 100<br><span class="hljs-meta"> </span><br><span class="hljs-meta"> #</span><span class="bash"> <span class="hljs-keyword">do</span> mapping</span><br> STAR --runThreadN 20 --genomeDir ~/reference/index/STAR/mm10/ \<br> --readFilesIn SRR3589959_1.fastq SRR3589959_2.fastq \<br> --outSAMtype BAM SortedByCoordinate \<br> --outFileNamePrefix ./SRR3589959<br> <br><span class="hljs-meta"> </span><br><span class="hljs-meta">#</span><span class="bash"> the problem is the processing seems never end!</span><br><span class="hljs-meta">#</span><span class="bash"> and STAR took too much storage</span><br><span class="hljs-meta">#</span><span class="bash"> change to hisat2</span><br><span class="hljs-meta">#</span><span class="bash"> ref genome</span><br><br>hisat2 -x /Users/keqinliu/Downloads/hg38/genome \<br>-1 /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH007/SRR1606795.fastq.gz \<br>-S /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH007/SRR1606795.sam <br><span class="hljs-meta">#</span><span class="bash"> report error</span><br><span class="hljs-meta">#</span><span class="bash"> Error: Must specify at least one <span class="hljs-built_in">read</span> input with -U/-1/-2</span><br><span class="hljs-meta">#</span><span class="bash"> Overall time: 00:00:00</span><br><span class="hljs-meta">#</span><span class="bash"> (ERR): hisat2-align exited with value 1</span><br><br>hisat2 -x /Users/keqinliu/Downloads/hg38/genome \<br>-U /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH007/SRR1606795.fastq.gz \<br>-S /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH007/SRR1606795.sam # it works<br><br><br></code></pre></td></tr></table></figure>

<p>Mac turn off/on </p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs shell">sudo pmset -b sleep 0; sudo pmset -b disablesleep 1 # sleep function off<br><br>sudo pmset -b sleep 5; sudo pmset -b disablesleep 0 # sleep function on<br></code></pre></td></tr></table></figure>

<p>report</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-comment"># SRR2432923.sam(child)</span><br><span class="hljs-attribute">2599467</span> reads; of these:<br>  <span class="hljs-attribute">2599467</span> (<span class="hljs-number">100</span>.<span class="hljs-number">00</span>%) were unpaired; of these:<br>    <span class="hljs-attribute">149548</span> (<span class="hljs-number">5</span>.<span class="hljs-number">75</span>%) aligned <span class="hljs-number">0</span> times<br>    <span class="hljs-attribute">1933123</span> (<span class="hljs-number">74</span>.<span class="hljs-number">37</span>%) aligned exactly <span class="hljs-number">1</span> time<br>    <span class="hljs-attribute">516796</span> (<span class="hljs-number">19</span>.<span class="hljs-number">88</span>%) aligned &gt;<span class="hljs-number">1</span> times<br><span class="hljs-attribute">94</span>.<span class="hljs-number">25</span>% overall alignment rate<br><br><span class="hljs-comment"># SRR1606795.sam(mother)</span><br><span class="hljs-attribute">3638957</span> reads; of these:<br>  <span class="hljs-attribute">3638957</span> (<span class="hljs-number">100</span>.<span class="hljs-number">00</span>%) were unpaired; of these:<br>    <span class="hljs-attribute">727664</span> (<span class="hljs-number">20</span>.<span class="hljs-number">00</span>%) aligned <span class="hljs-number">0</span> times<br>    <span class="hljs-attribute">2715042</span> (<span class="hljs-number">74</span>.<span class="hljs-number">61</span>%) aligned exactly <span class="hljs-number">1</span> time<br>    <span class="hljs-attribute">196251</span> (<span class="hljs-number">5</span>.<span class="hljs-number">39</span>%) aligned &gt;<span class="hljs-number">1</span> times<br><span class="hljs-attribute">80</span>.<span class="hljs-number">00</span>% overall alignment rate<br><br><span class="hljs-comment"># SRR1606537.sam(father)</span><br><br><span class="hljs-comment"># SRR1602073.sam (father)</span><br><span class="hljs-attribute">2507162</span> reads; of these:<br>  <span class="hljs-attribute">2507162</span> (<span class="hljs-number">100</span>.<span class="hljs-number">00</span>%) were unpaired; of these:<br>    <span class="hljs-attribute">302285</span> (<span class="hljs-number">12</span>.<span class="hljs-number">06</span>%) aligned <span class="hljs-number">0</span> times<br>    <span class="hljs-attribute">2065134</span> (<span class="hljs-number">82</span>.<span class="hljs-number">37</span>%) aligned exactly <span class="hljs-number">1</span> time<br>    <span class="hljs-attribute">139743</span> (<span class="hljs-number">5</span>.<span class="hljs-number">57</span>%) aligned &gt;<span class="hljs-number">1</span> times<br><span class="hljs-attribute">87</span>.<span class="hljs-number">94</span>% overall alignment rate<br><br><span class="hljs-comment"># SRR1602075.sam (mother)</span><br><span class="hljs-attribute">gzip</span>: /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH<span class="hljs-number">002</span>/SRR<span class="hljs-number">1602075</span>/SRR<span class="hljs-number">1602075</span>.fastq.gz: unexpected end of file<br><span class="hljs-attribute">gzip</span>: /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH<span class="hljs-number">002</span>/SRR<span class="hljs-number">1602075</span>/SRR<span class="hljs-number">1602075</span>.fastq.gz: uncompress failed<br><span class="hljs-attribute">6341028</span> reads; of these:<br>  <span class="hljs-attribute">6341028</span> (<span class="hljs-number">100</span>.<span class="hljs-number">00</span>%) were unpaired; of these:<br>    <span class="hljs-attribute">290072</span> (<span class="hljs-number">4</span>.<span class="hljs-number">57</span>%) aligned <span class="hljs-number">0</span> times<br>    <span class="hljs-attribute">5817396</span> (<span class="hljs-number">91</span>.<span class="hljs-number">74</span>%) aligned exactly <span class="hljs-number">1</span> time<br>    <span class="hljs-attribute">233560</span> (<span class="hljs-number">3</span>.<span class="hljs-number">68</span>%) aligned &gt;<span class="hljs-number">1</span> times<br><span class="hljs-attribute">95</span>.<span class="hljs-number">43</span>% overall alignment rate<br><br><span class="hljs-comment"># SRR5117454.sam(child)</span><br><br></code></pre></td></tr></table></figure>

<p>sam to bam</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs shell"><span class="hljs-meta">#</span><span class="bash"> SH007</span><br>samtools view -S SRR2432923.sam -b &gt; SRR2432923.bam #child<br>samtools view -S SRR1606795.sam -b &gt; SRR1606795.bam #mother<br>samtools view -S SRR1606537.sam -b &gt; SRR1606537.bam #father<br><span class="hljs-meta"></span><br><span class="hljs-meta">#</span><span class="bash"> SH002</span><br>samtools view -S SRR1602073.sam -b &gt; SRR1602073.bam #father<br>samtools view -S SRR1602075.sam -b &gt; SRR1602075.bam #mother<br>samtools view -S SRR5117454.sam -b &gt; SRR5117454.bam<br></code></pre></td></tr></table></figure>

<p>GATK hyplotypeCaller</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><code class="hljs shell">/Users/keqinliu/Downloads/gatk-4.1.9.0/gatk --java-options &quot;-Xmx4g&quot; HaplotypeCaller  \<br>   -R /Users/keqinliu/Downloads/resources_broad_hg38_v0_Homo_sapiens_assembly38.fasta \<br>   -I /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH007/SRX2432923/SRR2432923.bam \ # child<br>   -O /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH007/SRR2432923.g.vcf.gz \<br>   -ERC GVCF <br>   -G Standard \<br>   -G AS_Standard<br><span class="hljs-meta"></span><br><span class="hljs-meta">#</span><span class="bash"> report error!</span><br><span class="hljs-meta">#</span><span class="bash"> </span><br><span class="hljs-meta"></span><br><span class="hljs-meta">#</span><span class="bash"> <span class="hljs-keyword">then</span> try to check whether bam file is damaged</span><br>samtools view -c -f 1 -F 12 /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH007/SRX2432923/SRR2432923.bam<br><span class="hljs-meta">#</span><span class="bash"> ruturn 0</span><br>samtools view -c -f 1 -F 12 /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH002/SRR1602073.bam <br>samtools view -c -f 1 -F 12 /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH002/SRR5117454.bam <br><span class="hljs-meta">#</span><span class="bash"> all <span class="hljs-built_in">return</span> 0</span><br><span class="hljs-meta">#</span><span class="bash"> so bam file is damaged</span><br><span class="hljs-meta"></span><br><span class="hljs-meta">#</span><span class="bash"> check whether sam file is damaged</span><br>samtools view -c -f 1 -F 12 /Users/keqinliu/studying_document/bioinfo/human_RME/raw_data/SH002/SRR1602073.sam<br><span class="hljs-meta">#</span><span class="bash"> <span class="hljs-built_in">return</span> 0</span><br><span class="hljs-meta">#</span><span class="bash"> so sam file is dameged</span><br></code></pre></td></tr></table></figure>

<p>GATK GenotypeGVCFs</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs"><br></code></pre></td></tr></table></figure>



<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs awk">ascp -k <span class="hljs-number">1</span> -QT -l <span class="hljs-number">300</span>m -P33001 -i ~<span class="hljs-regexp">/.aspera/</span>connect<span class="hljs-regexp">/etc/</span>asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:<span class="hljs-variable">$SRR1606795</span>.<br><br>ascp -i asperaweb_id_dsa.openssh -Tr -Q -l <span class="hljs-number">6</span>M -P33001 -L- -k1 era-fasp@fasp.sra.ebi.ac.uk:<span class="hljs-regexp">/vol1/</span>fastq<span class="hljs-regexp">/SRR160/</span><span class="hljs-number">007</span><span class="hljs-regexp">/SRR1606537/</span>SRR1606537.fastq.gz<br><br>ascp -k <span class="hljs-number">1</span> -QT -l <span class="hljs-number">300</span>m -P33001 -i ~<span class="hljs-regexp">/.aspera/</span>connect<span class="hljs-regexp">/etc/</span>asperaweb_id_dsa.openssh anonftp@ftp-private.ncbi.nlm.nih.gov:<span class="hljs-regexp">//</span>sra<span class="hljs-regexp">/sra-instant/</span>reads<span class="hljs-regexp">/ByExp/</span>sra<span class="hljs-regexp">/SRX/</span>SRX727<span class="hljs-regexp">/SRX727828/</span>SRR1606795/SRR1606795.sra.<br><br>ascp -v -QT -l <span class="hljs-number">400</span>m -P33001 -k1 -i ~<span class="hljs-regexp">/.aspera/</span>connect<span class="hljs-regexp">/etc/</span>asperaweb_id_dsa.openssh anonftp@ftp.ncbi.nlm.nih.gov:<span class="hljs-regexp">/sra/</span>sra-instant<span class="hljs-regexp">/reads/</span>ByRun<span class="hljs-regexp">/sra/</span>SRR<span class="hljs-regexp">/SRR160/</span>SRR1606795/SRR1606795.sra. <br><br>ftp.sra.ebi.ac.uk<span class="hljs-regexp">/vol1/</span>fastq<span class="hljs-regexp">/SRR160/</span><span class="hljs-number">007</span><span class="hljs-regexp">/SRR1606537/</span>SRR1606537.fastq.gz<br></code></pre></td></tr></table></figure>



<figure class="highlight gradle"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs gradle"><span class="hljs-regexp">/data/m</span>ouse<span class="hljs-regexp">/human_rme/</span>sratoolkit.<span class="hljs-number">2.10</span>.<span class="hljs-number">8</span>-ubuntu64<span class="hljs-regexp">/bin/</span>fastq-<span class="hljs-keyword">dump</span> -I --split-files<br><br><span class="hljs-regexp">/data/m</span>ouse<span class="hljs-regexp">/human_rme/</span>sratoolkit.<span class="hljs-number">2.10</span>.<span class="hljs-number">8</span>-ubuntu64<span class="hljs-regexp">/bin/</span>prefetch SRR1602075<br></code></pre></td></tr></table></figure>



<figure class="highlight jboss-cli"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs jboss-cli">STAR <span class="hljs-params">--genomeDir</span> <span class="hljs-string">/data/mouse/human_rme/genome_index</span> \<br>    <span class="hljs-params">--runThreadN</span> 20 \<br>    <span class="hljs-params">--readFilesIn</span> sample_r1.fq.gz sample_r2.fq.gz \<br>    <span class="hljs-comment"># --readFilesCommand zcat \</span><br>    <span class="hljs-params">--outFileNamePrefix</span> sample \<br>    <span class="hljs-params">--outSAMtype</span> BAM SortedByCoordinate \<br>    <span class="hljs-params">--outBAMsortingThreadN</span> 10<br></code></pre></td></tr></table></figure>



<figure class="highlight awk"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs awk"><span class="hljs-regexp">/data/m</span>ouse<span class="hljs-regexp">/human_rme/g</span>enome_index<span class="hljs-regexp">/gatk-4.1.9.0/g</span>atk --java-options <span class="hljs-string">&quot;-Xmx4g&quot;</span> HaplotypeCaller  \<br>   -R <span class="hljs-regexp">/data/m</span>ouse<span class="hljs-regexp">/human_rme/g</span>enome_index/hg19.fa \<br>   -I <span class="hljs-regexp">/data/m</span>ouse<span class="hljs-regexp">/human_rme/</span>SH007/SRR5117458.<span class="hljs-number">1</span>.bam \ <span class="hljs-comment"># child</span><br>   -O <span class="hljs-regexp">/data/m</span>ouse<span class="hljs-regexp">/human_rme/</span>SH007/SRR5117458.<span class="hljs-number">1</span>.g.vcf.gz \<br>   -ERC GVCF <br>   -G Standard \<br>   -G AS_Standard<br><br>https:<span class="hljs-regexp">//gi</span>thub.com<span class="hljs-regexp">/samtools/</span>samtools<span class="hljs-regexp">/releases/</span>download<span class="hljs-regexp">/1.11/</span>samtools-<span class="hljs-number">1.11</span>.tar.bz2<br></code></pre></td></tr></table></figure>



<figure class="highlight less"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs less"><span class="hljs-variable">@RG</span> <span class="hljs-attribute">ID</span>:HG00423  <span class="hljs-attribute">PL</span>:illumina <span class="hljs-attribute">PU</span>:H0164ALXX140820.<span class="hljs-number">2</span>    <span class="hljs-attribute">LB</span>:Solexa-<span class="hljs-number">272222</span>    <span class="hljs-attribute">PI</span>:<span class="hljs-number">0</span>    <span class="hljs-attribute">DT</span>:<span class="hljs-number">2014</span><span class="hljs-attribute">-08-20T00</span>:<span class="hljs-number">00</span>:<span class="hljs-number">00</span>-<span class="hljs-number">0400</span> <span class="hljs-attribute">SM</span>:NA12878  <span class="hljs-attribute">CN</span>:BI<br></code></pre></td></tr></table></figure>



<p>use GATK to do with 1000genome bam file </p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs shell">/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk --java-options &quot;-Xmx4g&quot; HaplotypeCaller  \<br>-R /data/mouse/human_rme/genome_index/hg19.fa \<br>-I HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.20130415.bam \<br>-O /data/mouse/human_rme/SH007/HG00421.vcf.gz \<br>-ERC GVCF <br></code></pre></td></tr></table></figure>

<p>error report:</p>
<figure class="highlight dns"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs dns"><span class="hljs-keyword">A</span> USER ERROR has occurred: Input files reference and reads have incompatible contigs: No overlapping contigs found.<br>  reference contigs = [chr1, chr2, chr3, chr4, chr5, chr6, chr7, chrX, chr8, chr9, chr10, chr11]<br>  reads contigs = [<span class="hljs-number">1</span>, <span class="hljs-number">2</span>, <span class="hljs-number">3</span>, <span class="hljs-number">4</span>, <span class="hljs-number">5</span>, <span class="hljs-number">6</span>, <span class="hljs-number">7</span>, <span class="hljs-number">8</span>, <span class="hljs-number">9</span>, <span class="hljs-number">10</span>, <span class="hljs-number">11</span>, <span class="hljs-number">12</span>, <span class="hljs-number">13</span>, <span class="hljs-number">14</span>, <span class="hljs-number">15</span>, <span class="hljs-number">16</span>, <span class="hljs-number">17</span>, <span class="hljs-number">18</span>, <span class="hljs-number">19</span>, <span class="hljs-number">20</span>, <span class="hljs-number">21</span>, <span class="hljs-number">22</span>, X, Y, MT, GL<span class="hljs-number">000207.1</span>, GL<span class="hljs-number">000226.1</span>, GL<span class="hljs-number">000229.1</span>, GL<span class="hljs-number">000231.1</span>, GL<span class="hljs-number">000210.1</span>, GL<span class="hljs-number">000239.1</span>, GL<span class="hljs-number">000235.1</span>, GL<span class="hljs-number">000201.1</span>, GL<span class="hljs-number">000247.1</span>, GL<span class="hljs-number">000245.1</span>, GL<span class="hljs-number">000197.1</span>, GL<span class="hljs-number">000203.1</span>, GL<span class="hljs-number">000246.1</span>, GL<span class="hljs-number">000249.1</span>, GL<span class="hljs-number">000196.1</span>, GL<span class="hljs-number">000248.1</span>, GL<span class="hljs-number">000244.1</span>, GL<span class="hljs-number">000238.1</span>, GL<span class="hljs-number">000202.1</span>, GL<span class="hljs-number">000234.1</span>, GL<span class="hljs-number">000232.1</span>, GL<span class="hljs-number">000206.1</span>, GL<span class="hljs-number">000240.1</span>, GL<span class="hljs-number">000236.1</span>, GL<span class="hljs-number">000241.1</span>, GL<span class="hljs-number">000243.1</span>, GL<span class="hljs-number">000242.1</span>, GL<span class="hljs-number">000230.1</span>, GL<span class="hljs-number">000237.1</span>, GL<span class="hljs-number">000233.1</span>, GL<span class="hljs-number">000204.1</span>, GL<span class="hljs-number">000198.1</span>, GL<span class="hljs-number">000208.1</span>, GL<span class="hljs-number">000191.1</span>, GL<span class="hljs-number">000227.1</span>, GL<span class="hljs-number">000228.1</span>, GL<span class="hljs-number">000214.1</span>, GL<span class="hljs-number">000221.1</span>, GL<span class="hljs-number">000209.1</span>, GL<span class="hljs-number">000218.1</span>, GL<span class="hljs-number">000220.1</span>, GL<span class="hljs-number">000213.1</span>, GL<span class="hljs-number">000211.1</span>, GL<span class="hljs-number">000199.1</span>, GL<span class="hljs-number">000217.1</span>, GL<span class="hljs-number">000216.1</span>, GL<span class="hljs-number">000215.1</span>, GL<span class="hljs-number">000205.1</span>, GL<span class="hljs-number">000219.1</span>, GL<span class="hljs-number">000224.1</span>, GL<span class="hljs-number">000223.1</span>, GL<span class="hljs-number">000195.1</span>, GL<span class="hljs-number">000212.1</span>, GL<span class="hljs-number">000222.1</span>, GL<span class="hljs-number">000200.1</span>, GL<span class="hljs-number">000193.1</span>, GL<span class="hljs-number">000194.1</span>, GL<span class="hljs-number">000225.1</span>, GL<span class="hljs-number">000192.1</span>, NC_007605, hs37d5]<br></code></pre></td></tr></table></figure>

<p>Use hg19 ref genome</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs shell">/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk --java-options &quot;-Xmx4g&quot; HaplotypeCaller  \<br>-R /data/mouse/human_rme/genome_index/hg19.fa \<br>-I HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.20130415.bam \<br>-O /data/mouse/human_rme/SH007/HG00421.vcf.gz \<br>-ERC GVCF <br></code></pre></td></tr></table></figure>

<p>error report:</p>
<figure class="highlight dns"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs dns"><span class="hljs-keyword">A</span> USER ERROR has occurred: Input files reference and reads have incompatible contigs: No overlapping contigs found.<br>  reference contigs = [chr1, chr2, chr3, chr4, chr5, chr6, chr7, chrX, chr8, chr9, chr10, chr11]<br>  reads contigs = [<span class="hljs-number">1</span>, <span class="hljs-number">2</span>, <span class="hljs-number">3</span>, <span class="hljs-number">4</span>, <span class="hljs-number">5</span>, <span class="hljs-number">6</span>, <span class="hljs-number">7</span>, <span class="hljs-number">8</span>, <span class="hljs-number">9</span>, <span class="hljs-number">10</span>, <span class="hljs-number">11</span>, <span class="hljs-number">12</span>, <span class="hljs-number">13</span>, <span class="hljs-number">14</span>, <span class="hljs-number">15</span>, <span class="hljs-number">16</span>, <span class="hljs-number">17</span>, <span class="hljs-number">18</span>, <span class="hljs-number">19</span>, <span class="hljs-number">20</span>, <span class="hljs-number">21</span>, <span class="hljs-number">22</span>, X, Y, MT, GL<span class="hljs-number">000207.1</span>, GL<span class="hljs-number">000226.1</span>, GL<span class="hljs-number">000229.1</span>, GL<span class="hljs-number">000231.1</span>, GL<span class="hljs-number">000210.1</span>, GL<span class="hljs-number">000239.1</span>, GL<span class="hljs-number">000235.1</span>, GL<span class="hljs-number">000201.1</span>, GL<span class="hljs-number">000247.1</span>, GL<span class="hljs-number">000245.1</span>, GL<span class="hljs-number">000197.1</span>, GL<span class="hljs-number">000203.1</span>, GL<span class="hljs-number">000246.1</span>, GL<span class="hljs-number">000249.1</span>, GL<span class="hljs-number">000196.1</span>, GL<span class="hljs-number">000248.1</span>, GL<span class="hljs-number">000244.1</span>, GL<span class="hljs-number">000238.1</span>, GL<span class="hljs-number">000202.1</span>, GL<span class="hljs-number">000234.1</span>, GL<span class="hljs-number">000232.1</span>, GL<span class="hljs-number">000206.1</span>, GL<span class="hljs-number">000240.1</span>, GL<span class="hljs-number">000236.1</span>, GL<span class="hljs-number">000241.1</span>, GL<span class="hljs-number">000243.1</span>, GL<span class="hljs-number">000242.1</span>, GL<span class="hljs-number">000230.1</span>, GL<span class="hljs-number">000237.1</span>, GL<span class="hljs-number">000233.1</span>, GL<span class="hljs-number">000204.1</span>, GL<span class="hljs-number">000198.1</span>, GL<span class="hljs-number">000208.1</span>, GL<span class="hljs-number">000191.1</span>, GL<span class="hljs-number">000227.1</span>, GL<span class="hljs-number">000228.1</span>, GL<span class="hljs-number">000214.1</span>, GL<span class="hljs-number">000221.1</span>, GL<span class="hljs-number">000209.1</span>, GL<span class="hljs-number">000218.1</span>, GL<span class="hljs-number">000220.1</span>, GL<span class="hljs-number">000213.1</span>, GL<span class="hljs-number">000211.1</span>, GL<span class="hljs-number">000199.1</span>, GL<span class="hljs-number">000217.1</span>, GL<span class="hljs-number">000216.1</span>, GL<span class="hljs-number">000215.1</span>, GL<span class="hljs-number">000205.1</span>, GL<span class="hljs-number">000219.1</span>, GL<span class="hljs-number">000224.1</span>, GL<span class="hljs-number">000223.1</span>, GL<span class="hljs-number">000195.1</span>, GL<span class="hljs-number">000212.1</span>, GL<span class="hljs-number">000222.1</span>, GL<span class="hljs-number">000200.1</span>, GL<span class="hljs-number">000193.1</span>, GL<span class="hljs-number">000194.1</span>, GL<span class="hljs-number">000225.1</span>, GL<span class="hljs-number">000192.1</span>, NC_007605, hs37d5]<br></code></pre></td></tr></table></figure>

<p>use hg38 get error report</p>
<figure class="highlight smali"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs smali">A USER ERROR has occurred: Input files reference<span class="hljs-built_in"> and </span>reads have incompatible contigs: No overlapping contigs found.<br>  reference contigs = [chr1, chr10, chr11, chr11_KI270721v1_random, chr12, chr13, chr14, chr14_GL000009v2_random, chr14_GL000225v1_random, chr14_KI270722v1_random, chr14_GL000194v1_random, chr14_KI270723v1_random, chr14_KI270724v1_random, chr14_KI270725v1_random, chr14_KI270726v1_random, chr15, chr15_KI270727v1_random, chr16, chr16_KI270728v1_random, chr17, chr17_GL000205v2_random, chr17_KI270729v1_random, chr17_KI270730v1_random, chr18, chr19, chr1_KI270706v1_random, chr1_KI270707v1_random, chr1_KI270708v1_random, chr1_KI270709v1_random, chr1_KI270710v1_random, chr1_KI270711v1_random, chr1_KI270712v1_random, chr1_KI270713v1_random, chr1_KI270714v1_random, chr2, chr20, chr21, chr22, chr22_KI270731v1_random, chr22_KI270732v1_random, chr22_KI270733v1_random, chr22_KI270734v1_random, chr22_KI270735v1_random, chr22_KI270736v1_random, chr22_KI270737v1_random, chr22_KI270738v1_random, chr22_KI270739v1_random, chr2_KI270715v1_random, chr2_KI270716v1_random, chr3, chr3_GL000221v1_random, chr4, chr4_GL000008v2_random, chr5, chr5_GL000208v1_random, chr6, chr7, chr8, chr9, chr9_KI270717v1_random, chr9_KI270718v1_random, chr9_KI270719v1_random, chr9_KI270720v1_random, chr1_KI270762v1_alt, chr1_KI270766v1_alt, chr1_KI270760v1_alt, chr1_KI270765v1_alt, chr1_GL383518v1_alt, chr1_GL383519v1_alt, chr1_GL383520v2_alt, chr1_KI270764v1_alt, chr1_KI270763v1_alt, chr1_KI270759v1_alt, chr1_KI270761v1_alt, chr2_KI270770v1_alt, chr2_KI270773v1_alt, chr2_KI270774v1_alt, chr2_KI270769v1_alt, chr2_GL383521v1_alt, chr2_KI270772v1_alt, chr2_KI270775v1_alt, chr2_KI270771v1_alt, chr2_KI270768v1_alt, chr2_GL582966v2_alt, chr2_GL383522v1_alt, chr2_KI270776v1_alt, chr2_KI270767v1_alt, chr3_JH636055v2_alt, chr3_KI270783v1_alt, chr3_KI270780v1_alt, chr3_GL383526v1_alt, chr3_KI270777v1_alt, chr3_KI270778v1_alt, chr3_KI270781v1_alt, chr3_KI270779v1_alt, chr3_KI270782v1_alt, chr3_KI270784v1_alt, chr4_KI270790v1_alt, chr4_GL383528v1_alt, chr4_KI270787v1_alt, chr4_GL000257v2_alt, chr4_KI270788v1_alt, chr4_GL383527v1_alt, chr4_KI270785v1_alt, chr4_KI270789v1_alt, chr4_KI270786v1_alt, chr5_KI270793v1_alt, chr5_KI270792v1_alt, chr5_KI270791v1_alt, chr5_GL383532v1_alt, chr5_GL949742v1_alt, chr5_KI270794v1_alt, chr5_GL339449v2_alt, chr5_GL383530v1_alt, chr5_KI270796v1_alt, chr5_GL383531v1_alt, chr5_KI270795v1_alt, chr6_GL000250v2_alt, chr6_KI270800v1_alt, chr6_KI270799v1_alt, chr6_GL383533v1_alt, chr6_KI270801v1_alt, chr6_KI270802v1_alt, chr6_KB021644v2_alt, chr6_KI270797v1_alt, chr6_KI270798v1_alt, chr7_KI270804v1_alt, chr7_KI270809v1_alt, chr7_KI270806v1_alt, chr7_GL383534v2_alt, chr7_KI270803v1_alt, chr7_KI270808v1_alt, chr7_KI270807v1_alt, chr7_KI270805v1_alt, chr8_KI270818v1_alt, chr8_KI270812v1_alt, chr8_KI270811v1_alt, chr8_KI270821v1_alt, chr8_KI270813v1_alt, chr8_KI270822v1_alt, chr8_KI270814v1_alt, chr8_KI270810v1_alt, chr8_KI270819v1_alt, chr8_KI270820v1_alt, chr8_KI270817v1_alt, chr8_KI270816v1_alt, chr8_KI270815v1_alt, chr9_GL383539v1_alt, chr9_GL383540v1_alt, chr9_GL383541v1_alt, chr9_GL383542v1_alt, chr9_KI270823v1_alt, chr10_GL383545v1_alt, chr10_KI270824v1_alt, chr10_GL383546v1_alt, chr10_KI270825v1_alt, chr11_KI270832v1_alt, chr11_KI270830v1_alt, chr11_KI270831v1_alt, chr11_KI270829v1_alt, chr11_GL383547v1_alt, chr11_JH159136v1_alt, chr11_JH159137v1_alt, chr11_KI270827v1_alt, chr11_KI270826v1_alt, chr12_GL877875v1_alt, chr12_GL877876v1_alt, chr12_KI270837v1_alt, chr12_GL383549v1_alt, chr12_KI270835v1_alt, chr12_GL383550v2_alt, chr12_GL383552v1_alt, chr12_GL383553v2_alt, chr12_KI270834v1_alt, chr12_GL383551v1_alt, chr12_KI270833v1_alt, chr12_KI270836v1_alt, chr13_KI270840v1_alt, chr13_KI270839v1_alt, chr13_KI270843v1_alt, chr13_KI270841v1_alt, chr13_KI270838v1_alt, chr13_KI270842v1_alt, chr14_KI270844v1_alt, chr14_KI270847v1_alt, chr14_KI270845v1_alt, chr14_KI270846v1_alt, chr15_KI270852v1_alt, chr15_KI270851v1_alt, chr15_KI270848v1_alt, chr15_GL383554v1_alt, chr15_KI270849v1_alt, chr15_GL383555v2_alt, chr15_KI270850v1_alt, chr16_KI270854v1_alt, chr16_KI270856v1_alt, chr16_KI270855v1_alt, chr16_KI270853v1_alt, chr16_GL383556v1_alt, chr16_GL383557v1_alt, chr17_GL383563v3_alt, chr17_KI270862v1_alt, chr17_KI270861v1_alt, chr17_KI270857v1_alt, chr17_JH159146v1_alt, chr17_JH159147v1_alt, chr17_GL383564v2_alt, chr17_GL000258v2_alt, chr17_GL383565v1_alt, chr17_KI270858v1_alt, chr17_KI270859v1_alt, chr17_GL383566v1_alt, chr17_KI270860v1_alt, chr18_KI270864v1_alt, chr18_GL383567v1_alt, chr18_GL383570v1_alt, chr18_GL383571v1_alt, chr18_GL383568v1_alt, chr18_GL383569v1_alt, chr18_GL383572v1_alt, chr18_KI270863v1_alt, chr19_KI270868v1_alt, chr19_KI270865v1_alt, chr19_GL383573v1_alt, chr19_GL383575v2_alt, chr19_GL383576v1_alt, chr19_GL383574v1_alt, chr19_KI270866v1_alt, chr19_KI270867v1_alt, chr19_GL949746v1_alt, chr20_GL383577v2_alt, chr20_KI270869v1_alt, chr20_KI270871v1_alt, chr20_KI270870v1_alt, chr21_GL383578v2_alt, chr21_KI270874v1_alt, chr21_KI270873v1_alt, chr21_GL383579v2_alt, chr21_GL383580v2_alt, chr21_GL383581v2_alt, chr21_KI270872v1_alt, chr22_KI270875v1_alt, chr22_KI270878v1_alt, chr22_KI270879v1_alt, chr22_KI270876v1_alt, chr22_KI270877v1_alt, chr22_GL383583v2_alt, chr22_GL383582v2_alt, chrX_KI270880v1_alt, chrX_KI270881v1_alt, chr19_KI270882v1_alt, chr19_KI270883v1_alt, chr19_KI270884v1_alt, chr19_KI270885v1_alt, chr19_KI270886v1_alt, chr19_KI270887v1_alt, chr19_KI270888v1_alt, chr19_KI270889v1_alt, chr19_KI270890v1_alt, chr19_KI270891v1_alt, chr1_KI270892v1_alt, chr2_KI270894v1_alt, chr2_KI270893v1_alt, chr3_KI270895v1_alt, chr4_KI270896v1_alt, chr5_KI270897v1_alt, chr5_KI270898v1_alt, chr6_GL000251v2_alt, chr7_KI270899v1_alt, chr8_KI270901v1_alt, chr8_KI270900v1_alt, chr11_KI270902v1_alt, chr11_KI270903v1_alt, chr12_KI270904v1_alt, chr15_KI270906v1_alt, chr15_KI270905v1_alt, chr17_KI270907v1_alt, chr17_KI270910v1_alt, chr17_KI270909v1_alt, chr17_JH159148v1_alt, chr17_KI270908v1_alt, chr18_KI270912v1_alt, chr18_KI270911v1_alt, chr19_GL949747v2_alt, chr22_KB663609v1_alt, chrX_KI270913v1_alt, chr19_KI270914v1_alt, chr19_KI270915v1_alt, chr19_KI270916v1_alt, chr19_KI270917v1_alt, chr19_KI270918v1_alt, chr19_KI270919v1_alt, chr19_KI270920v1_alt, chr19_KI270921v1_alt, chr19_KI270922v1_alt, chr19_KI270923v1_alt, chr3_KI270924v1_alt, chr4_KI270925v1_alt, chr6_GL000252v2_alt, chr8_KI270926v1_alt, chr11_KI270927v1_alt, chr19_GL949748v2_alt, chr22_KI270928v1_alt, chr19_KI270929v1_alt, chr19_KI270930v1_alt, chr19_KI270931v1_alt, chr19_KI270932v1_alt, chr19_KI270933v1_alt, chr19_GL000209v2_alt, chr3_KI270934v1_alt, chr6_GL000253v2_alt, chr19_GL949749v2_alt, chr3_KI270935v1_alt, chr6_GL000254v2_alt, chr19_GL949750v2_alt, chr3_KI270936v1_alt, chr6_GL000255v2_alt, chr19_GL949751v2_alt, chr3_KI270937v1_alt, chr6_GL000256v2_alt, chr19_GL949752v1_alt, chr6_KI270758v1_alt, chr19_GL949753v2_alt, chr19_KI270938v1_alt, chrM, chrUn_KI270302v1, chrUn_KI270304v1, chrUn_KI270303v1, chrUn_KI270305v1, chrUn_KI270322v1, chrUn_KI270320v1, chrUn_KI270310v1, chrUn_KI270316v1, chrUn_KI270315v1, chrUn_KI270312v1, chrUn_KI270311v1, chrUn_KI270317v1, chrUn_KI270412v1, chrUn_KI270411v1, chrUn_KI270414v1, chrUn_KI270419v1, chrUn_KI270418v1, chrUn_KI270420v1, chrUn_KI270424v1, chrUn_KI270417v1, chrUn_KI270422v1, chrUn_KI270423v1, chrUn_KI270425v1, chrUn_KI270429v1, chrUn_KI270442v1, chrUn_KI270466v1, chrUn_KI270465v1, chrUn_KI270467v1, chrUn_KI270435v1, chrUn_KI270438v1, chrUn_KI270468v1, chrUn_KI270510v1, chrUn_KI270509v1, chrUn_KI270518v1, chrUn_KI270508v1, chrUn_KI270516v1, chrUn_KI270512v1, chrUn_KI270519v1, chrUn_KI270522v1, chrUn_KI270511v1, chrUn_KI270515v1, chrUn_KI270507v1, chrUn_KI270517v1, chrUn_KI270529v1, chrUn_KI270528v1, chrUn_KI270530v1, chrUn_KI270539v1, chrUn_KI270538v1, chrUn_KI270544v1, chrUn_KI270548v1, chrUn_KI270583v1, chrUn_KI270587v1, chrUn_KI270580v1, chrUn_KI270581v1, chrUn_KI270579v1, chrUn_KI270589v1, chrUn_KI270590v1, chrUn_KI270584v1, chrUn_KI270582v1, chrUn_KI270588v1, chrUn_KI270593v1, chrUn_KI270591v1, chrUn_KI270330v1, chrUn_KI270329v1, chrUn_KI270334v1, chrUn_KI270333v1, chrUn_KI270335v1, chrUn_KI270338v1, chrUn_KI270340v1, chrUn_KI270336v1, chrUn_KI270337v1, chrUn_KI270363v1, chrUn_KI270364v1, chrUn_KI270362v1, chrUn_KI270366v1, chrUn_KI270378v1, chrUn_KI270379v1, chrUn_KI270389v1, chrUn_KI270390v1, chrUn_KI270387v1, chrUn_KI270395v1, chrUn_KI270396v1, chrUn_KI270388v1, chrUn_KI270394v1, chrUn_KI270386v1, chrUn_KI270391v1, chrUn_KI270383v1, chrUn_KI270393v1, chrUn_KI270384v1, chrUn_KI270392v1, chrUn_KI270381v1, chrUn_KI270385v1, chrUn_KI270382v1, chrUn_KI270376v1, chrUn_KI270374v1, chrUn_KI270372v1, chrUn_KI270373v1, chrUn_KI270375v1, chrUn_KI270371v1, chrUn_KI270448v1, chrUn_KI270521v1, chrUn_GL000195v1, chrUn_GL000219v1, chrUn_GL000220v1, chrUn_GL000224v1, chrUn_KI270741v1, chrUn_GL000226v1, chrUn_GL000213v1, chrUn_KI270743v1, chrUn_KI270744v1, chrUn_KI270745v1, chrUn_KI270746v1, chrUn_KI270747v1, chrUn_KI270748v1, chrUn_KI270749v1, chrUn_KI270750v1, chrUn_KI270751v1, chrUn_KI270752v1, chrUn_KI270753v1, chrUn_KI270754v1, chrUn_KI270755v1, chrUn_KI270756v1, chrUn_KI270757v1, chrUn_GL000214v1, chrUn_KI270742v1, chrUn_GL000216v2, chrUn_GL000218v1, chrX, chrY, chrY_KI270740v1_random, chr1_KQ031383v1_fix, chr1_KQ983255v1_alt, chr1_KN538361v1_fix, chr1_KQ458383v1_alt, chr1_KN196473v1_fix, chr1_KZ208904v1_alt, chr1_KN196472v1_fix, chr1_KZ208905v1_alt, chr1_KQ458382v1_alt, chr1_KV880763v1_alt, chr1_KN196474v1_fix, chr1_KN538360v1_fix, chr1_KZ208906v1_fix, chr1_KQ458384v1_alt, chr2_KQ031384v1_fix, chr2_KZ208907v1_alt, chr2_KQ983256v1_alt, chr2_KZ208908v1_alt, chr2_KN538363v1_fix, chr2_KN538362v1_fix, chr3_KV766192v1_fix, chr3_KN196475v1_fix, chr3_KQ031385v1_fix, chr3_KN538364v1_fix, chr3_KZ208909v1_alt, chr3_KQ031386v1_fix, chr3_KN196476v1_fix, chr4_KQ090013v1_alt, chr4_KQ090014v1_alt, chr4_KQ090015v1_alt, chr4_KV766193v1_alt, chr4_KQ983257v1_fix, chr4_KQ983258v1_alt, chr5_KZ208910v1_alt, chr5_KN196477v1_alt, chr5_KV575243v1_alt, chr5_KV575244v1_fix, chr6_KZ208911v1_fix, chr6_KQ090017v1_alt, chr6_KQ031387v1_fix, chr6_KN196478v1_fix, chr6_KQ090016v1_fix, chr6_KV766194v1_fix, chr7_KV880764v1_fix, chr7_KV880765v1_fix, chr7_KZ208912v1_fix, chr7_KZ208913v1_alt, chr7_KQ031388v1_fix, chr8_KZ208915v1_fix, chr8_KV880767v1_fix, chr8_KV880766v1_fix, chr8_KZ208914v1_fix, chr9_KQ090018v1_alt, chr9_KQ090019v1_alt, chr9_KN196479v1_fix, chr10_KN538367v1_fix, chr10_KQ090020v1_alt, chr10_KN196480v1_fix, chr10_KQ090021v1_fix, chr10_KN538366v1_fix, chr10_KN538365v1_fix, chr11_KQ759759v1_fix, chr11_KN538368v1_alt, chr11_KV766195v1_fix, chr11_KQ090022v1_fix, chr11_KN196481v1_fix, chr12_KQ090023v1_alt, chr12_KZ208916v1_fix, chr12_KN538369v1_fix, chr12_KN196482v1_fix, chr12_KZ208918v1_alt, chr12_KQ759760v1_fix, chr12_KZ208917v1_fix, chr12_KN538370v1_fix, chr13_KN538372v1_fix, chr13_KQ090024v1_alt, chr13_KN196483v1_fix, chr13_KN538373v1_fix, chr13_KQ090025v1_alt, chr13_KN538371v1_fix, chr14_KZ208920v1_fix, chr14_KZ208919v1_alt, chr15_KN538374v1_fix, chr15_KQ031389v1_alt, chr16_KQ090026v1_alt, chr16_KV880768v1_fix, chr16_KQ090027v1_alt, chr16_KZ208921v1_alt, chr16_KQ031390v1_alt, chr17_KV766196v1_fix, chr17_KV575245v1_fix, chr17_KV766198v1_alt, chr17_KV766197v1_alt, chr18_KQ458385v1_alt, chr18_KQ090028v1_fix, chr18_KZ208922v1_fix, chr19_KQ458386v1_fix, chr19_KN196484v1_fix, chr19_KV575246v1_alt, chr19_KV575247v1_alt, chr19_KV575248v1_alt, chr19_KV575249v1_alt, chr19_KV575250v1_alt, chr19_KV575251v1_alt, chr19_KV575252v1_alt, chr19_KV575253v1_alt, chr19_KV575254v1_alt, chr19_KV575255v1_alt, chr19_KV575256v1_alt, chr19_KV575257v1_alt, chr19_KV575259v1_alt, chr19_KV575260v1_alt, chr19_KV575258v1_alt, chr22_KN196485v1_alt, chr22_KQ458387v1_alt, chr22_KQ458388v1_alt, chr22_KN196486v1_alt, chr22_KQ759761v1_alt, chr22_KQ759762v1_fix, chrX_KV766199v1_alt, chrY_KZ208923v1_fix, chrY_KZ208924v1_fix, chrY_KN196487v1_fix, chr1_KZ559100v1_fix, chr3_KZ559104v1_fix, chr3_KZ559105v1_alt, chr3_KZ559103v1_alt, chr3_KZ559102v1_alt, chr3_KZ559101v1_alt, chr7_KZ559106v1_alt, chr8_KZ559107v1_alt, chr11_KZ559109v1_fix, chr11_KZ559108v1_fix, chr11_KZ559111v1_alt, chr11_KZ559110v1_alt, chr12_KZ559112v1_alt, chr16_KZ559113v1_fix, chr17_KZ559114v1_alt, chr18_KZ559116v1_alt, chr18_KZ559115v1_fix]<br>  reads contigs = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, X, Y, MT, GL000207.1, GL000226.1, GL000229.1, GL000231.1, GL000210.1, GL000239.1, GL000235.1, GL000201.1, GL000247.1, GL000245.1, GL000197.1, GL000203.1, GL000246.1, GL000249.1, GL000196.1, GL000248.1, GL000244.1, GL000238.1, GL000202.1, GL000234.1, GL000232.1, GL000206.1, GL000240.1, GL000236.1, GL000241.1, GL000243.1, GL000242.1, GL000230.1, GL000237.1, GL000233.1, GL000204.1, GL000198.1, GL000208.1, GL000191.1, GL000227.1, GL000228.1, GL000214.1, GL000221.1, GL000209.1, GL000218.1, GL000220.1, GL000213.1, GL000211.1, GL000199.1, GL000217.1, GL000216.1, GL000215.1, GL000205.1, GL000219.1, GL000224.1, GL000223.1, GL000195.1, GL000212.1, GL000222.1, GL000200.1, GL000193.1, GL000194.1, GL000225.1, GL000192.1, NC_007605, hs37d5]<br><br>***********************************************************************<br>Set the<span class="hljs-keyword"> system</span> property GATK_STACKTRACE_ON_USER_EXCEPTION (--java-options &#x27;-DGATK_STACKTRACE_ON_USER_EXCEPTION=true&#x27;) to print the stack trace.<br></code></pre></td></tr></table></figure>

<p>Use human_g1k_v37.fasta as reference genome</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs shell">/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk --java-options &quot;-Xmx4g&quot; HaplotypeCaller  \<br>-R /data/mouse/human_rme/genome_index/human_g1k_v37.fasta \<br>-I HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.20130415.bam \<br>-O /data/mouse/human_rme/SH007/HG00421.vcf.gz \<br>-ERC GVCF <br></code></pre></td></tr></table></figure>

<p>error report</p>
<figure class="highlight vim"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs vim">A USER ERROR <span class="hljs-built_in">has</span> occurred: Traversal by intervals was requested but some <span class="hljs-built_in">input</span> <span class="hljs-keyword">files</span> are not indexed.<br>Please <span class="hljs-built_in">index</span> <span class="hljs-keyword">all</span> <span class="hljs-built_in">input</span> <span class="hljs-keyword">file</span><span class="hljs-variable">s:</span><br><br>samtools <span class="hljs-built_in">index</span> /data/mouse/human_rme/SH007/HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.<span class="hljs-number">20130415</span>.bam<br></code></pre></td></tr></table></figure>

<p>then try to index by samtools</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs shell">samtools index /data/mouse/human_rme/SH007/HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.20130415.bam<br></code></pre></td></tr></table></figure>

<p>Error report</p>
<figure class="highlight pgsql"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs pgsql">[W::bam_hdr_read] EOF marker <span class="hljs-keyword">is</span> absent. The <span class="hljs-keyword">input</span> <span class="hljs-keyword">is</span> probably truncated<br>[E::bgzf_read_block] Failed <span class="hljs-keyword">to</span> <span class="hljs-keyword">read</span> BGZF block data at <span class="hljs-keyword">offset</span> <span class="hljs-number">4612181361</span> expected <span class="hljs-number">18560</span> bytes; hread returned <span class="hljs-number">12925</span><br>[E::bgzf_read] <span class="hljs-keyword">Read</span> block operation failed <span class="hljs-keyword">with</span> error <span class="hljs-number">4</span> <span class="hljs-keyword">after</span> <span class="hljs-number">0</span> <span class="hljs-keyword">of</span> <span class="hljs-number">4</span> bytes<br>samtools <span class="hljs-keyword">index</span>: failed <span class="hljs-keyword">to</span> <span class="hljs-keyword">create</span> <span class="hljs-keyword">index</span> <span class="hljs-keyword">for</span> &quot;/data/mouse/human_rme/SH007/HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.20130415.bam&quot;: <span class="hljs-keyword">No</span> such file <span class="hljs-keyword">or</span> directory<br></code></pre></td></tr></table></figure>

<p>Guess: bam file is probably truncated, check bam file</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs shell">tail HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.20130415.bam | hexdump -C<br></code></pre></td></tr></table></figure>

<p>return</p>
<figure class="highlight tap"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br></pre></td><td class="code"><pre><code class="hljs tap">00000000  c1 c5<span class="hljs-number"> 58 </span>42 f1 3a a6<span class="hljs-number"> 89 </span><span class="hljs-number"> 41 </span>9b<span class="hljs-number"> 13 </span>31 bb da cd 6a  |..XB.:..A..1...j|<br>00000010  eb<span class="hljs-number"> 60 </span>5d cc 8a 1e<span class="hljs-number"> 33 </span>ec  cf cd<span class="hljs-number"> 36 </span>53<span class="hljs-number"> 54 </span>e0 ac c1  |.`]...3...6ST...|<br>00000020  cd 6f<span class="hljs-number"> 09 </span>1c f8<span class="hljs-number"> 32 </span>db c1 <span class="hljs-number"> 99 </span>6d<span class="hljs-number"> 91 </span>c6<span class="hljs-number"> 02 </span>b2 b2<span class="hljs-number"> 55 </span> |.o...2...m.....U|<br>00000030 <span class="hljs-number"> 64 </span>ab 0e<span class="hljs-number"> 63 </span>25 d2<span class="hljs-number"> 70 </span>d3 <span class="hljs-number"> 61 </span>19<span class="hljs-number"> 37 </span>84 d8 b7 d9 8d  |d..c%.p.a.7.....|<br>00000040  5a b8<span class="hljs-number"> 02 </span>f1<span class="hljs-number"> 31 </span>fe<span class="hljs-number"> 33 </span>3f <span class="hljs-number"> 07 </span>68 e7<span class="hljs-number"> 79 </span>94 2b ff f6  |Z...1.3?.h.y.+..|<br>00000050  8a a2<span class="hljs-number"> 03 </span>2a<span class="hljs-number"> 47 </span>e8<span class="hljs-number"> 45 </span>85  9d<span class="hljs-number"> 79 </span>39 8e 4b 8e fe<span class="hljs-number"> 91 </span> |...*G.E..y9.K...|<br>00000060  e9<span class="hljs-number"> 74 </span>72 6e cf<span class="hljs-number"> 35 </span>bd 3e  1f 9c 3b a3 fe<span class="hljs-number"> 00 </span>98<span class="hljs-number"> 43 </span> |.trn.5.&gt;..;....C|<br>00000070 <span class="hljs-number"> 97 </span>69 bb<span class="hljs-number"> 07 </span>12<span class="hljs-number"> 40 </span>64<span class="hljs-number"> 98 </span> 2e<span class="hljs-number"> 00 </span>70<span class="hljs-number"> 59 </span>b6 8a db<span class="hljs-number"> 86 </span> |.i...@d...pY....|<br>00000080  0a<span class="hljs-number"> 65 </span>bb<span class="hljs-number"> 34 </span>c4 8a a1 e9 <span class="hljs-number"> 34 </span>6b 9c<span class="hljs-number"> 36 </span>e3<span class="hljs-number"> 45 </span>22 b6  |.e.4....4k.6.E&quot;.|<br>00000090  c6<span class="hljs-number"> 83 </span>d9<span class="hljs-number"> 40 </span>a9<span class="hljs-number"> 32 </span>5d<span class="hljs-number"> 08 </span> 1c<span class="hljs-number"> 07 </span>40 8d 3a<span class="hljs-number"> 05 </span>17 4d  |...@.2]...@.:..M|<br>000000a0  d1<span class="hljs-number"> 14 </span>a8<span class="hljs-number"> 18 </span>16<span class="hljs-number"> 96 </span>e1 f7  ae ef aa e9<span class="hljs-number"> 49 </span>f4 f6<span class="hljs-number"> 77 </span> |............I..w|<br>000000b0 <span class="hljs-number"> 56 </span>a4<span class="hljs-number"> 65 </span>11 e8 c5 c9 ee <span class="hljs-number"> 79 </span>39 8e eb 9a<span class="hljs-number"> 33 </span>77<span class="hljs-number"> 22 </span> |V.e.....y9...3w&quot;|<br>000000c0  e4 5c cc a5<span class="hljs-number"> 44 </span>43 cc 0b <span class="hljs-number"> 88 </span>db 1d<span class="hljs-number"> 00 </span>99<span class="hljs-number"> 18 </span>bd 2d  |.\..DC.........-|<br>000000d0 <span class="hljs-number"> 13 </span>1b db<span class="hljs-number"> 02 </span>85 0b 4f b1  a1 c4 f6 a3<span class="hljs-number"> 27 </span>cc<span class="hljs-number"> 76 </span>eb  |......O.....&#x27;.v.|<br>000000e0 <span class="hljs-number"> 59 </span>49 4a cd<span class="hljs-number"> 70 </span>87<span class="hljs-number"> 93 </span>15 <span class="hljs-number"> 38 </span>10<span class="hljs-number"> 01 </span>44 7f 5c e0<span class="hljs-number"> 45 </span> |YIJ.p...8..D.\.E|<br>000000f0 <span class="hljs-number"> 53 </span>5a f9<span class="hljs-number"> 08 </span>b3<span class="hljs-number"> 14 </span>c0 bb  e4 da<span class="hljs-number"> 96 </span>85<span class="hljs-number"> 27 </span>6b<span class="hljs-number"> 58 </span>be  |SZ..........&#x27;kX.|<br>00000100  3e<span class="hljs-number"> 47 </span>71<span class="hljs-number"> 27 </span>de f6 b3 ef  4c<span class="hljs-number"> 27 </span>b0<span class="hljs-number"> 26 </span>de b6 dd<span class="hljs-number"> 84 </span> |&gt;Gq&#x27;....L&#x27;.&amp;....|<br>00000110 <span class="hljs-number"> 38 </span>b9<span class="hljs-number"> 51 </span>1b af<span class="hljs-number"> 61 </span>b9 0a  ce 0c<span class="hljs-number"> 13 </span>06<span class="hljs-number"> 46 </span>c5<span class="hljs-number"> 39 </span>8e  |8.Q..a......F.9.|<br>00000120 <span class="hljs-number"> 35 </span>b1<span class="hljs-number"> 24 </span>2a 2b<span class="hljs-number"> 94 </span>ea<span class="hljs-number"> 06 </span><span class="hljs-number"> 17 </span>9a<span class="hljs-number"> 98 </span>28 d2<span class="hljs-number"> 81 </span>77<span class="hljs-number"> 48 </span> |5.$*+......(..wH|<br>00000130  3d<span class="hljs-number"> 57 </span>88<span class="hljs-number"> 40 </span>a4<span class="hljs-number"> 79 </span>96 f3  7c 0f<span class="hljs-number"> 62 </span>05 4a e2<span class="hljs-number"> 44 </span>c7  |=W.@.y..|.b.J.D.|<br>00000140 <span class="hljs-number"> 34 </span>a4 0e d9 df 0b 0a<span class="hljs-number"> 49 </span><span class="hljs-number"> 18 </span>07 7c cf f5 fd d2 3c  |4......I..|....&lt;|<br>00000150  ab ba a2<span class="hljs-number"> 80 </span>f2 fb 8a d5 <span class="hljs-number"> 24 </span>d2 fd e5<span class="hljs-number"> 15 </span>19 1c<span class="hljs-number"> 94 </span> |........$.......|<br>00000160 <span class="hljs-number"> 33 </span>59 dc<span class="hljs-number"> 55 </span>38<span class="hljs-number"> 95 </span>2e c1 <span class="hljs-number"> 15 </span>2c<span class="hljs-number"> 28 </span>fc d4<span class="hljs-number"> 35 </span>d7 e1  |3Y.U8....,(..5..|<br>00000170 <span class="hljs-number"> 01 </span>23 b0 ce 4b d9<span class="hljs-number"> 92 </span>08 <span class="hljs-number"> 71 </span>69 6c<span class="hljs-number"> 82 </span>1b<span class="hljs-number"> 41 </span>22<span class="hljs-number"> 88 </span> |.<span class="hljs-comment">#..K...qil..A&quot;.|</span><br>00000180 <span class="hljs-number"> 74 </span>51<span class="hljs-number"> 15 </span>8a<span class="hljs-number"> 07 </span>52 f9 ea <span class="hljs-number"> 28 </span>94<span class="hljs-number"> 05 </span>3f 0c b0 f0 ce  |tQ...R..(..?....|<br>00000190 <span class="hljs-number"> 45 </span>6c<span class="hljs-number"> 94 </span>44<span class="hljs-number"> 59 </span>65<span class="hljs-number"> 42 </span>12 <span class="hljs-number"> 85 </span>a2 df 9e 2e<span class="hljs-number"> 45 </span>52<span class="hljs-number"> 15 </span> |El.DYeB......ER.|<br>000001a0 <span class="hljs-number"> 88 </span>17<span class="hljs-number"> 72 </span>e6 1d 6c dd<span class="hljs-number"> 99 </span> 9f 7d<span class="hljs-number"> 45 </span>61 a8<span class="hljs-number"> 56 </span>f3 fe  |..r..l...&#125;Ea.V..|<br>000001b0  6a 7d<span class="hljs-number"> 87 </span>e7 1a e9 fe ca  0a<span class="hljs-number"> 77 </span>8b 5a<span class="hljs-number"> 26 </span>72<span class="hljs-number"> 51 </span>76  |j&#125;.......w.Z&amp;rQv|<br>000001c0 <span class="hljs-number"> 72 </span>5e b7 e4 ca<span class="hljs-number"> 77 </span>bf ec <span class="hljs-number"> 19 </span>08 d3 cc a8 1f<span class="hljs-number"> 45 </span>0e  |r^...w........E.|<br>000001d0  ca 6a<span class="hljs-number"> 01 </span>85 f2 0b<span class="hljs-number"> 45 </span>08  b4<span class="hljs-number"> 24 </span>2b d0<span class="hljs-number"> 58 </span>c4 a2 e8  |.j....E..$+.X...|<br>000001e0  c6<span class="hljs-number"> 64 </span>cb f3<span class="hljs-number"> 85 </span>36<span class="hljs-number"> 41 </span>b8  0d cf<span class="hljs-number"> 00 </span>d2<span class="hljs-number"> 89 </span>00 8e e6  |.d...6A.........|<br>000001f0  ef<span class="hljs-number"> 72 </span>16 e1 f6 e8<span class="hljs-number"> 84 </span>fb  be 1f<span class="hljs-number"> 45 </span>c5<span class="hljs-number"> 62 </span>d1<span class="hljs-number"> 09 </span>0b  |.r........E.b...|<br>00000200  3b bb 9e<span class="hljs-number"> 50 </span>14<span class="hljs-number"> 42 </span>8c 4d <span class="hljs-number"> 96 </span>bf<span class="hljs-number"> 41 </span>e9 dd bb<span class="hljs-number"> 91 </span>88  |;..P.B.M..A.....|<br>00000210  c2 dd 5d<span class="hljs-number"> 42 </span>bd<span class="hljs-number"> 48 </span>f1 b5  9d<span class="hljs-number"> 09 </span>9a 8c<span class="hljs-number"> 75 </span>a3 1c 1f  |..]B.H......u...|<br>00000220  4e<span class="hljs-number"> 50 </span>dc 9a<span class="hljs-number"> 33 </span>6d<span class="hljs-number"> 94 </span>ed <span class="hljs-number"> 23 </span>8b<span class="hljs-number"> 01 </span>d8<span class="hljs-number"> 44 </span>a2 ef<span class="hljs-number"> 47 </span> |NP..3m..<span class="hljs-comment">#...D..G|</span><br>00000230  6f 6d 6c fc<span class="hljs-number"> 02 </span>90 2e<span class="hljs-number"> 07 </span><span class="hljs-number"> 27 </span>5a<span class="hljs-number"> 46 </span>c3 fe<span class="hljs-number"> 68 </span>30 0e  |oml.....&#x27;ZF..h0.|<br>00000240 <span class="hljs-number"> 75 </span>73 7d<span class="hljs-number"> 38 </span>bf 4c 8f c3  9f<span class="hljs-number"> 51 </span>87 f5 c6 e7 5d 0f  |us&#125;8.L...Q....].|<br>00000250  ce bb f3 c1 e4 bc eb 3f <span class="hljs-number"> 21 </span>07 a3 fb 2a<span class="hljs-number"> 74 </span>c4 0b  |.......?!...*t..|<br>00000260 <span class="hljs-number"> 85 </span>00 4e<span class="hljs-number"> 47 </span>11 ed 6e<span class="hljs-number"> 17 </span> b6<span class="hljs-number"> 44 </span>bc<span class="hljs-number"> 15 </span>3b 5b 3b 9b  |..NG..n..D..;[;.|<br>00000270  f0 d8<span class="hljs-number"> 75 </span>24<span class="hljs-number"> 77 </span>15 f3 fd  e0 ee 7e<span class="hljs-number"> 29 </span>a4<span class="hljs-number"> 39 </span>30 eb  |..u$w.....~).90.|<br>00000280  0f<span class="hljs-number"> 20 </span>d8 cd<span class="hljs-number"> 07 </span>ed 7b<span class="hljs-number"> 27 </span> 0f 4e ee<span class="hljs-number"> 55 </span>2f<span class="hljs-number"> 66 </span>28<span class="hljs-number"> 71 </span> |. ....&#123;&#x27;.N.U/f(q|<br>00000290 <span class="hljs-number"> 88 </span>e2 f6 dc<span class="hljs-number"> 10 </span>c5 2b cf <span class="hljs-number"> 46 </span>e6 7f<span class="hljs-number"> 06 </span>fa af 3d bc  |......+.F.....=.|<br>000002a0 <span class="hljs-number"> 28 </span>16 f3 a3 ef 2c 8a<span class="hljs-number"> 16 </span><span class="hljs-number"> 34 </span>20<span class="hljs-number"> 18 </span>6b b4 4e 9a<span class="hljs-number"> 67 </span> |(....,..4 .k.N.g|<br>000002b0  1b<span class="hljs-number"> 27 </span>77<span class="hljs-number"> 32 </span>bc 1f c9<span class="hljs-number"> 21 </span> 7a f3 1c a2 2b bb<span class="hljs-number"> 83 </span>61  |.&#x27;w2...!z...+..a|<br>000002c0  ae 0b de a4 d7 c9 6d<span class="hljs-number"> 87 </span><span class="hljs-number"> 77 </span>dd cd 1b<span class="hljs-number"> 47 </span>bb<span class="hljs-number"> 31 </span>67  |......m.w...G.1g|<br>000002d0  b9<span class="hljs-number"> 98 </span>c5<span class="hljs-number"> 07 </span>71 a9<span class="hljs-number"> 78 </span>58 <span class="hljs-number"> 22 </span>bb 3b<span class="hljs-number"> 81 </span>5f<span class="hljs-number"> 88 </span>f8<span class="hljs-number"> 81 </span> |....q.xX&quot;.;._...|<br>000002e0 <span class="hljs-number"> 47 </span>92<span class="hljs-number"> 42 </span>71<span class="hljs-number"> 87 </span>45<span class="hljs-number"> 31 </span>c0  aa b7 5c ae<span class="hljs-number"> 76 </span>b9 b8<span class="hljs-number"> 01 </span> |G.Bq.E1...\.v...|<br>000002f0  a1<span class="hljs-number"> 05 </span>b0 bc<span class="hljs-number"> 30 </span>74 e2<span class="hljs-number"> 03 </span> 6c<span class="hljs-number"> 26 </span>01 5f 9e<span class="hljs-number"> 73 </span>e0 3c  |....0t..l&amp;._.s.&lt;|<br>00000300  8b<span class="hljs-number"> 25 </span>7d a1<span class="hljs-number"> 74 </span>70 e7 c8 <span class="hljs-number"> 15 </span>b1<span class="hljs-number"> 24 </span>a2<span class="hljs-number"> 28 </span>19 1c 9f  |.%&#125;.tp....$.(...|<br>00000310  4c<span class="hljs-number"> 92 </span>85 c0 e2<span class="hljs-number"> 75 </span>bb e5  fb<span class="hljs-number"> 41 </span>ad<span class="hljs-number"> 75 </span>bf b6 1e<span class="hljs-number"> 52 </span> |L....u...A.u...R|<br>00000320 <span class="hljs-number"> 48 </span>48 5c a6 0b<span class="hljs-number"> 03 </span>be<span class="hljs-number"> 39 </span> d7<span class="hljs-number"> 07 </span>35<span class="hljs-number"> 39 </span>f0<span class="hljs-number"> 50 </span>cc af  |HH\....9..59.P..|<br>00000330  b0<span class="hljs-number"> 10 </span>01 4f<span class="hljs-number"> 84 </span>fb f0 bc  bb 9a<span class="hljs-number"> 33 </span>d9 cd<span class="hljs-number"> 47 </span>1c f0  |...O......3..G..|<br>00000340  3c d8<span class="hljs-number"> 73 </span>5c<span class="hljs-number"> 56 </span>88<span class="hljs-number"> 14 </span>d9  f3<span class="hljs-number"> 75 </span>24<span class="hljs-number"> 72 </span>81<span class="hljs-number"> 70 </span>9c 2d  |&lt;.s\V....u$r.p.-|<br>00000350  a2 a4<span class="hljs-number"> 91 </span>c2 8f 4a<span class="hljs-number"> 48 </span>bd  8a 3a f5 f7<span class="hljs-number"> 62 </span>e6 8b<span class="hljs-number"> 94 </span> |.....JH..:..b...|<br>00000360  ee 0a d7 a3 5b<span class="hljs-number"> 41 </span>bc b5  e7<span class="hljs-number"> 09 </span>f0<span class="hljs-number"> 40 </span>94<span class="hljs-number"> 14 </span>73 3e  |....[A.....@..s&gt;|<br>00000370 <span class="hljs-number"> 09 </span>c1<span class="hljs-number"> 69 </span>3f cf dd 0a 9d <span class="hljs-number"> 19 </span>07 fb e6 0a 0e f6 c5  |..i?............|<br>00000380  0b 4e<span class="hljs-number"> 61 </span>22<span class="hljs-number"> 66 </span>64 c5<span class="hljs-number"> 95 </span> af e3<span class="hljs-number"> 77 </span>87 3d c7 e9<span class="hljs-number"> 31 </span> |.Na&quot;fd....w.=..1|<br>00000390 <span class="hljs-number"> 26 </span>34<span class="hljs-number"> 66 </span>b7<span class="hljs-number"> 44 </span>c2<span class="hljs-number"> 50 </span>46 <span class="hljs-number"> 80 </span>a9<span class="hljs-number"> 58 </span>94 b2<span class="hljs-number"> 85 </span>8e d8  |&amp;4f.D.PF..X.....|<br>000003a0  cd<span class="hljs-number"> 64 </span>58 1c<span class="hljs-number"> 10 </span>29 ac d4 <span class="hljs-number"> 11 </span>4a d0 2a ec 3a<span class="hljs-number"> 60 </span>b6  |.dX..)...J.*.:`.|<br>000003b0  af<span class="hljs-number"> 29 </span>29<span class="hljs-number"> 27 </span>2c<span class="hljs-number"> 10 </span>28 b0 <span class="hljs-number"> 01 </span>d1 b1 4c 7c cc 4a a2  |.))&#x27;,.(....L|.J.|<br>000003c0  f2 a7 b0 5d ea<span class="hljs-number"> 89 </span>48 6d <span class="hljs-number"> 07 </span>02 c4 d1<span class="hljs-number"> 97 </span>35 f7 af  |...]..Hm.....5..|<br>000003d0 <span class="hljs-number"> 47 </span>9a f1 b7<span class="hljs-number"> 57 </span>64 bf<span class="hljs-number"> 50 </span> d0<span class="hljs-number"> 64 </span>49 5c<span class="hljs-number"> 79 </span>5e bc e4  |G...Wd.P.dI\y^..|<br>000003e0  ca f8<span class="hljs-number"> 62 </span>4f aa<span class="hljs-number"> 93 </span>77<span class="hljs-number"> 87 </span> 1d<span class="hljs-number"> 81 </span>8d cf<span class="hljs-number"> 38 </span>c6<span class="hljs-number"> 63 </span>a7  |..bO..w.....8.c.|<br>000003f0  a3 6c a1 b2 6c 7b bd b0  0f<span class="hljs-number"> 21 </span>d3<span class="hljs-number"> 58 </span>4f 6d<span class="hljs-number"> 89 </span>d2  |.l..l&#123;...!.XOm..|<br>00000400 <span class="hljs-number"> 58 </span>d9<span class="hljs-number"> 01 </span>9c<span class="hljs-number"> 59 </span>01 a6 cb  b3<span class="hljs-number"> 65 </span>b1<span class="hljs-number"> 65 </span>0c 2e 2a c7  |X...Y....e.e..*.|<br>00000410  c7<span class="hljs-number"> 38 </span>eb<span class="hljs-number"> 32 </span>d9<span class="hljs-number"> 27 </span>3b<span class="hljs-number"> 99 </span> a6 ca<span class="hljs-number"> 76 </span>d8 e0 5a df b5  |.8.2.&#x27;;...v..Z..|<br>00000420  9d a6 fa d2 8d<span class="hljs-number"> 99 </span>b4 c0 <span class="hljs-number"> 32 </span>d0<span class="hljs-number"> 28 </span>41<span class="hljs-number"> 20 </span>16 9b 4d  |........2.(A ..M|<br>00000430 <span class="hljs-number"> 27 </span>72<span class="hljs-number"> 03 </span>b6 d9<span class="hljs-number"> 74 </span>98 b9  d3 de cc 9d<span class="hljs-number"> 76 </span>7a d8<span class="hljs-number"> 81 </span> |&#x27;r...t......vz..|<br>00000440  da e9 b0<span class="hljs-number"> 51 </span>cf e9 e0<span class="hljs-number"> 70 </span><span class="hljs-number"> 77 </span>7f<span class="hljs-number"> 34 </span>34<span class="hljs-number"> 03 </span>d3 e1<span class="hljs-number"> 72 </span> |...Q...pw.44...r|<br>00000450  c8 fa 5d<span class="hljs-number"> 87 </span>71 5e a2<span class="hljs-number"> 31 </span> f1<span class="hljs-number"> 71 </span>6d<span class="hljs-number"> 15 </span>c4<span class="hljs-number"> 65 </span>01<span class="hljs-number"> 86 </span> |..].q^.1.qm..e..|<br>00000460  0d d8 ad c7<span class="hljs-number"> 53 </span>23 8d dd  aa 8b<span class="hljs-number"> 18 </span>26<span class="hljs-number"> 00 </span>b2 6d d5  |....S<span class="hljs-comment">#.....&amp;..m.|</span><br>00000470  b3<span class="hljs-number"> 43 </span>6c b8 e2<span class="hljs-number"> 02 </span>30<span class="hljs-number"> 56 </span> d8 a3 5a b6 ab<span class="hljs-number"> 84 </span>50 4f  |.Cl...0V..Z...PO|<br>00000480 <span class="hljs-number"> 19 </span>c7<span class="hljs-number"> 81 </span>34<span class="hljs-number"> 70 </span>66 0c 8c <span class="hljs-number"> 51 </span>5a<span class="hljs-number"> 39 </span>b1<span class="hljs-number"> 55 </span>67 bd<span class="hljs-number"> 28 </span> |...4pf..QZ9.Ug.(|<br>00000490 <span class="hljs-number"> 44 </span>5b bf<span class="hljs-number"> 77 </span>d6 ba<span class="hljs-number"> 52 </span>57  cd<span class="hljs-number"> 47 </span>3e f2<span class="hljs-number"> 91 </span>a7 8f f4  |D[.w..RW.G&gt;.....|<br>000004a0  c4 a4 e5 db<span class="hljs-number"> 17 </span>47 5f 7f  fe 9d e5<span class="hljs-number"> 55 </span>41 cd<span class="hljs-number"> 56 </span>bd  |.....G_....UA.V.|<br>000004b0 <span class="hljs-number"> 56 </span>ad 9d 8d 9b f3 1e 6d  bb<span class="hljs-number"> 80 </span>33 7c 3b 1a<span class="hljs-number"> 75 </span>47  |V......m..3|;.uG|<br>000004c0  7c<span class="hljs-number"> 98 </span>e7<span class="hljs-number"> 80 </span>f1<span class="hljs-number"> 90 </span>75 4c <span class="hljs-number"> 77 </span>14<span class="hljs-number"> 62 </span>9f<span class="hljs-number"> 02 </span>a6<span class="hljs-number"> 14 </span>88  ||.....uLw.b.....|<br>000004d0  c6 cc<span class="hljs-number"> 39 </span>e6 1e ca<span class="hljs-number"> 76 </span>9d <span class="hljs-number"> 12 </span>6e 2a 4e<span class="hljs-number"> 55 </span>12<span class="hljs-number"> 43 </span>50  |..9...v..n*NU.CP|<br>000004e0  0c 4e<span class="hljs-number"> 05 </span>f7 8a<span class="hljs-number"> 61 </span>24 8c <span class="hljs-number"> 95 </span>76<span class="hljs-number"> 34 </span>75 cd<span class="hljs-number"> 13 </span>c1<span class="hljs-number"> 29 </span> |.N...a$..v4u...)|<br>000004f0  4e<span class="hljs-number"> 14 </span>49 df e3<span class="hljs-number"> 10 </span>b5<span class="hljs-number"> 01 </span><span class="hljs-number"> 43 </span>66<span class="hljs-number"> 56 </span>d0<span class="hljs-number"> 52 </span>79 4c<span class="hljs-number"> 65 </span> |N.I.....CfV.RyLe|<br>00000500  8b<span class="hljs-number"> 85 </span>ca<span class="hljs-number"> 29 </span>8b<span class="hljs-number"> 50 </span>ac e7 <span class="hljs-number"> 12 </span>c1 b9 e7<span class="hljs-number"> 39 </span>10 b0 0a  |...).P......9...|<br>00000510  e8<span class="hljs-number"> 99 </span>8f c6<span class="hljs-number"> 01 </span>ef 0f 2e <span class="hljs-number"> 49 </span>95 cc 0d<span class="hljs-number"> 73 </span>5f 1f<span class="hljs-number"> 68 </span> |........I...s_.h|<br>00000520  6d f7 2f a5<span class="hljs-number"> 29 </span>b3 b6<span class="hljs-number"> 89 </span> ed 0b<span class="hljs-number"> 65 </span>74 af a8<span class="hljs-number"> 22 </span>95  |m./.).....et..&quot;.|<br>00000530  b5 3c d8<span class="hljs-number"> 64 </span>50 c5 b6<span class="hljs-number"> 96 </span> e2 c7 6c 7a db<span class="hljs-number"> 26 </span>29<span class="hljs-number"> 92 </span> |.&lt;.dP.....lz.&amp;).|<br>00000540  2c<span class="hljs-number"> 92 </span>c6 b6 a6 6c<span class="hljs-number"> 08 </span>ce  a0 5c 3e b5 5d<span class="hljs-number"> 10 </span>da 2e  |,....l...\&gt;.]...|<br>00000550  e3 b5 ed a6 fc<span class="hljs-number"> 12 </span>dd ca  f5<span class="hljs-number"> 68 </span>37 fd ae b7 2f d6  |.........h7.../.|<br>00000560 <span class="hljs-number"> 82 </span>7e fd 9d<span class="hljs-number"> 65 </span>a9<span class="hljs-number"> 92 </span>7a  f5 b4 7a d6 a8 8f 4f c3  |.~..e..z..z...O.|<br>00000570  2b fb 8c fc<span class="hljs-number"> 10 </span>57 f5<span class="hljs-number"> 76 </span> fa<span class="hljs-number"> 21 </span>41<span class="hljs-number"> 59 </span>76 ad<span class="hljs-number"> 43 </span>a2  |+....W.v.!AYv.C.|<br>00000580 <span class="hljs-number"> 70 </span>be d8<span class="hljs-number"> 76 </span>e5<span class="hljs-number"> 00 </span>58 dc  6a<span class="hljs-number"> 71 </span>05 dc<span class="hljs-number"> 58 </span>f5 2e 5c  |p..v..X.jq..X..\|<br>00000590  8c<span class="hljs-number"> 25 </span>50<span class="hljs-number"> 29 </span>5f a6 4a<span class="hljs-number"> 85 </span><span class="hljs-number"> 12 </span>eb<span class="hljs-number"> 15 </span>26<span class="hljs-number"> 64 </span>d2<span class="hljs-number"> 30 </span>14  |.%P)_.J....&amp;d.0.|<br>000005a0  0c b7<span class="hljs-number"> 63 </span>56<span class="hljs-number"> 60 </span>c8 a8<span class="hljs-number"> 62 </span><span class="hljs-number"> 80 </span>89 0c 9f<span class="hljs-number"> 69 </span>db<span class="hljs-number"> 82 </span>aa  |..cV`..b....i...|<br>000005b0  a5<span class="hljs-number"> 56 </span>a9 cc 0b b6 b6<span class="hljs-number"> 02 </span> a1<span class="hljs-number"> 13 </span>a0 1b 2b 5c c7 5f  |.V..........+\._|<br>000005c0  7c<span class="hljs-number"> 67 </span>59 f1 b6 de 3c ad  d7 cf c6 a3 6e<span class="hljs-number"> 57 </span>b7<span class="hljs-number"> 64 </span> ||gY...&lt;.....nW.d|<br>000005d0 <span class="hljs-number"> 86 </span>6b b2 4c ae<span class="hljs-number"> 17 </span>71 6e  db<span class="hljs-number"> 77 </span>8c 5d 2f<span class="hljs-number"> 62 </span>a5 ab  |.k.L..qn.w.]/b..|<br>000005e0 <span class="hljs-number"> 33 </span>ef 2a<span class="hljs-number"> 75 </span>2a 6c d3 3a  6e 8f 4d<span class="hljs-number"> 15 </span>b3 da e0 d8  |3.*u*l.:n.M.....|<br>000005f0  ad<span class="hljs-number"> 43 </span>68<span class="hljs-number"> 56 </span>4e 4b c7<span class="hljs-number"> 83 </span><span class="hljs-number"> 70 </span>01<span class="hljs-number"> 45 </span>7c a5 cc<span class="hljs-number"> 96 </span>6b  |.ChVNK..p.E|...k|<br>00000600 <span class="hljs-number"> 80 </span>a9 0b 1c<span class="hljs-number"> 30 </span>c6<span class="hljs-number"> 46 </span>00 <span class="hljs-number"> 15 </span>70 dc c2 b1 9e<span class="hljs-number"> 52 </span>5e  |....0.F..p....R^|<br>00000610  f7 7a<span class="hljs-number"> 53 </span>a0 db 2b 2c fa  e7 2f<span class="hljs-number"> 16 </span>36 4e da<span class="hljs-number"> 67 </span>a7  |.zS..+,../.6N.g.|<br>00000620  b5 7a e6 3a ae<span class="hljs-number"> 65 </span>cc dd <span class="hljs-number"> 73 </span>a7 d3 0d<span class="hljs-number"> 85 </span>e2 a8 0e  |.z.:.e..s.......|<br>00000630  8c 4d<span class="hljs-number"> 26 </span>06<span class="hljs-number"> 05 </span>d1<span class="hljs-number"> 62 </span>b0  6f<span class="hljs-number"> 29 </span>b0 6f<span class="hljs-number"> 11 </span>55<span class="hljs-number"> 70 </span>13  |.M&amp;...b.o).o.Up.|<br>00000640  eb<span class="hljs-number"> 77 </span>c5<span class="hljs-number"> 58 </span>e8 a1 5c f6 <span class="hljs-number"> 89 </span>56 e3 5c<span class="hljs-number"> 27 </span>ae<span class="hljs-number"> 32 </span>4e  |.w.X..\..V.\&#x27;.2N|<br>00000650  a3<span class="hljs-number"> 08 </span>65 d1 0c b3 5b<span class="hljs-number"> 35 </span><span class="hljs-number"> 70 </span>f7<span class="hljs-number"> 37 </span>b3<span class="hljs-number"> 63 </span>dc f0 2b  |..e...[5p.7.c..+|<br>00000660 <span class="hljs-number"> 18 </span>c7<span class="hljs-number"> 82 </span>73 a2<span class="hljs-number"> 04 </span>5f 4f  a0 5f 9c<span class="hljs-number"> 25 </span>8d bf 7b<span class="hljs-number"> 85 </span> |...s.._O._.%..&#123;.|<br>00000670 <span class="hljs-number"> 45 </span>ff 0c<span class="hljs-number"> 00 </span>fd 4b 4b<span class="hljs-number"> 65 </span> e4 b3<span class="hljs-number"> 56 </span>eb<span class="hljs-number"> 64 </span>5c<span class="hljs-number"> 46 </span>be  |E....KKe..V.d\F.|<br>00000680  2e cc<span class="hljs-number"> 79 </span>16<span class="hljs-number"> 31 </span>2c<span class="hljs-number"> 14 </span>63 <span class="hljs-number"> 43 </span>19<span class="hljs-number"> 90 </span>08 6c<span class="hljs-number"> 41 </span>07<span class="hljs-number"> 18 </span> |..y.1,.cC...lA..|<br>00000690  2b<span class="hljs-number"> 70 </span>f2 d9<span class="hljs-number"> 45 </span>d2<span class="hljs-number"> 76 </span>7b <span class="hljs-number"> 11 </span>a0 4a 8d<span class="hljs-number"> 04 </span>9a<span class="hljs-number"> 17 </span>33  |+p..E.v&#123;..J....3|<br>000006a0  6d<span class="hljs-number"> 67 </span>61<span class="hljs-number"> 70 </span>0e<span class="hljs-number"> 00 </span>d0 2c <span class="hljs-number"> 53 </span>4c 7a<span class="hljs-number"> 26 </span>40<span class="hljs-number"> 10 </span>31<span class="hljs-number"> 09 </span> |mgap...,SLz&amp;@.1.|<br>000006b0  8a<span class="hljs-number"> 29 </span>7b 6e b2 bc<span class="hljs-number"> 73 </span>aa <span class="hljs-number"> 65 </span>b9 9c<span class="hljs-number"> 09 </span>b9 db f1 8c  |.)&#123;n..s.e.......|<br>000006c0 <span class="hljs-number"> 75 </span>07 ba b3 e2<span class="hljs-number"> 30 </span>44<span class="hljs-number"> 11 </span> 0e b3<span class="hljs-number"> 74 </span>18 ce<span class="hljs-number"> 09 </span>6e 5c  |u....0D...t...n\|<br>000006d0 <span class="hljs-number"> 13 </span>65<span class="hljs-number"> 70 </span>27 d9<span class="hljs-number"> 38 </span>dc<span class="hljs-number"> 78 </span><span class="hljs-number"> 96 </span>c8 a6 df<span class="hljs-number"> 11 </span>1c 5c<span class="hljs-number"> 09 </span> |.ep&#x27;.8.x......\.|<br>000006e0 <span class="hljs-number"> 69 </span>8b<span class="hljs-number"> 18 </span>65 7b<span class="hljs-number"> 40 </span>1e<span class="hljs-number"> 57 </span><span class="hljs-number"> 80 </span>fd<span class="hljs-number"> 99 </span>63 7b<span class="hljs-number"> 56 </span>aa 0a  |i..e&#123;@.W...c&#123;V..|<br>000006f0  5a ad<span class="hljs-number"> 55 </span>0c c4 d1<span class="hljs-number"> 22 </span>6c  f3 c5<span class="hljs-number"> 45 </span>bd<span class="hljs-number"> 49 </span>36 2d 0e  |Z.U...&quot;l..E.I6-.|<br>00000700  8c 5b e9 d4 2e<span class="hljs-number"> 81 </span>c4 a5  5e<span class="hljs-number"> 98 </span>ad 8f d6<span class="hljs-number"> 74 </span>c8 a5  |.[......^....t..|<br>00000710  3b 1d<span class="hljs-number"> 09 </span>f8 ec db<span class="hljs-number"> 17 </span>55 <span class="hljs-number"> 00 </span>51 a4 ce 5b 8c<span class="hljs-number"> 58 </span>26  |;......U.Q..[.X&amp;|<br>00000720 <span class="hljs-number"> 82 </span>74 ce ed ac<span class="hljs-number"> 46 </span>b7<span class="hljs-number"> 32 </span><span class="hljs-number"> 48 </span>31<span class="hljs-number"> 17 </span>bc<span class="hljs-number"> 76 </span>7f e6 b5  |.t...F.2H1..v...|<br>00000730  9f<span class="hljs-number"> 30 </span>55 b4 f4<span class="hljs-number"> 88 </span>02 9c <span class="hljs-number"> 46 </span>2c e5 ef 6e b9 e4 ee  |.0U.....F,..n...|<br>00000740 <span class="hljs-number"> 91 </span>bf                                             |..|<br>00000742<br></code></pre></td></tr></table></figure>

<p>seems 28 byte empty BGZF block as an EOF marker is absent</p>
<p>Try </p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs shell">samtools view -b -q 20 HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.20130415.bam &gt; HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.20130415.new.bam<br></code></pre></td></tr></table></figure>

<p>return</p>
<figure class="highlight applescript"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs applescript">[W::bam_hdr_read] EOF marker <span class="hljs-keyword">is</span> absent. The input <span class="hljs-keyword">is</span> probably truncated<br>[E::bgzf_read_block] Failed <span class="hljs-keyword">to</span> <span class="hljs-built_in">read</span> BGZF block data <span class="hljs-keyword">at</span> <span class="hljs-built_in">offset</span> <span class="hljs-number">4612181361</span> expected <span class="hljs-number">18560</span> bytes; hread returned <span class="hljs-number">12925</span><br>[E::bgzf_read] Read block operation failed <span class="hljs-keyword">with</span> <span class="hljs-keyword">error</span> <span class="hljs-number">4</span> <span class="hljs-keyword">after</span> <span class="hljs-number">0</span> <span class="hljs-keyword">of</span> <span class="hljs-number">4</span> bytes<br>[main_samview] truncated <span class="hljs-built_in">file</span>.<br></code></pre></td></tr></table></figure>

<p>then try to use HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.20130415.new.bam by samtools index:</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs shell">samtools index HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.20130415.new.bam<br></code></pre></td></tr></table></figure>

<p>(No error return!)</p>
<p>try GATK</p>
<figure class="highlight gradle"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs gradle"><span class="hljs-regexp">/data/m</span>ouse<span class="hljs-regexp">/human_rme/g</span>enome_index<span class="hljs-regexp">/gatk-4.1.9.0/g</span>atk --java-<span class="hljs-keyword">options</span> <span class="hljs-string">&quot;-Xmx4g&quot;</span> HaplotypeCaller  \<br>-R <span class="hljs-regexp">/data/m</span>ouse<span class="hljs-regexp">/human_rme/g</span>enome_index/human_g1k_v37.fasta \<br>-I HG00421.mapped.ILLUMINA.bwa.CHS.low_coverage.<span class="hljs-number">20130415</span>.<span class="hljs-keyword">new</span>.bam \<br>-O <span class="hljs-regexp">/data/m</span>ouse<span class="hljs-regexp">/human_rme/</span>SH007/HG00421_new.vcf.gz \<br>-ERC GVCF <br></code></pre></td></tr></table></figure>



<p>mapping by STAR (没有gtf文件时的做法：)</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><code class="hljs shell">STAR --runThreadN 6 --runMode genomeGenerate \<br>--genomeDir /data/mouse/human_rme/genome_index \<br>--genomeFastaFiles /data/mouse/human_rme/genome_index/human_g1k_v37.fasta #generate index<br><br>STAR --genomeDir /data/mouse/human_rme/genome_index \<br>    --runThreadN 20 \<br>    --readFilesIn sample_r1.fq.gz sample_r2.fq.gz \<br>    # --readFilesCommand zcat \<br>    --outFileNamePrefix sample \<br>    --outSAMtype BAM SortedByCoordinate \<br>    --outBAMsortingThreadN 10 #do mapping<br></code></pre></td></tr></table></figure>



<p>12.2 进度：</p>
<p><strong>SH007</strong></p>
<p>genotypeGVCF done HG00422  (5110)</p>
<p>genotypeGVCF HG00421  ready</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs shell"><span class="hljs-meta">#</span><span class="bash">GATK genotypeGVCF</span><br>/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk --java-options &quot;-Xmx4g&quot; GenotypeGVCFs \<br>-R /data/mouse/human_rme/genome_index/human_g1k_v37.fasta \<br>-V HG00422_new.vcf.gz \<br>-O HG00422_genotypeGVCF.vcf.gz<br></code></pre></td></tr></table></figure>



<p>HG00423 can’t do gatk missing RG </p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs shell">samtools view -H HG00444Aligned.sortedByCoord.out.bam | grep &#x27;^@RG&#x27;<br></code></pre></td></tr></table></figure>

<p>return : none！</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs shell">java -jar /data/mouse/human_rme/genome_index/picard.jar <br></code></pre></td></tr></table></figure>

<p>try AddOrReplaceReadGroups</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><code class="hljs shell">java -jar /data/mouse/human_rme/genome_index/picard.jar AddOrReplaceReadGroups \<br>I=HG00423Aligned.sortedByCoord.out.bam \<br>O=HG00423Aligned.sortedByCoord.out.new.bam \<br>RGID=4 \<br>RGLB=lib1 \<br>RGPL=illumina \<br>RGPU=unit1 \<br>RGSM=20<br></code></pre></td></tr></table></figure>

<p>then try samtools index</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs shell">samtools index HG00423Aligned.sortedByCoord.out.new.bam<br></code></pre></td></tr></table></figure>

<p>(it seems work!!!) </p>
<p>gatk doing HG00423 65457 (shuting down)</p>
<figure class="highlight reasonml"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br></pre></td><td class="code"><pre><code class="hljs reasonml">/data/mouse/human_rme/genome_index/gatk-<span class="hljs-number">4.1</span>.<span class="hljs-number">9.0</span>/gatk --java-options <span class="hljs-string">&quot;-Xmx4g&quot;</span> HaplotypeCaller  \<br>-R /data/mouse/human_rme/genome_index/human_g1k_v37.fasta \<br>-I <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">HG00423Aligned</span>.</span></span>sortedByCoord.out.<span class="hljs-keyword">new</span>.bam \<br>-O /data/mouse/human_rme/SH007/<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">HG00423_new</span>.</span></span>vcf.gz \<br>-ERC GVCF<br><br># return<br>Using GATK jar /data/mouse/human_rme/genome_index/gatk-<span class="hljs-number">4.1</span>.<span class="hljs-number">9.0</span>/gatk-package-<span class="hljs-number">4.1</span>.<span class="hljs-number">9.0</span>-local.jar<br>Running:<br>    java -<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Dsamjdk</span>.</span></span>use_async_io_read_samtools=<span class="hljs-literal">false</span> -<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Dsamjdk</span>.</span></span>use_async_io_write_samtools=<span class="hljs-literal">true</span> -<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Dsamjdk</span>.</span></span>use_async_io_write_tribble=<span class="hljs-literal">false</span> -<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Dsamjdk</span>.</span></span>compression_level=<span class="hljs-number">2</span> -Xmx4g -jar /data/mouse/human_rme/genome_index/gatk-<span class="hljs-number">4.1</span>.<span class="hljs-number">9.0</span>/gatk-package-<span class="hljs-number">4.1</span>.<span class="hljs-number">9.0</span>-local.jar HaplotypeCaller -R /data/mouse/human_rme/genome_index/human_g1k_v37.fasta -I <span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">HG00423Aligned</span>.</span></span>sortedByCoord.out.<span class="hljs-keyword">new</span>.bam -O /data/mouse/human_rme/SH007/<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">HG00423_new</span>.</span></span>vcf.gz -ERC GVCF<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.494</span> INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/mouse/human_rme/genome_index/gatk-<span class="hljs-number">4.1</span>.<span class="hljs-number">9.0</span>/gatk-package-<span class="hljs-number">4.1</span>.<span class="hljs-number">9.0</span>-local.jar!/com/intel/gkl/native/libgkl_compression.so<br>Dec <span class="hljs-number">02</span>, <span class="hljs-number">2020</span> <span class="hljs-number">3</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05</span> PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine<br>INFO: Failed <span class="hljs-keyword">to</span> detect whether we are running on Google Compute Engine.<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.775</span> INFO  HaplotypeCaller - ------------------------------------------------------------<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.776</span> INFO  HaplotypeCaller - The Genome Analysis Toolkit (GATK) v4.<span class="hljs-number">1.9</span>.<span class="hljs-number">0</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.776</span> INFO  HaplotypeCaller - For support <span class="hljs-keyword">and</span> documentation go <span class="hljs-keyword">to</span> https:<span class="hljs-comment">//software.broadinstitute.org/gatk/</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.832</span> INFO  HaplotypeCaller - Executing <span class="hljs-keyword">as</span> zhuyufei@bdp-svr05 on Linux v5.<span class="hljs-number">3.1</span>-<span class="hljs-number">1.</span>el7.elrepo.x86_64 amd64<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.832</span> INFO  HaplotypeCaller - Java runtime: OpenJDK <span class="hljs-number">64</span>-Bit Server VM v1.<span class="hljs-number">8.0_152</span>-release-<span class="hljs-number">1056</span>-b12<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.832</span> INFO  HaplotypeCaller - Start Date/Time: <span class="hljs-number">2020</span>年<span class="hljs-number">12</span>月<span class="hljs-number">2</span>日 下午<span class="hljs-number">03</span>时<span class="hljs-number">21</span>分<span class="hljs-number">05</span>秒<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.832</span> INFO  HaplotypeCaller - ------------------------------------------------------------<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.832</span> INFO  HaplotypeCaller - ------------------------------------------------------------<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.833</span> INFO  HaplotypeCaller - HTSJDK Version: <span class="hljs-number">2.23</span>.<span class="hljs-number">0</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.834</span> INFO  HaplotypeCaller - Picard Version: <span class="hljs-number">2.23</span>.<span class="hljs-number">3</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.834</span> INFO  HaplotypeCaller - HTSJDK Defaults.COMPRESSION_LEVEL : <span class="hljs-number">2</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.834</span> INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : <span class="hljs-literal">false</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.834</span> INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : <span class="hljs-literal">true</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.834</span> INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : <span class="hljs-literal">false</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.834</span> INFO  HaplotypeCaller - Deflater: IntelDeflater<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.834</span> INFO  HaplotypeCaller - Inflater: IntelInflater<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.834</span> INFO  HaplotypeCaller - GCS max retries/reopens: <span class="hljs-number">20</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.835</span> INFO  HaplotypeCaller - Requester pays: disabled<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">05.835</span> INFO  HaplotypeCaller - Initializing engine<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.227</span> INFO  HaplotypeCaller - Done initializing engine<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.228</span> INFO  HaplotypeCallerEngine - Tool is <span class="hljs-keyword">in</span> reference confidence mode <span class="hljs-keyword">and</span> the annotation, the following changes will be made <span class="hljs-keyword">to</span> any specified annotations: &#x27;StrandBiasBySample&#x27; will be enabled. &#x27;ChromosomeCounts&#x27;, &#x27;FisherStrand&#x27;, &#x27;StrandOddsRatio&#x27; <span class="hljs-keyword">and</span> &#x27;QualByDepth&#x27; annotations have been disabled<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.235</span> INFO  HaplotypeCallerEngine - Standard Emitting <span class="hljs-keyword">and</span> Calling confidence set <span class="hljs-keyword">to</span> <span class="hljs-number">0.0</span> <span class="hljs-keyword">for</span> reference-model confidence output<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.235</span> INFO  HaplotypeCallerEngine - All sites annotated <span class="hljs-keyword">with</span> PLs forced <span class="hljs-keyword">to</span> <span class="hljs-literal">true</span> <span class="hljs-keyword">for</span> reference-model confidence output<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.248</span> INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/data/mouse/human_rme/genome_index/gatk-<span class="hljs-number">4.1</span>.<span class="hljs-number">9.0</span>/gatk-package-<span class="hljs-number">4.1</span>.<span class="hljs-number">9.0</span>-local.jar!/com/intel/gkl/native/libgkl_utils.so<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.249</span> INFO  NativeLibraryLoader - Loading libgkl_pairhmm_omp.so from jar:file:/data/mouse/human_rme/genome_index/gatk-<span class="hljs-number">4.1</span>.<span class="hljs-number">9.0</span>/gatk-package-<span class="hljs-number">4.1</span>.<span class="hljs-number">9.0</span>-local.jar!/com/intel/gkl/native/libgkl_pairhmm_omp.so<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.301</span> INFO  IntelPairHmm - Using CPU-supported AVX-<span class="hljs-number">512</span> instructions<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.301</span> INFO  IntelPairHmm - Flush-<span class="hljs-keyword">to</span>-zero (FTZ) is enabled <span class="hljs-keyword">when</span> running PairHMM<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.301</span> INFO  IntelPairHmm - Available threads: <span class="hljs-number">64</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.301</span> INFO  IntelPairHmm - Requested threads: <span class="hljs-number">4</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.301</span> INFO  PairHMM - Using the OpenMP multi-threaded AVX-accelerated native PairHMM implementation<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.336</span> INFO  ProgressMeter - Starting traversal<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">06.337</span> INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Regions Processed   Regions/Minute<br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">16.341</span> INFO  ProgressMeter -           <span class="hljs-number">1</span>:<span class="hljs-number">11753701</span>              <span class="hljs-number">0.2</span>                 <span class="hljs-number">39180</span>         <span class="hljs-number">235056.5</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">26.339</span> INFO  ProgressMeter -           <span class="hljs-number">1</span>:<span class="hljs-number">28001701</span>              <span class="hljs-number">0.3</span>                 <span class="hljs-number">93340</span>         <span class="hljs-number">279992.0</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">36.340</span> INFO  ProgressMeter -           <span class="hljs-number">1</span>:<span class="hljs-number">42413701</span>              <span class="hljs-number">0.5</span>                <span class="hljs-number">141380</span>         <span class="hljs-number">282731.7</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">46.341</span> INFO  ProgressMeter -           <span class="hljs-number">1</span>:<span class="hljs-number">57197701</span>              <span class="hljs-number">0.7</span>                <span class="hljs-number">190660</span>         <span class="hljs-number">285961.4</span><br><span class="hljs-number">15</span>:<span class="hljs-number">21</span>:<span class="hljs-number">59.455</span> INFO  ProgressMeter -           <span class="hljs-number">1</span>:<span class="hljs-number">70118701</span>              <span class="hljs-number">0.9</span>                <span class="hljs-number">233730</span>         <span class="hljs-number">264017.2</span><br><span class="hljs-number">15</span>:<span class="hljs-number">22</span>:<span class="hljs-number">09.454</span> INFO  ProgressMeter -           <span class="hljs-number">1</span>:<span class="hljs-number">84668701</span>              <span class="hljs-number">1.1</span>                <span class="hljs-number">282230</span>         <span class="hljs-number">268292.2</span><br><span class="hljs-number">15</span>:<span class="hljs-number">22</span>:<span class="hljs-number">19.454</span> INFO  ProgressMeter -           <span class="hljs-number">1</span>:<span class="hljs-number">99485701</span>              <span class="hljs-number">1.2</span>                <span class="hljs-number">331620</span>         <span class="hljs-number">272128.2</span><br><span class="hljs-number">15</span>:<span class="hljs-number">22</span>:<span class="hljs-number">29.454</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">111821701</span>              <span class="hljs-number">1.4</span>                <span class="hljs-number">372740</span>         <span class="hljs-number">269071.3</span><br><span class="hljs-number">15</span>:<span class="hljs-number">22</span>:<span class="hljs-number">39.455</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">125609701</span>              <span class="hljs-number">1.6</span>                <span class="hljs-number">418700</span>         <span class="hljs-number">269786.7</span><br><span class="hljs-number">15</span>:<span class="hljs-number">22</span>:<span class="hljs-number">49.617</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">139055701</span>              <span class="hljs-number">1.7</span>                <span class="hljs-number">463520</span>         <span class="hljs-number">269279.6</span><br><span class="hljs-number">15</span>:<span class="hljs-number">22</span>:<span class="hljs-number">59.643</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">153017701</span>              <span class="hljs-number">1.9</span>                <span class="hljs-number">510060</span>         <span class="hljs-number">270106.4</span><br><span class="hljs-number">15</span>:<span class="hljs-number">23</span>:<span class="hljs-number">09.768</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">163760701</span>              <span class="hljs-number">2.1</span>                <span class="hljs-number">545870</span>         <span class="hljs-number">265350.4</span><br><span class="hljs-number">15</span>:<span class="hljs-number">23</span>:<span class="hljs-number">19.767</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">176396701</span>              <span class="hljs-number">2.2</span>                <span class="hljs-number">587990</span>         <span class="hljs-number">264403.8</span><br><span class="hljs-number">15</span>:<span class="hljs-number">23</span>:<span class="hljs-number">29.983</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">188480701</span>              <span class="hljs-number">2.4</span>                <span class="hljs-number">628270</span>         <span class="hljs-number">262424.3</span><br><span class="hljs-number">15</span>:<span class="hljs-number">23</span>:<span class="hljs-number">39.984</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">201497701</span>              <span class="hljs-number">2.6</span>                <span class="hljs-number">671660</span>         <span class="hljs-number">262288.6</span><br><span class="hljs-number">15</span>:<span class="hljs-number">23</span>:<span class="hljs-number">50.582</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">213197701</span>              <span class="hljs-number">2.7</span>                <span class="hljs-number">710660</span>         <span class="hljs-number">259609.7</span><br><span class="hljs-number">15</span>:<span class="hljs-number">24</span>:<span class="hljs-number">00.582</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">225782701</span>              <span class="hljs-number">2.9</span>                <span class="hljs-number">752610</span>         <span class="hljs-number">259155.8</span><br><span class="hljs-number">15</span>:<span class="hljs-number">24</span>:<span class="hljs-number">10.583</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">236975701</span>              <span class="hljs-number">3.1</span>                <span class="hljs-number">789920</span>         <span class="hljs-number">257240.1</span><br><span class="hljs-number">15</span>:<span class="hljs-number">24</span>:<span class="hljs-number">20.582</span> INFO  ProgressMeter -          <span class="hljs-number">1</span>:<span class="hljs-number">247268701</span>              <span class="hljs-number">3.2</span>                <span class="hljs-number">824230</span>         <span class="hljs-number">254595.0</span><br><span class="hljs-number">15</span>:<span class="hljs-number">24</span>:<span class="hljs-number">46.185</span> INFO  HaplotypeCaller - Shutting down engine<br><span class="hljs-literal">[<span class="hljs-number">2020</span>年<span class="hljs-number">12</span>月<span class="hljs-number">2</span>日 下午<span class="hljs-number">03</span>时<span class="hljs-number">24</span>分<span class="hljs-number">46</span>秒]</span> org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller <span class="hljs-keyword">done</span>. Elapsed time: <span class="hljs-number">3.68</span> minutes.<br><span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Runtime</span>.</span></span>total<span class="hljs-constructor">Memory()</span>=<span class="hljs-number">4291821568</span><br>Exception <span class="hljs-keyword">in</span> thread <span class="hljs-string">&quot;main&quot;</span> java.lang.OutOfMemoryError: Java heap space<br>        at java.util.stream.Nodes$DoubleArrayNode.&lt;init&gt;(<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Nodes</span>.</span></span>java:<span class="hljs-number">1429</span>)<br>        at java.util.stream.Nodes$DoubleFixedNodeBuilder.&lt;init&gt;(<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Nodes</span>.</span></span>java:<span class="hljs-number">1589</span>)<br>        at java.util.stream.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Nodes</span>.</span></span>double<span class="hljs-constructor">Builder(Nodes.<span class="hljs-params">java</span>:279)</span><br>        at java.util.stream.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">DoublePipeline</span>.</span></span>make<span class="hljs-constructor">NodeBuilder(DoublePipeline.<span class="hljs-params">java</span>:164)</span><br>        at java.util.stream.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">AbstractPipeline</span>.</span></span>evaluate(<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">AbstractPipeline</span>.</span></span>java:<span class="hljs-number">543</span>)<br>        at java.util.stream.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">AbstractPipeline</span>.</span></span>evaluate<span class="hljs-constructor">ToArrayNode(AbstractPipeline.<span class="hljs-params">java</span>:260)</span><br>        at java.util.stream.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">DoublePipeline</span>.</span></span><span class="hljs-keyword">to</span><span class="hljs-constructor">Array(DoublePipeline.<span class="hljs-params">java</span>:506)</span><br>        at org.broadinstitute.hellbender.utils.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">MathUtils</span>.</span></span>median(<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">MathUtils</span>.</span></span>java:<span class="hljs-number">841</span>)<br>        at org.broadinstitute.hellbender.utils.variant.writers.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">GVCFBlock</span>.</span></span>get<span class="hljs-constructor">MedianDP(GVCFBlock.<span class="hljs-params">java</span>:75)</span><br>        at org.broadinstitute.hellbender.utils.variant.writers.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">HomRefBlock</span>.</span></span>create<span class="hljs-constructor">HomRefGenotype(HomRefBlock.<span class="hljs-params">java</span>:73)</span><br>        at org.broadinstitute.hellbender.utils.variant.writers.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">GVCFBlock</span>.</span></span><span class="hljs-keyword">to</span><span class="hljs-constructor">VariantContext(GVCFBlock.<span class="hljs-params">java</span>:49)</span><br>        at org.broadinstitute.hellbender.utils.variant.writers.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">GVCFBlockCombiner</span>.</span></span>emit<span class="hljs-constructor">CurrentBlock(GVCFBlockCombiner.<span class="hljs-params">java</span>:177)</span><br>        at org.broadinstitute.hellbender.utils.variant.writers.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">GVCFBlockCombiner</span>.</span></span>signal<span class="hljs-constructor">EndOfInput(GVCFBlockCombiner.<span class="hljs-params">java</span>:227)</span><br>        at org.broadinstitute.hellbender.utils.variant.writers.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">GVCFWriter</span>.</span></span>close(<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">GVCFWriter</span>.</span></span>java:<span class="hljs-number">70</span>)<br>        at org.broadinstitute.hellbender.tools.walkers.haplotypecaller.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">HaplotypeCaller</span>.</span></span>close<span class="hljs-constructor">Tool(HaplotypeCaller.<span class="hljs-params">java</span>:216)</span><br>        at org.broadinstitute.hellbender.engine.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">GATKTool</span>.</span></span><span class="hljs-keyword">do</span><span class="hljs-constructor">Work(GATKTool.<span class="hljs-params">java</span>:1053)</span><br>        at org.broadinstitute.hellbender.cmdline.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">CommandLineProgram</span>.</span></span>run<span class="hljs-constructor">Tool(CommandLineProgram.<span class="hljs-params">java</span>:140)</span><br>        at org.broadinstitute.hellbender.cmdline.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">CommandLineProgram</span>.</span></span>instance<span class="hljs-constructor">MainPostParseArgs(CommandLineProgram.<span class="hljs-params">java</span>:192)</span><br>        at org.broadinstitute.hellbender.cmdline.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">CommandLineProgram</span>.</span></span>instance<span class="hljs-constructor">Main(CommandLineProgram.<span class="hljs-params">java</span>:211)</span><br>        at org.broadinstitute.hellbender.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Main</span>.</span></span>run<span class="hljs-constructor">CommandLineProgram(Main.<span class="hljs-params">java</span>:160)</span><br>        at org.broadinstitute.hellbender.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Main</span>.</span></span>main<span class="hljs-constructor">Entry(Main.<span class="hljs-params">java</span>:203)</span><br>        at org.broadinstitute.hellbender.<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Main</span>.</span></span>main(<span class="hljs-module-access"><span class="hljs-module"><span class="hljs-identifier">Main</span>.</span></span>java:<span class="hljs-number">289</span>)<br><br></code></pre></td></tr></table></figure>





<p><strong>SH014</strong></p>
<p>HG00442 vcf ready</p>
<p>HG00443 vcf ready</p>
<p>gatk HaplotypeCaller doing HG00444 but shutting down (1299)</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br></pre></td><td class="code"><pre><code class="hljs shell">/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk --java-options &quot;-Xmx4g&quot; HaplotypeCaller  \<br>-R /data/mouse/human_rme/genome_index/human_g1k_v37.fasta \<br>-I HG00444Aligned.sortedByCoord.out.new.bam \<br>-O /data/mouse/human_rme/SH014/HG00444_new.vcf.gz \<br>-ERC GVCF<br><span class="hljs-meta"></span><br><span class="hljs-meta">#</span><span class="bash"> <span class="hljs-built_in">return</span></span><br>Using GATK jar /data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar<br>Running:<br>    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx4g -jar /data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar HaplotypeCaller -R /data/mouse/human_rme/genome_index/human_g1k_v37.fasta -I HG00444Aligned.sortedByCoord.out.new.bam -O /data/mouse/human_rme/SH014/HG00444_new.vcf.gz -ERC GVCF<br>15:51:47.944 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar!/com/intel/gkl/native/libgkl_compression.so<br>Dec 02, 2020 3:51:48 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine<br>INFO: Failed to detect whether we are running on Google Compute Engine.<br>15:51:48.116 INFO  HaplotypeCaller - ------------------------------------------------------------<br>15:51:48.116 INFO  HaplotypeCaller - The Genome Analysis Toolkit (GATK) v4.1.9.0<br>15:51:48.117 INFO  HaplotypeCaller - For support and documentation go to https://software.broadinstitute.org/gatk/<br>15:51:48.225 INFO  HaplotypeCaller - Executing as zhuyufei@bdp-svr05 on Linux v5.3.1-1.el7.elrepo.x86_64 amd64<br>15:51:48.225 INFO  HaplotypeCaller - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_152-release-1056-b12<br>15:51:48.225 INFO  HaplotypeCaller - Start Date/Time: 2020年12月2日 下午03时51分47秒<br>15:51:48.225 INFO  HaplotypeCaller - ------------------------------------------------------------<br>15:51:48.225 INFO  HaplotypeCaller - ------------------------------------------------------------<br>15:51:48.226 INFO  HaplotypeCaller - HTSJDK Version: 2.23.0<br>15:51:48.226 INFO  HaplotypeCaller - Picard Version: 2.23.3<br>15:51:48.227 INFO  HaplotypeCaller - HTSJDK Defaults.COMPRESSION_LEVEL : 2<br>15:51:48.227 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false<br>15:51:48.227 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true<br>15:51:48.227 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false<br>15:51:48.227 INFO  HaplotypeCaller - Deflater: IntelDeflater<br>15:51:48.227 INFO  HaplotypeCaller - Inflater: IntelInflater<br>15:51:48.227 INFO  HaplotypeCaller - GCS max retries/reopens: 20<br>15:51:48.228 INFO  HaplotypeCaller - Requester pays: disabled<br>15:51:48.228 INFO  HaplotypeCaller - Initializing engine<br>15:51:48.606 INFO  HaplotypeCaller - Done initializing engine<br>15:51:48.608 INFO  HaplotypeCallerEngine - Tool is in reference confidence mode and the annotation, the following changes will be made to any specified annotations: &#x27;StrandBiasBySample&#x27; will be enabled. &#x27;ChromosomeCounts&#x27;, &#x27;FisherStrand&#x27;, &#x27;StrandOddsRatio&#x27; and &#x27;QualByDepth&#x27; annotations have been disabled<br>15:51:48.615 INFO  HaplotypeCallerEngine - Standard Emitting and Calling confidence set to 0.0 for reference-model confidence output<br>15:51:48.615 INFO  HaplotypeCallerEngine - All sites annotated with PLs forced to true for reference-model confidence output<br>15:51:48.631 INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar!/com/intel/gkl/native/libgkl_utils.so<br>15:51:48.632 INFO  NativeLibraryLoader - Loading libgkl_pairhmm_omp.so from jar:file:/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar!/com/intel/gkl/native/libgkl_pairhmm_omp.so<br>15:51:48.687 INFO  IntelPairHmm - Using CPU-supported AVX-512 instructions<br>15:51:48.687 INFO  IntelPairHmm - Flush-to-zero (FTZ) is enabled when running PairHMM<br>15:51:48.688 INFO  IntelPairHmm - Available threads: 64<br>15:51:48.688 INFO  IntelPairHmm - Requested threads: 4<br>15:51:48.688 INFO  PairHMM - Using the OpenMP multi-threaded AVX-accelerated native PairHMM implementation<br>15:51:48.725 INFO  ProgressMeter - Starting traversal<br>15:51:48.725 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Regions Processed   Regions/Minute<br>15:51:58.727 INFO  ProgressMeter -           1:11600701              0.2                 38670         232020.0<br>15:52:08.726 INFO  ProgressMeter -           1:27983701              0.3                 93280         279826.0<br>15:52:18.727 INFO  ProgressMeter -           1:42827701              0.5                142760         285501.0<br>15:52:28.747 INFO  ProgressMeter -           1:57899701              0.7                193000         289340.9<br>15:52:39.822 INFO  ProgressMeter -           1:70169701              0.9                233900         274654.1<br>15:52:49.822 INFO  ProgressMeter -           1:85727701              1.0                285760         280629.2<br>15:52:59.823 INFO  ProgressMeter -          1:101075701              1.2                336920         284328.7<br>15:53:09.824 INFO  ProgressMeter -          1:113090701              1.4                376970         278896.2<br>15:53:19.906 INFO  ProgressMeter -          1:127433701              1.5                424780         279518.8<br>15:53:29.910 INFO  ProgressMeter -          1:142109701              1.7                473700         280899.8<br>15:53:39.985 INFO  ProgressMeter -          1:155774701              1.9                519250         280019.8<br>15:53:49.996 INFO  ProgressMeter -          1:166265701              2.0                554220         274205.7<br>15:54:00.097 INFO  ProgressMeter -          1:178859701              2.2                596200         272297.5<br>15:54:10.227 INFO  ProgressMeter -          1:191456701              2.4                638190         270608.7<br>15:54:20.303 INFO  ProgressMeter -          1:204050701              2.5                680170         269237.4<br>15:54:30.432 INFO  ProgressMeter -          1:216647701              2.7                722160         267951.3<br>15:54:40.525 INFO  ProgressMeter -          1:229241701              2.9                764140         266870.8<br>15:54:50.526 INFO  ProgressMeter -          1:240734701              3.0                802450         264835.0<br>15:55:26.303 INFO  HaplotypeCaller - Shutting down engine<br>[2020年12月2日 下午03时55分26秒] org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller done. Elapsed time: 3.64 minutes.<br>Runtime.totalMemory()=4291821568<br>Exception in thread &quot;main&quot; java.lang.OutOfMemoryError: Java heap space<br>        at java.util.stream.Nodes$DoubleArrayNode.&lt;init&gt;(Nodes.java:1429)<br>        at java.util.stream.Nodes$DoubleFixedNodeBuilder.&lt;init&gt;(Nodes.java:1589)<br>        at java.util.stream.Nodes.doubleBuilder(Nodes.java:279)<br>        at java.util.stream.DoublePipeline.makeNodeBuilder(DoublePipeline.java:164)<br>        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:543)<br>        at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)<br>        at java.util.stream.DoublePipeline.toArray(DoublePipeline.java:506)<br>        at org.broadinstitute.hellbender.utils.MathUtils.median(MathUtils.java:841)<br>        at org.broadinstitute.hellbender.utils.variant.writers.GVCFBlock.getMedianDP(GVCFBlock.java:75)<br>        at org.broadinstitute.hellbender.utils.variant.writers.HomRefBlock.createHomRefGenotype(HomRefBlock.java:73)<br>        at org.broadinstitute.hellbender.utils.variant.writers.GVCFBlock.toVariantContext(GVCFBlock.java:49)<br>        at org.broadinstitute.hellbender.utils.variant.writers.GVCFBlockCombiner.emitCurrentBlock(GVCFBlockCombiner.java:177)<br>        at org.broadinstitute.hellbender.utils.variant.writers.GVCFBlockCombiner.signalEndOfInput(GVCFBlockCombiner.java:227)<br>        at org.broadinstitute.hellbender.utils.variant.writers.GVCFWriter.close(GVCFWriter.java:70)<br>        at org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller.closeTool(HaplotypeCaller.java:216)<br>        at org.broadinstitute.hellbender.engine.GATKTool.doWork(GATKTool.java:1053)<br>        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:140)<br>        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:192)<br>        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:211)<br>        at org.broadinstitute.hellbender.Main.runCommandLineProgram(Main.java:160)<br>        at org.broadinstitute.hellbender.Main.mainEntry(Main.java:203)<br>        at org.broadinstitute.hellbender.Main.main(Main.java:289)<br></code></pre></td></tr></table></figure>

<p>Solution: java程序溢出</p>
<figure class="highlight diff"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs diff"><span class="hljs-deletion">-Xmx75000M</span><br></code></pre></td></tr></table></figure>



<p><strong>SH021</strong></p>
<p>HG00463 vcf ready</p>
<p>gatk doing HG00464 but shutting down (633) </p>
<figure class="highlight julia"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs julia"><span class="hljs-built_in">Exception</span> <span class="hljs-keyword">in</span> thread <span class="hljs-string">&quot;main&quot;</span> java.lang.<span class="hljs-built_in">OutOfMemoryError</span>: Java heap space<br></code></pre></td></tr></table></figure>



<p>gatk HaplotypeCaller doing HG00465 but shutting down(729)</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br></pre></td><td class="code"><pre><code class="hljs shell">/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk --java-options &quot;-Xmx4g&quot; HaplotypeCaller  \<br>-R /data/mouse/human_rme/genome_index/human_g1k_v37.fasta \<br>-I HG00465Aligned.sortedByCoord.out.new.bam \<br>-O /data/mouse/human_rme/SH021/HG00465_new.vcf.gz \<br>-ERC GVCF<br><span class="hljs-meta"></span><br><span class="hljs-meta">#</span><span class="bash"> <span class="hljs-built_in">return</span></span><br>Using GATK jar /data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar<br>Running:<br>    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx4g -jar /data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar HaplotypeCaller -R /data/mouse/human_rme/genome_index/human_g1k_v37.fasta -I HG00465Aligned.sortedByCoord.out.new.bam -O /data/mouse/human_rme/SH021/HG00465_new.vcf.gz -ERC GVCF<br>15:35:01.927 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar!/com/intel/gkl/native/libgkl_compression.so<br>Dec 02, 2020 3:35:02 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine<br>INFO: Failed to detect whether we are running on Google Compute Engine.<br>15:35:02.092 INFO  HaplotypeCaller - ------------------------------------------------------------<br>15:35:02.092 INFO  HaplotypeCaller - The Genome Analysis Toolkit (GATK) v4.1.9.0<br>15:35:02.093 INFO  HaplotypeCaller - For support and documentation go to https://software.broadinstitute.org/gatk/<br>15:35:02.192 INFO  HaplotypeCaller - Executing as zhuyufei@bdp-svr05 on Linux v5.3.1-1.el7.elrepo.x86_64 amd64<br>15:35:02.192 INFO  HaplotypeCaller - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_152-release-1056-b12<br>15:35:02.192 INFO  HaplotypeCaller - Start Date/Time: 2020年12月2日 下午03时35分01秒<br>15:35:02.192 INFO  HaplotypeCaller - ------------------------------------------------------------<br>15:35:02.193 INFO  HaplotypeCaller - ------------------------------------------------------------<br>15:35:02.194 INFO  HaplotypeCaller - HTSJDK Version: 2.23.0<br>15:35:02.194 INFO  HaplotypeCaller - Picard Version: 2.23.3<br>15:35:02.194 INFO  HaplotypeCaller - HTSJDK Defaults.COMPRESSION_LEVEL : 2<br>15:35:02.194 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false<br>15:35:02.194 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true<br>15:35:02.194 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false<br>15:35:02.195 INFO  HaplotypeCaller - Deflater: IntelDeflater<br>15:35:02.195 INFO  HaplotypeCaller - Inflater: IntelInflater<br>15:35:02.195 INFO  HaplotypeCaller - GCS max retries/reopens: 20<br>15:35:02.195 INFO  HaplotypeCaller - Requester pays: disabled<br>15:35:02.195 INFO  HaplotypeCaller - Initializing engine<br>15:35:02.568 INFO  HaplotypeCaller - Done initializing engine<br>15:35:02.569 INFO  HaplotypeCallerEngine - Tool is in reference confidence mode and the annotation, the following changes will be made to any specified annotations: &#x27;StrandBiasBySample&#x27; will be enabled. &#x27;ChromosomeCounts&#x27;, &#x27;FisherStrand&#x27;, &#x27;StrandOddsRatio&#x27; and &#x27;QualByDepth&#x27; annotations have been disabled<br>15:35:02.576 INFO  HaplotypeCallerEngine - Standard Emitting and Calling confidence set to 0.0 for reference-model confidence output<br>15:35:02.576 INFO  HaplotypeCallerEngine - All sites annotated with PLs forced to true for reference-model confidence output<br>15:35:02.589 INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar!/com/intel/gkl/native/libgkl_utils.so<br>15:35:02.590 INFO  NativeLibraryLoader - Loading libgkl_pairhmm_omp.so from jar:file:/data/mouse/human_rme/genome_index/gatk-4.1.9.0/gatk-package-4.1.9.0-local.jar!/com/intel/gkl/native/libgkl_pairhmm_omp.so<br>15:35:02.642 INFO  IntelPairHmm - Using CPU-supported AVX-512 instructions<br>15:35:02.642 INFO  IntelPairHmm - Flush-to-zero (FTZ) is enabled when running PairHMM<br>15:35:02.642 INFO  IntelPairHmm - Available threads: 64<br>15:35:02.642 INFO  IntelPairHmm - Requested threads: 4<br>15:35:02.642 INFO  PairHMM - Using the OpenMP multi-threaded AVX-accelerated native PairHMM implementation<br>15:35:02.671 INFO  ProgressMeter - Starting traversal<br>15:35:02.672 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Regions Processed   Regions/Minute<br>15:35:12.675 INFO  ProgressMeter -           1:13535701              0.2                 45120         270720.0<br>15:35:22.673 INFO  ProgressMeter -           1:28421701              0.3                 94740         284220.0<br>15:35:32.672 INFO  ProgressMeter -           1:42599701              0.5                142000         284000.0<br>15:35:42.673 INFO  ProgressMeter -           1:56069701              0.7                186900         280343.0<br>15:35:59.292 INFO  ProgressMeter -           1:70610701              0.9                235370         249425.1<br>15:36:09.292 INFO  ProgressMeter -           1:83174701              1.1                277250         249699.8<br>15:36:19.293 INFO  ProgressMeter -           1:97466701              1.3                324890         254413.3<br>15:36:29.294 INFO  ProgressMeter -          1:108578701              1.4                361930         250696.1<br>15:36:39.295 INFO  ProgressMeter -          1:121907701              1.6                406360         252337.4<br>15:36:49.407 INFO  ProgressMeter -          1:135599701              1.8                452000         254087.2<br>15:36:59.411 INFO  ProgressMeter -          1:148913701              1.9                496380         255131.7<br>15:37:10.510 INFO  ProgressMeter -          1:157865701              2.1                526220         246978.2<br>15:37:20.743 INFO  ProgressMeter -          1:168830701              2.3                562770         244556.8<br>15:37:30.823 INFO  ProgressMeter -          1:180791701              2.5                602640         244064.5<br>15:37:40.934 INFO  ProgressMeter -          1:192755701              2.6                642520         243591.0<br>15:37:50.934 INFO  ProgressMeter -          1:205016701              2.8                683390         243687.8<br>15:38:00.935 INFO  ProgressMeter -          1:217073701              3.0                723580         243543.5<br>15:38:10.935 INFO  ProgressMeter -          1:228584701              3.1                761950         242835.8<br>15:38:20.935 INFO  ProgressMeter -          1:239054701              3.3                796850         241149.4<br>15:38:49.666 INFO  HaplotypeCaller - Shutting down engine<br>[2020年12月2日 下午03时38分49秒] org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller done. Elapsed time: 3.80 minutes.<br>Runtime.totalMemory()=4292345856<br>Exception in thread &quot;main&quot; java.lang.OutOfMemoryError: Java heap space<br>        at java.util.stream.Nodes$DoubleArrayNode.&lt;init&gt;(Nodes.java:1429)<br>        at java.util.stream.Nodes$DoubleFixedNodeBuilder.&lt;init&gt;(Nodes.java:1589)<br>        at java.util.stream.Nodes.doubleBuilder(Nodes.java:279)<br>        at java.util.stream.DoublePipeline.makeNodeBuilder(DoublePipeline.java:164)<br>        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:543)<br>        at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)<br>        at java.util.stream.DoublePipeline.toArray(DoublePipeline.java:506)<br>        at org.broadinstitute.hellbender.utils.MathUtils.median(MathUtils.java:841)<br>        at org.broadinstitute.hellbender.utils.variant.writers.GVCFBlock.getMedianDP(GVCFBlock.java:75)<br>        at org.broadinstitute.hellbender.utils.variant.writers.HomRefBlock.createHomRefGenotype(HomRefBlock.java:73)<br>        at org.broadinstitute.hellbender.utils.variant.writers.GVCFBlock.toVariantContext(GVCFBlock.java:49)<br>        at org.broadinstitute.hellbender.utils.variant.writers.GVCFBlockCombiner.emitCurrentBlock(GVCFBlockCombiner.java:177)<br>        at org.broadinstitute.hellbender.utils.variant.writers.GVCFBlockCombiner.signalEndOfInput(GVCFBlockCombiner.java:227)<br>        at org.broadinstitute.hellbender.utils.variant.writers.GVCFWriter.close(GVCFWriter.java:70)<br>        at org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller.closeTool(HaplotypeCaller.java:216)<br>        at org.broadinstitute.hellbender.engine.GATKTool.doWork(GATKTool.java:1053)<br>        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:140)<br>        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:192)<br>        at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:211)<br>        at org.broadinstitute.hellbender.Main.runCommandLineProgram(Main.java:160)<br>        at org.broadinstitute.hellbender.Main.mainEntry(Main.java:203)<br>        at org.broadinstitute.hellbender.Main.main(Main.java:289)<br></code></pre></td></tr></table></figure>



<p><strong>SH002</strong></p>
<p>gatk HaplotypeCaller doing HG00407 (1060)</p>
<p>gatk HaplotypeCaller doing HG00406(1182) (but shutting down at Current Locus 1:246638732; try agin then shutting Current Locus down 1:241940732)</p>
<p>gatk HaplotypeCaller doing HG00408 (1455) (but shutting down at Current Locus 1:240860701 )</p>
<h1 id="imprinting-gene"><a href="#imprinting-gene" class="headerlink" title="imprinting gene"></a>imprinting gene</h1><p>基因铭印\parencite{reik2001genomic}是在真兽类哺乳动物中观察到的一种表观遗传现象。对于大多数常染色体基因，两个亲本拷贝都是转录或沉默。然而，在一小组基因中，其中一个拷贝以亲本特有的方式被关闭，从而导致单等位基因表达。这些基因被称为“印迹”，因为沉默的基因拷贝在卵子或精子中具有表观遗传标记或印迹。</p>
<p>印迹基因在胎儿和胎盘组织中发挥重要作用，在产前和产后的发育和生长中发挥重要作用\parencite{morison2005census}。有趣的是，母方表达的基因会限制胚胎的生长，而父方表达的基因会促进胚胎的生长。Igf2和 Igf2r在小鼠体内的拮抗作用是这种突出场景的一个典型案例。父方表达Igf2基因的缺失导致了宫内生长迟缓。另一方面，缺失母方表达的Igf2r基因，导致过度生长\parencite{lau1994loss}。</p>
<p>母方和父方表达的基因的拮抗作用引发了一系列进化理论的争论，旨在解释在“自然选择”过程中遗传印记的起源。目前最为科学接受的理论是亲属理论\parencite{haig1989selective, moore1991genomic}。简单地说，这个理论认为在多配偶的哺乳动物物种中，沉默来自母系的生长抑制基因可以导致胚胎的生长。这与营养需求增加有关，从而导致以后代为代价开发母亲资源，而后代可能是另一个男性的后代。</p>
<p>一个基因调控机制的进化优先沉默一个基因的亲本等位基因意味着父方和母方表达基因在进化过程中经历不同的选择压力。这一假设得到了两个群体揭示了不同的序列保守模式的支持。父系表达基因的蛋白质编码 DNA 序列在不同哺乳动物中保存得很好，而母系表达基因则差异很大。是否父方和母方表达的基因在分子功能和基因调控上也不同，这是一个尚未详细研究的问题。许多研究表明，印迹基因不仅在胚胎发育过程中起重要作用，而且还具有出生后的功能。因此，以产前发育为中心的亲属理论或许可以解释基因铭印进化的某些方面，但不是全部\parencite{hamed2012cellular}。</p>
<p>在出生后的发育过程中，基因铭印会影响内分泌网络、能量代谢和行为。对小鼠的敲除研究表明，表达 Peg1和 Peg3基因的两个父方基因具有明显的行为表型\parencite{lefebvre1998abnormal}。从父亲那里遗传到这些基因的非等位基因的雌性表现出缺乏母性照顾行为，包括吞食胎盘和筑巢以及幼崽聚集。</p>
<p>另一个有趣的事实是，胎盘哺乳动物如老鼠和人类，以及有袋动物如负鼠和袋鼠，都有基因铭印。卵生哺乳动物，如鸭嘴兽和针鼹鼠，似乎缺乏印记基因。胎盘哺乳动物和有袋类动物区别于卵生哺乳动物的一个生殖策略是允许胚胎直接影响用于自身生长的母性资源的数量。相反，在卵子中发育的胚胎不能直接影响母体资源。大多数无脊椎动物和脊椎动物使用产卵繁殖策略。值得注意的是，它们也可以进行孤雌生殖ー一种繁殖形式，即雌配子不经过雄配子受精而发育成一个新的二倍体个体(注意，孤雌生殖胚胎来自同一母体基因组的复制，而图2中描述的雌核发育胚胎来自两个不同的母体基因组)。生物体进行单性生殖的能力很可能表明基因铭印的完全缺失，因为这表明父系基因组是可有可无的。然而，在哺乳动物中，印迹基因表达控制胎儿生长的直接结果是孤雌生殖是不可能的。双亲都必须产生可存活的后代，使哺乳动物完全依赖有性生殖进行繁殖。因此，哺乳动物几乎不存在孤雌生殖现象\parencite{renfree2009evolution}。</p>
<p>为什么基因铭印只在一些哺乳动物中进化，而在脊椎动物中却没有？基因组印记的三个特征ーー许多印记基因的生长调节功能，印记基因对胎盘哺乳动物和有袋哺乳动物的限制，以及父系基因组对胎儿发育的必要性，为两个同样有吸引力的假说提供了证据。</p>
<p>第一个假设提出，基因铭印的进化是为了回应“父母冲突”的情况\parencite{moore1991genomic}。这源于母体和父体基因组的对立利益: 胚胎的发育依赖于父母一方，但是受到胚胎的影响，胚胎的基因组来自父母双方。父方表达的印记基因被认为可以促进胚胎发育，从而最大限度地提高拥有特定父方基因组的个体后代的适应性。母方表达的印迹基因被认为可以抑制胎儿的生长。这将使得母系资源更平等地分配给所有后代，并增加母系基因组向多个后代的传递，这些后代可能有不同的父系基因组。</p>
<p>第二个假设被称为“滋养层防御”\parencite{varmuza1994genomic}。这就提出，如果自发的卵母细胞激活导致胚胎的完全发育，那么母体基因组就有可能因为具备内部繁殖的解剖学条件而面临风险。由于男性缺乏必要的内部繁殖解剖设备，他们不共享相同的风险，应自发激活精子发生。因此，印记被认为可以抑制母染色体上促进胎盘发育的基因，或者激活限制这一过程的基因。因此，胎盘侵入母体子宫血管所必需的基因只能在受精后由父体基因组表达。</p>
<p>这两种假说都指出印迹基因在调节胎盘发育和功能方面的作用，然而，无论是亲代冲突还是滋养层防御模型都不能为所有的数据提供完整的解释\parencite{wilkins2003good}。有趣的是，植物胚乳中也发现了印迹基因，这种组织被比作胎盘，因为它将营养资源从亲本植物转移到胚胎中\parencite{grossniklaus2014transcriptional}。这一发现加强了关于基因铭印是作为调节父母和子女之间营养物质转移的手段进化而来的论点。有可能的是，并不是一个集群中的所有基因都是刻意为之的印记机制的目标，有些基因可能只是这个过程的“无辜旁观者”，而且它们的功能也不能提供有关基因铭印的信息。受印迹机制影响的无辜旁观者基因的存在可以令人满意地解释印迹基因的奇特丰富性，但在发育过程中没有明显的生物学功能\parencite{bartolomei1997genomic}。</p>
<h2 id="method-design"><a href="#method-design" class="headerlink" title="method design"></a>method design</h2><p>artical: <a target="_blank" rel="noopener" href="https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-019-0674-0">https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-019-0674-0</a></p>
<p>这篇文章的主要思路是通过SNV来确定子代基因来自于父母双方的哪一方。</p>
<p>印记基因假说：</p>
<blockquote>
<p>Genomic imprinting is a special case of mono-allelic expression where genes are expressed in a parent-of-origin (PofO)-specific manner. Although several hypotheses exist to explain why genomic imprinting occurs, the parental conflict hypothesis that imprinted genes evolved from a parental battle between males and females to influence the allocation of maternal resources to offspring. This type of mono-allelic expression can be observed in mammals at different developmental stages and is dependent on stage, cell, and tissue type.</p>
</blockquote>
<h3 id="Material"><a href="#Material" class="headerlink" title="Material"></a>Material</h3><p>165 trio from HapMap/1000 Genomes Projects with RNA-Seq data from lymphoblastoid cell lines (LCLs) and 131 trios from the Genome-of-the-Netherlands </p>
<h3 id="Focus"><a href="#Focus" class="headerlink" title="Focus"></a>Focus</h3><p>complete imprinting (exclusive expression of the paternal or maternal allele) and incomplete imprinting (bias in expression towards the maternal or paternal allele)</p>
<h3 id="Method"><a href="#Method" class="headerlink" title="Method"></a>Method</h3><p>allele-specific RNA-Seq analysis of parent-offspring trios: </p>
<p>used phased genotypes to compute the relative expression from the maternal and paternal alleles in RNA-Seq reads at expressed heterozygous single nucleotide variants (SNVs);</p>
<p>summed the paternal and maternal counts for all heterozygous SNVs contained in a gene(irrespective of their exonic or intronic nature);</p>
<p>statistical tests to check for consistent parental expression bias of autosomal genes within the populations: Wilcoxon signed-rank (WSR) test and ShrinkBayes (SB);</p>
<ol>
<li><p>Strand-specific RNA-Seq in lymphoblastoid cell lines</p>
<ol>
<li><p>quality control by fastqc (version 0.11.2)</p>
</li>
<li><p>Over-represented sequences were removed by trimmomatic (version 0.32) [reads ≥ 30 bp in length were kept]</p>
</li>
<li><p>Cleaned reads were mapped to the human reference genome (hg19) with Gencode v16 annotations by STAR aligner (version 2.3.0) [yielding a mean of 79% uniquely mapped reads]</p>
</li>
<li><p>intermediate BAM file processing such as add read groups and sorting and merging BAM files of the same samples by Picard (version 1.112)</p>
</li>
<li><p>correct for mapping errors and biases which can result in false-positive allele-specific read assignments by WASP software (version 0.1)<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4626402/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4626402/</a> [resulting in the removal of a mean of 36% of reads that overlapped SNVs in each sample] {WASP input: bam; then identifies mapped reads that overlap known polymorphisms}</p>
</li>
<li><p>SNVs in each offspring were assigned on parental origin </p>
</li>
<li><p>determine allele-specific expression by heterozygous sites</p>
</li>
<li><p>quantified reference and alternate RNA-Seq reads mapped at heterozygous loci by AlleleCounter (v0.2, <a target="_blank" rel="noopener" href="https://github.com/secastel/allelecounter">https://github.com/secastel/allelecounter</a>) </p>
</li>
<li><p>reference and alternate allele counts were used with PofO information to assign counts to the maternal and paternal alleles at each heterozygous site [Reads that did not uniquely map, or had base quality ≤ 10, were discarded.]</p>
</li>
<li><p>Use filter to reduce the mapping errors:</p>
<p>① removing heterozygous SNVs that had a mappability score &lt; 1 (based on the “CRG GEM Alignability of 50mers with no more than 2 mismatches” track, downloaded from UCSC genome browser)</p>
<p>② removing heterozygous SNVs that overlapped CNVs with MAF ≥ 5% identified in samples from the 1000 Genomes and HapMap Projects (ftp://<a target="_blank" rel="noopener" href="http://ftp.1000genomes.ebi.ac.uk/vol1/withdrawn/phase3/integrated_sv_map/">ftp.1000genomes.ebi.ac.uk/vol1/withdrawn/phase3/integrated_sv_map/</a> and common CNVs</p>
<p>③ removing heterozygous SNVs that are segmental duplications</p>
<p>④ removing heterozygous SNVs that are simple repeats (both downloaded from “Variation and Repeats” track group of the UCSC genome browser). </p>
<p>[These filters resulted in the removal of 21% of heterozygous sites, leaving ~ 3.1 million sites for downstream analysis.]</p>
</li>
<li><p>Genotyping DNA: </p>
<p>BEAGLE  (<a target="_blank" rel="noopener" href="https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-019-0674-0#ref-CR57">https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-019-0674-0#ref-CR57</a>) and IMPUTE2. Using GATK:UnifiedGenotyper as input for BEAGLE, treating all samples as unrelated. SHAPEIT2 and MVNcall19 were then used along with trio information to phase the complete set of SNVs. Each haplotype transmitted to the offspring, and therefore, allelic parental origin was then obtained from the phased haplotypes</p>
</li>
</ol>
</li>
<li><p>overall Genotype data processing</p>
<ol>
<li><p>get genotype data from smples</p>
<p> The <strong>GATK</strong> joint genotyping workflow can be applied in genotyping, here list the steps:</p>
<p> ①Versions 3.0 and above of GATK offer the possibility of calling DNA variants on cohorts of samples using the <strong>HaplotypeCaller</strong> algorithm in Genomic Variant Call Format (GVCF) mode.</p>
<p> input: RNA data</p>
<p> output: one GVCF file per sample</p>
<p> ②variants are called from the GVCF files through a joint genotyping analysis. </p>
</li>
<li><p>quality control: resolving strand inconsistencies, removing multi-allelic SNVs and indels, removing SNVs not present in the 1000 Genomes data</p>
</li>
<li><p>Use PLINK (versions 1.07 and 1.9), vcftools (version 0.1.15) and Beagle Utilities to convert coordinates from hg18 to hg19</p>
</li>
</ol>
</li>
</ol>
<h1 id="ref"><a href="#ref" class="headerlink" title="ref"></a>ref</h1><figure class="highlight tex"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br><span class="line">232</span><br><span class="line">233</span><br><span class="line">234</span><br><span class="line">235</span><br><span class="line">236</span><br><span class="line">237</span><br><span class="line">238</span><br><span class="line">239</span><br><span class="line">240</span><br><span class="line">241</span><br><span class="line">242</span><br><span class="line">243</span><br><span class="line">244</span><br><span class="line">245</span><br><span class="line">246</span><br><span class="line">247</span><br><span class="line">248</span><br><span class="line">249</span><br><span class="line">250</span><br><span class="line">251</span><br><span class="line">252</span><br><span class="line">253</span><br><span class="line">254</span><br><span class="line">255</span><br><span class="line">256</span><br><span class="line">257</span><br><span class="line">258</span><br><span class="line">259</span><br><span class="line">260</span><br><span class="line">261</span><br><span class="line">262</span><br><span class="line">263</span><br><span class="line">264</span><br><span class="line">265</span><br><span class="line">266</span><br><span class="line">267</span><br><span class="line">268</span><br><span class="line">269</span><br><span class="line">270</span><br><span class="line">271</span><br><span class="line">272</span><br><span class="line">273</span><br><span class="line">274</span><br><span class="line">275</span><br><span class="line">276</span><br><span class="line">277</span><br><span class="line">278</span><br><span class="line">279</span><br><span class="line">280</span><br><span class="line">281</span><br><span class="line">282</span><br><span class="line">283</span><br><span class="line">284</span><br><span class="line">285</span><br><span class="line">286</span><br><span class="line">287</span><br><span class="line">288</span><br><span class="line">289</span><br><span class="line">290</span><br><span class="line">291</span><br><span class="line">292</span><br><span class="line">293</span><br><span class="line">294</span><br><span class="line">295</span><br><span class="line">296</span><br><span class="line">297</span><br><span class="line">298</span><br><span class="line">299</span><br><span class="line">300</span><br><span class="line">301</span><br><span class="line">302</span><br><span class="line">303</span><br><span class="line">304</span><br><span class="line">305</span><br><span class="line">306</span><br><span class="line">307</span><br><span class="line">308</span><br><span class="line">309</span><br><span class="line">310</span><br><span class="line">311</span><br><span class="line">312</span><br><span class="line">313</span><br><span class="line">314</span><br><span class="line">315</span><br><span class="line">316</span><br><span class="line">317</span><br><span class="line">318</span><br><span class="line">319</span><br><span class="line">320</span><br><span class="line">321</span><br><span class="line">322</span><br><span class="line">323</span><br><span class="line">324</span><br><span class="line">325</span><br><span class="line">326</span><br><span class="line">327</span><br><span class="line">328</span><br><span class="line">329</span><br><span class="line">330</span><br><span class="line">331</span><br><span class="line">332</span><br><span class="line">333</span><br><span class="line">334</span><br><span class="line">335</span><br><span class="line">336</span><br><span class="line">337</span><br><span class="line">338</span><br><span class="line">339</span><br><span class="line">340</span><br><span class="line">341</span><br><span class="line">342</span><br><span class="line">343</span><br><span class="line">344</span><br><span class="line">345</span><br><span class="line">346</span><br><span class="line">347</span><br><span class="line">348</span><br><span class="line">349</span><br><span class="line">350</span><br><span class="line">351</span><br><span class="line">352</span><br><span class="line">353</span><br><span class="line">354</span><br><span class="line">355</span><br><span class="line">356</span><br><span class="line">357</span><br><span class="line">358</span><br><span class="line">359</span><br><span class="line">360</span><br><span class="line">361</span><br><span class="line">362</span><br><span class="line">363</span><br><span class="line">364</span><br><span class="line">365</span><br><span class="line">366</span><br><span class="line">367</span><br><span class="line">368</span><br><span class="line">369</span><br><span class="line">370</span><br><span class="line">371</span><br><span class="line">372</span><br><span class="line">373</span><br><span class="line">374</span><br><span class="line">375</span><br><span class="line">376</span><br><span class="line">377</span><br><span class="line">378</span><br><span class="line">379</span><br><span class="line">380</span><br><span class="line">381</span><br><span class="line">382</span><br><span class="line">383</span><br><span class="line">384</span><br><span class="line">385</span><br><span class="line">386</span><br><span class="line">387</span><br><span class="line">388</span><br><span class="line">389</span><br><span class="line">390</span><br><span class="line">391</span><br><span class="line">392</span><br><span class="line">393</span><br><span class="line">394</span><br><span class="line">395</span><br><span class="line">396</span><br><span class="line">397</span><br><span class="line">398</span><br><span class="line">399</span><br><span class="line">400</span><br><span class="line">401</span><br><span class="line">402</span><br><span class="line">403</span><br><span class="line">404</span><br><span class="line">405</span><br><span class="line">406</span><br><span class="line">407</span><br><span class="line">408</span><br><span class="line">409</span><br><span class="line">410</span><br><span class="line">411</span><br><span class="line">412</span><br><span class="line">413</span><br><span class="line">414</span><br><span class="line">415</span><br><span class="line">416</span><br><span class="line">417</span><br><span class="line">418</span><br><span class="line">419</span><br><span class="line">420</span><br><span class="line">421</span><br><span class="line">422</span><br><span class="line">423</span><br><span class="line">424</span><br><span class="line">425</span><br><span class="line">426</span><br><span class="line">427</span><br><span class="line">428</span><br><span class="line">429</span><br><span class="line">430</span><br><span class="line">431</span><br><span class="line">432</span><br><span class="line">433</span><br><span class="line">434</span><br><span class="line">435</span><br><span class="line">436</span><br><span class="line">437</span><br><span class="line">438</span><br><span class="line">439</span><br><span class="line">440</span><br><span class="line">441</span><br><span class="line">442</span><br><span class="line">443</span><br><span class="line">444</span><br><span class="line">445</span><br><span class="line">446</span><br><span class="line">447</span><br><span class="line">448</span><br><span class="line">449</span><br><span class="line">450</span><br><span class="line">451</span><br><span class="line">452</span><br><span class="line">453</span><br><span class="line">454</span><br><span class="line">455</span><br><span class="line">456</span><br><span class="line">457</span><br><span class="line">458</span><br><span class="line">459</span><br><span class="line">460</span><br><span class="line">461</span><br><span class="line">462</span><br><span class="line">463</span><br><span class="line">464</span><br><span class="line">465</span><br><span class="line">466</span><br><span class="line">467</span><br><span class="line">468</span><br><span class="line">469</span><br><span class="line">470</span><br><span class="line">471</span><br><span class="line">472</span><br><span class="line">473</span><br><span class="line">474</span><br><span class="line">475</span><br><span class="line">476</span><br><span class="line">477</span><br><span class="line">478</span><br><span class="line">479</span><br><span class="line">480</span><br><span class="line">481</span><br><span class="line">482</span><br><span class="line">483</span><br><span class="line">484</span><br><span class="line">485</span><br><span class="line">486</span><br><span class="line">487</span><br><span class="line">488</span><br><span class="line">489</span><br><span class="line">490</span><br><span class="line">491</span><br><span class="line">492</span><br><span class="line">493</span><br><span class="line">494</span><br><span class="line">495</span><br><span class="line">496</span><br><span class="line">497</span><br><span class="line">498</span><br><span class="line">499</span><br><span class="line">500</span><br><span class="line">501</span><br><span class="line">502</span><br><span class="line">503</span><br><span class="line">504</span><br><span class="line">505</span><br><span class="line">506</span><br><span class="line">507</span><br><span class="line">508</span><br><span class="line">509</span><br><span class="line">510</span><br><span class="line">511</span><br><span class="line">512</span><br><span class="line">513</span><br><span class="line">514</span><br><span class="line">515</span><br><span class="line">516</span><br><span class="line">517</span><br><span class="line">518</span><br><span class="line">519</span><br><span class="line">520</span><br><span class="line">521</span><br><span class="line">522</span><br><span class="line">523</span><br><span class="line">524</span><br><span class="line">525</span><br><span class="line">526</span><br><span class="line">527</span><br><span class="line">528</span><br><span class="line">529</span><br><span class="line">530</span><br><span class="line">531</span><br><span class="line">532</span><br><span class="line">533</span><br><span class="line">534</span><br><span class="line">535</span><br><span class="line">536</span><br><span class="line">537</span><br><span class="line">538</span><br><span class="line">539</span><br><span class="line">540</span><br><span class="line">541</span><br><span class="line">542</span><br><span class="line">543</span><br><span class="line">544</span><br><span class="line">545</span><br><span class="line">546</span><br><span class="line">547</span><br><span class="line">548</span><br><span class="line">549</span><br><span class="line">550</span><br><span class="line">551</span><br><span class="line">552</span><br><span class="line">553</span><br><span class="line">554</span><br><span class="line">555</span><br><span class="line">556</span><br><span class="line">557</span><br><span class="line">558</span><br><span class="line">559</span><br><span class="line">560</span><br><span class="line">561</span><br><span class="line">562</span><br><span class="line">563</span><br><span class="line">564</span><br><span class="line">565</span><br><span class="line">566</span><br><span class="line">567</span><br><span class="line">568</span><br><span class="line">569</span><br><span class="line">570</span><br><span class="line">571</span><br><span class="line">572</span><br><span class="line">573</span><br><span class="line">574</span><br><span class="line">575</span><br><span class="line">576</span><br><span class="line">577</span><br><span class="line">578</span><br><span class="line">579</span><br><span class="line">580</span><br><span class="line">581</span><br><span class="line">582</span><br><span class="line">583</span><br><span class="line">584</span><br><span class="line">585</span><br><span class="line">586</span><br><span class="line">587</span><br><span class="line">588</span><br><span class="line">589</span><br><span class="line">590</span><br><span class="line">591</span><br><span class="line">592</span><br><span class="line">593</span><br><span class="line">594</span><br><span class="line">595</span><br><span class="line">596</span><br><span class="line">597</span><br><span class="line">598</span><br><span class="line">599</span><br><span class="line">600</span><br><span class="line">601</span><br><span class="line">602</span><br><span class="line">603</span><br><span class="line">604</span><br><span class="line">605</span><br><span class="line">606</span><br><span class="line">607</span><br><span class="line">608</span><br><span class="line">609</span><br><span class="line">610</span><br><span class="line">611</span><br><span class="line">612</span><br><span class="line">613</span><br><span class="line">614</span><br><span class="line">615</span><br><span class="line">616</span><br><span class="line">617</span><br><span class="line">618</span><br><span class="line">619</span><br><span class="line">620</span><br><span class="line">621</span><br><span class="line">622</span><br><span class="line">623</span><br><span class="line">624</span><br><span class="line">625</span><br><span class="line">626</span><br><span class="line">627</span><br><span class="line">628</span><br><span class="line">629</span><br><span class="line">630</span><br><span class="line">631</span><br><span class="line">632</span><br><span class="line">633</span><br><span class="line">634</span><br><span class="line">635</span><br><span class="line">636</span><br><span class="line">637</span><br><span class="line">638</span><br><span class="line">639</span><br><span class="line">640</span><br><span class="line">641</span><br><span class="line">642</span><br><span class="line">643</span><br><span class="line">644</span><br><span class="line">645</span><br><span class="line">646</span><br><span class="line">647</span><br><span class="line">648</span><br><span class="line">649</span><br><span class="line">650</span><br><span class="line">651</span><br><span class="line">652</span><br><span class="line">653</span><br><span class="line">654</span><br><span class="line">655</span><br><span class="line">656</span><br><span class="line">657</span><br><span class="line">658</span><br><span class="line">659</span><br><span class="line">660</span><br><span class="line">661</span><br><span class="line">662</span><br><span class="line">663</span><br><span class="line">664</span><br><span class="line">665</span><br><span class="line">666</span><br><span class="line">667</span><br><span class="line">668</span><br><span class="line">669</span><br><span class="line">670</span><br><span class="line">671</span><br><span class="line">672</span><br><span class="line">673</span><br><span class="line">674</span><br><span class="line">675</span><br><span class="line">676</span><br><span class="line">677</span><br><span class="line">678</span><br><span class="line">679</span><br><span class="line">680</span><br><span class="line">681</span><br><span class="line">682</span><br><span class="line">683</span><br><span class="line">684</span><br><span class="line">685</span><br><span class="line">686</span><br><span class="line">687</span><br><span class="line">688</span><br><span class="line">689</span><br><span class="line">690</span><br><span class="line">691</span><br><span class="line">692</span><br><span class="line">693</span><br><span class="line">694</span><br><span class="line">695</span><br><span class="line">696</span><br><span class="line">697</span><br><span class="line">698</span><br><span class="line">699</span><br><span class="line">700</span><br><span class="line">701</span><br><span class="line">702</span><br><span class="line">703</span><br><span class="line">704</span><br><span class="line">705</span><br><span class="line">706</span><br><span class="line">707</span><br><span class="line">708</span><br><span class="line">709</span><br><span class="line">710</span><br><span class="line">711</span><br><span class="line">712</span><br><span class="line">713</span><br><span class="line">714</span><br><span class="line">715</span><br><span class="line">716</span><br><span class="line">717</span><br><span class="line">718</span><br><span class="line">719</span><br><span class="line">720</span><br><span class="line">721</span><br><span class="line">722</span><br><span class="line">723</span><br><span class="line">724</span><br><span class="line">725</span><br><span class="line">726</span><br><span class="line">727</span><br><span class="line">728</span><br><span class="line">729</span><br><span class="line">730</span><br><span class="line">731</span><br><span class="line">732</span><br><span class="line">733</span><br><span class="line">734</span><br><span class="line">735</span><br><span class="line">736</span><br><span class="line">737</span><br><span class="line">738</span><br><span class="line">739</span><br><span class="line">740</span><br><span class="line">741</span><br><span class="line">742</span><br><span class="line">743</span><br><span class="line">744</span><br><span class="line">745</span><br><span class="line">746</span><br><span class="line">747</span><br><span class="line">748</span><br><span class="line">749</span><br><span class="line">750</span><br><span class="line">751</span><br><span class="line">752</span><br><span class="line">753</span><br><span class="line">754</span><br><span class="line">755</span><br><span class="line">756</span><br><span class="line">757</span><br><span class="line">758</span><br><span class="line">759</span><br><span class="line">760</span><br><span class="line">761</span><br><span class="line">762</span><br><span class="line">763</span><br><span class="line">764</span><br><span class="line">765</span><br><span class="line">766</span><br><span class="line">767</span><br><span class="line">768</span><br><span class="line">769</span><br><span class="line">770</span><br><span class="line">771</span><br><span class="line">772</span><br><span class="line">773</span><br><span class="line">774</span><br><span class="line">775</span><br><span class="line">776</span><br><span class="line">777</span><br><span class="line">778</span><br><span class="line">779</span><br><span class="line">780</span><br><span class="line">781</span><br><span class="line">782</span><br><span class="line">783</span><br><span class="line">784</span><br><span class="line">785</span><br><span class="line">786</span><br><span class="line">787</span><br><span class="line">788</span><br><span class="line">789</span><br><span class="line">790</span><br><span class="line">791</span><br><span class="line">792</span><br><span class="line">793</span><br><span class="line">794</span><br><span class="line">795</span><br><span class="line">796</span><br><span class="line">797</span><br><span class="line">798</span><br><span class="line">799</span><br><span class="line">800</span><br><span class="line">801</span><br><span class="line">802</span><br><span class="line">803</span><br><span class="line">804</span><br><span class="line">805</span><br><span class="line">806</span><br><span class="line">807</span><br><span class="line">808</span><br><span class="line">809</span><br><span class="line">810</span><br><span class="line">811</span><br><span class="line">812</span><br><span class="line">813</span><br><span class="line">814</span><br><span class="line">815</span><br><span class="line">816</span><br><span class="line">817</span><br><span class="line">818</span><br><span class="line">819</span><br><span class="line">820</span><br><span class="line">821</span><br><span class="line">822</span><br><span class="line">823</span><br><span class="line">824</span><br><span class="line">825</span><br><span class="line">826</span><br><span class="line">827</span><br><span class="line">828</span><br><span class="line">829</span><br><span class="line">830</span><br><span class="line">831</span><br><span class="line">832</span><br><span class="line">833</span><br><span class="line">834</span><br><span class="line">835</span><br><span class="line">836</span><br><span class="line">837</span><br><span class="line">838</span><br><span class="line">839</span><br><span class="line">840</span><br><span class="line">841</span><br><span class="line">842</span><br><span class="line">843</span><br><span class="line">844</span><br><span class="line">845</span><br><span class="line">846</span><br><span class="line">847</span><br><span class="line">848</span><br><span class="line">849</span><br><span class="line">850</span><br><span class="line">851</span><br><span class="line">852</span><br><span class="line">853</span><br><span class="line">854</span><br><span class="line">855</span><br><span class="line">856</span><br><span class="line">857</span><br><span class="line">858</span><br><span class="line">859</span><br><span class="line">860</span><br><span class="line">861</span><br><span class="line">862</span><br><span class="line">863</span><br><span class="line">864</span><br><span class="line">865</span><br><span class="line">866</span><br><span class="line">867</span><br><span class="line">868</span><br><span class="line">869</span><br><span class="line">870</span><br><span class="line">871</span><br><span class="line">872</span><br><span class="line">873</span><br><span class="line">874</span><br><span class="line">875</span><br><span class="line">876</span><br><span class="line">877</span><br><span class="line">878</span><br><span class="line">879</span><br><span class="line">880</span><br><span class="line">881</span><br><span class="line">882</span><br><span class="line">883</span><br><span class="line">884</span><br><span class="line">885</span><br><span class="line">886</span><br><span class="line">887</span><br><span class="line">888</span><br><span class="line">889</span><br><span class="line">890</span><br><span class="line">891</span><br><span class="line">892</span><br><span class="line">893</span><br><span class="line">894</span><br><span class="line">895</span><br><span class="line">896</span><br><span class="line">897</span><br><span class="line">898</span><br><span class="line">899</span><br><span class="line">900</span><br><span class="line">901</span><br><span class="line">902</span><br><span class="line">903</span><br><span class="line">904</span><br><span class="line">905</span><br><span class="line">906</span><br><span class="line">907</span><br><span class="line">908</span><br><span class="line">909</span><br><span class="line">910</span><br><span class="line">911</span><br><span class="line">912</span><br><span class="line">913</span><br><span class="line">914</span><br><span class="line">915</span><br><span class="line">916</span><br><span class="line">917</span><br><span class="line">918</span><br><span class="line">919</span><br><span class="line">920</span><br><span class="line">921</span><br><span class="line">922</span><br><span class="line">923</span><br><span class="line">924</span><br><span class="line">925</span><br><span class="line">926</span><br><span class="line">927</span><br><span class="line">928</span><br><span class="line">929</span><br><span class="line">930</span><br><span class="line">931</span><br><span class="line">932</span><br><span class="line">933</span><br><span class="line">934</span><br><span class="line">935</span><br><span class="line">936</span><br><span class="line">937</span><br><span class="line">938</span><br><span class="line">939</span><br><span class="line">940</span><br><span class="line">941</span><br><span class="line">942</span><br><span class="line">943</span><br><span class="line">944</span><br><span class="line">945</span><br><span class="line">946</span><br><span class="line">947</span><br><span class="line">948</span><br><span class="line">949</span><br><span class="line">950</span><br><span class="line">951</span><br><span class="line">952</span><br><span class="line">953</span><br><span class="line">954</span><br><span class="line">955</span><br><span class="line">956</span><br><span class="line">957</span><br><span class="line">958</span><br><span class="line">959</span><br><span class="line">960</span><br><span class="line">961</span><br><span class="line">962</span><br><span class="line">963</span><br><span class="line">964</span><br><span class="line">965</span><br><span class="line">966</span><br><span class="line">967</span><br><span class="line">968</span><br><span class="line">969</span><br><span class="line">970</span><br><span class="line">971</span><br><span class="line">972</span><br><span class="line">973</span><br></pre></td><td class="code"><pre><code class="hljs tex">@article&#123;wutz2011gene,<br>  title=&#123;Gene silencing in X-chromosome inactivation: advances in understanding facultative heterochromatin formation&#125;,<br>  author=&#123;Wutz, Anton&#125;,<br>  journal=&#123;Nature Reviews Genetics&#125;,<br>  volume=&#123;12&#125;,<br>  number=&#123;8&#125;,<br>  pages=&#123;542--553&#125;,<br>  year=&#123;2011&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;barlow2011genomic,<br>  title=&#123;Genomic imprinting: a mammalian epigenetic discovery model&#125;,<br>  author=&#123;Barlow, Denise P&#125;,<br>  journal=&#123;Annual review of genetics&#125;,<br>  volume=&#123;45&#125;,<br>  pages=&#123;379--403&#125;,<br>  year=&#123;2011&#125;,<br>  publisher=&#123;Annual Reviews&#125;<br>&#125;<br><br>@article&#123;ferron2011postnatal,<br>  title=&#123;Postnatal loss of Dlk1 imprinting in stem cells and niche astrocytes regulates neurogenesis&#125;,<br>  author=&#123;Ferr&#123;<span class="hljs-keyword">\&#x27;</span>o&#125;n, Sacri R and Charalambous, Marika and Radford, Elizabeth and McEwen, Kirsten and Wildner, Hendrik and Hind, Eleanor and Morante-Redolat, Jose Manuel and Laborda, Jorge and Guillemot, Francois and Bauer, Steven R and others&#125;,<br>  journal=&#123;Nature&#125;,<br>  volume=&#123;475&#125;,<br>  number=&#123;7356&#125;,<br>  pages=&#123;381--385&#125;,<br>  year=&#123;2011&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;lo2003allelic,<br>  title=&#123;Allelic variation in gene expression is common in the human genome&#125;,<br>  author=&#123;Lo, H Shuen and Wang, Zhining and Hu, Ying and Yang, Howard H and Gere, Sheryl and Buetow, Kenneth H and Lee, Maxwell P&#125;,<br>  journal=&#123;Genome research&#125;,<br>  volume=&#123;13&#125;,<br>  number=&#123;8&#125;,<br>  pages=&#123;1855--1862&#125;,<br>  year=&#123;2003&#125;,<br>  publisher=&#123;Cold Spring Harbor Lab&#125;<br>&#125;<br><br>@misc&#123;bartolomei25mammalian,<br>  title=&#123;Mammalian genomic imprinting. Cold Spring Harb Perspect Biol. 2011; 3 (7)&#125;,<br>  author=&#123;Bartolomei, MS and Ferguson-Smith, AC&#125;,<br>  publisher=&#123;Epub 2011/05/18. doi: 10.1101/cshperspect. a002592. PubMed PMID: 21576252&#125;<br>&#125;<br><br>@article&#123;heijmans2008persistent,<br>  title=&#123;Persistent epigenetic differences associated with prenatal exposure to famine in humans&#125;,<br>  author=&#123;Heijmans, Bastiaan T and Tobi, Elmar W and Stein, Aryeh D and Putter, Hein and Blauw, Gerard J and Susser, Ezra S and Slagboom, P Eline and Lumey, LH&#125;,<br>  journal=&#123;Proceedings of the National Academy of Sciences&#125;,<br>  volume=&#123;105&#125;,<br>  number=&#123;44&#125;,<br>  pages=&#123;17046--17049&#125;,<br>  year=&#123;2008&#125;,<br>  publisher=&#123;National Acad Sciences&#125;<br>&#125;<br><br>@article&#123;deng2014single,<br>  title=&#123;Single-cell RNA-seq reveals dynamic, random monoallelic gene expression in mammalian cells&#125;,<br>  author=&#123;Deng, Qiaolin and Ramsk&#123;<span class="hljs-keyword">\&quot;</span>o&#125;ld, Daniel and Reinius, Bj&#123;<span class="hljs-keyword">\&quot;</span>o&#125;rn and Sandberg, Rickard&#125;,<br>  journal=&#123;Science&#125;,<br>  volume=&#123;343&#125;,<br>  number=&#123;6167&#125;,<br>  pages=&#123;193--196&#125;,<br>  year=&#123;2014&#125;,<br>  publisher=&#123;American Association for the Advancement of Science&#125;<br>&#125;<br><br>@article&#123;gendrel2014developmental,<br>  title=&#123;Developmental dynamics and disease potential of random monoallelic gene expression&#125;,<br>  author=&#123;Gendrel, Anne-Valerie and Attia, Mikael and Chen, Chong-Jian and Diabangouaya, Patricia and Servant, Nicolas and Barillot, Emmanuel and Heard, Edith&#125;,<br>  journal=&#123;Developmental cell&#125;,<br>  volume=&#123;28&#125;,<br>  number=&#123;4&#125;,<br>  pages=&#123;366--380&#125;,<br>  year=&#123;2014&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;chess2012mechanisms,<br>  title=&#123;Mechanisms and consequences of widespread random monoallelic expression&#125;,<br>  author=&#123;Chess, Andrew&#125;,<br>  journal=&#123;Nature Reviews Genetics&#125;,<br>  volume=&#123;13&#125;,<br>  number=&#123;6&#125;,<br>  pages=&#123;421--428&#125;,<br>  year=&#123;2012&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;tycko2010allele,<br>  title=&#123;Allele-specific DNA methylation: beyond imprinting&#125;,<br>  author=&#123;Tycko, Benjamin&#125;,<br>  journal=&#123;Human molecular genetics&#125;,<br>  volume=&#123;19&#125;,<br>  number=&#123;R2&#125;,<br>  pages=&#123;R210--R220&#125;,<br>  year=&#123;2010&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;ohlsson2007widespread,<br>  title=&#123;Widespread monoallelic expression&#125;,<br>  author=&#123;Ohlsson, Rolf&#125;,<br>  journal=&#123;Science&#125;,<br>  volume=&#123;318&#125;,<br>  number=&#123;5853&#125;,<br>  pages=&#123;1077--1078&#125;,<br>  year=&#123;2007&#125;,<br>  publisher=&#123;American Association for the Advancement of Science&#125;<br>&#125;<br><br>@article&#123;esteller2000promoter,<br>  title=&#123;Promoter hypermethylation and BRCA1 inactivation in sporadic breast and ovarian tumors&#125;,<br>  author=&#123;Esteller, Manel and Silva, Jose Maria and Dominguez, Gemma and Bonilla, Felix and Matias-Guiu, Xavier and Lerma, Enrique and Bussaglia, Elena and Prat, Jaime and Harkes, I Clara and Repasky, Elizabeth A and others&#125;,<br>  journal=&#123;JNCI: Journal of the National Cancer Institute&#125;,<br>  volume=&#123;92&#125;,<br>  number=&#123;7&#125;,<br>  pages=&#123;564--569&#125;,<br>  year=&#123;2000&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;thrall2006brca1,<br>  title=&#123;BRCA1 expression in a large series of sporadic ovarian carcinomas: a Gynecologic Oncology Group study&#125;,<br>  author=&#123;Thrall, M and Gallion, HH and Kryscio, R and Kapali, M and Armstrong, DK and Deloia, Julie A&#125;,<br>  journal=&#123;International Journal of Gynecologic Cancer&#125;,<br>  volume=&#123;16&#125;,<br>  number=&#123;Suppl 1&#125;,<br>  pages=&#123;166--171&#125;,<br>  year=&#123;2006&#125;,<br>  publisher=&#123;BMJ Specialist Journals&#125;<br>&#125;<br><br>@article&#123;yan2002allelic,<br>  title=&#123;Allelic variation in human gene expression&#125;,<br>  author=&#123;Yan, Hai and Yuan, Weishi and Velculescu, Victor E and Vogelstein, Bert and Kinzler, Kenneth W and others&#125;,<br>  journal=&#123;Science&#125;,<br>  volume=&#123;297&#125;,<br>  number=&#123;5584&#125;,<br>  pages=&#123;1143--1143&#125;,<br>  year=&#123;2002&#125;<br>&#125;<br><br>@article&#123;cedar2008choreography,<br>  title=&#123;Choreography of Ig allelic exclusion&#125;,<br>  author=&#123;Cedar, Howard and Bergman, Yehudit&#125;,<br>  journal=&#123;Current opinion in immunology&#125;,<br>  volume=&#123;20&#125;,<br>  number=&#123;3&#125;,<br>  pages=&#123;308--317&#125;,<br>  year=&#123;2008&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;borel2015biased,<br>  title=&#123;Biased allelic expression in human primary fibroblast single cells&#125;,<br>  author=&#123;Borel, Christelle and Ferreira, Pedro G and Santoni, Federico and Delaneau, Olivier and Fort, Alexandre and Popadin, Konstantin Y and Garieri, Marco and Falconnet, Emilie and Ribaux, Pascale and Guipponi, Michel and others&#125;,<br>  journal=&#123;The American Journal of Human Genetics&#125;,<br>  volume=&#123;96&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;70--80&#125;,<br>  year=&#123;2015&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;risso2014normalization,<br>  title=&#123;Normalization of RNA-seq data using factor analysis of control genes or samples&#125;,<br>  author=&#123;Risso, Davide and Ngai, John and Speed, Terence P and Dudoit, Sandrine&#125;,<br>  journal=&#123;Nature biotechnology&#125;,<br>  volume=&#123;32&#125;,<br>  number=&#123;9&#125;,<br>  pages=&#123;896--902&#125;,<br>  year=&#123;2014&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;lappalainen2013transcriptome,<br>  title=&#123;Transcriptome and genome sequencing uncovers functional variation in humans&#125;,<br>  author=&#123;Lappalainen, Tuuli and Sammeth, Michael and Friedl&#123;<span class="hljs-keyword">\&quot;</span>a&#125;nder, Marc R and Ac‘t Hoen, Peter and Monlong, Jean and Rivas, Manuel A and Gonzalez-Porta, Mar and Kurbatova, Natalja and Griebel, Thasso and Ferreira, Pedro G and others&#125;,<br>  journal=&#123;Nature&#125;,<br>  volume=&#123;501&#125;,<br>  number=&#123;7468&#125;,<br>  pages=&#123;506--511&#125;,<br>  year=&#123;2013&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;li2009sequence,<br>  title=&#123;The sequence alignment/map format and SAMtools&#125;,<br>  author=&#123;Li, Heng and Handsaker, Bob and Wysoker, Alec and Fennell, Tim and Ruan, Jue and Homer, Nils and Marth, Gabor and Abecasis, Goncalo and Durbin, Richard&#125;,<br>  journal=&#123;Bioinformatics&#125;,<br>  volume=&#123;25&#125;,<br>  number=&#123;16&#125;,<br>  pages=&#123;2078--2079&#125;,<br>  year=&#123;2009&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;trapnell2012differential,<br>  title=&#123;Differential gene and transcript expression analysis of RNA-seq experiments with TopHat and Cufflinks&#125;,<br>  author=&#123;Trapnell, Cole and Roberts, Adam and Goff, Loyal and Pertea, Geo and Kim, Daehwan and Kelley, David R and Pimentel, Harold and Salzberg, Steven L and Rinn, John L and Pachter, Lior&#125;,<br>  journal=&#123;Nature protocols&#125;,<br>  volume=&#123;7&#125;,<br>  number=&#123;3&#125;,<br>  pages=&#123;562--578&#125;,<br>  year=&#123;2012&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;trapnell2010transcript,<br>  title=&#123;Transcript assembly and quantification by RNA-Seq reveals unannotated transcripts and isoform switching during cell differentiation&#125;,<br>  author=&#123;Trapnell, Cole and Williams, Brian A and Pertea, Geo and Mortazavi, Ali and Kwan, Gordon and Van Baren, Marijke J and Salzberg, Steven L and Wold, Barbara J and Pachter, Lior&#125;,<br>  journal=&#123;Nature biotechnology&#125;,<br>  volume=&#123;28&#125;,<br>  number=&#123;5&#125;,<br>  pages=&#123;511--515&#125;,<br>  year=&#123;2010&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;harrow2012gencode,<br>  title=&#123;GENCODE: the reference human genome annotation for The ENCODE Project&#125;,<br>  author=&#123;Harrow, Jennifer and Frankish, Adam and Gonzalez, Jose M and Tapanari, Electra and Diekhans, Mark and Kokocinski, Felix and Aken, Bronwen L and Barrell, Daniel and Zadissa, Amonida and Searle, Stephen and others&#125;,<br>  journal=&#123;Genome research&#125;,<br>  volume=&#123;22&#125;,<br>  number=&#123;9&#125;,<br>  pages=&#123;1760--1774&#125;,<br>  year=&#123;2012&#125;,<br>  publisher=&#123;Cold Spring Harbor Lab&#125;<br>&#125;<br><br>@article&#123;xie2019modeling,<br>  title=&#123;Modeling allele-specific expression at the gene and SNP levels simultaneously by a Bayesian logistic mixed regression model&#125;,<br>  author=&#123;Xie, Jing and Ji, Tieming and Ferreira, Marco AR and Li, Yahan and Patel, Bhaumik N and Rivera, Rocio M&#125;,<br>  journal=&#123;BMC bioinformatics&#125;,<br>  volume=&#123;20&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;530&#125;,<br>  year=&#123;2019&#125;,<br>  publisher=&#123;Springer&#125;<br>&#125;<br><br><br>@article&#123;mckenna2010genome,<br>  title=&#123;The Genome Analysis Toolkit: a MapReduce framework for analyzing next-generation DNA sequencing data&#125;,<br>  author=&#123;McKenna, Aaron and Hanna, Matthew and Banks, Eric and Sivachenko, Andrey and Cibulskis, Kristian and Kernytsky, Andrew and Garimella, Kiran and Altshuler, David and Gabriel, Stacey and Daly, Mark and others&#125;,<br>  journal=&#123;Genome research&#125;,<br>  volume=&#123;20&#125;,<br>  number=&#123;9&#125;,<br>  pages=&#123;1297--1303&#125;,<br>  year=&#123;2010&#125;,<br>  publisher=&#123;Cold Spring Harbor Lab&#125;<br>&#125;<br><br>@article&#123;li2011statistical,<br>  title=&#123;A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data&#125;,<br>  author=&#123;Li, Heng&#125;,<br>  journal=&#123;Bioinformatics&#125;,<br>  volume=&#123;27&#125;,<br>  number=&#123;21&#125;,<br>  pages=&#123;2987--2993&#125;,<br>  year=&#123;2011&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;kim2015hisat,<br>  title=&#123;HISAT: a fast spliced aligner with low memory requirements&#125;,<br>  author=&#123;Kim, Daehwan and Langmead, Ben and Salzberg, Steven L&#125;,<br>  journal=&#123;Nature methods&#125;,<br>  volume=&#123;12&#125;,<br>  number=&#123;4&#125;,<br>  pages=&#123;357--360&#125;,<br>  year=&#123;2015&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;li2009fast,<br>  title=&#123;Fast and accurate short read alignment with Burrows--Wheeler transform&#125;,<br>  author=&#123;Li, Heng and Durbin, Richard&#125;,<br>  journal=&#123;bioinformatics&#125;,<br>  volume=&#123;25&#125;,<br>  number=&#123;14&#125;,<br>  pages=&#123;1754--1760&#125;,<br>  year=&#123;2009&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;mayba2014mbased,<br>  title=&#123;MBASED: allele-specific expression detection in cancer tissues and cell lines&#125;,<br>  author=&#123;Mayba, Oleg and Gilbert, Houston N and Liu, Jinfeng and Haverty, Peter M and Jhunjhunwala, Suchit and Jiang, Zhaoshi and Watanabe, Colin and Zhang, Zemin&#125;,<br>  journal=&#123;Genome biology&#125;,<br>  volume=&#123;15&#125;,<br>  number=&#123;8&#125;,<br>  pages=&#123;405&#125;,<br>  year=&#123;2014&#125;,<br>  publisher=&#123;Springer&#125;<br>&#125;<br><br>@article&#123;edsgard2016geneiase,<br>  title=&#123;GeneiASE: Detection of condition-dependent and static allele-specific expression from RNA-seq data without haplotype information&#125;,<br>  author=&#123;Edsg&#123;<span class="hljs-keyword">\&quot;</span>a&#125;rd, Daniel and Iglesias, Maria Jesus and Reilly, Sarah-Jayne and Hamsten, Anders and Tornvall, Per and Odeberg, Jacob and Emanuelsson, Olof&#125;,<br>  journal=&#123;Scientific reports&#125;,<br>  volume=&#123;6&#125;,<br>  pages=&#123;21134&#125;,<br>  year=&#123;2016&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;harvey2015quasar,<br>  title=&#123;QuASAR: quantitative allele-specific analysis of reads&#125;,<br>  author=&#123;Harvey, Chris T and Moyerbrailean, Gregory A and Davis, Gordon O and Wen, Xiaoquan and Luca, Francesca and Pique-Regi, Roger&#125;,<br>  journal=&#123;Bioinformatics&#125;,<br>  volume=&#123;31&#125;,<br>  number=&#123;8&#125;,<br>  pages=&#123;1235--1242&#125;,<br>  year=&#123;2015&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;fan2020asep,<br>  title=&#123;ASEP: Gene-based detection of allele-specific expression across individuals in a population by RNA sequencing&#125;,<br>  author=&#123;Fan, Jiaxin and Hu, Jian and Xue, Chenyi and Zhang, Hanrui and Susztak, Katalin and Reilly, Muredach P and Xiao, Rui and Li, Mingyao&#125;,<br>  journal=&#123;PLoS Genetics&#125;,<br>  volume=&#123;16&#125;,<br>  number=&#123;5&#125;,<br>  pages=&#123;e1008786&#125;,<br>  year=&#123;2020&#125;,<br>  publisher=&#123;Public Library of Science&#125;<br>&#125;<br><br>@article&#123;castel2016rare,<br>  title=&#123;Rare variant phasing and haplotypic expression from RNA sequencing with phASER&#125;,<br>  author=&#123;Castel, Stephane E and Mohammadi, Pejman and Chung, Wendy K and Shen, Yufeng and Lappalainen, Tuuli&#125;,<br>  journal=&#123;Nature communications&#125;,<br>  volume=&#123;7&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;1--6&#125;,<br>  year=&#123;2016&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;li2009fast,<br>  title=&#123;Fast and accurate short read alignment with Burrows--Wheeler transform&#125;,<br>  author=&#123;Li, Heng and Durbin, Richard&#125;,<br>  journal=&#123;bioinformatics&#125;,<br>  volume=&#123;25&#125;,<br>  number=&#123;14&#125;,<br>  pages=&#123;1754--1760&#125;,<br>  year=&#123;2009&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;li2009sequence,<br>  title=&#123;The sequence alignment/map format and SAMtools&#125;,<br>  author=&#123;Li, Heng and Handsaker, Bob and Wysoker, Alec and Fennell, Tim and Ruan, Jue and Homer, Nils and Marth, Gabor and Abecasis, Goncalo and Durbin, Richard&#125;,<br>  journal=&#123;Bioinformatics&#125;,<br>  volume=&#123;25&#125;,<br>  number=&#123;16&#125;,<br>  pages=&#123;2078--2079&#125;,<br>  year=&#123;2009&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;poplin2017scaling,<br>  title=&#123;Scaling accurate genetic variant discovery to tens of thousands of samples&#125;,<br>  author=&#123;Poplin, Ryan and Ruano-Rubio, Valentin and DePristo, Mark A and Fennell, Tim J and Carneiro, Mauricio O and Van der Auwera, Geraldine A and Kling, David E and Gauthier, Laura D and Levy-Moonshine, Ami and Roazen, David and others&#125;,<br>  journal=&#123;BioRxiv&#125;,<br>  pages=&#123;201178&#125;,<br>  year=&#123;2017&#125;,<br>  publisher=&#123;Cold Spring Harbor Laboratory&#125;<br>&#125;<br><br>@article&#123;sherry2001dbsnp,<br>  title=&#123;dbSNP: the NCBI database of genetic variation&#125;,<br>  author=&#123;Sherry, Stephen T and Ward, M-H and Kholodov, M and Baker, J and Phan, Lon and Smigielski, Elizabeth M and Sirotkin, Karl&#125;,<br>  journal=&#123;Nucleic acids research&#125;,<br>  volume=&#123;29&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;308--311&#125;,<br>  year=&#123;2001&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;cingolani2012program,<br>  title=&#123;A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118; iso-2; iso-3&#125;,<br>  author=&#123;Cingolani, Pablo and Platts, Adrian and Wang, Le Lily and Coon, Melissa and Nguyen, Tung and Wang, Luan and Land, Susan J and Lu, Xiangyi and Ruden, Douglas M&#125;,<br>  journal=&#123;Fly&#125;,<br>  volume=&#123;6&#125;,<br>  number=&#123;2&#125;,<br>  pages=&#123;80--92&#125;,<br>  year=&#123;2012&#125;,<br>  publisher=&#123;Taylor <span class="hljs-keyword">\&amp;</span> Francis&#125;<br>&#125;<br><br>@article&#123;ruden2012using,<br>  title=&#123;Using Drosophila melanogaster as a model for genotoxic chemical mutational studies with a new program, SnpSift&#125;,<br>  author=&#123;Ruden, Douglas Mark and Cingolani, Pablo and Patel, Viral M and Coon, Melissa and Nguyen, Tung and Land, Susan J and Lu, Xiangyi&#125;,<br>  journal=&#123;Frontiers in genetics&#125;,<br>  volume=&#123;3&#125;,<br>  pages=&#123;35&#125;,<br>  year=&#123;2012&#125;,<br>  publisher=&#123;Frontiers&#125;<br>&#125;<br><br>@article&#123;ahn2019analysis,<br>  title=&#123;Analysis of allele-specific expression using RNA-seq of the Korean native pig and Landrace reciprocal cross&#125;,<br>  author=&#123;Ahn, Byeongyong and Choi, Min-Kyeung and Yum, Joori and Cho, In-Cheol and Kim, Jin-Hoi and Park, Chankyu&#125;,<br>  journal=&#123;Asian-Australasian journal of animal sciences&#125;,<br>  volume=&#123;32&#125;,<br>  number=&#123;12&#125;,<br>  pages=&#123;1816&#125;,<br>  year=&#123;2019&#125;,<br>  publisher=&#123;Asian-Australasian Association of Animal Production Societies (AAAP)&#125;<br>&#125;<br><br>@article&#123;jadhav2019rna,<br>  title=&#123;RNA-Seq in 296 phased trios provides a high-resolution map of genomic imprinting&#125;,<br>  author=&#123;Jadhav, Bharati and Monajemi, Ramin and Gagalova, Kristina K and Ho, Daniel and Draisma, Harmen HM and van de Wiel, Mark A and Franke, Lude and Heijmans, Bastiaan T and van Meurs, Joyce and Jansen, Rick and others&#125;,<br>  journal=&#123;BMC biology&#125;,<br>  volume=&#123;17&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;50&#125;,<br>  year=&#123;2019&#125;,<br>  publisher=&#123;BioMed Central&#125;<br>&#125;<br><br>@article&#123;barallon2010recommendation,<br>  title=&#123;Recommendation of short tandem repeat profiling for authenticating human cell lines, stem cells, and tissues&#125;,<br>  author=&#123;Barallon, Rita and Bauer, Steven R and Butler, John and Capes-Davis, Amanda and Dirks, Wilhelm G and Elmore, Eugene and Furtado, Manohar and Kline, Margaret C and Kohara, Arihiro and Los, Georgyi V and others&#125;,<br>  journal=&#123;In Vitro Cellular <span class="hljs-keyword">\&amp;</span> Developmental Biology-Animal&#125;,<br>  volume=&#123;46&#125;,<br>  number=&#123;9&#125;,<br>  pages=&#123;727--732&#125;,<br>  year=&#123;2010&#125;,<br>  publisher=&#123;Springer&#125;<br>&#125;<br><br>@article&#123;dumache2018chimerism,<br>  title=&#123;Chimerism Monitoring by Short Tandem Repeat (STR) Markers in Allogeneic Stem Cell Transplantation.&#125;,<br>  author=&#123;Dumache, Raluca and Enache, Alexandra and Barbarii, Ligia and Constantinescu, Carmen and Pascalau, Andreea and Jinca, Cristian and Arghirescu, Smaranda&#125;,<br>  journal=&#123;Clinical laboratory&#125;,<br>  volume=&#123;64&#125;,<br>  number=&#123;9&#125;,<br>  pages=&#123;1535--1543&#125;,<br>  year=&#123;2018&#125;<br>&#125;<br><br>@article&#123;nims2010short,<br>  title=&#123;Short tandem repeat profiling: part of an overall strategy for reducing the frequency of cell misidentification&#125;,<br>  author=&#123;Nims, Raymond W and Sykes, Greg and Cottrill, Karin and Ikonomi, Pranvera and Elmore, Eugene&#125;,<br>  journal=&#123;In Vitro Cellular <span class="hljs-keyword">\&amp;</span> Developmental Biology-Animal&#125;,<br>  volume=&#123;46&#125;,<br>  number=&#123;10&#125;,<br>  pages=&#123;811--819&#125;,<br>  year=&#123;2010&#125;,<br>  publisher=&#123;Springer&#125;<br>&#125;<br><br>@article&#123;huang2002mutation,<br>  title=&#123;Mutation patterns at dinucleotide microsatellite loci in humans&#125;,<br>  author=&#123;Huang, Qing-Yang and Xu, Fu-Hua and Shen, Hui and Deng, Hong-Yi and Liu, Yong-Jun and Liu, Yao-Zhong and Li, Jin-Long and Recker, Robert R and Deng, Hong-Wen&#125;,<br>  journal=&#123;The American Journal of Human Genetics&#125;,<br>  volume=&#123;70&#125;,<br>  number=&#123;3&#125;,<br>  pages=&#123;625--634&#125;,<br>  year=&#123;2002&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;di1998heterogeneity,<br>  title=&#123;Heterogeneity of microsatellite mutations within and between loci, and implications for human demographic histories&#125;,<br>  author=&#123;Di Rienzo, Anna and Donnelly, Peter and Toomajian, Chris and Sisk, Bronwyn and Hill, Adrian and Petzl-Erler, Maria Luiza and Haines, G Ken and Barch, David H&#125;,<br>  journal=&#123;Genetics&#125;,<br>  volume=&#123;148&#125;,<br>  number=&#123;3&#125;,<br>  pages=&#123;1269--1284&#125;,<br>  year=&#123;1998&#125;,<br>  publisher=&#123;Genetics Soc America&#125;<br>&#125;<br><br>@article&#123;hastbacka1992linkage,<br>  title=&#123;Linkage disequilibrium mapping in isolated founder populations: diastrophic dysplasia in Finland&#125;,<br>  author=&#123;H&#123;<span class="hljs-keyword">\&quot;</span>a&#125;stbacka, Johanna and de la Chapelle, Albert and Kaitila, Ilkka and Sistonen, Pertti and Weaver, Alix and Lander, Eric&#125;,<br>  journal=&#123;Nature genetics&#125;,<br>  volume=&#123;2&#125;,<br>  number=&#123;3&#125;,<br>  pages=&#123;204--211&#125;,<br>  year=&#123;1992&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;holtkemper2001mutation,<br>  title=&#123;Mutation rates at two human Y-chromosomal microsatellite loci using small pool PCR techniques&#125;,<br>  author=&#123;Holtkemper, Ulrike and Rolf, Burkhard and Hohoff, Carsten and Forster, Peter and Brinkmann, Bernd&#125;,<br>  journal=&#123;Human molecular genetics&#125;,<br>  volume=&#123;10&#125;,<br>  number=&#123;6&#125;,<br>  pages=&#123;629--633&#125;,<br>  year=&#123;2001&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;huang2002mutation,<br>  title=&#123;Mutation patterns at dinucleotide microsatellite loci in humans&#125;,<br>  author=&#123;Huang, Qing-Yang and Xu, Fu-Hua and Shen, Hui and Deng, Hong-Yi and Liu, Yong-Jun and Liu, Yao-Zhong and Li, Jin-Long and Recker, Robert R and Deng, Hong-Wen&#125;,<br>  journal=&#123;The American Journal of Human Genetics&#125;,<br>  volume=&#123;70&#125;,<br>  number=&#123;3&#125;,<br>  pages=&#123;625--634&#125;,<br>  year=&#123;2002&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;fan2007brief,<br>  title=&#123;A brief review of short tandem repeat mutation&#125;,<br>  author=&#123;Fan, Hao and Chu, Jia-You&#125;,<br>  journal=&#123;Genomics, proteomics <span class="hljs-keyword">\&amp;</span> bioinformatics&#125;,<br>  volume=&#123;5&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;7--14&#125;,<br>  year=&#123;2007&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;benson1999tandem,<br>  title=&#123;Tandem repeats finder: a program to analyze DNA sequences&#125;,<br>  author=&#123;Benson, Gary&#125;,<br>  journal=&#123;Nucleic acids research&#125;,<br>  volume=&#123;27&#125;,<br>  number=&#123;2&#125;,<br>  pages=&#123;573--580&#125;,<br>  year=&#123;1999&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;gymrek2012lobstr,<br>  title=&#123;lobSTR: a short tandem repeat profiler for personal genomes&#125;,<br>  author=&#123;Gymrek, Melissa and Golan, David and Rosset, Saharon and Erlich, Yaniv&#125;,<br>  journal=&#123;Genome research&#125;,<br>  volume=&#123;22&#125;,<br>  number=&#123;6&#125;,<br>  pages=&#123;1154--1162&#125;,<br>  year=&#123;2012&#125;,<br>  publisher=&#123;Cold Spring Harbor Lab&#125;<br>&#125;<br><br>@article&#123;tang2017strscan,<br>  title=&#123;STRScan: targeted profiling of short tandem repeats in whole-genome sequencing data&#125;,<br>  author=&#123;Tang, Haixu and Nzabarushimana, Etienne&#125;,<br>  journal=&#123;BMC bioinformatics&#125;,<br>  volume=&#123;18&#125;,<br>  number=&#123;11&#125;,<br>  pages=&#123;31--36&#125;,<br>  year=&#123;2017&#125;,<br>  publisher=&#123;BioMed Central&#125;<br>&#125;<br><br>@misc&#123;andrews2010fastqc,<br>  title=&#123;FastQC: a quality control tool for high throughput sequence data&#125;,<br>  author=&#123;Andrews, Simon and others&#125;,<br>  year=&#123;2010&#125;,<br>  publisher=&#123;Babraham Bioinformatics, Babraham Institute, Cambridge, United Kingdom&#125;<br>&#125;<br><br>@article&#123;kent2002human,<br>  title=&#123;The human genome browser at UCSC&#125;,<br>  author=&#123;Kent, W James and Sugnet, Charles W and Furey, Terrence S and Roskin, Krishna M and Pringle, Tom H and Zahler, Alan M and Haussler, David&#125;,<br>  journal=&#123;Genome research&#125;,<br>  volume=&#123;12&#125;,<br>  number=&#123;6&#125;,<br>  pages=&#123;996--1006&#125;,<br>  year=&#123;2002&#125;,<br>  publisher=&#123;Cold Spring Harbor Lab&#125;<br>&#125;<br><br>@article&#123;richard2008comparative,<br>  title=&#123;Comparative genomics and molecular dynamics of DNA repeats in eukaryotes&#125;,<br>  author=&#123;Richard, Guy-Franck and Kerrest, Alix and Dujon, Bernard&#125;,<br>  journal=&#123;Microbiology and molecular biology reviews&#125;,<br>  volume=&#123;72&#125;,<br>  number=&#123;4&#125;,<br>  pages=&#123;686--727&#125;,<br>  year=&#123;2008&#125;,<br>  publisher=&#123;Am Soc Microbiol&#125;<br>&#125;<br><br>@article&#123;gulcher2012microsatellite,<br>  title=&#123;Microsatellite markers for linkage and association studies&#125;,<br>  author=&#123;Gulcher, Jeffrey&#125;,<br>  journal=&#123;Cold Spring Harbor Protocols&#125;,<br>  volume=&#123;2012&#125;,<br>  number=&#123;4&#125;,<br>  pages=&#123;pdb--top068510&#125;,<br>  year=&#123;2012&#125;,<br>  publisher=&#123;Cold Spring Harbor Laboratory Press&#125;<br>&#125;<br><br>@article&#123;brinkmann1998mutation,<br>  title=&#123;Mutation rate in human microsatellites: influence of the structure and length of the tandem repeat&#125;,<br>  author=&#123;Brinkmann, Bernd and Klintschar, Michael and Neuhuber, Franz and H&#123;<span class="hljs-keyword">\&quot;</span>u&#125;hne, Julia and Rolf, Burkhard&#125;,<br>  journal=&#123;The American Journal of Human Genetics&#125;,<br>  volume=&#123;62&#125;,<br>  number=&#123;6&#125;,<br>  pages=&#123;1408--1415&#125;,<br>  year=&#123;1998&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;kit1961equilibrium,<br>  title=&#123;Equilibrium sedimentation in density gradients of DNA preparations from animal tissues&#125;,<br>  author=&#123;Kit, Saul&#125;,<br>  journal=&#123;Journal of molecular biology&#125;,<br>  volume=&#123;3&#125;,<br>  number=&#123;6&#125;,<br>  pages=&#123;711--IN2&#125;,<br>  year=&#123;1961&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@book&#123;turnpenny2016emery,<br>  title=&#123;Emery&#x27;s Elements of Medical Genetics E-Book&#125;,<br>  author=&#123;Turnpenny, Peter D and Ellard, Sian&#125;,<br>  year=&#123;2016&#125;,<br>  publisher=&#123;Elsevier Health Sciences&#125;<br>&#125;<br><br>@article&#123;king1997evolutionary,<br>  title=&#123;Evolutionary tuning knobs&#125;,<br>  author=&#123;King, David G and Soller, Morris and Kashi, Yechezkel&#125;,<br>  journal=&#123;Endeavour&#125;,<br>  volume=&#123;21&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;36--40&#125;,<br>  year=&#123;1997&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;chistiakov2006microsatellites,<br>  title=&#123;Microsatellites and their genomic distribution, evolution, function and applications: a review with special reference to fish genetics&#125;,<br>  author=&#123;Chistiakov, Dimitry A and Hellemans, Bart and Volckaert, Filip AM&#125;,<br>  journal=&#123;Aquaculture&#125;,<br>  volume=&#123;255&#125;,<br>  number=&#123;1-4&#125;,<br>  pages=&#123;1--29&#125;,<br>  year=&#123;2006&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;pearson2005repeat,<br>  title=&#123;Repeat instability: mechanisms of dynamic mutations&#125;,<br>  author=&#123;Pearson, Christopher E and Edamura, Kerrie Nichol and Cleary, John D&#125;,<br>  journal=&#123;Nature Reviews Genetics&#125;,<br>  volume=&#123;6&#125;,<br>  number=&#123;10&#125;,<br>  pages=&#123;729--742&#125;,<br>  year=&#123;2005&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;klintschar2004haplotype,<br>  title=&#123;Haplotype studies support slippage as the mechanism of germline mutations in short tandem repeats&#125;,<br>  author=&#123;Klintschar, Michael and Dauber, Eva-Maria and Ricci, Ugo and Cerri, Nicoletta and Immel, Uta-Dorothee and Kleiber, Manfred and Mayr, Wolfgang R&#125;,<br>  journal=&#123;Electrophoresis&#125;,<br>  volume=&#123;25&#125;,<br>  number=&#123;20&#125;,<br>  pages=&#123;3344--3348&#125;,<br>  year=&#123;2004&#125;,<br>  publisher=&#123;Wiley Online Library&#125;<br>&#125;<br><br>@article&#123;brinkmann1998mutation,<br>  title=&#123;Mutation rate in human microsatellites: influence of the structure and length of the tandem repeat&#125;,<br>  author=&#123;Brinkmann, Bernd and Klintschar, Michael and Neuhuber, Franz and H&#123;<span class="hljs-keyword">\&quot;</span>u&#125;hne, Julia and Rolf, Burkhard&#125;,<br>  journal=&#123;The American Journal of Human Genetics&#125;,<br>  volume=&#123;62&#125;,<br>  number=&#123;6&#125;,<br>  pages=&#123;1408--1415&#125;,<br>  year=&#123;1998&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;gymrek2016abundant,<br>  title=&#123;Abundant contribution of short tandem repeats to gene expression variation in humans&#125;,<br>  author=&#123;Gymrek, Melissa and Willems, Thomas and Guilmatre, Audrey and Zeng, Haoyang and Markus, Barak and Georgiev, Stoyan and Daly, Mark J and Price, Alkes L and Pritchard, Jonathan K and Sharp, Andrew J and others&#125;,<br>  journal=&#123;Nature genetics&#125;,<br>  volume=&#123;48&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;22--29&#125;,<br>  year=&#123;2016&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;marcotte1999census,<br>  title=&#123;A census of protein repeats&#125;,<br>  author=&#123;Marcotte, Edward M and Pellegrini, Matteo and Yeates, Todd O and Eisenberg, David&#125;,<br>  journal=&#123;Journal of molecular biology&#125;,<br>  volume=&#123;293&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;151--160&#125;,<br>  year=&#123;1999&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;sutherland1995simple,<br>  title=&#123;Simple tandem DNA repeats and human genetic disease.&#125;,<br>  author=&#123;Sutherland, Grant R and Richards, Robert I&#125;,<br>  journal=&#123;Proceedings of the National Academy of Sciences&#125;,<br>  volume=&#123;92&#125;,<br>  number=&#123;9&#125;,<br>  pages=&#123;3636--3641&#125;,<br>  year=&#123;1995&#125;,<br>  publisher=&#123;National Acad Sciences&#125;<br>&#125;<br><br>@article&#123;hancock2005simple,<br>  title=&#123;Simple sequence repeats in proteins and their significance for network evolution&#125;,<br>  author=&#123;Hancock, John M and Simon, Michelle&#125;,<br>  journal=&#123;Gene&#125;,<br>  volume=&#123;345&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;113--118&#125;,<br>  year=&#123;2005&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;fondon2004molecular,<br>  title=&#123;Molecular origins of rapid and continuous morphological evolution&#125;,<br>  author=&#123;Fondon, John W and Garner, Harold R&#125;,<br>  journal=&#123;Proceedings of the National Academy of Sciences&#125;,<br>  volume=&#123;101&#125;,<br>  number=&#123;52&#125;,<br>  pages=&#123;18058--18063&#125;,<br>  year=&#123;2004&#125;,<br>  publisher=&#123;National Acad Sciences&#125;<br>&#125;<br><br>@article&#123;verstrepen2005intragenic,<br>  title=&#123;Intragenic tandem repeats generate functional variability&#125;,<br>  author=&#123;Verstrepen, Kevin J and Jansen, An and Lewitter, Fran and Fink, Gerald R&#125;,<br>  journal=&#123;Nature genetics&#125;,<br>  volume=&#123;37&#125;,<br>  number=&#123;9&#125;,<br>  pages=&#123;986--990&#125;,<br>  year=&#123;2005&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;moxon1994adaptive,<br>  title=&#123;Adaptive evolution of highly mutable loci in pathogenic bacteria&#125;,<br>  author=&#123;Moxon, E Richard and Rainey, Paul B and Nowak, Martin A and Lenski, Richard E&#125;,<br>  journal=&#123;Current biology&#125;,<br>  volume=&#123;4&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;24--33&#125;,<br>  year=&#123;1994&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;grunewald2015chimeric,<br>  title=&#123;Chimeric EWSR1-FLI1 regulates the Ewing sarcoma susceptibility gene EGR2 via a GGAA microsatellite&#125;,<br>  author=&#123;Gr&#123;<span class="hljs-keyword">\&quot;</span>u&#125;newald, Thomas GP and Bernard, Virginie and Gilardi-Hebenstreit, Pascale and Raynal, Virginie and Surdez, Didier and Aynaud, Marie-Ming and Mirabeau, Olivier and Cidre-Aranaz, Florencia and Tirode, Franck and Zaidi, Sakina and others&#125;,<br>  journal=&#123;Nature genetics&#125;,<br>  volume=&#123;47&#125;,<br>  number=&#123;9&#125;,<br>  pages=&#123;1073--1078&#125;,<br>  year=&#123;2015&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;musa2019cooperation,<br>  title=&#123;Cooperation of cancer drivers with regulatory germline variants shapes clinical outcomes&#125;,<br>  author=&#123;Musa, Julian and Cidre-Aranaz, Florencia and Aynaud, Marie-Ming and Orth, Martin F and Knott, Maximilian ML and Mirabeau, Olivier and Mazor, Gal and Varon, Mor and H&#123;<span class="hljs-keyword">\&quot;</span>o&#125;lting, Tilman LB and Grosset&#123;<span class="hljs-keyword">\^</span>e&#125;te, Sandrine and others&#125;,<br>  journal=&#123;Nature communications&#125;,<br>  volume=&#123;10&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;1--10&#125;,<br>  year=&#123;2019&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;forster2015elevated,<br>  title=&#123;Elevated germline mutation rate in teenage fathers&#125;,<br>  author=&#123;Forster, Peter and Hohoff, Carsten and Dunkelmann, Bettina and Sch&#123;<span class="hljs-keyword">\&quot;</span>u&#125;renkamp, Marianne and Pfeiffer, Heidi and Neuhuber, Franz and Brinkmann, Bernd&#125;,<br>  journal=&#123;Proceedings of the Royal Society B: Biological Sciences&#125;,<br>  volume=&#123;282&#125;,<br>  number=&#123;1803&#125;,<br>  pages=&#123;20142898&#125;,<br>  year=&#123;2015&#125;,<br>  publisher=&#123;The Royal Society&#125;<br>&#125;<br><br>@article&#123;weber1993mutation,<br>  title=&#123;Mutation of human short tandem repeats&#125;,<br>  author=&#123;Weber, James L and Wong, Carmen&#125;,<br>  journal=&#123;Human molecular genetics&#125;,<br>  volume=&#123;2&#125;,<br>  number=&#123;8&#125;,<br>  pages=&#123;1123--1128&#125;,<br>  year=&#123;1993&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;jarne1996microsatellites,<br>  title=&#123;Microsatellites, from molecules to populations and back&#125;,<br>  author=&#123;Jarne, Philippe and Lagoda, Pierre JL&#125;,<br>  journal=&#123;Trends in ecology <span class="hljs-keyword">\&amp;</span> evolution&#125;,<br>  volume=&#123;11&#125;,<br>  number=&#123;10&#125;,<br>  pages=&#123;424--429&#125;,<br>  year=&#123;1996&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;kruglyak1998equilibrium,<br>  title=&#123;Equilibrium distributions of microsatellite repeat length resulting from a balance between slippage events and point mutations&#125;,<br>  author=&#123;Kruglyak, Semyon and Durrett, Richard T and Schug, Malcolm D and Aquadro, Charles F&#125;,<br>  journal=&#123;Proceedings of the National Academy of Sciences&#125;,<br>  volume=&#123;95&#125;,<br>  number=&#123;18&#125;,<br>  pages=&#123;10774--10778&#125;,<br>  year=&#123;1998&#125;,<br>  publisher=&#123;National Acad Sciences&#125;<br>&#125;<br><br>@article&#123;reik2001genomic,<br>  title=&#123;Genomic imprinting: parental influence on the genome&#125;,<br>  author=&#123;Reik, Wolf and Walter, J&#123;<span class="hljs-keyword">\&quot;</span>o&#125;rn&#125;,<br>  journal=&#123;Nature Reviews Genetics&#125;,<br>  volume=&#123;2&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;21--32&#125;,<br>  year=&#123;2001&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;morison2005census,<br>  title=&#123;A census of mammalian imprinting&#125;,<br>  author=&#123;Morison, Ian M and Ramsay, Joshua P and Spencer, Hamish G&#125;,<br>  journal=&#123;TRENDS in Genetics&#125;,<br>  volume=&#123;21&#125;,<br>  number=&#123;8&#125;,<br>  pages=&#123;457--465&#125;,<br>  year=&#123;2005&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;lau1994loss,<br>  title=&#123;Loss of the imprinted IGF2/cation-independent mannose 6-phosphate receptor results in fetal overgrowth and perinatal lethality.&#125;,<br>  author=&#123;Lau, MM and Stewart, CE and Liu, Ziying and Bhatt, Harshida and Rotwein, Peter and Stewart, Colin L&#125;,<br>  journal=&#123;Genes <span class="hljs-keyword">\&amp;</span> Development&#125;,<br>  volume=&#123;8&#125;,<br>  number=&#123;24&#125;,<br>  pages=&#123;2953--2963&#125;,<br>  year=&#123;1994&#125;,<br>  publisher=&#123;Cold Spring Harbor Lab&#125;<br>&#125;<br><br>@article&#123;haig1989selective,<br>  title=&#123;Selective forces in the emergence of the seed habit&#125;,<br>  author=&#123;Haig, David and Westoby, Mark&#125;,<br>  journal=&#123;Biological Journal of the Linnean Society&#125;,<br>  volume=&#123;38&#125;,<br>  number=&#123;3&#125;,<br>  pages=&#123;215--238&#125;,<br>  year=&#123;1989&#125;,<br>  publisher=&#123;Oxford University Press&#125;<br>&#125;<br><br>@article&#123;moore1991genomic,<br>  title=&#123;Genomic imprinting in mammalian development: a parental tug-of-war&#125;,<br>  author=&#123;Moore, Tom and Haig, David&#125;,<br>  journal=&#123;Trends in genetics&#125;,<br>  volume=&#123;7&#125;,<br>  number=&#123;2&#125;,<br>  pages=&#123;45--49&#125;,<br>  year=&#123;1991&#125;,<br>  publisher=&#123;Elsevier Current Trends&#125;<br>&#125;<br><br>@article&#123;lefebvre1998abnormal,<br>  title=&#123;Abnormal maternal behaviour and growth retardation associated with loss of the imprinted gene Mest&#125;,<br>  author=&#123;Lefebvre, Louis and Viville, St&#123;<span class="hljs-keyword">\&#x27;</span>e&#125;phane and Barton, Sheila C and Ishino, Fumitoshi and Keverne, Eric B and Surani, M Azim&#125;,<br>  journal=&#123;Nature genetics&#125;,<br>  volume=&#123;20&#125;,<br>  number=&#123;2&#125;,<br>  pages=&#123;163--169&#125;,<br>  year=&#123;1998&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;hamed2012cellular,<br>  title=&#123;Cellular functions of genetically imprinted genes in human and mouse as annotated in the gene ontology&#125;,<br>  author=&#123;Hamed, Mohamed and Ismael, Siba and Paulsen, Martina and Helms, Volkhard&#125;,<br>  journal=&#123;PloS one&#125;,<br>  volume=&#123;7&#125;,<br>  number=&#123;11&#125;,<br>  pages=&#123;e50285&#125;,<br>  year=&#123;2012&#125;,<br>  publisher=&#123;Public Library of Science&#125;<br>&#125;<br><br>@article&#123;renfree2009evolution,<br>  title=&#123;Evolution of genomic imprinting: insights from marsupials and monotremes&#125;,<br>  author=&#123;Renfree, Marilyn B and Hore, Timothy A and Shaw, Geoffrey and Marshall Graves, Jennifer A and Pask, Andrew J&#125;,<br>  journal=&#123;Annual review of genomics and human genetics&#125;,<br>  volume=&#123;10&#125;,<br>  pages=&#123;241--262&#125;,<br>  year=&#123;2009&#125;,<br>  publisher=&#123;Annual Reviews&#125;<br>&#125;<br><br>@article&#123;moore1991genomic,<br>  title=&#123;Genomic imprinting in mammalian development: a parental tug-of-war&#125;,<br>  author=&#123;Moore, Tom and Haig, David&#125;,<br>  journal=&#123;Trends in genetics&#125;,<br>  volume=&#123;7&#125;,<br>  number=&#123;2&#125;,<br>  pages=&#123;45--49&#125;,<br>  year=&#123;1991&#125;,<br>  publisher=&#123;Elsevier Current Trends&#125;<br>&#125;<br><br>@article&#123;varmuza1994genomic,<br>  title=&#123;Genomic imprinting—defusing the ovarian time bomb&#125;,<br>  author=&#123;Varmuza, Sue and Mann, Mellissa&#125;,<br>  journal=&#123;Trends in Genetics&#125;,<br>  volume=&#123;10&#125;,<br>  number=&#123;4&#125;,<br>  pages=&#123;118--123&#125;,<br>  year=&#123;1994&#125;,<br>  publisher=&#123;Elsevier&#125;<br>&#125;<br><br>@article&#123;wilkins2003good,<br>  title=&#123;What good is genomic imprinting: the function of parent-specific gene expression&#125;,<br>  author=&#123;Wilkins, Jon F and Haig, David&#125;,<br>  journal=&#123;Nature Reviews Genetics&#125;,<br>  volume=&#123;4&#125;,<br>  number=&#123;5&#125;,<br>  pages=&#123;359--368&#125;,<br>  year=&#123;2003&#125;,<br>  publisher=&#123;Nature Publishing Group&#125;<br>&#125;<br><br>@article&#123;grossniklaus2014transcriptional,<br>  title=&#123;Transcriptional silencing by polycomb-group proteins&#125;,<br>  author=&#123;Grossniklaus, Ueli and Paro, Renato&#125;,<br>  journal=&#123;Cold Spring Harbor perspectives in biology&#125;,<br>  volume=&#123;6&#125;,<br>  number=&#123;11&#125;,<br>  pages=&#123;a019331&#125;,<br>  year=&#123;2014&#125;,<br>  publisher=&#123;Cold Spring Harbor Lab&#125;<br>&#125;<br><br>@article&#123;bartolomei1997genomic,<br>  title=&#123;Genomic imprinting in mammals&#125;,<br>  author=&#123;Bartolomei, Marisa S and Tilghman, Shirley M&#125;,<br>  journal=&#123;Annual review of genetics&#125;,<br>  volume=&#123;31&#125;,<br>  number=&#123;1&#125;,<br>  pages=&#123;493--525&#125;,<br>  year=&#123;1997&#125;,<br>  publisher=&#123;Annual Reviews 4139 El Camino Way, PO Box 10139, Palo Alto, CA 94303-0139, USA&#125;<br>&#125;<br><br><br><br><br></code></pre></td></tr></table></figure>
            </div>
            <hr>
            <div>
              <div class="post-metas mb-3">
                
                  <div class="post-meta mr-3">
                    <i class="iconfont icon-category"></i>
                    
                      <a class="hover-with-bg" href="/categories/BIOINFO-EXPERIMENT-RECORDS/">BIOINFO EXPERIMENT RECORDS</a>
                    
                  </div>
                
                
                  <div class="post-meta">
                    <i class="iconfont icon-tags"></i>
                    
                      <a class="hover-with-bg" href="/tags/RME/">RME</a>
                    
                  </div>
                
              </div>
              
                <p class="note note-warning">
                  
                    本博客所有文章除特别声明外，均采用 <a target="_blank" href="https://creativecommons.org/licenses/by-sa/4.0/deed.zh" rel="nofollow noopener noopener">CC BY-SA 4.0 协议</a> ，转载请注明出处！
                  
                </p>
              
              
                <div class="post-prevnext">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/2021/06/17/20210617%E7%94%9F%E7%89%A9%E7%BB%9F%E8%AE%A1%E5%AD%A6%E7%8C%9C%E9%A2%98/">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">20210617生物统计学猜题</span>
                        <span class="visible-mobile">上一篇</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2021/01/19/20210119UCSD_Bioinfomatics/">
                        <span class="hidden-mobile">UCSD Bioinfomatics</span>
                        <span class="visible-mobile">下一篇</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
              <!-- Comments -->
              <article class="comments" id="comments">
                
                  
                
                
  <div class="disqus" style="width:100%">
    <div id="disqus_thread"></div>
    <script>
      Fluid.utils.waitElementVisible('disqus_thread', function() {
        Fluid.utils.createCssLink('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqusjs.css');
        Fluid.utils.createScript('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqus.js', function () {
          new DisqusJS({
            shortname: 'fluid',
            apikey: ''
          });
        });
      });
    </script>
    <noscript>Please enable JavaScript to view the
      <a target="_blank" href="https://disqus.com/?ref_noscript" rel="nofollow noopener noopener">comments powered by Disqus.</a>
    </noscript>
  </div>


              </article>
            
          </article>
        </div>
      </div>
    </div>
    
      <div class="d-none d-lg-block col-lg-2 toc-container" id="toc-ctn">
        <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;目录</p>
  <div class="toc-body" id="toc-body"></div>
</div>

      </div>
    
  </div>
</div>

<!-- Custom -->


    

    
      <a id="scroll-top-button" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">搜索</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v"
                 for="local-search-input">关键词</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>
    

    
  </main>

  <footer class="text-center mt-5 py-3">
  <div class="footer-content">
     <a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a> <i class="iconfont icon-love"></i> <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener"><span>Fluid</span></a> 
  </div>
  

  

  
</footer>

<!-- SCRIPTS -->

  <script  src="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" ></script>
<script  src="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/js/bootstrap.min.js" ></script>
<script  src="/js/debouncer.js" ></script>
<script  src="/js/events.js" ></script>
<script  src="/js/plugins.js" ></script>

<!-- Plugins -->


  
    <script  src="/js/lazyload.js" ></script>
  



  



  <script  src="https://cdn.jsdelivr.net/npm/tocbot@4.12.0/dist/tocbot.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/anchor-js@4.3.0/anchor.min.js" ></script>



  <script defer src="https://cdn.jsdelivr.net/npm/clipboard@2.0.6/dist/clipboard.min.js" ></script>






  <script  src="https://cdn.jsdelivr.net/npm/typed.js@2.0.11/lib/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var title = document.getElementById('subtitle').title;
      
      typing(title)
      
    })(window, document);
  </script>



  <script  src="/js/local-search.js" ></script>
  <script>
    (function () {
      var path = "/local-search.xml";
      $('#local-search-input').on('click', function() {
        searchFunc(path, 'local-search-input', 'local-search-result');
      });
      $('#modalSearch').on('shown.bs.modal', function() {
        $('#local-search-input').focus();
      });
    })()
  </script>















<!-- 主题的启动项 保持在最底部 -->
<script  src="/js/boot.js" ></script>



</body>
</html>
