

<!DOCTYPE html>
<html lang="zh-CN" data-default-color-scheme=&#34;auto&#34;>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/favicon.png">
  <link rel="icon" href="/img/favicon.png">
  <meta name="viewport"
        content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  
    <meta name="description" content="">
  
  <meta name="author" content="John Doe">
  <meta name="keywords" content="">
  
  <title>2021.9.27实验记录iPSYCH_PGC_ASD_Nov2017 - Hexo</title>

  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" />


  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/github-markdown-css@4.0.0/github-markdown.min.css" />
  <link  rel="stylesheet" href="/lib/hint/hint.min.css" />

  
    
    
      
      <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/highlight.js@10.4.0/styles/github-gist.min.css" />
    
  

  
    <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.css" />
  



<!-- 主题依赖的图标库，不要自行修改 -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_ba1fz6golrf.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_kmeydafke9r.css">


<link  rel="stylesheet" href="/css/main.css" />

<!-- 自定义样式保持在最底部 -->


  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    var CONFIG = {"hostname":"example.com","root":"/","version":"1.8.8","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"right","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"copy_btn":true,"image_zoom":{"enable":true},"toc":{"enable":true,"headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"onlypost":false},"web_analytics":{"enable":false,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":null,"app_key":null,"server_url":null}}};
  </script>
  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
<meta name="generator" content="Hexo 5.4.0"></head>


<body>
  <header style="height: 70vh;">
    <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand"
       href="/">&nbsp;<strong>Liu Keqin</strong>&nbsp;</a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/">
                <i class="iconfont icon-home-fill"></i>
                首页
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/">
                <i class="iconfont icon-archive-fill"></i>
                归档
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/">
                <i class="iconfont icon-category-fill"></i>
                分类
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/">
                <i class="iconfont icon-tags-fill"></i>
                标签
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/">
                <i class="iconfont icon-user-fill"></i>
                关于
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" data-toggle="modal" data-target="#modalSearch">&nbsp;<i
                class="iconfont icon-search"></i>&nbsp;</a>
          </li>
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" href="javascript:">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

    <div class="banner" id="banner" parallax=true
         style="background: url('/img/default.png') no-repeat center center;
           background-size: cover;">
      <div class="full-bg-img">
        <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
          <div class="page-header text-center fade-in-up">
            <span class="h2" id="subtitle" title="2021.9.27实验记录iPSYCH_PGC_ASD_Nov2017">
              
            </span>

            
              <div class="mt-3">
  
  
    <span class="post-meta">
      <i class="iconfont icon-date-fill" aria-hidden="true"></i>
      <time datetime="2021-09-27 17:18" pubdate>
        2021年9月27日 下午
      </time>
    </span>
  
</div>

<div class="mt-1">
  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-chart"></i>
      3.1k 字
    </span>
  

  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-clock-fill"></i>
      
      
      59
       分钟
    </span>
  

  
  
</div>

            
          </div>

          
        </div>
      </div>
    </div>
  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="d-none d-lg-block col-lg-2"></div>
    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div class="py-5" id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">2021.9.27实验记录iPSYCH_PGC_ASD_Nov2017</h1>
            
            <div class="markdown-body">
              <h1 id="2021-9-27实验记录iPSYCH-PGC-ASD-Nov2017"><a href="#2021-9-27实验记录iPSYCH-PGC-ASD-Nov2017" class="headerlink" title="2021.9.27实验记录iPSYCH_PGC_ASD_Nov2017"></a>2021.9.27实验记录iPSYCH_PGC_ASD_Nov2017</h1><p><strong>参考文献：</strong></p>
<blockquote>
<p>Alonso-Gonzalez, Aitana, et al. “Novel gene-based analysis of ASD GWAS: Insight into the biological role of associated genes.” <em>Frontiers in genetics</em> 10 (2019): 733.</p>
</blockquote>
<p><strong>数据来源：</strong></p>
<p>PGC数据库：<a target="_blank" rel="noopener" href="https://www.med.unc.edu/pgc/download-results/">https://www.med.unc.edu/pgc/download-results/</a></p>
<p><strong>数据描述：</strong></p>
<table>
<thead>
<tr>
<th align="left">pgcGroup</th>
<th align="left">phenotype</th>
<th align="left">publication</th>
<th align="left">journal</th>
<th align="left">pubMedIDlink</th>
<th align="left">Data DOI</th>
</tr>
</thead>
<tbody><tr>
<td align="left">ASD</td>
<td align="left">Autism spectrum disorder</td>
<td align="left">asd2019</td>
<td align="left">Nature Genetics</td>
<td align="left"><a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/30804558">30804558</a></td>
<td align="left">10.6084/m9.figshare.14671989</td>
</tr>
</tbody></table>
<p><strong>运行环境：</strong></p>
<p>MacOS Catalina 10.15.7</p>
<p><strong>软件及网页工具：</strong></p>
<p>R version 4.0.2 (2020-06-22) </p>
<p>Pascal (Pathway scoring algorithm) (<a target="_blank" rel="noopener" href="https://www2.unil.ch/cbg/index.php?title=PASCAL">https://www2.unil.ch/cbg/index.php?title=PASCAL</a>)</p>
<p>FunCoup v.4.0 (<a target="_blank" rel="noopener" href="http://funcoup.sbc.su.se/search/">http://funcoup.sbc.su.se/search/</a>)</p>
<p>PathwAX II (<a target="_blank" rel="noopener" href="https://pathwax.sbc.su.se/">https://pathwax.sbc.su.se/</a>)</p>
<p>GENE2FUNC (<a target="_blank" rel="noopener" href="http://fuma.ctglab.nl/">http://fuma.ctglab.nl/</a>)</p>
<p>dbMDEGA (<a target="_blank" rel="noopener" href="https://dbmdega.shinyapps.io/dbMDEGA/">https://dbmdega.shinyapps.io/dbMDEGA/</a>)</p>
<p>LocusZoom tool (<a target="_blank" rel="noopener" href="http://locuszoom.org/">http://locuszoom.org/</a>) </p>
<h2 id="1-查看PGC统计摘要文件格式"><a href="#1-查看PGC统计摘要文件格式" class="headerlink" title="1.查看PGC统计摘要文件格式"></a>1.查看PGC统计摘要文件格式</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs R">iPSYCH_PGC_ASD_Nov2017 &lt;- read.delim(<span class="hljs-string">&quot;/Users/keqinliu/Downloads/14671989/iPSYCH_PGC_ASD_Nov2017&quot;</span>)<br>print(head(iPSYCH_PGC_ASD_Nov2017))<br><span class="hljs-comment">##   CHR        SNP        BP A1 A2  INFO      OR     SE        P</span><br><span class="hljs-comment">## 1   8 rs62513865 101592213  T  C 0.949 1.00652 0.0270 0.808600</span><br><span class="hljs-comment">## 2   8 rs79643588 106973048  A  G 0.997 1.01786 0.0240 0.460600</span><br><span class="hljs-comment">## 3   8 rs17396518 108690829  T  G 0.987 0.96127 0.0140 0.004647</span><br><span class="hljs-comment">## 4   8   rs983166 108681675  A  C 0.998 0.97990 0.0139 0.145200</span><br><span class="hljs-comment">## 5   8 rs28842593 103044620  T  C 0.857 0.99591 0.0203 0.841500</span><br><span class="hljs-comment">## 6   8 rs35107696 109712249  A AT 0.999 1.01308 0.0165 0.430200</span><br></code></pre></td></tr></table></figure>



<h2 id="2-运行PASCAL"><a href="#2-运行PASCAL" class="headerlink" title="2.运行PASCAL"></a>2.运行PASCAL</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">(base) KeqindeMacBook-Air:PASCAL keqinliu$ Pascal --pval=/Users/keqinliu/Downloads/iPSYCH_PGC_ASD_Nov2017 --runpathway=on --mergedistance=1<br></code></pre></td></tr></table></figure>

<p>出现报错：</p>
<figure class="highlight asciidoc"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><code class="hljs asciidoc">Hello I&#x27;m Pascal<br>WARN: Unknown option (ignored): --mergedistance=1<br>genescoring not set. set to sum as default.<br>2021/09/27_10:29:05:N<br><span class="hljs-section">SETTINGS FILE</span><br><span class="hljs-section">-------------</span><br><br><span class="hljs-bullet">- </span>No settings file specified<br><span class="hljs-bullet">- </span>Loading default settings<br><br>random seed:<br>2<br>Writing file: output/settingsOut.txt<br>Class found = ch.unil.genescore.main.Settings<br>Package = package ch.unil.genescore.main<br><span class="hljs-section">LOADING INPUT FILES</span><br><span class="hljs-section">-------------------</span><br><br>Load annotation: resources/annotation/ucsc/ucsc<span class="hljs-emphasis">_known_genes_2013-09-03.txt</span><br><span class="hljs-emphasis">Loading annotation: resources/annotation/ucsc/ucsc_known_genes_2013-09-03.txt</span><br><span class="hljs-emphasis">Reading file: resources/annotation/ucsc/ucsc_known_genes_2013-09-03.txt</span><br><span class="hljs-emphasis">Reading file: resources/annotation/hla/hlaGenesEntrezIds.txt</span><br><span class="hljs-emphasis">- 205 genes removed because they are in the excluded genes file</span><br><span class="hljs-emphasis">Reading file: /Users/keqinliu/Downloads/iPSYCH_PGC_ASD_</span>Nov2017<br><span class="hljs-symbol">WARNING: </span>snp CHRhas no valid p-value (use NA or NaN for omitted values).<br><span class="hljs-symbol">WARNING: </span>snp 8has no valid p-value (use NA or NaN for omitted values).<br><span class="hljs-symbol">WARNING: </span>snp 8has no valid p-value (use NA or NaN for omitted values).<br></code></pre></td></tr></table></figure>

<p>检查报错，发现Pascal提示snp无有效p-value</p>
<p>于是检查Pascal自带的示例文件 EUR.CARDIoGRAM_2010_lipids.HDL_ONE.txt</p>
<figure class="highlight apache"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><code class="hljs apache"><span class="hljs-attribute">rs3764261</span>	<span class="hljs-number">7</span>.<span class="hljs-number">10</span>e-<span class="hljs-number">380</span><br><span class="hljs-attribute">rs173539</span>	<span class="hljs-number">2</span>.<span class="hljs-number">50</span>e-<span class="hljs-number">379</span><br><span class="hljs-attribute">rs247616</span>	<span class="hljs-number">8</span>.<span class="hljs-number">54</span>e-<span class="hljs-number">378</span><br><span class="hljs-attribute">rs7205804</span>	<span class="hljs-number">2</span>.<span class="hljs-number">00</span>e-<span class="hljs-number">329</span><br><span class="hljs-attribute">rs1532624</span>	<span class="hljs-number">2</span>.<span class="hljs-number">93</span>e-<span class="hljs-number">327</span><br><span class="hljs-attribute">rs11508026</span>	<span class="hljs-number">2</span>.<span class="hljs-number">08</span>e-<span class="hljs-number">323</span><br><span class="hljs-attribute">rs1800775</span>	<span class="hljs-number">2</span>.<span class="hljs-number">06</span>e-<span class="hljs-number">306</span><br><span class="hljs-attribute">rs711752</span>	<span class="hljs-number">2</span>.<span class="hljs-number">07</span>e-<span class="hljs-number">297</span><br><span class="hljs-attribute">rs1864163</span>	<span class="hljs-number">3</span>.<span class="hljs-number">69</span>e-<span class="hljs-number">281</span><br><span class="hljs-attribute">rs11076175</span>	<span class="hljs-number">3</span>.<span class="hljs-number">99</span>e-<span class="hljs-number">272</span><br><span class="hljs-attribute">rs9939224</span>	<span class="hljs-number">5</span>.<span class="hljs-number">61</span>e-<span class="hljs-number">272</span><br><span class="hljs-attribute">rs7499892</span>	<span class="hljs-number">1</span>.<span class="hljs-number">17</span>e-<span class="hljs-number">271</span><br><span class="hljs-attribute">rs7203984</span>	<span class="hljs-number">2</span>.<span class="hljs-number">34</span>e-<span class="hljs-number">242</span><br><span class="hljs-attribute">rs9926440</span>	<span class="hljs-number">4</span>.<span class="hljs-number">43</span>e-<span class="hljs-number">215</span><br><span class="hljs-attribute">rs9929488</span>	<span class="hljs-number">4</span>.<span class="hljs-number">26</span>e-<span class="hljs-number">190</span><br><span class="hljs-attribute">rs12448528</span>	<span class="hljs-number">5</span>.<span class="hljs-number">60</span>e-<span class="hljs-number">170</span><br><span class="hljs-attribute">rs9989419</span>	<span class="hljs-number">2</span>.<span class="hljs-number">96</span>e-<span class="hljs-number">162</span><br><span class="hljs-attribute">rs12708967</span>	<span class="hljs-number">6</span>.<span class="hljs-number">48</span>e-<span class="hljs-number">160</span><br><span class="hljs-attribute">rs289714</span>	<span class="hljs-number">3</span>.<span class="hljs-number">05</span>e-<span class="hljs-number">153</span><br><span class="hljs-attribute">rs17368435</span>	<span class="hljs-number">8</span>.<span class="hljs-number">57</span>e-<span class="hljs-number">121</span><br></code></pre></td></tr></table></figure>

<p>于是决定将iPSYCH_PGC_ASD_Nov2017统计摘要格式更改为Pascal示例文件的格式：</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><code class="hljs r">iPSYCH_PGC_ASD_Nov2017 &lt;- read.delim(<span class="hljs-string">&quot;/Users/keqinliu/Downloads/iPSYCH_PGC_ASD_Nov2017&quot;</span>)<br>print(head(iPSYCH_PGC_ASD_Nov2017))<br><br>new &lt;- cbind(iPSYCH_PGC_ASD_Nov2017$SNP, iPSYCH_PGC_ASD_Nov2017$P)<br>print(head(new))<br><span class="hljs-comment">##     [,1]         [,2]      </span><br><span class="hljs-comment">## [1,] &quot;rs62513865&quot; &quot;0.8086&quot;  </span><br><span class="hljs-comment">## [2,] &quot;rs79643588&quot; &quot;0.4606&quot;  </span><br><span class="hljs-comment">## [3,] &quot;rs17396518&quot; &quot;0.004647&quot;</span><br><span class="hljs-comment">## [4,] &quot;rs983166&quot;   &quot;0.1452&quot;  </span><br><span class="hljs-comment">## [5,] &quot;rs28842593&quot; &quot;0.8415&quot;  </span><br><span class="hljs-comment">## [6,] &quot;rs35107696&quot; &quot;0.4302&quot;  </span><br></code></pre></td></tr></table></figure>

<p>将修改后的文件保存为：</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs R">write_excel_csv(new, file=<span class="hljs-string">&quot;iPSYCH_PGC_ASD_Nov2017_new&quot;</span>, delim = <span class="hljs-string">&quot; &quot;</span>)<br></code></pre></td></tr></table></figure>

<p>运行Pascal</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">(base) KeqindeMacBook-Air:PASCAL keqinliu$ Pascal --pval=/Users/keqinliu/iPSYCH_PGC_ASD_Nov2017_new --runpathway=on --chr=8<br></code></pre></td></tr></table></figure>

<figure class="highlight crystal"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs crystal">Writing <span class="hljs-symbol">file:</span> output/iPSYCH_PGC_ASD_Nov2017_new.sum.numSnpError.chr8.txt<br><br>Reading <span class="hljs-symbol">file:</span> resources/genesets/msigdb/msigBIOCARTA_KEGG_REACTOME.gmt<br><br><span class="hljs-symbol">Loaded:</span><br>- <span class="hljs-number">1077</span> gene sets<br>- <span class="hljs-number">799</span> genes <span class="hljs-keyword">with</span> pathway / gene set <span class="hljs-keyword">annotation</span><br><span class="hljs-symbol">Ignored:</span><br>- <span class="hljs-number">7584</span> genes not found in loaded genome <span class="hljs-keyword">annotation</span> (<span class="hljs-title">UCSC</span> <span class="hljs-title">known</span> <span class="hljs-title">genes</span><br></code></pre></td></tr></table></figure>

<figure class="highlight asciidoc"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><code class="hljs asciidoc"><span class="hljs-section">CREATING META-GENES</span><br><span class="hljs-section">-------------------</span><br><br>Merging genes of the same pathway that are &lt;1.0mb apart:<br><span class="hljs-bullet">- </span>42 meta-genes created<br><span class="hljs-bullet">- </span>5.023809523809524 genes per meta-gene on average (min=2, max=8)<br><br>Computing gene scores for meta-genes ...<br>Writing file: output/iPSYCH<span class="hljs-emphasis">_PGC_ASD_Nov2017_new.sum.fusion.genescores.chr8.txt</span><br><span class="hljs-emphasis">Format: gene_</span>id, symbol, #snps, runtime, [warnings]<br></code></pre></td></tr></table></figure>

<figure class="highlight gams"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs gams">- Runtime: <span class="hljs-number">0</span>h <span class="hljs-number">0</span><span class="hljs-built_in">min</span> <span class="hljs-number">14</span>s <span class="hljs-number">139</span>ms<br>- <span class="hljs-number">1077</span> gene <span class="hljs-keyword">sets</span> with <span class="hljs-comment">p-value &lt; 1.0</span><br><br>Writing <span class="hljs-comment">file: output</span>/iPSYCH_PGC_ASD_Nov2017_new.PathwaySet--msigBIOCARTA_KEGG_REACTOME--<span class="hljs-keyword">sum</span>.chr8.txt<br></code></pre></td></tr></table></figure>



<h2 id="3-Pascal生成文件"><a href="#3-Pascal生成文件" class="headerlink" title="3.Pascal生成文件"></a>3.Pascal生成文件</h2><p>共5个：</p>
<p>iPSYCH_PGC_ASD_Nov2017_new.PathwaySet–msigBIOCARTA_KEGG_REACTOME–sum.chr8.txt</p>
<p>iPSYCH_PGC_ASD_Nov2017_new.sum.genescores.chr8.txt</p>
<p>iPSYCH_PGC_ASD_Nov2017_new.sum.fusion.genescores.chr8.txt</p>
<p>iPSYCH_PGC_ASD_Nov2017_new.sum.numSnpError.chr8.txt</p>
<p>settingsOut.txt</p>
<h3 id="3-1-检查生成的sum-genescores"><a href="#3-1-检查生成的sum-genescores" class="headerlink" title="3.1.检查生成的sum_genescores"></a>3.1.检查生成的sum_genescores</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs r">sum_genescores &lt;- read.delim(<span class="hljs-string">&quot;/Users/keqinliu/iPSYCH_PGC_ASD_Nov2017_new.sum.genescores.chr8.txt&quot;</span>)<br>print(head(sum_genescores))<br>sum_genescores[order(sum_genescores$pvalue),] <span class="hljs-comment">#按照pvalue升序排列，筛选前10位。</span><br></code></pre></td></tr></table></figure>

<table>
<thead>
<tr>
<th>chromosome</th>
<th>start</th>
<th>end</th>
<th>strand</th>
<th>gene_id</th>
<th>gene_symbol</th>
<th>numSnps</th>
<th>pvalue</th>
</tr>
</thead>
<tbody><tr>
<td>chr8</td>
<td>10530146</td>
<td>10558103</td>
<td>+</td>
<td>203076</td>
<td>C8orf74</td>
<td>314</td>
<td>1.773485e-06</td>
</tr>
<tr>
<td>chr8</td>
<td>10581277</td>
<td>10697299</td>
<td>-</td>
<td>83595</td>
<td>SOX7</td>
<td>752</td>
<td>4.839478e-06</td>
</tr>
<tr>
<td>chr8</td>
<td>10622883</td>
<td>10697299</td>
<td>-</td>
<td>54984</td>
<td>PINX1</td>
<td>684</td>
<td>7.776723e-06</td>
</tr>
<tr>
<td>chr8</td>
<td>8641998</td>
<td>8751131</td>
<td>-</td>
<td>9258</td>
<td>MFHAS1</td>
<td>853</td>
<td>7.811383e-06</td>
</tr>
<tr>
<td>chr8</td>
<td>10682882</td>
<td>10682953</td>
<td>-</td>
<td>100302166</td>
<td>MIR1322</td>
<td>429</td>
<td>2.032506e-05</td>
</tr>
<tr>
<td>chr8</td>
<td>11278972</td>
<td>11324276</td>
<td>-</td>
<td>83648</td>
<td>FAM167A</td>
<td>557</td>
<td>2.310428e-05</td>
</tr>
<tr>
<td>chr8</td>
<td>11225910</td>
<td>11296166</td>
<td>+</td>
<td>83656</td>
<td>C8orf12</td>
<td>622</td>
<td>2.538882e-05</td>
</tr>
<tr>
<td>chr8</td>
<td>8860313</td>
<td>8890849</td>
<td>+</td>
<td>90459</td>
<td>ERI1</td>
<td>504</td>
<td>3.097496e-05</td>
</tr>
<tr>
<td>chr8</td>
<td>11351520</td>
<td>11422108</td>
<td>+</td>
<td>640</td>
<td>BLK</td>
<td>582</td>
<td>3.772496e-05</td>
</tr>
<tr>
<td>chr8</td>
<td>8993763</td>
<td>9009152</td>
<td>-</td>
<td>79660</td>
<td>PPP1R3B</td>
<td>287</td>
<td>3.868200e-05</td>
</tr>
</tbody></table>
<h3 id="3-2-检查生成的pathwayset"><a href="#3-2-检查生成的pathwayset" class="headerlink" title="3.2.检查生成的pathwayset"></a>3.2.检查生成的pathwayset</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs R">PathwaySet &lt;- read.delim(<span class="hljs-string">&quot;/Volumes/My\ Passport\ for\ Mac/biosoftware/PASCAL/output/iPSYCH_PGC_ASD_Nov2017_new.PathwaySet--msigBIOCARTA_KEGG_REACTOME--sum.chr8.txt&quot;</span>)<br>print(head(PathwaySet))<br>PathwaySet[order(PathwaySet$chi2Pvalue),] <span class="hljs-comment">#按照chi1Pvalue升序排列，筛选pvalue均＜0.05</span><br></code></pre></td></tr></table></figure>

<table>
<thead>
<tr>
<th>Name</th>
<th>chi2Pvalue</th>
<th>empPvalue</th>
</tr>
</thead>
<tbody><tr>
<td>REACTOME_ACTIVATION_OF_KAINATE_RECEPTORS_UPON_GLUTAMATE_BINDING</td>
<td>0.02750000</td>
<td>0.02767</td>
</tr>
<tr>
<td>REACTOME_IONOTROPIC_ACTIVITY_OF_KAINATE_RECEPTORS</td>
<td>0.02750000</td>
<td>0.02720</td>
</tr>
<tr>
<td>REACTOME_SIGNAL_TRANSDUCTION_BY_L1</td>
<td>0.03125000</td>
<td>0.03123</td>
</tr>
<tr>
<td>REACTOME_ANTIGEN_ACTIVATES_B_CELL_RECEPTOR_LEADING_TO_GENERATION_OF_SECOND_MESSENGERS</td>
<td>0.03131834</td>
<td>0.02722</td>
</tr>
<tr>
<td>BIOCARTA_ALK_PATHWAY</td>
<td>0.03875000</td>
<td>0.03838</td>
</tr>
<tr>
<td>REACTOME_SYNTHESIS_SECRETION_AND_INACTIVATION_OF_GIP</td>
<td>0.03875000</td>
<td>0.03918</td>
</tr>
<tr>
<td>REACTOME_INCRETIN_SYNTHESIS_SECRETION_AND_INACTIVATION</td>
<td>0.03875000</td>
<td>0.03938</td>
</tr>
</tbody></table>
<h2 id="4-funcoup5检查基因网络"><a href="#4-funcoup5检查基因网络" class="headerlink" title="4.funcoup5检查基因网络"></a>4.funcoup5检查基因网络</h2><p><a target="_blank" rel="noopener" href="https://funcoup5.scilifelab.se/">https://funcoup5.scilifelab.se/</a></p>
<p>查询：C8orf74 SOX7 PINX1 MFHAS1 MIR1322 FAM167A C8orf12 ERI1 BLK PPP1R3B</p>
<p><img src="FunCoup_subnetwork.png" srcset="/img/loading.gif" alt="FunCoup_subnetwork (1)"></p>
<h2 id="5-PathwAX-II查询"><a href="#5-PathwAX-II查询" class="headerlink" title="5.PathwAX II查询"></a>5.PathwAX II查询</h2><p><a target="_blank" rel="noopener" href="https://pathwax.sbc.su.se/">https://pathwax.sbc.su.se/</a></p>
<p><strong>查询内容：</strong></p>
<blockquote>
<p>C8orf74 SOX7 PINX1 MFHAS1 MIR1322 FAM167A C8orf12 ERI1 BLK PPP1R3B</p>
</blockquote>
<p><img src="kegg_pathway.png" srcset="/img/loading.gif" alt="截屏2021-09-27 下午2.12.56"></p>
<table>
<thead>
<tr>
<th><strong>Pathway</strong></th>
<th><strong>PathwaySize</strong></th>
<th><strong>CrossTalk</strong></th>
<th><strong>type</strong></th>
<th><strong>pValue</strong></th>
<th><strong>FWER</strong></th>
</tr>
</thead>
<tbody><tr>
<td><strong>Focal adhesion</strong></td>
<td>198</td>
<td>77</td>
<td>enrichment</td>
<td>2.89E-15</td>
<td>8.44E-13</td>
</tr>
<tr>
<td><strong>Pathways in cancer</strong></td>
<td>533</td>
<td>128</td>
<td>enrichment</td>
<td>1.91E-14</td>
<td>5.57E-12</td>
</tr>
<tr>
<td><strong>Regulation of actin cytoskeleton</strong></td>
<td>211</td>
<td>81</td>
<td>enrichment</td>
<td>4.4E-13</td>
<td>1.28E-10</td>
</tr>
<tr>
<td><strong>Proteoglycans in cancer</strong></td>
<td>203</td>
<td>75</td>
<td>enrichment</td>
<td>6.16E-11</td>
<td>1.8E-08</td>
</tr>
<tr>
<td><strong>PI3K-Akt signaling pathway</strong></td>
<td>356</td>
<td>85</td>
<td>enrichment</td>
<td>2.94E-10</td>
<td>8.6E-08</td>
</tr>
<tr>
<td><strong>Cellular senescence</strong></td>
<td>160</td>
<td>81</td>
<td>enrichment</td>
<td>6.69E-10</td>
<td>1.95E-07</td>
</tr>
<tr>
<td><strong>Rap1 signaling pathway</strong></td>
<td>213</td>
<td>57</td>
<td>enrichment</td>
<td>1.61E-09</td>
<td>4.7E-07</td>
</tr>
<tr>
<td><strong>Oocyte meiosis</strong></td>
<td>126</td>
<td>81</td>
<td>enrichment</td>
<td>2.33E-09</td>
<td>6.8E-07</td>
</tr>
<tr>
<td><strong>Cell cycle</strong></td>
<td>126</td>
<td>90</td>
<td>enrichment</td>
<td>1.08E-08</td>
<td>0.00000315</td>
</tr>
<tr>
<td><strong>Gap junction</strong></td>
<td>90</td>
<td>55</td>
<td>enrichment</td>
<td>1.46E-08</td>
<td>0.00000426</td>
</tr>
<tr>
<td><strong>Human cytomegalovirus infection</strong></td>
<td>225</td>
<td>64</td>
<td>enrichment</td>
<td>3.09E-08</td>
<td>0.00000903</td>
</tr>
<tr>
<td><strong>ECM-receptor interaction</strong></td>
<td>88</td>
<td>17</td>
<td>enrichment</td>
<td>3.18E-08</td>
<td>0.00000927</td>
</tr>
<tr>
<td><strong>Axon guidance</strong></td>
<td>182</td>
<td>42</td>
<td>enrichment</td>
<td>3.94E-08</td>
<td>0.0000115</td>
</tr>
<tr>
<td><strong>Pathogenic Escherichia coli infection</strong></td>
<td>202</td>
<td>89</td>
<td>enrichment</td>
<td>9.19E-08</td>
<td>0.0000268</td>
</tr>
<tr>
<td><strong>FoxO signaling pathway</strong></td>
<td>132</td>
<td>45</td>
<td>enrichment</td>
<td>1.06E-07</td>
<td>0.0000309</td>
</tr>
<tr>
<td><strong>Estrogen signaling pathway</strong></td>
<td>139</td>
<td>60</td>
<td>enrichment</td>
<td>1.09E-07</td>
<td>0.0000318</td>
</tr>
<tr>
<td><strong>Cushing syndrome</strong></td>
<td>155</td>
<td>40</td>
<td>enrichment</td>
<td>1.23E-07</td>
<td>0.0000359</td>
</tr>
<tr>
<td><strong>Shigellosis</strong></td>
<td>205</td>
<td>81</td>
<td>enrichment</td>
<td>1.35E-07</td>
<td>0.0000394</td>
</tr>
<tr>
<td><strong>Small cell lung cancer</strong></td>
<td>93</td>
<td>29</td>
<td>enrichment</td>
<td>2.45E-07</td>
<td>0.0000715</td>
</tr>
<tr>
<td><strong>Viral carcinogenesis</strong></td>
<td>201</td>
<td>115</td>
<td>enrichment</td>
<td>2.71E-07</td>
<td>0.0000791</td>
</tr>
<tr>
<td><strong>Leukocyte transendothelial migration</strong></td>
<td>113</td>
<td>39</td>
<td>enrichment</td>
<td>4.52E-07</td>
<td>0.000132</td>
</tr>
<tr>
<td><strong>Yersinia infection</strong></td>
<td>120</td>
<td>41</td>
<td>enrichment</td>
<td>6.16E-07</td>
<td>0.000180</td>
</tr>
<tr>
<td><strong>Prostate cancer</strong></td>
<td>98</td>
<td>42</td>
<td>enrichment</td>
<td>9.26E-07</td>
<td>0.000271</td>
</tr>
<tr>
<td><strong>ErbB signaling pathway</strong></td>
<td>87</td>
<td>31</td>
<td>enrichment</td>
<td>9.57E-07</td>
<td>0.000279</td>
</tr>
<tr>
<td><strong>Vascular smooth muscle contraction</strong></td>
<td>133</td>
<td>39</td>
<td>enrichment</td>
<td>9.63E-07</td>
<td>0.000281</td>
</tr>
<tr>
<td><strong>Bacterial invasion of epithelial cells</strong></td>
<td>72</td>
<td>44</td>
<td>enrichment</td>
<td>0.00000102</td>
<td>0.000298</td>
</tr>
<tr>
<td><strong>cAMP signaling pathway</strong></td>
<td>219</td>
<td>51</td>
<td>enrichment</td>
<td>0.00000104</td>
<td>0.000305</td>
</tr>
<tr>
<td><strong>Human papillomavirus infection</strong></td>
<td>334</td>
<td>81</td>
<td>enrichment</td>
<td>0.00000125</td>
<td>0.000364</td>
</tr>
<tr>
<td><strong>Hepatitis B</strong></td>
<td>165</td>
<td>56</td>
<td>enrichment</td>
<td>0.00000159</td>
<td>0.000465</td>
</tr>
<tr>
<td><strong>Adherens junction</strong></td>
<td>69</td>
<td>35</td>
<td>enrichment</td>
<td>0.00000197</td>
<td>0.000575</td>
</tr>
<tr>
<td><strong>Platelet activation</strong></td>
<td>124</td>
<td>41</td>
<td>enrichment</td>
<td>0.00000198</td>
<td>0.000577</td>
</tr>
<tr>
<td><strong>GnRH signaling pathway</strong></td>
<td>95</td>
<td>34</td>
<td>enrichment</td>
<td>0.00000215</td>
<td>0.000628</td>
</tr>
<tr>
<td><strong>Fluid shear stress and atherosclerosis</strong></td>
<td>137</td>
<td>55</td>
<td>enrichment</td>
<td>0.00000220</td>
<td>0.000642</td>
</tr>
<tr>
<td><strong>AGE-RAGE signaling pathway in diabetic complications</strong></td>
<td>102</td>
<td>29</td>
<td>enrichment</td>
<td>0.00000245</td>
<td>0.000715</td>
</tr>
<tr>
<td><strong>Neurotrophin signaling pathway</strong></td>
<td>120</td>
<td>47</td>
<td>enrichment</td>
<td>0.00000271</td>
<td>0.000793</td>
</tr>
<tr>
<td><strong>Glioma</strong></td>
<td>77</td>
<td>35</td>
<td>enrichment</td>
<td>0.00000296</td>
<td>0.000864</td>
</tr>
<tr>
<td><strong>Gastric cancer</strong></td>
<td>149</td>
<td>34</td>
<td>enrichment</td>
<td>0.00000298</td>
<td>0.000870</td>
</tr>
<tr>
<td><strong>Oxytocin signaling pathway</strong></td>
<td>154</td>
<td>47</td>
<td>enrichment</td>
<td>0.00000315</td>
<td>0.000919</td>
</tr>
<tr>
<td><strong>VEGF signaling pathway</strong></td>
<td>61</td>
<td>27</td>
<td>enrichment</td>
<td>0.00000367</td>
<td>0.00107</td>
</tr>
<tr>
<td><strong>Growth hormone synthesis, secretion and action</strong></td>
<td>123</td>
<td>34</td>
<td>enrichment</td>
<td>0.00000567</td>
<td>0.00166</td>
</tr>
<tr>
<td><strong>Non-small cell lung cancer</strong></td>
<td>68</td>
<td>27</td>
<td>enrichment</td>
<td>0.00000570</td>
<td>0.00166</td>
</tr>
<tr>
<td><strong>Human immunodeficiency virus 1 infection</strong></td>
<td>212</td>
<td>67</td>
<td>enrichment</td>
<td>0.00000617</td>
<td>0.00180</td>
</tr>
<tr>
<td><strong>Phospholipase D signaling pathway</strong></td>
<td>149</td>
<td>30</td>
<td>enrichment</td>
<td>0.00000917</td>
<td>0.00268</td>
</tr>
<tr>
<td><strong>Progesterone-mediated oocyte maturation</strong></td>
<td>97</td>
<td>43</td>
<td>enrichment</td>
<td>0.00000923</td>
<td>0.00269</td>
</tr>
<tr>
<td><strong>Tight junction</strong></td>
<td>171</td>
<td>61</td>
<td>enrichment</td>
<td>0.0000120</td>
<td>0.00350</td>
</tr>
<tr>
<td><strong>Pertussis</strong></td>
<td>81</td>
<td>31</td>
<td>enrichment</td>
<td>0.0000155</td>
<td>0.00453</td>
</tr>
<tr>
<td><strong>p53 signaling pathway</strong></td>
<td>73</td>
<td>25</td>
<td>enrichment</td>
<td>0.0000162</td>
<td>0.00472</td>
</tr>
<tr>
<td><strong>Fc gamma R-mediated phagocytosis</strong></td>
<td>94</td>
<td>36</td>
<td>enrichment</td>
<td>0.0000182</td>
<td>0.00531</td>
</tr>
<tr>
<td><strong>Chemokine signaling pathway</strong></td>
<td>190</td>
<td>39</td>
<td>enrichment</td>
<td>0.0000198</td>
<td>0.00579</td>
</tr>
<tr>
<td><strong>Glycolysis / Gluconeogenesis</strong></td>
<td>68</td>
<td>49</td>
<td>enrichment</td>
<td>0.0000209</td>
<td>0.00611</td>
</tr>
<tr>
<td><strong>Long-term potentiation</strong></td>
<td>69</td>
<td>35</td>
<td>enrichment</td>
<td>0.0000231</td>
<td>0.00676</td>
</tr>
<tr>
<td><strong>Apelin signaling pathway</strong></td>
<td>138</td>
<td>37</td>
<td>enrichment</td>
<td>0.0000265</td>
<td>0.00773</td>
</tr>
<tr>
<td><strong>Melanogenesis</strong></td>
<td>102</td>
<td>31</td>
<td>enrichment</td>
<td>0.0000271</td>
<td>0.00791</td>
</tr>
<tr>
<td><strong>Ras signaling pathway</strong></td>
<td>233</td>
<td>44</td>
<td>enrichment</td>
<td>0.0000395</td>
<td>0.0115</td>
</tr>
<tr>
<td><strong>Bladder cancer</strong></td>
<td>41</td>
<td>20</td>
<td>enrichment</td>
<td>0.0000413</td>
<td>0.0121</td>
</tr>
<tr>
<td><strong>Chronic myeloid leukemia</strong></td>
<td>77</td>
<td>34</td>
<td>enrichment</td>
<td>0.0000428</td>
<td>0.0125</td>
</tr>
<tr>
<td><strong>Adrenergic signaling in cardiomyocytes</strong></td>
<td>151</td>
<td>45</td>
<td>enrichment</td>
<td>0.0000495</td>
<td>0.0144</td>
</tr>
<tr>
<td><strong>Hepatitis C</strong></td>
<td>159</td>
<td>54</td>
<td>enrichment</td>
<td>0.0000526</td>
<td>0.0153</td>
</tr>
<tr>
<td><strong>HIF-1 signaling pathway</strong></td>
<td>111</td>
<td>47</td>
<td>enrichment</td>
<td>0.0000530</td>
<td>0.0155</td>
</tr>
<tr>
<td><strong>Renal cell carcinoma</strong></td>
<td>70</td>
<td>33</td>
<td>enrichment</td>
<td>0.0000626</td>
<td>0.0183</td>
</tr>
<tr>
<td><strong>Relaxin signaling pathway</strong></td>
<td>130</td>
<td>30</td>
<td>enrichment</td>
<td>0.0000653</td>
<td>0.0191</td>
</tr>
<tr>
<td><strong>cGMP-PKG signaling pathway</strong></td>
<td>168</td>
<td>50</td>
<td>enrichment</td>
<td>0.0000727</td>
<td>0.0212</td>
</tr>
<tr>
<td><strong>Melanoma</strong></td>
<td>73</td>
<td>21</td>
<td>enrichment</td>
<td>0.0000757</td>
<td>0.0221</td>
</tr>
<tr>
<td><strong>T cell receptor signaling pathway</strong></td>
<td>104</td>
<td>28</td>
<td>enrichment</td>
<td>0.0000766</td>
<td>0.0224</td>
</tr>
<tr>
<td><strong>Hepatocellular carcinoma</strong></td>
<td>168</td>
<td>41</td>
<td>enrichment</td>
<td>0.0000805</td>
<td>0.0235</td>
</tr>
<tr>
<td><strong>MicroRNAs in cancer</strong></td>
<td>164</td>
<td>45</td>
<td>enrichment</td>
<td>0.0000844</td>
<td>0.0246</td>
</tr>
<tr>
<td><strong>Hippo signaling pathway</strong></td>
<td>153</td>
<td>46</td>
<td>enrichment</td>
<td>0.0000869</td>
<td>0.0254</td>
</tr>
<tr>
<td><strong>Kaposi sarcoma-associated herpesvirus infection</strong></td>
<td>188</td>
<td>44</td>
<td>enrichment</td>
<td>0.000106</td>
<td>0.0311</td>
</tr>
<tr>
<td><strong>Natural killer cell mediated cytotoxicity</strong></td>
<td>124</td>
<td>24</td>
<td>enrichment</td>
<td>0.000106</td>
<td>0.0311</td>
</tr>
<tr>
<td><strong>Prolactin signaling pathway</strong></td>
<td>72</td>
<td>23</td>
<td>enrichment</td>
<td>0.000119</td>
<td>0.0346</td>
</tr>
<tr>
<td><strong>Circadian entrainment</strong></td>
<td>99</td>
<td>26</td>
<td>enrichment</td>
<td>0.000127</td>
<td>0.0370</td>
</tr>
</tbody></table>
<blockquote>
<p> Failed query mapping: C20ORF20, FLJ20674</p>
</blockquote>
<p><img src="gene_network.png" srcset="/img/loading.gif" alt="截屏2021-09-27 下午2.13.35"></p>
<h2 id="6-GENE2FUNC-功能注释"><a href="#6-GENE2FUNC-功能注释" class="headerlink" title="6.GENE2FUNC 功能注释"></a>6.GENE2FUNC 功能注释</h2><p><a target="_blank" rel="noopener" href="https://fuma.ctglab.nl/snp2gene">https://fuma.ctglab.nl/snp2gene</a></p>
<p>先使用fuma工具，为snp添加基因注释</p>
<blockquote>
<p>用户名称：Liu Keqin</p>
<p>登录邮箱：<a href="mailto:&#108;&#x69;&#117;&#107;&#x65;&#113;&#x69;&#x6e;&#x32;&#x30;&#50;&#x30;&#64;&#x73;&#105;&#98;&#x73;&#x2e;&#97;&#x63;&#x2e;&#x63;&#x6e;">&#108;&#x69;&#117;&#107;&#x65;&#113;&#x69;&#x6e;&#x32;&#x30;&#50;&#x30;&#64;&#x73;&#105;&#98;&#x73;&#x2e;&#97;&#x63;&#x2e;&#x63;&#x6e;</a></p>
</blockquote>
<p><strong>运行参数记录</strong></p>
<blockquote>
<table>
<thead>
<tr>
<th>created_at</th>
<th>2021-09-27 09:44:20</th>
</tr>
</thead>
<tbody><tr>
<td>title</td>
<td>20210927</td>
</tr>
<tr>
<td>FUMA</td>
<td>v1.3.6d</td>
</tr>
<tr>
<td>MAGMA</td>
<td>v1.08</td>
</tr>
<tr>
<td>GWAScatalog</td>
<td>e104_r2021-09-15</td>
</tr>
<tr>
<td>ANNOVAR</td>
<td>2017-07-17</td>
</tr>
<tr>
<td>gwasfile</td>
<td>iPSYCH_PGC_ASD_Nov2017</td>
</tr>
<tr>
<td>chrcol</td>
<td>CHR</td>
</tr>
<tr>
<td>poscol</td>
<td>BP</td>
</tr>
<tr>
<td>rsIDcol</td>
<td>SNP</td>
</tr>
<tr>
<td>pcol</td>
<td>P</td>
</tr>
<tr>
<td>eacol</td>
<td>A1</td>
</tr>
<tr>
<td>neacol</td>
<td>A2</td>
</tr>
<tr>
<td>orcol</td>
<td>OR</td>
</tr>
<tr>
<td>becol</td>
<td>NA</td>
</tr>
<tr>
<td>secol</td>
<td>SE</td>
</tr>
<tr>
<td>leadSNPsfile</td>
<td>NA</td>
</tr>
<tr>
<td>addleadSNPs</td>
<td>1</td>
</tr>
<tr>
<td>regionsfile</td>
<td>NA</td>
</tr>
<tr>
<td>N</td>
<td>18381</td>
</tr>
<tr>
<td>Ncol</td>
<td>NA</td>
</tr>
<tr>
<td>exMHC</td>
<td>1</td>
</tr>
<tr>
<td>MHCopt</td>
<td>annot</td>
</tr>
<tr>
<td>extMHC</td>
<td>NA</td>
</tr>
<tr>
<td>ensembl</td>
<td>v92</td>
</tr>
<tr>
<td>genetype</td>
<td>protein_coding</td>
</tr>
<tr>
<td>leadP</td>
<td>5e-8</td>
</tr>
<tr>
<td>gwasP</td>
<td>0.05</td>
</tr>
<tr>
<td>r2</td>
<td>0.6</td>
</tr>
<tr>
<td>r2_2</td>
<td>0.1</td>
</tr>
<tr>
<td>refpanel</td>
<td>1KG/Phase3</td>
</tr>
<tr>
<td>pop</td>
<td>EUR</td>
</tr>
<tr>
<td>MAF</td>
<td>0</td>
</tr>
<tr>
<td>refSNPs</td>
<td>1</td>
</tr>
<tr>
<td>mergeDist</td>
<td>250</td>
</tr>
<tr>
<td>magma</td>
<td>1</td>
</tr>
<tr>
<td>magma_window</td>
<td>0</td>
</tr>
<tr>
<td>magma_exp</td>
<td>GTEx/v8/gtex_v8_ts_avg_log2TPM, GTEx/v8/gtex_v8_ts_general_avg_log2TPM</td>
</tr>
<tr>
<td>posMap</td>
<td>1</td>
</tr>
<tr>
<td>posMapWindowSize</td>
<td>10</td>
</tr>
<tr>
<td>posMapAnnot</td>
<td>NA</td>
</tr>
<tr>
<td>posMapCADDth</td>
<td>0</td>
</tr>
<tr>
<td>posMapRDBth</td>
<td>NA</td>
</tr>
<tr>
<td>posMapChr15</td>
<td>NA</td>
</tr>
<tr>
<td>posMapChr15Max</td>
<td>NA</td>
</tr>
<tr>
<td>posMapChr15Meth</td>
<td>NA</td>
</tr>
<tr>
<td>posMapAnnoDs</td>
<td>NA</td>
</tr>
<tr>
<td>posMapAnnoMeth</td>
<td>NA</td>
</tr>
<tr>
<td>eqtlMap</td>
<td>0</td>
</tr>
<tr>
<td>eqtlMaptss</td>
<td>NA</td>
</tr>
<tr>
<td>eqtlMapSig</td>
<td>1</td>
</tr>
<tr>
<td>eqtlMapP</td>
<td>1</td>
</tr>
<tr>
<td>eqtlMapCADDth</td>
<td>0</td>
</tr>
<tr>
<td>eqtlMapRDBth</td>
<td>NA</td>
</tr>
<tr>
<td>eqtlMapChr15</td>
<td>NA</td>
</tr>
<tr>
<td>eqtlMapChr15Max</td>
<td>NA</td>
</tr>
<tr>
<td>eqtlMapChr15Meth</td>
<td>NA</td>
</tr>
<tr>
<td>eqtlMapAnnoDs</td>
<td>NA</td>
</tr>
<tr>
<td>eqtlMapAnnoMeth</td>
<td>NA</td>
</tr>
<tr>
<td>ciMap</td>
<td>0</td>
</tr>
<tr>
<td>ciMapBuiltin</td>
<td>NA</td>
</tr>
<tr>
<td>ciMapFileN</td>
<td>0</td>
</tr>
<tr>
<td>ciMapFiles</td>
<td>NA</td>
</tr>
<tr>
<td>ciMapFDR</td>
<td>NA</td>
</tr>
<tr>
<td>ciMapPromWindow</td>
<td>NA</td>
</tr>
<tr>
<td>ciMapRoadmap</td>
<td>NA</td>
</tr>
<tr>
<td>ciMapEnhFilt</td>
<td>0</td>
</tr>
<tr>
<td>ciMapPromFilt</td>
<td>0</td>
</tr>
<tr>
<td>ciMapCADDth</td>
<td>0</td>
</tr>
<tr>
<td>ciMapRDBth</td>
<td>NA</td>
</tr>
<tr>
<td>ciMapChr15</td>
<td>NA</td>
</tr>
<tr>
<td>ciMapChr15Max</td>
<td>NA</td>
</tr>
<tr>
<td>ciMapChr15Meth</td>
<td>NA</td>
</tr>
<tr>
<td>ciMapAnnoDs</td>
<td>NA</td>
</tr>
<tr>
<td>ciMapAnnoMeth</td>
<td>NA</td>
</tr>
</tbody></table>
</blockquote>
<h3 id="Manhattan-Plot-GWAS-summary-statistics"><a href="#Manhattan-Plot-GWAS-summary-statistics" class="headerlink" title="Manhattan Plot (GWAS summary statistics)"></a>Manhattan Plot (GWAS summary statistics)</h3><p>For plotting, overlapping data points are not drawn (filtering was performed only for SNPs with P-value ≥ 1e-5, see tutorial for details).</p>
<p><img src="manhattan_FUMA_jobs147024.png" srcset="/img/loading.gif" alt="manhattan_FUMA_jobs147024"></p>
<h3 id="Mahattan-Plot-gene-based-test"><a href="#Mahattan-Plot-gene-based-test" class="headerlink" title="Mahattan Plot (gene-based test)"></a>Mahattan Plot (gene-based test)</h3><p>Input SNPs were mapped to 18937 protein coding genes. Genome wide significance (red dashed line in the plot) was defined at P = 0.05/18937 = 2.640e-6.</p>
<p>标注top20的基因：</p>
<p><img src="geneManhattan_FUMA_jobs147024.png" srcset="/img/loading.gif" alt="geneManhattan_FUMA_jobs147024"></p>
<h3 id="QQ-plot-GWAS-summary-statisics"><a href="#QQ-plot-GWAS-summary-statisics" class="headerlink" title="QQ plot (GWAS summary statisics)"></a>QQ plot (GWAS summary statisics)</h3><p>For plotting purposes, overlapping data points are not drawn (filtering was performed only for SNPs with P-value ≥ 1e-5, see tutorial for details).</p>
<p><img src="QQplot_FUMA_jobs147024.png" srcset="/img/loading.gif" alt="QQplot_FUMA_jobs147024"></p>
<h3 id="QQ-plot-gene-based-test"><a href="#QQ-plot-gene-based-test" class="headerlink" title="QQ plot (gene-based test)"></a>QQ plot (gene-based test)</h3><p>This is a Q-Q plot of the gene-based test computed by MAGMA.</p>
<p><img src="geneQQplot_FUMA_jobs147024.png" srcset="/img/loading.gif" alt="geneQQplot_FUMA_jobs147024"></p>
<h3 id="MAGMA-Gene-Set-Analysis"><a href="#MAGMA-Gene-Set-Analysis" class="headerlink" title="MAGMA Gene-Set Analysis"></a>MAGMA Gene-Set Analysis</h3><p>MAGMA gene-set analysis is performed for curated gene sets and GO terms obtained from MsigDB (see here for details).<br>The table displays either significant gene sets with Pbon &lt; 0.05 or top 10 gene sets when there are less than 10 significant gene sets. Full results are downloadable from “Download” tab.<br>Note that MAGMA gene-set analyses uses the full distribution of SNP p-values and is different from a pathway enrichment test as implemented in GENE2FUNC that only tests for enrichment of prioritized genes.</p>
<table>
<thead>
<tr>
<th align="left">Gene Set</th>
<th align="left">N genes</th>
<th align="left">Beta</th>
<th align="left">Beta STD</th>
<th align="left">SE</th>
<th align="left">P</th>
<th align="left">Pbon</th>
</tr>
</thead>
<tbody><tr>
<td align="left">GO_bp:go_macrophage_colony_stimulating_factor_production</td>
<td align="left">5</td>
<td align="left">1.394</td>
<td align="left">0.022649</td>
<td align="left">0.35169</td>
<td align="left">3.7034e-05</td>
<td align="left">0.573434456</td>
</tr>
<tr>
<td align="left">GO_cc:go_extrinsic_component_of_postsynaptic_membrane</td>
<td align="left">8</td>
<td align="left">1.057</td>
<td align="left">0.021721</td>
<td align="left">0.28832</td>
<td align="left">0.00012356</td>
<td align="left">1</td>
</tr>
<tr>
<td align="left">GO_bp:go_granulocyte_colony_stimulating_factor_production</td>
<td align="left">5</td>
<td align="left">1.3071</td>
<td align="left">0.021236</td>
<td align="left">0.36585</td>
<td align="left">0.00017718</td>
<td align="left">1</td>
</tr>
<tr>
<td align="left">GO_mf:go_neurotransmitter_receptor_activity</td>
<td align="left">108</td>
<td align="left">0.3108</td>
<td align="left">0.023405</td>
<td align="left">0.088422</td>
<td align="left">0.00022052</td>
<td align="left">1</td>
</tr>
<tr>
<td align="left">GO_cc:go_extrinsic_component_of_synaptic_membrane</td>
<td align="left">12</td>
<td align="left">0.87453</td>
<td align="left">0.022008</td>
<td align="left">0.25263</td>
<td align="left">0.00026897</td>
<td align="left">1</td>
</tr>
<tr>
<td align="left">GO_bp:go_extraembryonic_membrane_development</td>
<td align="left">9</td>
<td align="left">1.1923</td>
<td align="left">0.025987</td>
<td align="left">0.35019</td>
<td align="left">0.00033198</td>
<td align="left">1</td>
</tr>
<tr>
<td align="left">Curated_gene_sets:pid_ncadherin_pathway</td>
<td align="left">33</td>
<td align="left">0.49317</td>
<td align="left">0.02057</td>
<td align="left">0.1451</td>
<td align="left">0.00033919</td>
<td align="left">1</td>
</tr>
<tr>
<td align="left">GO_bp:go_histone_h3_k27_demethylation</td>
<td align="left">2</td>
<td align="left">1.65</td>
<td align="left">0.016957</td>
<td align="left">0.49</td>
<td align="left">0.0003802</td>
<td align="left">1</td>
</tr>
<tr>
<td align="left">GO_mf:go_histone_demethylase_activity_h3_k27_specific_</td>
<td align="left">2</td>
<td align="left">1.65</td>
<td align="left">0.016957</td>
<td align="left">0.49</td>
<td align="left">0.0003802</td>
<td align="left">1</td>
</tr>
<tr>
<td align="left">GO_cc:go_t_tubule</td>
<td align="left">52</td>
<td align="left">0.43768</td>
<td align="left">0.022904</td>
<td align="left">0.13185</td>
<td align="left">0.00045174</td>
<td align="left">1</td>
</tr>
</tbody></table>
<h3 id="MAGMA-Tissue-Expression-Analysis"><a href="#MAGMA-Tissue-Expression-Analysis" class="headerlink" title="MAGMA Tissue Expression Analysis"></a>MAGMA Tissue Expression Analysis</h3><p>MAGMA gene-property analysis is performed for gene expression of user selected data sets.<br>Note that MAGMA gene-property analyses uses the full distribution of SNP p-values and is different from a enrichment test of DEG (differentially expressed genes) as implemented in GENE2FUNC that only tests for enrichment of prioritised genes.</p>
<p><img src="magma_exp_gtex_v8_ts_avg_log2TPM_FUMA_jobs147024.png" srcset="/img/loading.gif" alt="magma_exp_gtex_v8_ts_avg_log2TPM_FUMA_jobs147024"></p>
<p><img src="magma_exp_gtex_v8_ts_general_avg_log2TPM_FUMA_jobs147024.png" srcset="/img/loading.gif" alt="magma_exp_gtex_v8_ts_general_avg_log2TPM_FUMA_jobs147024"></p>
<h3 id="Summary-of-SNPs-and-mapped-genes"><a href="#Summary-of-SNPs-and-mapped-genes" class="headerlink" title="Summary of SNPs and mapped genes"></a>Summary of SNPs and mapped genes</h3><table>
<thead>
<tr>
<th>item</th>
<th>number</th>
</tr>
</thead>
<tbody><tr>
<td>Genomic risk loci</td>
<td>3</td>
</tr>
<tr>
<td>lead SNPs</td>
<td>3</td>
</tr>
<tr>
<td>Ind. Sig. SNPs</td>
<td>7</td>
</tr>
<tr>
<td>candidate SNPs</td>
<td>447</td>
</tr>
<tr>
<td>candidate GWAS tagged SNPs</td>
<td>422</td>
</tr>
<tr>
<td>mapped genes</td>
<td>8</td>
</tr>
</tbody></table>
<h3 id="Functional-consequences-of-SNPs-on-genes"><a href="#Functional-consequences-of-SNPs-on-genes" class="headerlink" title="Functional consequences of SNPs on genes"></a>Functional consequences of SNPs on genes</h3><p><img src="snpAnnotPlot_FUMA_jobs147024.png" srcset="/img/loading.gif" alt="snpAnnotPlot_FUMA_jobs147024"></p>
<h3 id="Summary-per-genomic-risk-locus"><a href="#Summary-per-genomic-risk-locus" class="headerlink" title="Summary per genomic risk locus"></a>Summary per genomic risk locus</h3><p><img src="lociPlot_FUMA_jobs147024.png" srcset="/img/loading.gif" alt="lociPlot_FUMA_jobs147024"></p>
<h3 id="Genomic-Locus"><a href="#Genomic-Locus" class="headerlink" title="Genomic Locus"></a>Genomic Locus</h3><table>
<thead>
<tr>
<th align="left">Genomic Locus</th>
<th align="left">uniqID</th>
<th align="left">rsID</th>
<th align="left">chr</th>
<th align="left">pos</th>
<th align="left">P-value</th>
<th align="left">start</th>
<th align="left">end</th>
<th align="left">nSNPs</th>
<th align="left">nGWASSNPs</th>
<th align="left">nIndSigSNPs</th>
<th align="left">IndSigSNPs</th>
<th align="left">nLeadSNPs</th>
<th align="left">LeadSNPs</th>
</tr>
</thead>
<tbody><tr>
<td align="left">1</td>
<td align="left">8:10576775:C:G</td>
<td align="left">rs10099100</td>
<td align="left">8</td>
<td align="left">10576775</td>
<td align="left">1.065e-08</td>
<td align="left">10571591</td>
<td align="left">10583506</td>
<td align="left">6</td>
<td align="left">6</td>
<td align="left">1</td>
<td align="left">rs10099100</td>
<td align="left">1</td>
<td align="left">rs10099100</td>
</tr>
<tr>
<td align="left">2</td>
<td align="left">20:14836243:G:GTTTTTTT</td>
<td align="left">rs71190156</td>
<td align="left">20</td>
<td align="left">14836243</td>
<td align="left">2.752e-08</td>
<td align="left">14697386</td>
<td align="left">14836243</td>
<td align="left">49</td>
<td align="left">49</td>
<td align="left">1</td>
<td align="left">rs71190156</td>
<td align="left">1</td>
<td align="left">rs71190156</td>
</tr>
<tr>
<td align="left">3</td>
<td align="left">20:21248116:A:G</td>
<td align="left">rs910805</td>
<td align="left">20</td>
<td align="left">21248116</td>
<td align="left">2.041e-09</td>
<td align="left">20118473</td>
<td align="left">22105020</td>
<td align="left">392</td>
<td align="left">367</td>
<td align="left">5</td>
<td align="left">rs910805;rs12625304;rs6035820</td>
<td align="left">1</td>
<td align="left">rs910805</td>
</tr>
</tbody></table>
<h3 id="lead-SNPs"><a href="#lead-SNPs" class="headerlink" title="lead SNPs"></a>lead SNPs</h3><table>
<thead>
<tr>
<th align="left">No</th>
<th align="left">Genomic Locus</th>
<th align="left">uniqID</th>
<th align="left">rsID</th>
<th align="left">chr</th>
<th align="left">pos</th>
<th align="left">P-value</th>
<th align="left">nIndSigSNPs</th>
<th align="left">IndSigSNPs</th>
</tr>
</thead>
<tbody><tr>
<td align="left">1</td>
<td align="left">1</td>
<td align="left">8:10576775:C:G</td>
<td align="left">rs10099100</td>
<td align="left">8</td>
<td align="left">10576775</td>
<td align="left">1.065e-08</td>
<td align="left">1</td>
<td align="left">rs10099100</td>
</tr>
<tr>
<td align="left">2</td>
<td align="left">2</td>
<td align="left">20:14836243:G:GTTTTTTT</td>
<td align="left">rs71190156</td>
<td align="left">20</td>
<td align="left">14836243</td>
<td align="left">2.752e-08</td>
<td align="left">1</td>
<td align="left">rs71190156</td>
</tr>
<tr>
<td align="left">3</td>
<td align="left">3</td>
<td align="left">20:21248116:A:G</td>
<td align="left">rs910805</td>
<td align="left">20</td>
<td align="left">21248116</td>
<td align="left">2.041e-09</td>
<td align="left">5</td>
<td align="left">rs910805;rs12625304;rs6035820</td>
</tr>
</tbody></table>
<h3 id="Ind-Sig-SNPs"><a href="#Ind-Sig-SNPs" class="headerlink" title="Ind.Sig.SNPs"></a>Ind.Sig.SNPs</h3><table>
<thead>
<tr>
<th align="left">No</th>
<th align="left">Genomic Locus</th>
<th align="left">uniqID</th>
<th align="left">rsID</th>
<th align="left">chr</th>
<th align="left">pos</th>
<th align="left">P-value</th>
<th align="left">nSNPs</th>
<th align="left">nGWASSNPs</th>
</tr>
</thead>
<tbody><tr>
<td align="left">1</td>
<td align="left">1</td>
<td align="left">8:10576775:C:G</td>
<td align="left">rs10099100</td>
<td align="left">8</td>
<td align="left">10576775</td>
<td align="left">1.065e-08</td>
<td align="left">6</td>
<td align="left">6</td>
</tr>
<tr>
<td align="left">2</td>
<td align="left">2</td>
<td align="left">20:14836243:G:GTTTTTTT</td>
<td align="left">rs71190156</td>
<td align="left">20</td>
<td align="left">14836243</td>
<td align="left">2.752e-08</td>
<td align="left">49</td>
<td align="left">49</td>
</tr>
<tr>
<td align="left">4</td>
<td align="left">3</td>
<td align="left">20:21182388:C:T</td>
<td align="left">rs12625304</td>
<td align="left">20</td>
<td align="left">21182388</td>
<td align="left">8.16e-09</td>
<td align="left">4</td>
<td align="left">4</td>
</tr>
<tr>
<td align="left">6</td>
<td align="left">3</td>
<td align="left">20:21241417:A:T</td>
<td align="left">rs6035820</td>
<td align="left">20</td>
<td align="left">21241417</td>
<td align="left">2.867e-08</td>
<td align="left">162</td>
<td align="left">162</td>
</tr>
<tr>
<td align="left">7</td>
<td align="left">3</td>
<td align="left">20:21248116:A:G</td>
<td align="left">rs910805</td>
<td align="left">20</td>
<td align="left">21248116</td>
<td align="left">2.041e-09</td>
<td align="left">182</td>
<td align="left">182</td>
</tr>
</tbody></table>
<h3 id="mapped-genes"><a href="#mapped-genes" class="headerlink" title="mapped genes"></a>mapped genes</h3><table>
<thead>
<tr>
<th align="left">Gene</th>
<th align="left">Symbol</th>
<th align="left">HUGO</th>
<th align="left">entrezID</th>
<th align="left">chr</th>
<th align="left">start</th>
<th align="left">end</th>
<th align="left">strand</th>
<th align="left">type</th>
<th align="left">pLI</th>
<th align="left">ncRVIS</th>
<th align="left">posMapSNPs</th>
<th align="left">posMapMaxCADD</th>
<th align="left">minGwasP</th>
<th align="left">Genomic Locus</th>
<th align="left">IndSigSNPs</th>
</tr>
</thead>
<tbody><tr>
<td align="left">ENSG00000088930</td>
<td align="left">XRN2</td>
<td align="left">XRN2</td>
<td align="left">22803</td>
<td align="left">20</td>
<td align="left">21283942</td>
<td align="left">21370463</td>
<td align="left">1</td>
<td align="left">protein_coding</td>
<td align="left">0.813170185835591</td>
<td align="left">-0.511318347</td>
<td align="left">103</td>
<td align="left">13.49</td>
<td align="left">1.589e-08</td>
<td align="left">3</td>
<td align="left">rs910805;rs6035820;rs12625304</td>
</tr>
<tr>
<td align="left">ENSG00000125816</td>
<td align="left">NKX2-4</td>
<td align="left">NKX2-4</td>
<td align="left">644524</td>
<td align="left">20</td>
<td align="left">21376005</td>
<td align="left">21378666</td>
<td align="left">-1</td>
<td align="left">protein_coding</td>
<td align="left">0.278447692496314</td>
<td align="left">-0.618232094</td>
<td align="left">14</td>
<td align="left">11.39</td>
<td align="left">2.126e-08</td>
<td align="left">3</td>
<td align="left">rs910805;rs6035820</td>
</tr>
<tr>
<td align="left">ENSG00000171056</td>
<td align="left">SOX7</td>
<td align="left">SOX7</td>
<td align="left">83595</td>
<td align="left">8</td>
<td align="left">10581278</td>
<td align="left">10697357</td>
<td align="left">-1</td>
<td align="left">protein_coding</td>
<td align="left">0.131241610927833</td>
<td align="left">0.638029508</td>
<td align="left">6</td>
<td align="left">17.85</td>
<td align="left">1.065e-08</td>
<td align="left">1</td>
<td align="left">rs10099100</td>
</tr>
<tr>
<td align="left">ENSG00000172264</td>
<td align="left">MACROD2</td>
<td align="left">MACROD2</td>
<td align="left">140733</td>
<td align="left">20</td>
<td align="left">13976015</td>
<td align="left">16033842</td>
<td align="left">1</td>
<td align="left">protein_coding</td>
<td align="left">0.0113567470210345</td>
<td align="left">3.34668599</td>
<td align="left">49</td>
<td align="left">20.4</td>
<td align="left">2.752e-08</td>
<td align="left">2</td>
<td align="left">rs71190156</td>
</tr>
<tr>
<td align="left">ENSG00000183638</td>
<td align="left">RP1L1</td>
<td align="left">RP1L1</td>
<td align="left">94137</td>
<td align="left">8</td>
<td align="left">10463859</td>
<td align="left">10569697</td>
<td align="left">-1</td>
<td align="left">protein_coding</td>
<td align="left">NA</td>
<td align="left">1.244966111</td>
<td align="left">5</td>
<td align="left">3.864</td>
<td align="left">1.065e-08</td>
<td align="left">1</td>
<td align="left">rs10099100</td>
</tr>
<tr>
<td align="left">ENSG00000258724</td>
<td align="left">SOX7</td>
<td align="left">SOX7</td>
<td align="left">83595</td>
<td align="left">8</td>
<td align="left">10582909</td>
<td align="left">10697357</td>
<td align="left">-1</td>
<td align="left">protein_coding</td>
<td align="left">NA</td>
<td align="left">0.638029508</td>
<td align="left">4</td>
<td align="left">17.85</td>
<td align="left">1.065e-08</td>
<td align="left">1</td>
<td align="left">rs10099100</td>
</tr>
</tbody></table>
<h2 id="7-使用Locus-Zoom绘制Regional-Plots"><a href="#7-使用Locus-Zoom绘制Regional-Plots" class="headerlink" title="7.使用Locus Zoom绘制Regional Plots"></a>7.使用Locus Zoom绘制Regional Plots</h2><p>首先检查每个染色体上包含SNPs的主要区域：</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><code class="hljs R">iPSYCH_PGC_ASD_Nov2017 &lt;- read.delim(<span class="hljs-string">&quot;/Volumes/My\ Passport\ for\ Mac/data/14671989/iPSYCH_PGC_ASD_Nov2017&quot;</span>)<br>print(head(iPSYCH_PGC_ASD_Nov2017))<br>subset_ch2 &lt;- subset(iPSYCH_PGC_ASD_Nov2017, CHR==<span class="hljs-string">&quot;2&quot;</span>)<br>subset_ch2[order(subset_ch2$P),]<br><span class="hljs-comment">##         CHR            SNP        BP               A1    A2  INFO      OR     SE         P</span><br><span class="hljs-comment">## 4810052   2     rs59566011 159385181               GA     G 0.947 1.07315 0.0144 9.304e-07</span><br><span class="hljs-comment">## 4805439   2     rs70994211 159385148                T    TA 0.990 0.93557 0.0143 3.410e-06</span><br><span class="hljs-comment">## 3955815   2      rs6430841 140318199                A     G 1.000 1.08275 0.0176 6.402e-06</span><br><span class="hljs-comment">## 349794    2      rs7578456 202235348                A     G 0.991 1.06556 0.0141 6.532e-06</span><br><span class="hljs-comment">## 346713    2      rs2110688 202223429                T     C 0.997 1.06492 0.0141 8.044e-06</span><br></code></pre></td></tr></table></figure>

<p>然后将iPSYCH_PGC_ASD_Nov2017作为输入数据，使用Locus Zoom网页工具绘制Regional Plots (<a target="_blank" rel="noopener" href="http://locuszoom.org/genform.php?type=yourdata">http://locuszoom.org/genform.php?type=yourdata</a>)</p>
<blockquote>
<p>值得注意的是，输入Locus Zoom的数据，需要选择PLINK生成的格式，选择以制表符作为分隔符，其他非必须参数可选择默认。</p>
<p>其中，绘图用的mark是SNP，p.value为SNP对应的P。</p>
</blockquote>
<p>根据step6生成的曼哈顿图，这里重点关注6，8，17，20号染色体。</p>
<p>发现6号染色体上的SNPs的对应P值均大于0.05，于是不予考虑。</p>
<h3 id="EUR-8号染色体-10mb-11mb"><a href="#EUR-8号染色体-10mb-11mb" class="headerlink" title="EUR-8号染色体-10mb-11mb"></a>EUR-8号染色体-10mb-11mb</h3><embed id="pdfPlayer" src="EUR.chr8.10-11mb.pdf" type="application/pdf" width="100%" height="600" >



<h3 id="EUR-17号染色体-44mb-45mb"><a href="#EUR-17号染色体-44mb-45mb" class="headerlink" title="EUR-17号染色体-44mb-45mb"></a>EUR-17号染色体-44mb-45mb</h3><embed id="pdfPlayer" src="EUR.chr17.44-45mb.pdf" type="application/pdf" width="100%" height="600" >



<h3 id="EUR-20号染色体-21mb-22mb"><a href="#EUR-20号染色体-21mb-22mb" class="headerlink" title="EUR-20号染色体-21mb-22mb"></a>EUR-20号染色体-21mb-22mb</h3><embed id="pdfPlayer" src="EUR.chr20.21-22mb.pdf" type="application/pdf" width="100%" height="600" >
            </div>
            <hr>
            <div>
              <div class="post-metas mb-3">
                
                  <div class="post-meta mr-3">
                    <i class="iconfont icon-category"></i>
                    
                      <a class="hover-with-bg" href="/categories/BIOINFO-EXPERIMENT-RECORDS/">BIOINFO EXPERIMENT RECORDS</a>
                    
                  </div>
                
                
                  <div class="post-meta">
                    <i class="iconfont icon-tags"></i>
                    
                      <a class="hover-with-bg" href="/tags/GWAS/">GWAS</a>
                    
                      <a class="hover-with-bg" href="/tags/SNP/">SNP</a>
                    
                      <a class="hover-with-bg" href="/tags/ASD/">ASD</a>
                    
                  </div>
                
              </div>
              
                <p class="note note-warning">
                  
                    本博客所有文章除特别声明外，均采用 <a target="_blank" href="https://creativecommons.org/licenses/by-sa/4.0/deed.zh" rel="nofollow noopener noopener">CC BY-SA 4.0 协议</a> ，转载请注明出处！
                  
                </p>
              
              
                <div class="post-prevnext">
                  <article class="post-prev col-6">
                    
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2021/09/22/20210625%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6%E8%80%83%E8%AF%95/">
                        <span class="hidden-mobile">20210625生物信息学考试</span>
                        <span class="visible-mobile">下一篇</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
              <!-- Comments -->
              <article class="comments" id="comments">
                
                  
                
                
  <div class="disqus" style="width:100%">
    <div id="disqus_thread"></div>
    <script>
      Fluid.utils.waitElementVisible('disqus_thread', function() {
        Fluid.utils.createCssLink('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqusjs.css');
        Fluid.utils.createScript('https://cdn.jsdelivr.net/npm/disqusjs@1.0/dist/disqus.js', function () {
          new DisqusJS({
            shortname: 'fluid',
            apikey: ''
          });
        });
      });
    </script>
    <noscript>Please enable JavaScript to view the
      <a target="_blank" href="https://disqus.com/?ref_noscript" rel="nofollow noopener noopener">comments powered by Disqus.</a>
    </noscript>
  </div>


              </article>
            
          </article>
        </div>
      </div>
    </div>
    
      <div class="d-none d-lg-block col-lg-2 toc-container" id="toc-ctn">
        <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;目录</p>
  <div class="toc-body" id="toc-body"></div>
</div>

      </div>
    
  </div>
</div>

<!-- Custom -->


    

    
      <a id="scroll-top-button" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">搜索</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v"
                 for="local-search-input">关键词</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>
    

    
  </main>

  <footer class="text-center mt-5 py-3">
  <div class="footer-content">
     <a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a> <i class="iconfont icon-love"></i> <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener"><span>Fluid</span></a> 
  </div>
  

  

  
</footer>

<!-- SCRIPTS -->

  <script  src="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://cdn.jsdelivr.net/npm/nprogress@0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" ></script>
<script  src="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/js/bootstrap.min.js" ></script>
<script  src="/js/debouncer.js" ></script>
<script  src="/js/events.js" ></script>
<script  src="/js/plugins.js" ></script>

<!-- Plugins -->


  
    <script  src="/js/lazyload.js" ></script>
  



  



  <script  src="https://cdn.jsdelivr.net/npm/tocbot@4.12.0/dist/tocbot.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.js" ></script>



  <script  src="https://cdn.jsdelivr.net/npm/anchor-js@4.3.0/anchor.min.js" ></script>



  <script defer src="https://cdn.jsdelivr.net/npm/clipboard@2.0.6/dist/clipboard.min.js" ></script>






  <script  src="https://cdn.jsdelivr.net/npm/typed.js@2.0.11/lib/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var title = document.getElementById('subtitle').title;
      
      typing(title)
      
    })(window, document);
  </script>



  <script  src="/js/local-search.js" ></script>
  <script>
    (function () {
      var path = "/local-search.xml";
      $('#local-search-input').on('click', function() {
        searchFunc(path, 'local-search-input', 'local-search-result');
      });
      $('#modalSearch').on('shown.bs.modal', function() {
        $('#local-search-input').focus();
      });
    })()
  </script>















<!-- 主题的启动项 保持在最底部 -->
<script  src="/js/boot.js" ></script>



</body>
</html>
